From b6c4d06bf21b565fa75a18a6724cb4f096b2bfce Mon Sep 17 00:00:00 2001
From: Ethan Wei <ethanwei@ami.com>
Date: Fri, 28 Jul 2023 03:32:02 -0700
Subject: [PATCH] Fix Compilation Werror on GZIP 	modified:  
 libdwfl/gzip.c
Upstream-Status: Pending
---
 libdwfl/gzip.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/libdwfl/gzip.c b/libdwfl/gzip.c
index ba8ecfb..4d51075 100644
--- a/libdwfl/gzip.c
+++ b/libdwfl/gzip.c
@@ -94,8 +94,9 @@ bigger_buffer (struct unzip_state *state, size_t start)
 {
   size_t more = state->size ? state->size * 2 : start;
   char *b = realloc (state->buffer, more);
-  while (unlikely (b == NULL) && more >= state->size + 1024)
-    b = realloc (state->buffer, more -= 1024);
+  if (unlikely (b == NULL))
+    while (unlikely (b == NULL) && more >= state->size + 1024)
+      b = realloc (state->buffer, more -= 1024);
   if (unlikely (b == NULL))
     return false;
   state->buffer = b;
-- 
2.25.1

