From 1596e5801386930fe4afc64392b51c226dbe8065 Mon Sep 17 00:00:00 2001
From: Revathir <revathir@ami.com>
Date: Fri, 6 Sep 2024 11:06:45 +0530
Subject: [PATCH] Fix to update DHCPv6 address in systemd file
Upstream-Status: Pending
---
 src/network/networkd-state-file.c | 16 ++++++++++++++--
 1 file changed, 14 insertions(+), 2 deletions(-)

diff --git a/src/network/networkd-state-file.c b/src/network/networkd-state-file.c
index 3a95ba8d8b..0db3c1bcf9 100644
--- a/src/network/networkd-state-file.c
+++ b/src/network/networkd-state-file.c
@@ -20,7 +20,7 @@
 #include "set.h"
 #include "strv.h"
 #include "tmpfile-util.h"
-
+#include "dhcp6-lease-internal.h"
 static int ordered_set_put_dns_servers(OrderedSet **s, int ifindex, struct in_addr_full **dns, unsigned n) {
         int r;
 
@@ -801,7 +801,19 @@ static int link_save(Link *link) {
         } else
                 (void) unlink(link->lease_file);
 
-        r = link_serialize_dhcp6_client(link, f);
+	FOREACH_DHCP6_ADDRESS(link->dhcp6_lease) {
+                /*This function call will only append DHCPv6 leased address related entries to the lease file*/
+                struct in6_addr address;
+                r = sd_dhcp6_lease_get_address(link->dhcp6_lease, &address);
+               if (r >= 0)
+               {
+                       fprintf(f,
+                       "DHCP6_ADDRESS=%s\n",
+                       IN6_ADDR_TO_STRING(&address));
+               }
+        }
+
+	r = link_serialize_dhcp6_client(link, f);
         if (r < 0)
                 return r;
 
