From ee7012be2e4630a296931e01e82a81617e5b6b6d Mon Sep 17 00:00:00 2001
From: sonan <sonan@ami.com>
Date: Wed, 27 May 2026 11:58:39 +0530
Subject: [PATCH] Handle NTPSec mode separately in timedated persist restore
 logic.

Upstream-Status: Pending [OneTree]

Signed-off-by: sonan <sonan@ami.com>
---
 src/timedate/timedated.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/timedate/timedated.c b/src/timedate/timedated.c
index 19c4c179b4..618fa92ec9 100644
--- a/src/timedate/timedated.c
+++ b/src/timedate/timedated.c
@@ -1163,6 +1163,10 @@ static int restore_ntp_state(Context* c, sd_bus* bus)
 	    }
         }
     }
+    else if (strstr(sync, "NTPSec") != NULL)
+     {
+         log_info("Persisted mode is NTPSec. Not modifying NTP services (managed by NTPSec manager).");
+     }
     else if (strstr(sync, "NTP") != NULL)
     {
         log_info("Persisted mode is NTP. Enabling/starting NTP services.");
-- 
2.34.1

