From e52815323d00339ba04cadb74b33d6b00d1fcc19 Mon Sep 17 00:00:00 2001
From: DanielYu <DanielYu@ami.com>
Date: Mon, 20 Apr 2026 15:10:47 +0800
Subject: [PATCH] Software Secure Boot workaround

---
 boot/image-fit.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/boot/image-fit.c b/boot/image-fit.c
index 3cc556b727..97b06845e9 100644
--- a/boot/image-fit.c
+++ b/boot/image-fit.c
@@ -2117,7 +2117,7 @@ int fit_image_load(struct bootm_headers *images, ulong addr,
 		if (image_type == IH_TYPE_KERNEL)
 			images->fit_uname_cfg = fit_base_uname_config;
 
-		if (FIT_IMAGE_ENABLE_VERIFY && images->verify) {
+		if (FIT_IMAGE_ENABLE_VERIFY) { //Temporarily remove `images->verify` from the if statement to enable SW secure boot
 			puts("   Verifying Hash Integrity ... ");
 			if (fit_config_verify(fit, cfg_noffset)) {
 				puts("Bad Data Hash\n");
-- 
2.43.0

