From 6efb8106430e49bc7c2db5132aeae96c5eeec47b Mon Sep 17 00:00:00 2001
From: arjuncr <arjuncr@ami.com>
Date: Mon, 1 May 2023 14:57:12 +0530
Subject: [PATCH] Setting NM_INITIALIZATION_MODE as 2

Signed-off-by: arjuncr <arjuncr@ami.com>
---
 provisioning.settings | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/provisioning.settings b/provisioning.settings
index 694f1ac..3d2ba90 100644
--- a/provisioning.settings
+++ b/provisioning.settings
@@ -91,7 +91,7 @@ set(PCIE_POWER_BUDGET 0)
 set(PROCHOT_ASSERTION_RATIO 0)
 
 #This option determines initialization behavior for OpenBMC Node Manager and its behavior when SPS Node Manager is detected in the platform.
-set(NM_INITIALIZATION_MODE 0)
+set(NM_INITIALIZATION_MODE 2)
 
 #Pre-configured interface of PCIe. Default option is PLDM, other is PECI_OVER_SMBUS.
 set(ACCELERATORS_INTERFACE "PLDM")
@@ -135,4 +135,4 @@ set(I2C_ADDR_MAX 104)
 set(I2C_ADDR_MIN 88)
 set(FORCE_SMBALERT_MASK_INTERVAL_TIME_MS 10000)
 set(REDUNDANCY_ENABLED true)
-set(SMART_ENABLED true)
\ No newline at end of file
+set(SMART_ENABLED true)
-- 
2.34.1

