From 066fa5cc12be7d3835d98de8db8510daa30bea64 Mon Sep 17 00:00:00 2001
From: Manoj Ashok <amanoj@ami.com>
Date: Mon, 26 Aug 2024 18:54:52 +0530
Subject: [PATCH] adding necessary bootargs to support the sw emmc boot

Signed-off-by: Manoj Ashok <amanoj@ami.com>
---
 include/configs/intel_ast2600.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/include/configs/intel_ast2600.h b/include/configs/intel_ast2600.h
index e40b56754a..7150e0af28 100644
--- a/include/configs/intel_ast2600.h
+++ b/include/configs/intel_ast2600.h
@@ -35,6 +35,10 @@
 	"bootspi=fdt addr 20100000 && fdt header get fitsize totalsize && cp.b 20100000 ${loadaddr} ${fitsize} && bootm; echo Error loading kernel FIT image\0"	\
 	"verify=yes\0"	\
 	"fieldmode=true\0" \
+	"setmmcargs=setenv bootargs ${bootargs} rootwait root=PARTLABEL=rofs\0" \
+	"bootmmcdev=0\0" \
+	"bootmmcpart=1\0" \
+	"bootemmc=setenv bootmmcdev 0;setenv bootpart 1; run setmmcargs; ext4load mmc ${bootmmcdev}:${bootmmcpart} ${loadaddr} fitImage && bootm; echo Error loading kernel FIT  image\0" \
 	""
 
 #endif	/* __CONFIG_H */
-- 
2.34.1

