From febf5fca1d92bd8f3f6b501436b0d67cbfb0613c Mon Sep 17 00:00:00 2001
From: kirubagharanas <kirubagharanas@ami.com>
Date: Tue, 26 Nov 2024 20:21:53 +0530
Subject: [PATCH] Save env variables before autoboot
 signature verfication failure

---
 common/main.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/common/main.c b/common/main.c
index 07b34bf2b0..749c3d36a3 100644
--- a/common/main.c
+++ b/common/main.c
@@ -12,6 +12,7 @@
 #include <cli.h>
 #include <console.h>
 #include <version.h>
+#include <environment.h>
 
 /*
  * Board-specific Platform code can reimplement show_boot_progress () if needed
@@ -58,6 +59,7 @@ void main_loop(void)
 	if (cli_process_fdt(&s))
 		cli_secure_boot_cmd(s);
 
+	env_save();
 	autoboot_command(s);
 
 	cli_loop();
-- 
2.34.1

