From 293df63536bcee5eb2fd3560aae23ab4ff4143b0 Mon Sep 17 00:00:00 2001
From: Manoj Ashok <amanoj@ami.com>
Date: Fri, 2 May 2025 20:44:17 +0530
Subject: [PATCH] update address of fdt location

Signed-off-by: Manoj Ashok <amanoj@ami.com>
---
 include/configs/intel_ast2600.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/include/configs/intel_ast2600.h b/include/configs/intel_ast2600.h
index e40b56754a..ccd3f1cd81 100644
--- a/include/configs/intel_ast2600.h
+++ b/include/configs/intel_ast2600.h
@@ -32,7 +32,8 @@
 #undef CONFIG_EXTRA_ENV_SETTINGS
 #define CONFIG_EXTRA_ENV_SETTINGS	\
 	"loadaddr=" STR(CONFIG_SYS_LOAD_ADDR) "\0"	\
-	"bootspi=fdt addr 20100000 && fdt header get fitsize totalsize && cp.b 20100000 ${loadaddr} ${fitsize} && bootm; echo Error loading kernel FIT image\0"	\
+	"bootspi=fdt addr ${ftdspiaddr} && fdt header get fitsize totalsize && cp.b ${ftdspiaddr} ${loadaddr} ${fitsize} && bootm; echo Error loading kernel FIT image\0"	\
+	"ftdspiaddr=21120000\0" \
 	"verify=yes\0"	\
 	"fieldmode=true\0" \
 	""
-- 
2.34.1

