diff -Naur a/include/configs/aspeed-common.h  b/include/configs/aspeed-common.h
--- a/include/configs/aspeed-common.h	2023-08-24 18:34:25.307682129 +0530
+++ b/include/configs/aspeed-common.h	2023-08-24 18:27:54.846637434 +0530
@@ -13,10 +13,11 @@
 
 #define CONFIG_BOOTFILE		"all.bin"
 
-#define CONFIG_GATEWAYIP	192.168.0.1
-#define CONFIG_NETMASK		255.255.255.0
-#define CONFIG_IPADDR		192.168.0.45
-#define CONFIG_SERVERIP		192.168.0.81
+//#define CONFIG_GATEWAYIP	192.168.0.1
+//#define CONFIG_NETMASK		255.255.255.0
+//#define CONFIG_IPADDR		192.168.0.45
+//#define CONFIG_SERVERIP		192.168.0.81
+#define CONFIG_SYS_AUTOLOAD      "no"
 
 #define CONFIG_STANDALONE_LOAD_ADDR 0x83000000
 
