From efd00850e3edd4938bbc807b80273399c460ccb4 Mon Sep 17 00:00:00 2001
From: Minal Upadhye <minalsunilu@ami.com>
Date: Tue, 22 Jul 2025 09:10:55 -0400
Subject: [PATCH] migrate-journal-events-to-dbus_x86

Signed-off-by: Minal Upadhye <minalsunilu@ami.com>
---
 src/power_control.cpp | 298 +++++++++++++++++++++++++++++++++---------
 1 file changed, 233 insertions(+), 65 deletions(-)

diff --git a/src/power_control.cpp b/src/power_control.cpp
index 272cc8b..cf0830e 100644
--- a/src/power_control.cpp
+++ b/src/power_control.cpp
@@ -816,55 +816,177 @@ static void setRestartCause()
 
 static void systemPowerGoodFailedLog()
 {
-    sd_journal_send(
-        "MESSAGE=PowerControl: system power good failed to assert (VR failure)",
-        "PRIORITY=%i", LOG_INFO, "REDFISH_MESSAGE_ID=%s",
-        "OpenBMC.0.1.SystemPowerGoodFailed", "REDFISH_MESSAGE_ARGS=%d",
-        TimerMap["SioPowerGoodWatchdogMs"], NULL);
+    std::string severity = "xyz.openbmc_project.Logging.Entry.Level.Critical";
+    auto bus = sdbusplus::bus::new_default_system();
+    sdbusplus::message::message m = bus.new_method_call(
+        "xyz.openbmc_project.Logging", "/xyz/openbmc_project/logging",
+        "xyz.openbmc_project.Logging.Create", "Create");
+    std::string messageID = "OpenBMC.0.1.SystemPowerGoodFailed";
+    std::string arg = std::to_string(TimerMap["SioPowerGoodWatchdogMs"]);
+    std::string journalMsg = messageID + "," + arg;
+
+    std::map<std::string, std::string> additionalData;
+    additionalData["REDFISH_MESSAGE_ID"] = messageID;
+    additionalData["REDFISH_MESSAGE_ARGS"] = arg;
+    additionalData["Additional_Data"] =
+        "PowerControl: system power good failed to assert (VR failure)";
+    m.append(journalMsg, severity, additionalData);
+    try
+    {
+        bus.call(m);
+    }
+    catch (const sdbusplus::exception_t& e)
+    {
+        lg2::error("Failed to create log entry: {ERROR}", "ERROR", e.what());
+    }
 }
 
 static void psPowerOKFailedLog()
 {
-    sd_journal_send(
-        "MESSAGE=PowerControl: power supply power good failed to assert",
-        "PRIORITY=%i", LOG_INFO, "REDFISH_MESSAGE_ID=%s",
-        "OpenBMC.0.1.PowerSupplyPowerGoodFailed", "REDFISH_MESSAGE_ARGS=%d",
-        TimerMap["PsPowerOKWatchdogMs"], NULL);
+    std::string severity = "xyz.openbmc_project.Logging.Entry.Level.Critical";
+    auto bus = sdbusplus::bus::new_default_system();
+    sdbusplus::message::message m = bus.new_method_call(
+        "xyz.openbmc_project.Logging", "/xyz/openbmc_project/logging",
+        "xyz.openbmc_project.Logging.Create", "Create");
+    std::string messageID = "OpenBMC.0.1.PowerSupplyPowerGoodFailed";
+    std::string arg = std::to_string(TimerMap["PsPowerOKWatchdogMs"]);
+    std::string journalMsg = messageID + "," + arg;
+
+    std::map<std::string, std::string> additionalData;
+    additionalData["REDFISH_MESSAGE_ID"] = messageID;
+    additionalData["REDFISH_MESSAGE_ARGS"] = arg;
+    additionalData["Additional_Data"] =
+        "PowerControl: power supply power good failed to asser";
+    m.append(journalMsg, severity, additionalData);
+    try
+    {
+        bus.call(m);
+    }
+    catch (const sdbusplus::exception_t& e)
+    {
+        lg2::error("Failed to create log entry: {ERROR}", "ERROR", e.what());
+    }
 }
 
 static void powerRestorePolicyLog()
 {
-    sd_journal_send("MESSAGE=PowerControl: power restore policy applied",
-                    "PRIORITY=%i", LOG_INFO, "REDFISH_MESSAGE_ID=%s",
-                    "OpenBMC.0.1.PowerRestorePolicyApplied", NULL);
+    std::string severity =
+        "xyz.openbmc_project.Logging.Entry.Level.Informational";
+    auto bus = sdbusplus::bus::new_default_system();
+    sdbusplus::message::message m = bus.new_method_call(
+        "xyz.openbmc_project.Logging", "/xyz/openbmc_project/logging",
+        "xyz.openbmc_project.Logging.Create", "Create");
+    std::string journalMsg = "OpenBMC.0.1.PowerRestorePolicyApplied";
+
+    std::map<std::string, std::string> additionalData;
+    additionalData["REDFISH_MESSAGE_ID"] =
+        "OpenBMC.0.1.PowerRestorePolicyApplied";
+    additionalData["Additional_Data"] =
+        "PowerControl: power restore policy applied";
+    m.append(journalMsg, severity, additionalData);
+    try
+    {
+        bus.call(m);
+    }
+    catch (const sdbusplus::exception_t& e)
+    {
+        lg2::info("Failed to create log entry: {ERROR}", "ERROR", e.what());
+    }
 }
 
 static void powerButtonPressLog()
 {
-    sd_journal_send("MESSAGE=PowerControl: power button pressed", "PRIORITY=%i",
-                    LOG_INFO, "REDFISH_MESSAGE_ID=%s",
-                    "OpenBMC.0.1.PowerButtonPressed", NULL);
+    std::string severity =
+        "xyz.openbmc_project.Logging.Entry.Level.Informational";
+    auto bus = sdbusplus::bus::new_default_system();
+    sdbusplus::message::message m = bus.new_method_call(
+        "xyz.openbmc_project.Logging", "/xyz/openbmc_project/logging",
+        "xyz.openbmc_project.Logging.Create", "Create");
+    std::string journalMsg = "OpenBMC.0.1.PowerButtonPressed";
+
+    std::map<std::string, std::string> additionalData;
+    additionalData["REDFISH_MESSAGE_ID"] = "OpenBMC.0.1.PowerButtonPressed";
+    additionalData["Additional_Data"] = "PowerControl: power button pressed";
+    m.append(journalMsg, severity, additionalData);
+    try
+    {
+        bus.call(m);
+    }
+    catch (const sdbusplus::exception_t& e)
+    {
+        lg2::info("Failed to create log entry: {ERROR}", "ERROR", e.what());
+    }
 }
 
 static void resetButtonPressLog()
 {
-    sd_journal_send("MESSAGE=PowerControl: reset button pressed", "PRIORITY=%i",
-                    LOG_INFO, "REDFISH_MESSAGE_ID=%s",
-                    "OpenBMC.0.1.ResetButtonPressed", NULL);
+    std::string severity =
+        "xyz.openbmc_project.Logging.Entry.Level.Informational";
+    auto bus = sdbusplus::bus::new_default_system();
+    sdbusplus::message::message m = bus.new_method_call(
+        "xyz.openbmc_project.Logging", "/xyz/openbmc_project/logging",
+        "xyz.openbmc_project.Logging.Create", "Create");
+    std::string journalMsg = "OpenBMC.0.1.ResetButtonPressed";
+
+    std::map<std::string, std::string> additionalData;
+    additionalData["REDFISH_MESSAGE_ID"] = "OpenBMC.0.1.ResetButtonPressed";
+    additionalData["Additional_Data"] = "PowerControl: reset button pressed";
+    m.append(journalMsg, severity, additionalData);
+    try
+    {
+        bus.call(m);
+    }
+    catch (const sdbusplus::exception_t& e)
+    {
+        lg2::info("Failed to create log entry: {ERROR}", "ERROR", e.what());
+    }
 }
 
 static void nmiButtonPressLog()
 {
-    sd_journal_send("MESSAGE=PowerControl: NMI button pressed", "PRIORITY=%i",
-                    LOG_INFO, "REDFISH_MESSAGE_ID=%s",
-                    "OpenBMC.0.1.NMIButtonPressed", NULL);
+    std::string severity = "xyz.openbmc_project.Logging.Entry.Level.Critical";
+    auto bus = sdbusplus::bus::new_default_system();
+    sdbusplus::message::message m = bus.new_method_call(
+        "xyz.openbmc_project.Logging", "/xyz/openbmc_project/logging",
+        "xyz.openbmc_project.Logging.Create", "Create");
+    std::string journalMsg = "OpenBMC.0.1.NMIButtonPressed";
+
+    std::map<std::string, std::string> additionalData;
+    additionalData["REDFISH_MESSAGE_ID"] = "OpenBMC.0.1.NMIButtonPressed";
+    additionalData["Additional_Data"] = "PowerControl: NMI button pressed";
+    m.append(journalMsg, severity, additionalData);
+    try
+    {
+        bus.call(m);
+    }
+    catch (const sdbusplus::exception_t& e)
+    {
+        lg2::error("Failed to create log entry: {ERROR}", "ERROR", e.what());
+    }
 }
 
 static void nmiDiagIntLog()
 {
-    sd_journal_send("MESSAGE=PowerControl: NMI Diagnostic Interrupt",
-                    "PRIORITY=%i", LOG_INFO, "REDFISH_MESSAGE_ID=%s",
-                    "OpenBMC.0.1.NMIDiagnosticInterrupt", NULL);
+    std::string severity = "xyz.openbmc_project.Logging.Entry.Level.Critical";
+    auto bus = sdbusplus::bus::new_default_system();
+    sdbusplus::message::message m = bus.new_method_call(
+        "xyz.openbmc_project.Logging", "/xyz/openbmc_project/logging",
+        "xyz.openbmc_project.Logging.Create", "Create");
+    std::string journalMsg = "OpenBMC.0.1.NMIDiagnosticInterrupt";
+
+    std::map<std::string, std::string> additionalData;
+    additionalData["REDFISH_MESSAGE_ID"] = "OpenBMC.0.1.NMIDiagnosticInterrupt";
+    additionalData["Additional_Data"] =
+        "PowerControl: NMI Diagnostic Interrupt";
+    m.append(journalMsg, severity, additionalData);
+    try
+    {
+        bus.call(m);
+    }
+    catch (const sdbusplus::exception_t& e)
+    {
+        lg2::error("Failed to create log entry: {ERROR}", "ERROR", e.what());
+    }
 }
 
 PersistentState::PersistentState()
@@ -1483,12 +1605,14 @@ static void forcePowerOff()
 
         lg2::error("Power-button override failed. Not sure what to do now.");
 
-        if((powerState == PowerState::transitionToOff) || (powerState == PowerState::transitionToCycleOff))
+        if ((powerState == PowerState::transitionToOff) ||
+            (powerState == PowerState::transitionToCycleOff))
         {
             powerState = PowerState::on;
             hostIface->set_property(
-            "CurrentHostState",
-            std::string("xyz.openbmc_project.State.Host.HostState.Running"));
+                "CurrentHostState",
+                std::string(
+                    "xyz.openbmc_project.State.Host.HostState.Running"));
         }
     });
 }
@@ -1714,10 +1838,29 @@ static void currentHostStateMonitor()
                 pohCounterTimerStart();
                 // Clear the restart cause set for the next restart
                 clearRestartCause();
-                sd_journal_send("MESSAGE=Host system DC power is on",
-                                "PRIORITY=%i", LOG_INFO,
-                                "REDFISH_MESSAGE_ID=%s",
-                                "OpenBMC.0.1.DCPowerOn", NULL);
+                std::string severity =
+                    "xyz.openbmc_project.Logging.Entry.Level.Informational";
+                auto bus = sdbusplus::bus::new_default_system();
+                sdbusplus::message::message m = bus.new_method_call(
+                    "xyz.openbmc_project.Logging",
+                    "/xyz/openbmc_project/logging",
+                    "xyz.openbmc_project.Logging.Create", "Create");
+                std::string journalMsg = "OpenBMC.0.1.DCPowerOn";
+
+                std::map<std::string, std::string> additionalData;
+                additionalData["REDFISH_MESSAGE_ID"] = "OpenBMC.0.1.DCPowerOn";
+                additionalData["Additional_Data"] =
+                    "Host system DC power is on";
+                m.append(journalMsg, severity, additionalData);
+                try
+                {
+                    bus.call(m);
+                }
+                catch (const sdbusplus::exception_t& e)
+                {
+                    lg2::info("Failed to create log entry: {ERROR}", "ERROR",
+                              e.what());
+                }
             }
             else
             {
@@ -1734,10 +1877,29 @@ static void currentHostStateMonitor()
 #ifdef USE_ACBOOT
                 resetACBootProperty();
 #endif // USE_ACBOOT
-                sd_journal_send("MESSAGE=Host system DC power is off",
-                                "PRIORITY=%i", LOG_INFO,
-                                "REDFISH_MESSAGE_ID=%s",
-                                "OpenBMC.0.1.DCPowerOff", NULL);
+                std::string severity =
+                    "xyz.openbmc_project.Logging.Entry.Level.Informational";
+                auto bus = sdbusplus::bus::new_default_system();
+                sdbusplus::message::message m = bus.new_method_call(
+                    "xyz.openbmc_project.Logging",
+                    "/xyz/openbmc_project/logging",
+                    "xyz.openbmc_project.Logging.Create", "Create");
+                std::string journalMsg = "OpenBMC.0.1.DCPowerOff";
+
+                std::map<std::string, std::string> additionalData;
+                additionalData["REDFISH_MESSAGE_ID"] = "OpenBMC.0.1.DCPowerOff";
+                additionalData["Additional_Data"] =
+                    "Host system DC power is off";
+                m.append(journalMsg, severity, additionalData);
+                try
+                {
+                    bus.call(m);
+                }
+                catch (const sdbusplus::exception_t& e)
+                {
+                    lg2::info("Failed to create log entry: {ERROR}", "ERROR",
+                              e.what());
+                }
             }
         });
 }
@@ -3913,45 +4075,51 @@ int main(int argc, char* argv[])
         "HostTransitionTimeOut", timeOut,
         [](const uint64_t& requested, uint64_t& propertyValue) {
             propertyValue = requested;
-           hostTimeOut = propertyValue;
+            hostTimeOut = propertyValue;
             return true;
         });
     osIface->initialize();
 
-// Task State Service
+    // Task State Service
     sdbusplus::asio::object_server taskServer =
         sdbusplus::asio::object_server(conn);
 
     // OS State Interface
-    taskIface = taskServer.add_interface(
-        "/xyz/openbmc_project/state/host0",
-        "xyz.openbmc_project.Common.Task");
+    taskIface = taskServer.add_interface("/xyz/openbmc_project/state/host0",
+                                         "xyz.openbmc_project.Common.Task");
 
     taskIface->register_property(
-         "Status", std::string("xyz.openbmc_project.Common.Task.OperationStatus.Completed"),
-         [](const std::string& requested, std::string& resp) {
-
-         if((requested == "xyz.openbmc_project.Common.Task.OperationStatus.Cancelled") && (chassisTimeOut != 0))
-         {
-                 chassisTransitionTimer.cancel();
-                         chassis_timer = true;
-         }
-
-         if((requested == "xyz.openbmc_project.Common.Task.OperationStatus.Cancelled") && (powerTimeOut != 0))
-         {
-               powerTransitionTimer.cancel();
-                       power_timer = true;
-         }
-
-         if((requested == "xyz.openbmc_project.Common.Task.OperationStatus.Cancelled") && (hostTimeOut != 0))
-         {
-                 hostTransitionTimer.cancel();
-                       host_timer = true;
-         }
-
-          resp = requested;
-          return 1;
-         });
+        "Status",
+        std::string(
+            "xyz.openbmc_project.Common.Task.OperationStatus.Completed"),
+        [](const std::string& requested, std::string& resp) {
+            if ((requested ==
+                 "xyz.openbmc_project.Common.Task.OperationStatus.Cancelled") &&
+                (chassisTimeOut != 0))
+            {
+                chassisTransitionTimer.cancel();
+                chassis_timer = true;
+            }
+
+            if ((requested ==
+                 "xyz.openbmc_project.Common.Task.OperationStatus.Cancelled") &&
+                (powerTimeOut != 0))
+            {
+                powerTransitionTimer.cancel();
+                power_timer = true;
+            }
+
+            if ((requested ==
+                 "xyz.openbmc_project.Common.Task.OperationStatus.Cancelled") &&
+                (hostTimeOut != 0))
+            {
+                hostTransitionTimer.cancel();
+                host_timer = true;
+            }
+
+            resp = requested;
+            return 1;
+        });
 
     taskIface->register_property(
         "TaskId", defaultId, sdbusplus::asio::PropertyPermission::readWrite);
-- 
2.34.1

