From ec725207672ba91164e9bd4305ffb0d16f483ff9 Mon Sep 17 00:00:00 2001
From: ramadevip <ramadevip@ami.com>
Date: Thu, 8 Feb 2024 13:24:19 +0530
Subject: [PATCH] Power operation for future time

Signed-off-by: ramadevip <ramadevip@ami.com>
---
 src/power_control.cpp | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/src/power_control.cpp b/src/power_control.cpp
index 732e1fc..f91d7dd 100644
--- a/src/power_control.cpp
+++ b/src/power_control.cpp
@@ -48,9 +48,9 @@ PowerRestoreController powerRestore(io);
 static std::string node = "0";
 static const std::string appName = "power-control";
 
-uint16_t powerTimeOut;
-uint16_t timeOut = 0;
-uint16_t prop = 0;
+uint64_t powerTimeOut;
+uint64_t timeOut = 0;
+uint64_t prop = 0;
 bool timerStarted_system = false;
 bool timerStarted = false;
 bool timerStarted_chassis = false;
@@ -3704,7 +3704,7 @@ int main(int argc, char* argv[])
 
     osIface->register_property(
         "ChassisHostTransitionTimeOut", timeOut,
-        [](const uint16_t& requested, uint16_t& propertyValue) {
+        [](const uint64_t& requested, uint64_t& propertyValue) {
         propertyValue = requested;
         powerTimeOut = propertyValue;
         return true;
@@ -3712,7 +3712,7 @@ int main(int argc, char* argv[])
 
     osIface->register_property(
         "PowerTransitionTimeOut", timeOut,
-        [](const uint16_t& requested, uint16_t& propertyValue) {
+        [](const uint64_t& requested, uint64_t& propertyValue) {
         propertyValue = requested;
         powerTimeOut = propertyValue;
         return true;
@@ -3720,7 +3720,7 @@ int main(int argc, char* argv[])
 
     osIface->register_property(
         "HostTransitionTimeOut", timeOut,
-        [](const uint16_t& requested, uint16_t& propertyValue) {
+        [](const uint64_t& requested, uint64_t& propertyValue) {
         propertyValue = requested;
         powerTimeOut = propertyValue;
         return true;
-- 
2.25.1

