From 9ac6032340519e881b0a3afc3e2f4139381752f7 Mon Sep 17 00:00:00 2001
From: ramadevip <ramadevip@ami.com>
Date: Mon, 5 May 2025 13:02:55 +0530
Subject: [PATCH] Adding proper timer checkout for chassis and system reset
 actions

Signed-off-by: ramadevip <ramadevip@ami.com>
---
 src/power_control.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/power_control.cpp b/src/power_control.cpp
index bf3c013..02ed48a 100644
--- a/src/power_control.cpp
+++ b/src/power_control.cpp
@@ -3434,7 +3434,7 @@ int main(int argc, char* argv[])
             else if (requested ==
                      "xyz.openbmc_project.State.Chassis.Transition.PowerCycle")
             {
-                if (timerStarted == false)
+                if (timerStarted_chassis == false)
                 {
                     timerStarted_chassis = true;
                     auto interval = std::chrono::seconds(chassisTimeOut);
-- 
2.43.0

