From e91ee7da35663e017753ebe6ef234142d55d1616 Mon Sep 17 00:00:00 2001
From: viswanathc <viswanathc@ami.com>
Date: Wed, 16 Jul 2025 01:03:17 -0400
Subject: [PATCH] Coverity-Fix

---
 src/power_control.cpp | 283 +++++++++++++++++++++++++++++++++++-------
 1 file changed, 235 insertions(+), 48 deletions(-)

diff --git a/src/power_control.cpp b/src/power_control.cpp
index 272cc8b..731cdd5 100644
--- a/src/power_control.cpp
+++ b/src/power_control.cpp
@@ -911,7 +911,15 @@ PersistentState::PersistentState()
 }
 PersistentState::~PersistentState()
 {
-    saveState();
+    try
+    {
+        saveState();
+    }
+    catch (const std::exception& e)
+    {
+        lg2::error("Failed to Save State: {ERROR_MSG}",
+                    "ERROR_MSG", e.what());
+    }
 }
 const std::string PersistentState::get(Params parameter)
 {
@@ -2811,12 +2819,28 @@ int main(int argc, char* argv[])
     lg2::info("Start Chassis power control service for host : {NODE}", "NODE",
               node);
 
-    conn = std::make_shared<sdbusplus::asio::connection>(io);
+    try
+    {
+        conn = std::make_shared<sdbusplus::asio::connection>(io);
+    }
+    catch (const std::exception& e)
+    {
+        lg2::error("standard exception caught: {ERROR}", "ERROR", e);
+        return -1;
+    }
 
     // Load GPIO's through json config file
-    if (loadConfigValues() == -1)
+    try
     {
-        lg2::error("Host{NODE}: Error in Parsing...", "NODE", node);
+        if (loadConfigValues() == -1)
+        {
+            lg2::error("Host{NODE}: Error in Parsing...", "NODE", node);
+        }
+    }
+    catch (const std::exception& e)
+    {
+        lg2::error("standard exception caught: {ERROR}", "ERROR", e);
+        return -1;
     }
     /* Currently for single host based systems additional busname is added
     with "0" at the end of the name ex : xyz.openbmc_project.State.Host0.
@@ -2827,12 +2851,20 @@ int main(int argc, char* argv[])
     if (node == "0")
     {
         // Request all the dbus names
+	try
+	{
         conn->request_name(hostDbusName.c_str());
         conn->request_name(chassisDbusName.c_str());
         conn->request_name(osDbusName.c_str());
         conn->request_name(buttonDbusName.c_str());
         conn->request_name(nmiDbusName.c_str());
         conn->request_name(rstCauseDbusName.c_str());
+	}
+    catch (const std::exception& e)
+    {
+        lg2::error("standard exception caught: {ERROR}", "ERROR", e);
+        return -1;
+    }
     }
 
     hostDbusName += node;
@@ -2843,12 +2875,20 @@ int main(int argc, char* argv[])
     rstCauseDbusName += node;
 
     // Request all the dbus names
-    conn->request_name(hostDbusName.c_str());
-    conn->request_name(chassisDbusName.c_str());
-    conn->request_name(osDbusName.c_str());
-    conn->request_name(buttonDbusName.c_str());
-    conn->request_name(nmiDbusName.c_str());
-    conn->request_name(rstCauseDbusName.c_str());
+    try
+    {
+        conn->request_name(hostDbusName.c_str());
+        conn->request_name(chassisDbusName.c_str());
+        conn->request_name(osDbusName.c_str());
+        conn->request_name(buttonDbusName.c_str());
+        conn->request_name(nmiDbusName.c_str());
+        conn->request_name(rstCauseDbusName.c_str());
+    }
+    catch (const std::exception& e)
+    {
+        lg2::error("standard exception caught: {ERROR}", "ERROR", e);
+        return -1;
+    }
 
     if (sioPwrGoodConfig.lineName.empty() ||
         sioOnControlConfig.lineName.empty() || sioS5Config.lineName.empty())
@@ -2860,9 +2900,17 @@ int main(int argc, char* argv[])
     // Request PS_PWROK GPIO events
     if (powerOkConfig.type == ConfigType::GPIO)
     {
-        if (!requestGPIOEvents(powerOkConfig.lineName, psPowerOKHandler,
-                               psPowerOKLine, psPowerOKEvent))
+        try
+        {
+            if (!requestGPIOEvents(powerOkConfig.lineName, psPowerOKHandler,
+                                psPowerOKLine, psPowerOKEvent))
+            {
+                return -1;
+            }
+        }
+        catch (const std::exception& e)
         {
+            lg2::error("standard exception caught: {ERROR}", "ERROR", e);
             return -1;
         }
     }
@@ -2882,10 +2930,18 @@ int main(int argc, char* argv[])
         // Request SIO_POWER_GOOD GPIO events
         if (sioPwrGoodConfig.type == ConfigType::GPIO)
         {
-            if (!requestGPIOEvents(sioPwrGoodConfig.lineName,
-                                   sioPowerGoodHandler, sioPowerGoodLine,
-                                   sioPowerGoodEvent))
+            try
             {
+                if (!requestGPIOEvents(sioPwrGoodConfig.lineName,
+                                    sioPowerGoodHandler, sioPowerGoodLine,
+                                    sioPowerGoodEvent))
+                {
+                    return -1;
+                }
+            }
+            catch (const std::exception& e)
+            {
+                lg2::error("standard exception caught: {ERROR}", "ERROR", e);
                 return -1;
             }
         }
@@ -2905,10 +2961,18 @@ int main(int argc, char* argv[])
         // Request SIO_ONCONTROL GPIO events
         if (sioOnControlConfig.type == ConfigType::GPIO)
         {
-            if (!requestGPIOEvents(sioOnControlConfig.lineName,
-                                   sioOnControlHandler, sioOnControlLine,
-                                   sioOnControlEvent))
+            try
+            {
+                if (!requestGPIOEvents(sioOnControlConfig.lineName,
+                                    sioOnControlHandler, sioOnControlLine,
+                                    sioOnControlEvent))
+                {
+                    return -1;
+                }
+            }
+            catch (const std::exception& e)
             {
+                lg2::error("standard exception caught: {ERROR}", "ERROR", e);
                 return -1;
             }
         }
@@ -2928,9 +2992,17 @@ int main(int argc, char* argv[])
         // Request SIO_S5 GPIO events
         if (sioS5Config.type == ConfigType::GPIO)
         {
-            if (!requestGPIOEvents(sioS5Config.lineName, sioS5Handler,
-                                   sioS5Line, sioS5Event))
+            try
             {
+                if (!requestGPIOEvents(sioS5Config.lineName, sioS5Handler,
+                                    sioS5Line, sioS5Event))
+                {
+                    return -1;
+                }
+            }
+            catch (const std::exception& e)
+            {
+                lg2::error("standard exception caught: {ERROR}", "ERROR", e);
                 return -1;
             }
         }
@@ -2949,9 +3021,17 @@ int main(int argc, char* argv[])
     // Request POWER_BUTTON GPIO events
     if (powerButtonConfig.type == ConfigType::GPIO)
     {
-        if (!requestGPIOEvents(powerButtonConfig.lineName, powerButtonHandler,
-                               powerButtonLine, powerButtonEvent))
+        try
         {
+            if (!requestGPIOEvents(powerButtonConfig.lineName, powerButtonHandler,
+                                powerButtonLine, powerButtonEvent))
+            {
+                return -1;
+            }
+        }
+        catch (const std::exception& e)
+        {
+            lg2::error("standard exception caught: {ERROR}", "ERROR", e);
             return -1;
         }
     }
@@ -2965,9 +3045,17 @@ int main(int argc, char* argv[])
     // Request RESET_BUTTON GPIO events
     if (resetButtonConfig.type == ConfigType::GPIO)
     {
-        if (!requestGPIOEvents(resetButtonConfig.lineName, resetButtonHandler,
-                               resetButtonLine, resetButtonEvent))
+        try
         {
+            if (!requestGPIOEvents(resetButtonConfig.lineName, resetButtonHandler,
+                                resetButtonLine, resetButtonEvent))
+            {
+                return -1;
+            }
+        }
+        catch (const std::exception& e)
+        {
+            lg2::error("standard exception caught: {ERROR}", "ERROR", e);
             return -1;
         }
     }
@@ -2981,10 +3069,18 @@ int main(int argc, char* argv[])
     // Request NMI_BUTTON GPIO events
     if (nmiButtonConfig.type == ConfigType::GPIO)
     {
-        if (!nmiButtonConfig.lineName.empty())
+        try
+        {
+            if (!nmiButtonConfig.lineName.empty())
+            {
+                requestGPIOEvents(nmiButtonConfig.lineName, nmiButtonHandler,
+                                nmiButtonLine, nmiButtonEvent);
+            }
+        }
+        catch (const std::exception& e)
         {
-            requestGPIOEvents(nmiButtonConfig.lineName, nmiButtonHandler,
-                              nmiButtonLine, nmiButtonEvent);
+            lg2::error("standard exception caught: {ERROR}", "ERROR", e);
+            return -1;
         }
     }
     else if (nmiButtonConfig.type == ConfigType::DBUS)
@@ -2996,10 +3092,18 @@ int main(int argc, char* argv[])
     // Request ID_BUTTON GPIO events
     if (idButtonConfig.type == ConfigType::GPIO)
     {
-        if (!idButtonConfig.lineName.empty())
+        try
         {
-            requestGPIOEvents(idButtonConfig.lineName, idButtonHandler,
-                              idButtonLine, idButtonEvent);
+            if (!idButtonConfig.lineName.empty())
+            {
+                requestGPIOEvents(idButtonConfig.lineName, idButtonHandler,
+                                idButtonLine, idButtonEvent);
+            }
+        }
+        catch (const std::exception& e)
+        {
+            lg2::error("standard exception caught: {ERROR}", "ERROR", e);
+            return -1;
         }
     }
     else if (idButtonConfig.type == ConfigType::DBUS)
@@ -3019,9 +3123,17 @@ int main(int argc, char* argv[])
     // Request POST_COMPLETE GPIO events
     if (postCompleteConfig.type == ConfigType::GPIO)
     {
-        if (!requestGPIOEvents(postCompleteConfig.lineName, postCompleteHandler,
-                               postCompleteLine, postCompleteEvent))
+        try
+        {
+            if (!requestGPIOEvents(postCompleteConfig.lineName, postCompleteHandler,
+                                postCompleteLine, postCompleteEvent))
+            {
+                return -1;
+            }
+        }
+        catch (const std::exception& e)
         {
+            lg2::error("standard exception caught: {ERROR}", "ERROR", e);
             return -1;
         }
     }
@@ -3041,8 +3153,11 @@ int main(int argc, char* argv[])
     // initialize NMI_OUT GPIO.
     if (!nmiOutConfig.lineName.empty())
     {
-        setGPIOOutput(nmiOutConfig.lineName, !nmiOutConfig.polarity,
-                      nmiOutLine);
+        if(!setGPIOOutput(nmiOutConfig.lineName, !nmiOutConfig.polarity,
+                      nmiOutLine))
+        {
+            return -1;
+        }
     }
 
     // Initialize POWER_OUT and RESET_OUT GPIO.
@@ -3093,15 +3208,31 @@ int main(int argc, char* argv[])
     }
     else
     {
-        if (getProperty(powerOkConfig))
+        try
         {
-            powerState = PowerState::on;
+            if (getProperty(powerOkConfig))
+            {
+                powerState = PowerState::on;
+            }
+        }
+        catch (const std::exception& e)
+        {
+            lg2::error("standard exception caught: {ERROR}", "ERROR", e);
+            return -1;
         }
     }
     // Check if we need to start the Power Restore policy
     if (powerState != PowerState::on && bmcBootCheck())
     {
-        powerRestore.run();
+        try
+        {
+            powerRestore.run();
+        }
+        catch (const std::exception& e)
+        {
+            lg2::error("standard exception caught: {ERROR}", "ERROR", e);
+            return -1;
+        }
     }
 
     if (nmiOutLine)
@@ -3527,8 +3658,9 @@ int main(int argc, char* argv[])
             }
 
             if (resp == requested)
+            {
                 sendSignal(chassisIface, "RequestedPowerTransition");
-
+            }
             resp = requested;
             return 1;
         });
@@ -3715,7 +3847,15 @@ int main(int argc, char* argv[])
         }
         else
         {
-            powerButtonPressed = getProperty(powerButtonConfig) == 0;
+            try
+            {
+                powerButtonPressed = getProperty(powerButtonConfig) == 0;
+            }
+            catch (const std::exception& e)
+            {
+                lg2::error("standard exception caught: {ERROR}", "ERROR", e);
+                return -1;
+            }
         }
 
         powerButtonIface->register_property("ButtonPressed",
@@ -3771,7 +3911,15 @@ int main(int argc, char* argv[])
         }
         else
         {
-            resetButtonPressed = getProperty(resetButtonConfig) == 0;
+            try
+            {
+                resetButtonPressed = getProperty(resetButtonConfig) == 0;
+            }
+            catch (const std::exception& e)
+            {
+                lg2::error("standard exception caught: {ERROR}", "ERROR", e);
+                return -1;
+            }
         }
 
         resetButtonIface->register_property("ButtonPressed",
@@ -3817,7 +3965,15 @@ int main(int argc, char* argv[])
         }
         else
         {
-            nmiButtonPressed = getProperty(nmiButtonConfig) == 0;
+            try
+            {
+                nmiButtonPressed = getProperty(nmiButtonConfig) == 0;
+            }
+            catch (const std::exception& e)
+            {
+                lg2::error("standard exception caught: {ERROR}", "ERROR", e);
+                return -1;
+            }
         }
 
         nmiButtonIface->register_property("ButtonPressed", nmiButtonPressed);
@@ -3854,7 +4010,15 @@ int main(int argc, char* argv[])
         }
         else
         {
-            idButtonPressed = getProperty(idButtonConfig) == 0;
+            try
+            {
+                idButtonPressed = getProperty(idButtonConfig) == 0;
+            }
+            catch (const std::exception& e)
+            {
+                lg2::error("standard exception caught: {ERROR}", "ERROR", e);
+                return -1;
+            }
         }
 
         idButtonIface->register_property("ButtonPressed", idButtonPressed);
@@ -3884,9 +4048,17 @@ int main(int argc, char* argv[])
     }
     else
     {
-        osState = getProperty(postCompleteConfig) > 0
-                      ? OperatingSystemStateStage::Standby
-                      : OperatingSystemStateStage::Inactive;
+        try
+        {
+            osState = getProperty(postCompleteConfig) > 0
+                        ? OperatingSystemStateStage::Standby
+                        : OperatingSystemStateStage::Inactive;
+        }
+        catch (const std::exception& e)
+        {
+            lg2::error("standard exception caught: {ERROR}", "ERROR", e);
+            return -1;
+        }
     }
 
     osIface->register_property(
@@ -3997,7 +4169,15 @@ int main(int argc, char* argv[])
 
     restartCauseIface->initialize();
 
-    currentHostStateMonitor();
+    try
+    {
+        currentHostStateMonitor();
+    }
+    catch (const std::exception& e)
+    {
+        lg2::error("standard exception caught: {ERROR}", "ERROR", e);
+        return -1;
+    }
 
     if (!hpmStbyEnConfig.lineName.empty())
     {
@@ -4010,8 +4190,15 @@ int main(int argc, char* argv[])
             return -1;
         }
     }
-
-    io.run();
+    try
+    {
+        io.run();
+    }
+    catch (const std::exception& e)
+    {
+        lg2::error("standard exception caught: {ERROR}", "ERROR", e);
+        return -1;
+    }
 
     return 0;
 }
-- 
2.34.1

