From 47491e3a51ea0c71fa511d495a68c79df287b627 Mon Sep 17 00:00:00 2001
From: pavithra <pavithrat@ami.com>
Date: Tue, 20 May 2025 11:19:58 +0530
Subject: [PATCH 47/47] patch_err_fix

---
 sock_channel.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/sock_channel.cpp b/sock_channel.cpp
index 5480fbd..6220e1e 100644
--- a/sock_channel.cpp
+++ b/sock_channel.cpp
@@ -45,7 +45,7 @@ std::tuple<int, buffer> Channel::read()
                                outputPtr,          // Buffer
                                bufferSize,         // Bytes requested
                                0,                  // Flags
-                               &address.sockAddr,  // Address
+                               (struct sockaddr*)&address.inAddr,  // Address
                                &address.addrSize); // Address Length
 
         if (readDataLen == 0) // Peer has performed an orderly shutdown
@@ -99,7 +99,7 @@ int Channel::write(buffer& inBuffer)
                                outputPtr,         // Message
                                bufferSize,        // Length
                                MSG_NOSIGNAL,      // Flags
-                               &address.sockAddr, // Destination Address
+                               (struct sockaddr*)&address.inAddr, // Destination Address
                                address.addrSize); // Address Length
 
                     if (writeDataLen < 0)
-- 
2.34.1

