From a1a5490bd029e3f82c02c923a110f96e133e22c6 Mon Sep 17 00:00:00 2001
From: Pranoy Goru <gorumanip@ami.com>
Date: Fri, 21 Oct 2022 12:44:31 +0530
Subject: [PATCH] Added srvcfg

---
 src/main.cpp | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/main.cpp b/src/main.cpp
index f104f22..15b878d 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -41,7 +41,7 @@ static std::unordered_map<std::string /* unitName */,
     managedServices = {{"phosphor-ipmi-net", false}, {"bmcweb", false},
                        {"phosphor-ipmi-kcs", false}, {"start-ipkvm", false},
                        {"obmc-console", false},      {"dropbear", true},
-                       {"obmc-console-ssh", true}};
+                       {"obmc-console-ssh", true},   {"xyz.openbmc_project.VirtualMedia", false}};
 
 enum class UnitType
 {
@@ -131,7 +131,9 @@ static inline void
             }
 
             std::string instantiatedUnitName =
-                unitName + addInstanceName(instanceName, "@");
+                unitName + addInstanceName(instanceName, "_40");
+            boost::replace_all(instantiatedUnitName, "-", "_2d");
+            boost::replace_all(instantiatedUnitName, ".", "_2e");
             const sdbusplus::message::object_path& objectPath =
                 std::get<static_cast<int>(ListUnitElements::objectPath)>(unit);
             // Group the service & socket units togther.. Same services
