From a817caad838e4abcb411974a30411fe8931c8d35 Mon Sep 17 00:00:00 2001
From: krishnar4 <krishnar@ami.com>
Date: Mon, 8 Apr 2024 13:03:29 +0530
Subject: [PATCH 2/2] Added-ipmb-service-to-service-Config-Manager

---
 src/main.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/main.cpp b/src/main.cpp
index 67b152e..80f9402 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -42,7 +42,8 @@ static std::unordered_map<std::string /* unitName */,
                        {"phosphor-ipmi-kcs", false}, {"start-ipkvm", false},
                        {"obmc-console", false},      {"dropbear", true},
                        {"obmc-console-ssh", true},   {"xyz.openbmc_project.Pmt", false},
-                       {"xyz.openbmc_project.VirtualMedia", false}};
+                       {"xyz.openbmc_project.VirtualMedia", false},
+                       {"ipmb", false}};
 
 enum class UnitType
 {
-- 
2.34.1

