From 0eab22ff4c52c67bc73200623a2ec238a2d9abd9 Mon Sep 17 00:00:00 2001
From: pandiarajt <pandiarajt@ami.com>
Date: Wed, 17 Dec 2025 01:55:44 +0530
Subject: [PATCH 77/77] Synchronize states at boot and fix persistence

---
 src/srvcfg_manager.cpp | 13 +++----------
 1 file changed, 3 insertions(+), 10 deletions(-)

diff --git a/src/srvcfg_manager.cpp b/src/srvcfg_manager.cpp
index df4b2d0..cc19a3e 100644
--- a/src/srvcfg_manager.cpp
+++ b/src/srvcfg_manager.cpp
@@ -447,16 +447,9 @@ void ServiceConfig::loadStateFile()
                 "Enabled property for {FILEPATH} not equal. Setting to {SETTING}",
                 "FILEPATH", stateFile, "SETTING", stateMap[srvCfgPropEnabled]);
             unitEnabledState = stateMap[srvCfgPropEnabled];
+            unitRunningState = stateMap[srvCfgPropEnabled];
             updatedFlag |=
                 (1 << static_cast<uint8_t>(UpdatedProp::enabledState));
-            startServiceRestartTimer();
-        }
-        if (stateMap[srvCfgPropRunning] != unitRunningState)
-        {
-            lg2::info(
-                "Running property for {FILEPATH} not equal. Setting to {SETTING}",
-                "FILEPATH", stateFile, "SETTING", stateMap[srvCfgPropRunning]);
-            unitRunningState = stateMap[srvCfgPropRunning];
             updatedFlag |=
                 (1 << static_cast<uint8_t>(UpdatedProp::runningState));
             startServiceRestartTimer();
@@ -643,6 +636,8 @@ void ServiceConfig::restartUnitConfig(boost::asio::yield_context yield)
 
 void ServiceConfig::startServiceRestartTimer()
 {
+    // Ensure our persistent files are updated with changes
+    writeStateFile();
     timer->expires_after(std::chrono::seconds(restartTimeout));
     timer->async_wait([this](const boost::system::error_code& ec) {
         if (ec == boost::asio::error::operation_aborted)
@@ -656,8 +651,6 @@ void ServiceConfig::startServiceRestartTimer()
             return;
         }
         updateInProgress = true;
-        // Ensure our persistent files are updated with changes
-        writeStateFile();
         boost::asio::spawn(
             conn->get_io_context(),
             [this](boost::asio::yield_context yield) {
-- 
2.34.1

