From 4f8e593cac58e0f3c9dc360c0e3080d8e34ee469 Mon Sep 17 00:00:00 2001
From: chandrus <chandrus@ami.com>
Date: Thu, 20 Mar 2025 18:08:33 +0530
Subject: [PATCH 63/63] Updated the Json file in Masked Property

---
 src/srvcfg_manager.cpp | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/src/srvcfg_manager.cpp b/src/srvcfg_manager.cpp
index e0f4264..6b4b73b 100644
--- a/src/srvcfg_manager.cpp
+++ b/src/srvcfg_manager.cpp
@@ -777,6 +777,17 @@ void ServiceConfig::registerProperties()
                 srvCfgIface->set_property(srvCfgPropEnabled, unitEnabledState);
                 srvCfgIface->set_property(srvCfgPropRunning, unitRunningState);
                 internalSet = false;
+                if(checkServicetoAddTimeOutandMaxSessProp(instantiatedUnitName))
+                {
+                    for (auto& service : global_data["services"])
+                    {
+                        if (service["name"] == instantiatedUnitName) {
+                            service["Enabled"] = unitEnabledState;
+                            break;
+                        }
+                    }
+                    updateFileFromGlobalData();
+                }
                 startServiceRestartTimer();
             }
             res = req;
-- 
2.34.1

