From ac72568a83c0b5c0a2f186426a01bae696a099f6 Mon Sep 17 00:00:00 2001
From: Abinaya L <abinayal@ami.com>
Date: Wed, 7 Jun 2023 01:09:27 +0530
Subject: [PATCH] moving cpld inventory for non pfr to software manager

---
 include/defaults.hpp | 20 --------------------
 1 file changed, 20 deletions(-)

diff --git a/include/defaults.hpp b/include/defaults.hpp
index 7616c4d..eab110f 100644
--- a/include/defaults.hpp
+++ b/include/defaults.hpp
@@ -187,30 +187,10 @@ inline void loadSettings(sdbusplus::asio::object_server &objectServer,
         "Activation",
         "xyz.openbmc_project.Software.Activation.Activations.Active");
 
-    setting = &settings.emplace_back(
-        objectServer, "/xyz/openbmc_project/software/cpld_active",
-        "xyz.openbmc_project.Software.Version");
-    setting->addProperty("Version", "NA");
-    setting->addProperty(
-        "Purpose", "xyz.openbmc_project.Software.Version.VersionPurpose.Other");
-
-    setting = &settings.emplace_back(
-        objectServer, "/xyz/openbmc_project/software/cpld_active",
-        "xyz.openbmc_project.Software.Activation");
-    setting->addProperty(
-        "Activation",
-        "xyz.openbmc_project.Software.Activation.Activations.Active");
-    setting->addProperty(
-        "RequestedActivation",
-        "xyz.openbmc_project.Software.Activation.RequestedActivations.None");
-
     std::vector<Association> associations;
     associations.push_back(
         Association("functional", "software_version",
                     "/xyz/openbmc_project/software/bios_active"));
-    associations.push_back(
-        Association("functional", "software_version",
-                    "/xyz/openbmc_project/software/cpld_active"));
 
     setting =
         &settings.emplace_back(objectServer, "/xyz/openbmc_project/software",
