From 8788c04d56f1719dcca9877fce8e45b961ef4f0f Mon Sep 17 00:00:00 2001
From: pandiarajt <pandiarajt@ami.com>
Date: Tue, 8 Aug 2023 15:51:11 +0530
Subject: [PATCH 45/45] Enable the SOL by default

---
 include/defaults.hpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/include/defaults.hpp b/include/defaults.hpp
index eab110f..fe87122 100644
--- a/include/defaults.hpp
+++ b/include/defaults.hpp
@@ -274,7 +274,7 @@ inline void loadSettings(sdbusplus::asio::object_server &objectServer,
                                      "xyz.openbmc_project.Ipmi.SOL");
 
     setting->addProperty("Progress", static_cast<uint8_t>(0), false);
-    setting->addProperty("Enable", false);
+    setting->addProperty("Enable", true);
     setting->addProperty("ForceEncryption", true);
     setting->addProperty("ForceAuthentication", true);
     setting->addProperty("Privilege", static_cast<uint8_t>(4));
@@ -288,7 +288,7 @@ inline void loadSettings(sdbusplus::asio::object_server &objectServer,
                                      "xyz.openbmc_project.Ipmi.SOL");
 
     setting->addProperty("Progress", static_cast<uint8_t>(0), false);
-    setting->addProperty("Enable", false);
+    setting->addProperty("Enable", true);
     setting->addProperty("ForceEncryption", true);
     setting->addProperty("ForceAuthentication", true);
     setting->addProperty("Privilege", static_cast<uint8_t>(4));
@@ -302,7 +302,7 @@ inline void loadSettings(sdbusplus::asio::object_server &objectServer,
                                      "xyz.openbmc_project.Ipmi.SOL");
 
     setting->addProperty("Progress", static_cast<uint8_t>(0), false);
-    setting->addProperty("Enable", false);
+    setting->addProperty("Enable", true);
     setting->addProperty("ForceEncryption", true);
     setting->addProperty("ForceAuthentication", true);
     setting->addProperty("Privilege", static_cast<uint8_t>(4));
-- 
2.25.1

