From c00decafefd1a8b9c9042ff1d4fb62210776db8a Mon Sep 17 00:00:00 2001
From: kirubagharanas <kirubagharanas@ami.com>
Date: Fri, 5 Apr 2024 18:17:44 +0530
Subject: [PATCH] Add support to Apply time property

---
 include/defaults.hpp | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/include/defaults.hpp b/include/defaults.hpp
index b9bfef3..448e88f 100644
--- a/include/defaults.hpp
+++ b/include/defaults.hpp
@@ -370,6 +370,12 @@ inline void loadSettings(sdbusplus::asio::object_server &objectServer,
     setting->addProperty(
         "RequestedApplyTime",
         "xyz.openbmc_project.Software.ApplyTime.RequestedApplyTimes.Immediate");
+    setting->addProperty(
+        "MaintenanceWindowDurationInSeconds",
+        static_cast<uint64_t>(0));
+    setting->addProperty(
+        "MaintenanceWindowStartTime",
+        static_cast<uint64_t>(0));
 
     setting = &settings.emplace_back(objectServer,
                                      "/xyz/openbmc_project/logging/settings",
