From 0a204af15189431ca97ac51e73075cb50d18fa9b Mon Sep 17 00:00:00 2001
From: dineshsv <dineshsv@ami.com>
Date: Sun, 23 Jun 2024 20:45:44 +0530
Subject: [PATCH] Added Dbus object path, interface and property for BMC Global
 enables.

Signed-off-by: dineshsv <dineshsv@ami.com>
---
 include/defaults.hpp | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/include/defaults.hpp b/include/defaults.hpp
index 448e88f..100d15e 100644
--- a/include/defaults.hpp
+++ b/include/defaults.hpp
@@ -91,6 +91,18 @@ inline void loadSettings(sdbusplus::asio::object_server &objectServer,
 
     setting->addProperty("PowerRestoreDelay", static_cast<uint64_t>(0));
 
+    setting = &settings.emplace_back(
+        objectServer, "/xyz/openbmc_project/control/globalenables",
+        "xyz.openbmc_project.Control.BMC.Globalenables");
+
+    setting->addProperty("RecvMsgQueueInterrupt", true);
+    setting->addProperty("EventmsgFullintr", false);
+    setting->addProperty("EventmsgBuf", false);
+    setting->addProperty("Sel", true);
+    setting->addProperty("OEM0", false);
+    setting->addProperty("OEM1", false);
+    setting->addProperty("OEM2", false);
+
     setting = &settings.emplace_back(
         objectServer, "/xyz/openbmc_project/control/systemlock",
         "xyz.openbmc_project.Control.Security.SystemLock");
-- 
2.25.1

