From 3128961cd8c2508c218fbf67c312e6c085e95c16 Mon Sep 17 00:00:00 2001
From: vipinc <vipinc@ami.com>
Date: Sun, 1 Sep 2024 15:24:49 +0530
Subject: [PATCH] Add SEL Limit Flags

Signed-off-by: vipinc <vipinc@ami.com>
---
 include/defaults.hpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/defaults.hpp b/include/defaults.hpp
index 8d12356..f4e28ed 100644
--- a/include/defaults.hpp
+++ b/include/defaults.hpp
@@ -553,6 +553,8 @@ inline void loadSettings(sdbusplus::asio::object_server &objectServer,
 
     setting->addProperty("QuiesceOnHwError", false);
     setting->addProperty("SelPolicy", "xyz.openbmc_project.Logging.Settings.Policy.Circular");
+    setting->addProperty("InfoFlag", false);
+    setting->addProperty("ErrorFlag", false);
 
     setting = &settings.emplace_back(objectServer,
                                      "/xyz/openbmc_project/pfr/last_events",
-- 
2.25.1

