From d2573c97ff762deabb4ed48aab83f768d8f5eb21 Mon Sep 17 00:00:00 2001
From: sujithrar <sujithrar@ami.com>
Date: Tue, 8 Apr 2025 10:56:05 +0530
Subject: [PATCH 59/59] sol-session-bond-ifc

---
 include/defaults.hpp | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/include/defaults.hpp b/include/defaults.hpp
index 8828da8..558be53 100644
--- a/include/defaults.hpp
+++ b/include/defaults.hpp
@@ -323,6 +323,20 @@ inline void loadSettings(sdbusplus::asio::object_server &objectServer,
 
     setting->addProperty("Limit", static_cast<double>(0));
 
+     setting = &settings.emplace_back(objectServer,
+                                  "/xyz/openbmc_project/ipmi/sol/bond0",
+                                  "xyz.openbmc_project.Ipmi.SOL");
+
+setting->addProperty("Progress", static_cast<uint8_t>(0), false);
+setting->addProperty("Enable", true);
+setting->addProperty("ForceEncryption", true);
+setting->addProperty("ForceAuthentication", true);
+setting->addProperty("Privilege", static_cast<uint8_t>(4));
+setting->addProperty("AccumulateIntervalMS", static_cast<uint8_t>(100));
+setting->addProperty("Threshold", static_cast<uint8_t>(96));
+setting->addProperty("RetryCount", static_cast<uint8_t>(6));
+setting->addProperty("RetryIntervalMS", static_cast<uint8_t>(20));
+
     setting = &settings.emplace_back(objectServer,
                                      "/xyz/openbmc_project/ipmi/sol/eth0",
                                      "xyz.openbmc_project.Ipmi.SOL");
-- 
2.34.1

