From 68b63cbc3b3588b69b06480856fa26e26f328113 Mon Sep 17 00:00:00 2001
From: maniv-ami <manikandanv@ami.com>
Date: Fri, 23 May 2025 21:16:11 +0530
Subject: [PATCH 60/60] Added additional boot override related parameters

---
 include/defaults.hpp | 65 ++++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 65 insertions(+)

diff --git a/include/defaults.hpp b/include/defaults.hpp
index 558be53..21d23f1 100644
--- a/include/defaults.hpp
+++ b/include/defaults.hpp
@@ -68,12 +68,77 @@ inline void loadSettings(sdbusplus::asio::object_server &objectServer,
 
     setting->addProperty("Enabled", false);
 
+    setting = &settings.emplace_back(objectServer,
+				     "/xyz/openbmc_project/control/host0/boot",
+				     "xyz.openbmc_project.Control.Boot.ServicePartitionSelect");
+
+    setting->addProperty("ServicePartition", static_cast<uint8_t>(0));
+
+    setting = &settings.emplace_back(objectServer, 
+				    "/xyz/openbmc_project/control/host0/boot",
+				    "xyz.openbmc_project.Control.Boot.ServicePartitionScan");
+
+    setting->addProperty("PartitionDiscovered", false);
+    setting->addProperty("RequestScan", false);
+
+    setting = &settings.emplace_back(objectServer,
+				    "/xyz/openbmc_project/control/host0/boot",
+				    "xyz.openbmc_project.Control.Boot.BootMailbox");
+
+    setting->addProperty("BlockSelector", static_cast<uint8_t>(0));
+    setting->addProperty("Mailbox0L", static_cast<uint64_t>(0));
+    setting->addProperty("Mailbox0U", static_cast<uint64_t>(0));
+    setting->addProperty("Mailbox1L", static_cast<uint64_t>(0));
+    setting->addProperty("Mailbox1U", static_cast<uint64_t>(0));
+    setting->addProperty("Mailbox2L", static_cast<uint64_t>(0));
+    setting->addProperty("Mailbox2U", static_cast<uint64_t>(0));
+    setting->addProperty("Mailbox3L", static_cast<uint64_t>(0));
+    setting->addProperty("Mailbox3U", static_cast<uint64_t>(0));
+    setting->addProperty("Mailbox4L", static_cast<uint64_t>(0));
+    setting->addProperty("Mailbox4U", static_cast<uint64_t>(0));
+
+    setting = &settings.emplace_back(objectServer,
+				    "/xyz/openbmc_project/control/host0/boot",
+				    "xyz.openbmc_project.Control.Boot.BootInitiatorInfo");
+
+    setting->addProperty("Channel", static_cast<uint8_t>(0));
+    setting->addProperty("SessionID", static_cast<uint32_t>(0));
+    setting->addProperty("Timestamp", static_cast<uint32_t>(0));
+
+    setting = &settings.emplace_back(objectServer,
+				    "/xyz/openbmc_project/control/host0/boot",
+				    "xyz.openbmc_project.Control.Boot.DeviceInstance");
+
+    setting->addProperty("DeviceInstance", static_cast<uint32_t>(0));
+
     setting = &settings.emplace_back(
         objectServer, "/xyz/openbmc_project/control/host0/boot/one_time",
         "xyz.openbmc_project.Object.Enable");
 
     setting->addProperty("Enabled", false);
 
+    setting = &settings.emplace_back(objectServer,
+				    "/xyz/openbmc_project/control/host0/boot/one_time",
+				    "xyz.openbmc_project.Control.Boot.Source");
+
+    setting->addProperty(
+	"BootSource",
+	"xyz.openbmc_project.Control.Boot.Source.Sources.Default");
+
+    setting = &settings.emplace_back(objectServer,
+				    "/xyz/openbmc_project/control/host0/boot/one_time",
+				    "xyz.openbmc_project.Control.Boot.Mode");
+
+    setting->addProperty(
+	"BootMode",
+	"xyz.openbmc_project.Control.Boot.Mode.Modes.Regular");
+
+    setting = &settings.emplace_back(objectServer,
+				    "/xyz/openbmc_project/control/host0/boot/one_time",
+				    "xyz.openbmc_project.Control.Boot.DeviceInstance");
+
+    setting->addProperty("DeviceInstance", static_cast<uint32_t>(0));
+
     setting = &settings.emplace_back(
         objectServer, "/xyz/openbmc_project/control/host0/power_cap",
         "xyz.openbmc_project.Control.Power.Cap");
-- 
2.41.0

