From b163f96fd8bc81568373a6debf90437fd57c0808 Mon Sep 17 00:00:00 2001
From: mickhuang <mickhuang@ami.com>
Date: Wed, 22 Oct 2025 16:01:34 +0800
Subject: [PATCH] Update new usb vhub dev node name

Signed-off-by: mickhuang <mickhuang@ami.com>
---
 src/main.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/main.cpp b/src/main.cpp
index 8a15860..d1b2814 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -21,7 +21,7 @@
 
 #define DBUS_PROPERTIES_INTERFACE "org.freedesktop.DBus.Properties"
 
-static constexpr auto powerSaveMode = "/sys/class/usb/usb_vhub/power_save_mode";
+static constexpr auto powerSaveMode = "/sys/class/usb/vhub0/power_save_mode";
 static constexpr auto ccNodeBusy = 0xC0;
 
 static const std::string sessMgrService = "xyz.openbmc_project.SessionManager";
-- 
2.34.1

