From d4406f3552441fee9bf8512fa0929eb127a2592d Mon Sep 17 00:00:00 2001
From: RobinChiu <robinchiu@ami.com>
Date: Wed, 5 Nov 2025 09:23:50 +0800
Subject: [PATCH] Add dual node support

---
 include/defaults.hpp | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/include/defaults.hpp b/include/defaults.hpp
index da7fe87..f1ee4e9 100644
--- a/include/defaults.hpp
+++ b/include/defaults.hpp
@@ -501,6 +501,13 @@ setting->addProperty("RetryIntervalMS", static_cast<uint8_t>(20));
     setting->addProperty("TemparatureLimit", static_cast<uint8_t>(0));
     setting->addProperty("ExceptionTime", static_cast<uint16_t>(0));
 
+    setting = &settings.emplace_back(objectServer,
+                                     "/xyz/openbmc_project/control/HostMode",
+                                     "xyz.openbmc_project.Control.HostMode");
+
+    setting->addProperty("CurrentMode", static_cast<uint16_t>(0));
+    setting->addProperty("Mode", static_cast<uint16_t>(0));
+
     for (SettingsInterface &s : settings)
     {
         s.initialize();
-- 
2.34.1

