From 0ee8e70070a364f9edb2f091c7ecfda274684579 Mon Sep 17 00:00:00 2001
From: hemanthkumarm <hemanthkumarm@ami.com>
Date: Fri, 30 Jan 2026 07:04:55 +0000
Subject: [PATCH] Remove bios_active and rot_fw_active software settings

---
 include/defaults.hpp | 24 ------------------------
 1 file changed, 24 deletions(-)

diff --git a/include/defaults.hpp b/include/defaults.hpp
index 16e6bff..cf56302 100644
--- a/include/defaults.hpp
+++ b/include/defaults.hpp
@@ -271,31 +271,7 @@ inline void loadSettings(sdbusplus::asio::object_server &objectServer,
     setting->addProperty("Mode", "Normal");
     setting->addProperty("IsolationInitialState", "Inactive");
 
-    setting = &settings.emplace_back(
-        objectServer, "/xyz/openbmc_project/software/bios_active",
-        "xyz.openbmc_project.Software.Version");
-    setting->addProperty("Version", "NA");
-    setting->addProperty(
-        "Purpose", "xyz.openbmc_project.Software.Version.VersionPurpose.Host");
-
-    setting = &settings.emplace_back(
-        objectServer, "/xyz/openbmc_project/software/bios_active",
-        "xyz.openbmc_project.Software.Activation");
-    setting->addProperty(
-        "Activation",
-        "xyz.openbmc_project.Software.Activation.Activations.Active");
-
     setting->addProperty(
         "RequestedActivation",
         "xyz.openbmc_project.Software.Activation.RequestedActivations.None");
-
-    std::vector<Association> associations;
-    associations.push_back(
-        Association("functional", "software_version",
-                    "/xyz/openbmc_project/software/bios_active"));
-
-    setting =
-        &settings.emplace_back(objectServer, "/xyz/openbmc_project/software",
-                               "xyz.openbmc_project.Association.Definitions");
-    setting->interface->register_property("Associations", associations);
 
     setting = &settings.emplace_back(
         objectServer, "/xyz/openbmc_project/control/processor_error_config",
