From db16e55beaa8af390b6be40e9c9d0ba390862d9f Mon Sep 17 00:00:00 2001
From: Dhineshkumar M <mdhineshkumar@ami.com>
Date: Tue, 24 Mar 2026 09:18:21 +0530
Subject: [PATCH] Block D-Bus SetUSBPowerSaveMode during BIOS-BMC communication 

Signed-off-by: Dhineshkumar M <mdhineshkumar@ami.com>
---
 src/main.cpp | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/main.cpp b/src/main.cpp
index 835e440..7047f2e 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -22,6 +22,7 @@
 #define DBUS_PROPERTIES_INTERFACE "org.freedesktop.DBus.Properties"
 
 static constexpr auto powerSaveMode = "/sys/class/usb/vhub0/power_save_mode";
+static constexpr auto biosCommFlag = "/var/tmp/bioscomm";
 static constexpr auto ccNodeBusy = 0xC0;
 
 static const std::string sessMgrService = "xyz.openbmc_project.SessionManager";
@@ -86,6 +87,12 @@ int setUSBPowerSaveMode(int cmd)
     // convert int command to char
     char value = cmd + '0';
 
+    // Block power save mode changes while BIOS BMC communication is in progress
+    if (std::filesystem::exists(biosCommFlag))
+    {
+        return ccNodeBusy;
+    }
+
     if (0 < isKVMInUSe() || 0 < isVirtualMediaInUse())
     {
         return ccNodeBusy;
-- 
2.25.1

