diff -Naur a/src/pldmd.cpp b/src/pldmd.cpp
--- a/src/pldmd.cpp	2023-09-11 21:12:22.136082037 +0530
+++ b/src/pldmd.cpp	2023-09-11 21:13:32.116688473 +0530
@@ -699,7 +699,7 @@
     // TODO - Read from entity manager about the transport bindings to be
     // supported by PLDM
     mctpw::MCTPConfiguration config(mctpw::MessageType::pldm,
-                                    mctpw::BindingType::mctpOverSmBus);
+				    mctpw::BindingType::mctpOverPcieVdm);
 
     pldm::mctpWrapper = std::make_unique<mctpw::MCTPWrapper>(
         conn, config, onDeviceUpdate, pldm::msgRecvCallback);
