diff -Naur a/src/firmware_update.cpp b/src/firmware_update.cpp
--- a/src/firmware_update.cpp	2023-09-11 21:17:03.097499493 +0530
+++ b/src/firmware_update.cpp	2023-09-11 21:21:53.368264309 +0530
@@ -40,6 +40,9 @@
 // Timeout in milliseconds in between fwu command
 constexpr uint16_t fdCmdTimeout = 5000;
 
+// Timeout in milliseconds in verify State
+constexpr uint32_t verify_timeout = 600000;
+
 // Maximum retry count
 constexpr size_t retryCount = 3;
 
@@ -2162,7 +2165,7 @@
 
         expectedCmd = PLDM_VERIFY_COMPLETE;
 
-        startTimer(yield, fdCmdTimeout);
+        startTimer(yield, verify_timeout);
 
         if (!fdReqMatched)
         {
