diff -Naur a/src/firmware_update.cpp b/src/firmware_update.cpp
--- a/src/firmware_update.cpp	2023-09-12 18:28:18.507243110 +0530
+++ b/src/firmware_update.cpp	2023-09-12 18:32:05.829115664 +0530
@@ -25,6 +25,8 @@
 #include <xyz/openbmc_project/PLDM/FWU/FWUBase/server.hpp>
 
 #include "utils.h"
+
+#define PLDM_MAX_TRANSFER_SIZE 1024
 namespace pldm
 {
 namespace fwu
@@ -198,7 +200,7 @@
 bool FWUpdate::prepareRequestUpdateCommand()
 {
     uint16_t tempShort = 0;
-    updateProperties.max_transfer_size = PLDM_FWU_BASELINE_TRANSFER_SIZE;
+    updateProperties.max_transfer_size = PLDM_MAX_TRANSFER_SIZE;
     applicableComponentsVal = getApplicableComponents();
     updateProperties.no_of_comp =
         getApplicableComponentsCount(applicableComponentsVal);
