diff -Naur a/src/firmware_update.cpp b/src/firmware_update.cpp
--- a/src/firmware_update.cpp	2023-09-12 18:32:59.697554772 +0530
+++ b/src/firmware_update.cpp	2023-09-12 18:33:26.013768763 +0530
@@ -337,7 +337,7 @@
         return false;
     }
     component.comp_image_size = tempLong;
-    component.update_option_flags = {};
+    component.update_option_flags = {1};
     if (!pldmImg->getCompProperty<uint8_t>(component.comp_ver_str_type,
                                            "CmpVerStrType", currentComp))
     {
