diff -Naur a/src/firmware_update.cpp b/src/firmware_update.cpp
--- a/src/firmware_update.cpp	2023-09-12 18:33:57.810026890 +0530
+++ b/src/firmware_update.cpp	2023-09-12 18:34:21.306217350 +0530
@@ -2276,7 +2276,7 @@
         return PLDM_ERROR;
     }
 
-    bool8_t selfContainedActivationReq = true;
+    bool8_t selfContainedActivationReq = false;
     uint16_t estimatedTimeForSelfContainedActivation = 0;
     retVal = processActivateFirmware(yield, selfContainedActivationReq,
                                      estimatedTimeForSelfContainedActivation);
