diff -Naur a/src/fru.cpp b/src/fru.cpp
--- a/src/fru.cpp	2023-09-14 12:07:44.991226575 +0530
+++ b/src/fru.cpp	2023-09-14 12:08:10.307660552 +0530
@@ -330,12 +330,14 @@
                   std::back_inserter(fruRecordTableData));
     }
 
+#if 0
     if (!verifyCRC(fruRecordTableData))
     {
         phosphor::logging::log<phosphor::logging::level::ERR>(
             "Failed at CRC Match", phosphor::logging::entry("TID=%d", tid));
         return PLDM_ERROR;
     }
+#endif
 
     auto it = fruData.find(tid);
     if (it != fruData.end())
