diff -Naur a/include/fru.hpp b/include/fru.hpp
--- a/include/fru.hpp	2024-09-13 12:59:32.150507819 +0800
+++ b/include/fru.hpp	2024-09-13 13:02:13.112489277 +0800
@@ -58,7 +58,7 @@
 {
   public:
     GetPLDMFRU() = delete;
-    GetPLDMFRU(boost::asio::yield_context yieldVal, const pldm_tid_t tidVal);
+    GetPLDMFRU(boost::asio::yield_context yieldVal, const pldm_tid_t tidVal, const mctpw_eid_t eidVal);
     ~GetPLDMFRU();
 
     /** @brief runs supported FRU commands
@@ -100,6 +100,7 @@
 
     boost::asio::yield_context yield;
     pldm_tid_t tid;
+	mctpw_eid_t eid;
     FRUMetadata fruMetadata;
 };
 
@@ -107,13 +108,14 @@
 {
   public:
     SetPLDMFRU() = delete;
-    explicit SetPLDMFRU(const pldm_tid_t tidVal);
+    explicit SetPLDMFRU(const pldm_tid_t tidVal, const mctpw_eid_t eidVal);
 
     int setFruRecordTableCmd(boost::asio::yield_context yield,
                              const std::vector<uint8_t>& setFruData);
 
   private:
     pldm_tid_t tid;
+	mctpw_eid_t eid;
 
     uint8_t getTransferFlag(const size_t offset, const size_t length,
                             const size_t dataSize);
diff -Naur a/include/fwu_inventory.hpp b/include/fwu_inventory.hpp
--- a/include/fwu_inventory.hpp	2024-09-13 12:59:32.150507819 +0800
+++ b/include/fwu_inventory.hpp	2024-09-13 13:02:13.112489277 +0800
@@ -28,7 +28,7 @@
 {
   public:
     FWInventoryInfo() = delete;
-    explicit FWInventoryInfo(const pldm_tid_t _tid);
+    explicit FWInventoryInfo(const pldm_tid_t _tid, const mctpw_eid_t eid);
 
     /** @brief runs inventory commands
      */
@@ -113,6 +113,7 @@
     std::string getInventoryName();
 
     pldm_tid_t tid;
+	mctpw_eid_t eid;
     std::shared_ptr<sdbusplus::asio::object_server> objServer;
     // map that holds the component properties of a terminus
     CompPropertiesMap compPropertiesMap;
diff -Naur a/include/numeric_effecter_handler.hpp b/include/numeric_effecter_handler.hpp
--- a/include/numeric_effecter_handler.hpp	2024-09-13 12:59:32.150507819 +0800
+++ b/include/numeric_effecter_handler.hpp	2024-09-13 13:02:13.112489277 +0800
@@ -39,7 +39,7 @@
     ~NumericEffecterHandler();
 
     NumericEffecterHandler(
-        const pldm_tid_t tid, const EffecterID effecterID,
+        const pldm_tid_t tid, const mctpw_eid_t eid, const EffecterID effecterID,
         const std::string& name,
         const std::shared_ptr<pldm_numeric_effecter_value_pdr>& pdr);
 
@@ -73,6 +73,7 @@
 
     /** @brief Terminus ID*/
     pldm_tid_t _tid;
+	mctpw_eid_t _eid;
 
     /** @brief Effecter ID*/
     EffecterID _effecterID;
diff -Naur a/include/numeric_sensor_handler.hpp b/include/numeric_sensor_handler.hpp
--- a/include/numeric_sensor_handler.hpp	2024-09-13 12:59:32.150507819 +0800
+++ b/include/numeric_sensor_handler.hpp	2024-09-13 13:02:13.112489277 +0800
@@ -39,7 +39,7 @@
     ~NumericSensorHandler() = default;
 
     NumericSensorHandler(
-        const pldm_tid_t tid, const SensorID sensorID, const std::string& name,
+        const pldm_tid_t tid, const mctpw_eid_t eid, const SensorID sensorID, const std::string& name,
         const std::shared_ptr<pldm_numeric_sensor_value_pdr>& pdr);
 
     /** @brief Init NumericSensorHandler*/
@@ -78,6 +78,8 @@
 
     /** @brief Terminus ID*/
     pldm_tid_t _tid;
+    
+    mctpw_eid_t _eid;
 
     /** @brief Sensor ID*/
     SensorID _sensorID;
diff -Naur a/include/pdr_manager.hpp b/include/pdr_manager.hpp
--- a/include/pdr_manager.hpp	2024-09-13 12:59:32.150507819 +0800
+++ b/include/pdr_manager.hpp	2024-09-13 13:02:13.112489277 +0800
@@ -98,7 +98,7 @@
     PDRManager& operator=(PDRManager&&) = delete;
     ~PDRManager();
 
-    PDRManager(const pldm_tid_t tid);
+    PDRManager(const mctpw_eid_t eid, const pldm_tid_t tid);
 
     bool pdrManagerInit(boost::asio::yield_context yield);
 
@@ -310,6 +310,7 @@
 
     /** @brief Terminus ID*/
     pldm_tid_t _tid;
+    mctpw_eid_t _eid;
 
     /** @brief Temporarily holds entity association nodes, used to create entity
      * association tree
diff -Naur a/include/platform.hpp b/include/platform.hpp
--- a/include/platform.hpp	2024-09-13 12:59:32.150507819 +0800
+++ b/include/platform.hpp	2024-09-13 13:02:13.112489277 +0800
@@ -34,14 +34,14 @@
 
 std::optional<UUID>
     getTerminusUID(boost::asio::yield_context yield, const pldm_tid_t tid,
-                   std::optional<mctpw_eid_t> eid = std::nullopt);
+				   const mctpw_eid_t eid);
 
 class Platform
 {
   public:
     void stopSensorPolling();
     void startSensorPolling();
-    bool initTerminus(boost::asio::yield_context yield, const pldm_tid_t tid,
+    bool initTerminus(boost::asio::yield_context yield, const mctpw_eid_t eid, const pldm_tid_t tid,
                       const pldm::base::CommandSupportTable& commandTable);
     bool deleteTerminus(const pldm_tid_t tid);
 
diff -Naur a/include/platform_terminus.hpp b/include/platform_terminus.hpp
--- a/include/platform_terminus.hpp	2024-09-13 12:59:32.150507819 +0800
+++ b/include/platform_terminus.hpp	2024-09-13 13:02:13.112489277 +0800
@@ -28,7 +28,7 @@
 struct PlatformTerminus
 {
   public:
-    PlatformTerminus(boost::asio::yield_context yield, const pldm_tid_t tid);
+    PlatformTerminus(boost::asio::yield_context yield, const mctpw_eid_t eid, const pldm_tid_t tid);
 
     std::unique_ptr<PDRManager> pdrManager;
     std::unordered_map<SensorID, std::unique_ptr<NumericSensorHandler>>
@@ -46,6 +46,7 @@
     bool initPDRs(boost::asio::yield_context yield);
 
     pldm_tid_t _tid;
+    mctpw_eid_t _eid;
 };
 } // namespace platform
-} // namespace pldm
\ No newline at end of file
+} // namespace pldm
diff -Naur a/include/pldm.hpp b/include/pldm.hpp
--- a/include/pldm.hpp	2024-09-13 12:59:32.150507819 +0800
+++ b/include/pldm.hpp	2024-09-13 13:02:13.112489277 +0800
@@ -42,6 +42,8 @@
 /** @brief flag to enable debugging*/
 extern bool debug;
 
+extern int fd;
+
 namespace pldm
 {
 using DBusInterfacePtr = std::shared_ptr<sdbusplus::asio::dbus_interface>;
@@ -246,7 +248,7 @@
  *
  * @return Status of the operation
  */
-bool platformInit(boost::asio::yield_context yield, const pldm_tid_t tid,
+bool platformInit(boost::asio::yield_context yield, const mctpw_eid_t eid, const pldm_tid_t tid,
                   const pldm::base::CommandSupportTable& commandTable);
 
 /** @brief Delete Platform Monitoring and Control
@@ -265,7 +267,7 @@
 namespace fru
 {
 
-bool fruInit(boost::asio::yield_context yield, const pldm_tid_t tid);
+bool fruInit(boost::asio::yield_context yield, const pldm_tid_t tid, const mctpw_eid_t eid);
 bool deleteFRUDevice(const pldm_tid_t tid);
 
 } // namespace fru
@@ -273,7 +275,7 @@
 namespace fwu
 {
 
-bool fwuInit(boost::asio::yield_context yield, const pldm_tid_t tid);
+bool fwuInit(boost::asio::yield_context yield, const pldm_tid_t tid, const mctpw_eid_t eid);
 bool deleteFWDevice(const pldm_tid_t tid);
 void pldmMsgRecvFwUpdCallback(const pldm_tid_t tid, const uint8_t msgTag,
                               const bool tagOwner,
diff -Naur a/include/state_effecter_handler.hpp b/include/state_effecter_handler.hpp
--- a/include/state_effecter_handler.hpp	2024-09-13 12:59:32.150507819 +0800
+++ b/include/state_effecter_handler.hpp	2024-09-13 13:02:13.112489277 +0800
@@ -36,7 +36,7 @@
     StateEffecterHandler& operator=(StateEffecterHandler&&) = delete;
     ~StateEffecterHandler() = default;
 
-    StateEffecterHandler(const pldm_tid_t tid, const EffecterID effecterID,
+    StateEffecterHandler(const pldm_tid_t tid, const mctpw_eid_t eid, const EffecterID effecterID,
                          const std::string& name,
                          const std::shared_ptr<StateEffecterPDR>& pdr);
 
@@ -87,6 +87,8 @@
 
     /** @brief Terminus ID*/
     pldm_tid_t _tid;
+	
+	mctpw_eid_t _eid;
 
     /** @brief Effecter ID*/
     EffecterID _effecterID;
diff -Naur a/include/state_sensor_handler.hpp b/include/state_sensor_handler.hpp
--- a/include/state_sensor_handler.hpp	2024-09-13 12:59:32.150507819 +0800
+++ b/include/state_sensor_handler.hpp	2024-09-13 13:02:13.112489277 +0800
@@ -36,7 +36,7 @@
     StateSensorHandler& operator=(StateSensorHandler&&) = delete;
     ~StateSensorHandler() = default;
 
-    StateSensorHandler(const pldm_tid_t tid, const SensorID sensorID,
+    StateSensorHandler(const pldm_tid_t tid, const mctpw_eid_t eid, const SensorID sensorID,
                        const std::string& name,
                        const std::shared_ptr<StateSensorPDR>& pdr);
 
@@ -90,6 +90,7 @@
 
     /** @brief Terminus ID*/
     pldm_tid_t _tid;
+    mctpw_eid_t _eid;
 
     /** @brief Sensor ID*/
     SensorID _sensorID;
diff -Naur a/src/base.cpp b/src/base.cpp
--- a/src/base.cpp	2024-09-13 12:59:32.154507868 +0800
+++ b/src/base.cpp	2024-09-13 13:02:13.112489277 +0800
@@ -200,6 +200,8 @@
                                                       0x00);
     auto msg = reinterpret_cast<pldm_msg*>(getSupportedPLDMTypesRequest.data());
     std::vector<uint8_t> getSupportedPLDMTypesResponse;
+	uint8_t* responseMsg = nullptr;
+	size_t responseMsgSize{};
 
     int rc = encode_get_types_req(instanceID, msg);
     if (!validateBaseReqEncode(eid, rc, "GetTypes"))
@@ -207,22 +209,37 @@
         return false;
     }
 
+
+	// Send PLDM request msg and wait for response
+	rc = pldm_send_recv(eid, fd, getSupportedPLDMTypesRequest.data(), getSupportedPLDMTypesRequest.size(),
+					&responseMsg, &responseMsgSize);
+	if (0 > rc)
+	{
+		std::cout << "Failed to send message/receive response. RC = " << rc
+						<< ", errno = " << errno << "\n";
+		return -1;
+	}
+	pldm_msg* response = reinterpret_cast<pldm_msg*>(responseMsg);
+
+
     // TID passed as 0 will be ignored since EID is present.
     if (!sendReceivePldmMessage(yield, defaultTID, timeOut, retryCount,
                                 getSupportedPLDMTypesRequest,
                                 getSupportedPLDMTypesResponse, eid))
     {
-        phosphor::logging::log<phosphor::logging::level::ERR>(
+#if 0
+  		    phosphor::logging::log<phosphor::logging::level::ERR>(
             "Send or receive error while getting supported PLDM Types",
             phosphor::logging::entry("EID=%d", eid));
-        return false;
+	   return false;
+#endif
     }
-
     uint8_t completionCode;
     rc = decode_get_types_resp(
-        reinterpret_cast<pldm_msg*>(getSupportedPLDMTypesResponse.data()),
-        getSupportedPLDMTypesResponse.size() - hdrSize, &completionCode,
+        response,
+        responseMsgSize - hdrSize, &completionCode,
         supportedTypes.data());
+
     if (!validateBaseRespDecode(eid, rc, completionCode, "GetTypes"))
     {
         return false;
@@ -245,6 +262,8 @@
     variable_field responseVersion;
     auto msg = reinterpret_cast<pldm_msg*>(getPLDMVersionsRequest.data());
     std::vector<uint8_t> versionDataBuffer;
+    uint8_t* responseMsg = nullptr;
+    size_t responseMsgSize{};
 
     supportedVersions.clear();
     while (PLDM_END != transferFlag && PLDM_START_AND_END != transferFlag)
@@ -263,23 +282,39 @@
             return false;
         }
 
+		
+		// Send PLDM request msg and wait for response
+		rc = pldm_send_recv(eid, fd, getPLDMVersionsRequest.data(), getPLDMVersionsRequest.size(),
+						&responseMsg, &responseMsgSize);
+		if (0 > rc)
+		{
+			std::cout << "Failed to send message/receive response. RC = " << rc
+						<< ", errno = " << errno << "\n";
+			return -1;
+		}
+		pldm_msg* response = reinterpret_cast<pldm_msg*>(responseMsg);
+
         // TID passed as 0 will be ignored since EID is present.
         if (!sendReceivePldmMessage(yield, defaultTID, timeOut, retryCount,
                                     getPLDMVersionsRequest,
                                     getPLDMVersionsResponse, eid))
         {
+#if 0
             phosphor::logging::log<phosphor::logging::level::ERR>(
                 "Send or receive error while getting supported PLDM Versions",
                 phosphor::logging::entry("EID=0x%X", eid));
             return false;
+#endif
         }
 
         // For multipart response next transfer handle will be updated in
         // transferHandle variable and it will be used in subsequent iterations
         rc = decode_get_version_resp(
-            reinterpret_cast<pldm_msg*>(getPLDMVersionsResponse.data()),
-            getPLDMVersionsResponse.size() - hdrSize, &completionCode,
-            &transferHandle, &transferFlag, &responseVersion);
+            response,
+            responseMsgSize  - hdrSize, &completionCode,
+            &transferHandle, &transferFlag, &responseVersion);       
+
+
         if (!validateBaseRespDecode(eid, rc, completionCode, "GetVersion"))
         {
             return false;
@@ -349,6 +384,8 @@
         sizeof(pldm_get_commands_req) + hdrSize, 0x00);
     auto msg = reinterpret_cast<pldm_msg*>(getCommandsRequest.data());
     std::vector<uint8_t> getCommandsResponse;
+    uint8_t* responseMsg = nullptr;
+    size_t responseMsgSize{};
 
     int rc = encode_get_commands_req(instanceID, pldmType, version, msg);
     if (!validateBaseReqEncode(eid, rc, "GetPLDMCommands"))
@@ -356,20 +393,36 @@
         return std::nullopt;
     }
 
+
+	// Send PLDM request msg and wait for response
+	rc = pldm_send_recv(eid, fd, getCommandsRequest.data(), getCommandsRequest.size(),
+					&responseMsg, &responseMsgSize);
+	if (0 > rc)
+	{
+		std::cout << "Failed to send message/receive response. RC = " << rc
+					<< ", errno = " << errno << "\n";
+		return std::nullopt;
+	}
+	pldm_msg* response = reinterpret_cast<pldm_msg*>(responseMsg);
+
     if (!sendReceivePldmMessage(yield, defaultTID, timeOut, retryCount,
                                 getCommandsRequest, getCommandsResponse, eid))
     {
+#if 0
         phosphor::logging::log<phosphor::logging::level::ERR>(
             "Send or receive error during GetPLDMCommands request",
             phosphor::logging::entry("EID=0x%X", eid));
         return std::nullopt;
+#endif
     }
 
     uint8_t completionCode;
     SupportedCommands commands;
-    rc = decode_get_commands_resp(
-        reinterpret_cast<pldm_msg*>(getCommandsResponse.data()),
-        getCommandsResponse.size() - hdrSize, &completionCode, commands.data());
+    
+	rc = decode_get_commands_resp(
+        response,
+        responseMsgSize - hdrSize, &completionCode, commands.data());
+
     if (!validateBaseRespDecode(eid, rc, completionCode, "GetPLDMCommands"))
     {
         return std::nullopt;
@@ -384,6 +437,8 @@
     std::vector<uint8_t> getTIDRequest(hdrSize, 0x00);
     auto msg = reinterpret_cast<pldm_msg*>(getTIDRequest.data());
     std::vector<uint8_t> getTIDResponse;
+    uint8_t* responseMsg = nullptr;
+    size_t responseMsgSize{};
 
     int rc = encode_get_tid_req(instanceID, msg);
     if (!validateBaseReqEncode(eid, rc, "GetTID"))
@@ -394,16 +449,30 @@
     if (!sendReceivePldmMessage(yield, defaultTID, timeOut, retryCount,
                                 getTIDRequest, getTIDResponse, eid))
     {
+#if 0
         phosphor::logging::log<phosphor::logging::level::ERR>(
-            "Send or receive error during GetTID request");
-        return std::nullopt;
+			"Send or receive error during GetTID request");
+	//        return std::nullopt;
+#endif
     }
 
+	// Send PLDM request msg and wait for response
+	rc = pldm_send_recv(eid, fd, getTIDRequest.data(), getTIDRequest.size(),
+					&responseMsg, &responseMsgSize);
+	if (0 > rc)
+	{
+		std::cout << "Failed to send message/receive response. RC = " << rc
+					<< ", errno = " << errno << "\n";
+		return std::nullopt;
+	}
+	pldm_msg* response = reinterpret_cast<pldm_msg*>(responseMsg);
+
     uint8_t completionCode;
     uint8_t tID;
-    rc = decode_get_tid_resp(reinterpret_cast<pldm_msg*>(getTIDResponse.data()),
-                             getTIDResponse.size() - hdrSize, &completionCode,
+    rc = decode_get_tid_resp(response,
+                             responseMsgSize - hdrSize, &completionCode,
                              &tID);
+
     if (!validateBaseRespDecode(eid, rc, completionCode, "GetTID"))
     {
         return std::nullopt;
@@ -441,6 +510,8 @@
                                        0x00);
     auto msg = reinterpret_cast<pldm_msg*>(setTIDRequest.data());
     std::vector<uint8_t> setTIDResponse;
+    uint8_t* responseMsg = nullptr;
+    size_t responseMsgSize{};
 
     int rc = encode_set_tid_req(instanceID, tid, msg);
     if (!validateBaseReqEncode(eid, rc, "SetTID"))
@@ -451,14 +522,28 @@
     if (!sendReceivePldmMessage(yield, defaultTID, timeOut, retryCount,
                                 setTIDRequest, setTIDResponse, eid))
     {
-        phosphor::logging::log<phosphor::logging::level::ERR>(
+#if 0        
+		phosphor::logging::log<phosphor::logging::level::ERR>(
             "Send or receive error during SetTID request");
-        return false;
+//        return false;
+#endif
     }
 
-    uint8_t completionCode;
-    rc = decode_set_tid_resp(reinterpret_cast<pldm_msg*>(setTIDResponse.data()),
-                             setTIDResponse.size() - hdrSize, &completionCode);
+	// Send PLDM request msg and wait for response
+	rc = pldm_send_recv(eid, fd, setTIDRequest.data(), setTIDRequest.size(),
+					&responseMsg, &responseMsgSize);
+	if (0 > rc)
+	{
+		std::cout << "Failed to send message/receive response. RC = " << rc
+				<< ", errno = " << errno << "\n";
+		return false;
+	}
+	pldm_msg* response = reinterpret_cast<pldm_msg*>(responseMsg);
+
+	uint8_t completionCode;
+    rc = decode_set_tid_resp(response,
+                             responseMsgSize - hdrSize, &completionCode);
+
     if (!validateBaseRespDecode(eid, rc, completionCode, "SetTID"))
     {
         return false;
diff -Naur a/src/firmware_update.cpp b/src/firmware_update.cpp
--- a/src/firmware_update.cpp	2024-09-13 12:59:32.154507868 +0800
+++ b/src/firmware_update.cpp	2024-09-13 13:02:13.116489326 +0800
@@ -68,6 +68,7 @@
 std::unique_ptr<FWUpdate> fwUpdate = nullptr;
 std::unique_ptr<sdbusplus::asio::dbus_interface> associationsIntf = nullptr;
 std::map<uint8_t, std::string> inventoryPaths;
+TIDMapper tidMapper;
 
 FWUpdate::FWUpdate(const pldm_tid_t _tid, const uint8_t _deviceIDRecord) :
     reserveBWTimer(
@@ -152,11 +153,30 @@
     if (!sendPldmMessage(yield, currentTid, retryCount, msgTag, false,
                          pldmResp))
     {
+#if 0
         phosphor::logging::log<phosphor::logging::level::ERR>(
             "sendError: Failed to send PLDM message",
             phosphor::logging::entry("TID=%d", currentTid));
         return false;
+#endif
     }
+	std::optional<mctpw_eid_t> eid = tidMapper.getMappedEID(currentTid);
+	// Open connection to MCTP socket
+	int fd = pldm_open();
+	if (-1 == fd)
+	{
+			std::cout << "Failed to init mctp"
+					<< "\n";
+			return -1;
+	}
+	// Send PLDM request msg and wait for response
+	int rc = pldm_send(eid.value(), fd, pldmResp.data(), pldmResp.size());
+	if (0 > rc)
+	{
+			std::cout << "Failed to send message/receive response. RC = " << rc
+					<< ", errno = " << errno << "\n";
+			return -1;
+	}
     return true;
 }
 
@@ -386,6 +406,7 @@
 {
 
     uint8_t instanceID = createInstanceId(currentTid);
+	std::optional<mctpw_eid_t> eid = tidMapper.getMappedEID(currentTid);
     std::vector<uint8_t> pldmReq(sizeof(struct PLDMEmptyRequest) +
                                  sizeof(struct request_update_req) +
                                  compImgSetVerStrn.length);
@@ -400,6 +421,8 @@
         return retVal;
     }
     std::vector<uint8_t> pldmResp;
+	uint8_t* responseMsg = nullptr;
+	size_t responseMsgSize{};
     size_t count = 0;
     do
     {
@@ -410,14 +433,27 @@
         if (!sendReceivePldmMessage(yield, currentTid, timeout, retryCount,
                                     pldmReq, pldmResp))
         {
+#if 0
             phosphor::logging::log<phosphor::logging::level::ERR>(
                 "requestUpdate: Failed to send or receive PLDM message",
                 phosphor::logging::entry("TID=%d", currentTid));
             return PLDM_ERROR;
+#endif
         }
-        auto msgResp = reinterpret_cast<pldm_msg*>(pldmResp.data());
+
+		// Send PLDM request msg and wait for response
+		int rc = pldm_send_recv(eid.value(), fd, pldmReq.data(), pldmReq.size(),
+						&responseMsg, &responseMsgSize);
+		if (0 > rc)
+		{
+				std::cout << "Failed to send message/receive response. RC = " << rc
+						<< ", errno = " << errno << "\n";
+				return -1;
+		}
+		pldm_msg* response = reinterpret_cast<pldm_msg*>(responseMsg);
+		//auto msgResp = reinterpret_cast<pldm_msg*>(pldmResp.data());
         retVal = decode_request_update_resp(
-            msgResp, pldmResp.size() - hdrSize, &completionCode,
+            response, responseMsgSize - hdrSize, &completionCode,
             &fwDeviceMetaDataLen, &fdWillSendGetPkgDataCmd);
     } while ((completionCode == RETRY_REQUEST_UPDATE) &&
              (++count < retryCount));
@@ -521,20 +557,36 @@
 
     // Send GetDeviceMetaData Command
     std::vector<uint8_t> pldmResp;
+	uint8_t* responseMsg = nullptr;
+	size_t responseMsgSize{};
+	std::optional<mctpw_eid_t> eid = tidMapper.getMappedEID(currentTid);
     if (!sendReceivePldmMessage(yield, currentTid, timeout, retryCount, pldmReq,
                                 pldmResp))
     {
+#if 0
         phosphor::logging::log<phosphor::logging::level::ERR>(
             "GetDeviceMetaData: Failed to send or receive PLDM message",
             phosphor::logging::entry("TID=%d", currentTid));
         return PLDM_ERROR;
+#endif
     }
     struct variable_field metaData = {};
-    auto msgResp = reinterpret_cast<pldm_msg*>(pldmResp.data());
+
+	// Send PLDM request msg and wait for response
+	int rc = pldm_send_recv(eid.value(), fd, pldmReq.data(), pldmReq.size(),
+					&responseMsg, &responseMsgSize);
+	if (0 > rc)
+	{
+			std::cout << "Failed to send message/receive response. RC = " << rc
+					<< ", errno = " << errno << "\n";
+			return -1;
+	}
+	pldm_msg* response = reinterpret_cast<pldm_msg*>(responseMsg);
+    //auto msgResp = reinterpret_cast<pldm_msg*>(pldmResp.data());
 
     // Decode received response
     retVal = decode_get_device_meta_data_resp(
-        msgResp, pldmResp.size() - hdrSize, &completionCode,
+        response, responseMsgSize - hdrSize, &completionCode,
         &nextDataTransferHandle, &transferFlag, &metaData);
     if (!validatePLDMRespDecode(currentTid, retVal, completionCode,
                                 "GetDeviceMetaData"))
@@ -750,10 +802,22 @@
     if (!sendPldmMessage(yield, currentTid, retryCount, msgTag, false,
                          pldmResp))
     {
+#if 0
         phosphor::logging::log<phosphor::logging::level::ERR>(
             "sendMetaData: Failed to send PLDM message");
         return PLDM_ERROR;
+#endif
     }
+	std::optional<mctpw_eid_t> eid = tidMapper.getMappedEID(currentTid);
+
+	// Send PLDM request msg and wait for response
+	int rc = pldm_send(eid.value(), fd, pldmResp.data(), pldmResp.size());
+	if (0 > rc)
+	{
+			std::cout << "Failed to send message/receive response. RC = " << rc
+					<< ", errno = " << errno << "\n";
+			return -1;
+	}
 
     return PLDM_SUCCESS;
 }
@@ -842,17 +906,34 @@
         return retVal;
     }
     std::vector<uint8_t> pldmResp;
+	uint8_t* responseMsg = nullptr;
+	size_t responseMsgSize{};
+	std::optional<mctpw_eid_t> eid = tidMapper.getMappedEID(currentTid);
+
     if (!sendReceivePldmMessage(yield, currentTid, timeout, retryCount, pldmReq,
                                 pldmResp))
     {
+#if 0
         phosphor::logging::log<phosphor::logging::level::ERR>(
             "passComponentTable: Failed to send or receive PLDM message",
             phosphor::logging::entry("TID=%d", currentTid));
         return PLDM_ERROR;
+#endif
     }
-    auto msgResp = reinterpret_cast<pldm_msg*>(pldmResp.data());
+
+	// Send PLDM request msg and wait for response
+	int rc = pldm_send_recv(eid.value(), fd, pldmReq.data(), pldmReq.size(),
+					&responseMsg, &responseMsgSize);
+	if (0 > rc)
+	{
+			std::cout << "Failed to send message/receive response. RC = " << rc
+					<< ", errno = " << errno << "\n";
+			return -1;
+	}
+	pldm_msg* response = reinterpret_cast<pldm_msg*>(responseMsg);
+//    auto msgResp = reinterpret_cast<pldm_msg*>(pldmResp.data());
     retVal = decode_pass_component_table_resp(
-        msgResp, pldmResp.size() - hdrSize, &completionCode, &compResp,
+        response, responseMsgSize - hdrSize, &completionCode, &compResp,
         &compRespCode);
     if (!validatePLDMRespDecode(currentTid, retVal, completionCode,
                                 std::string("PassComponentTable")))
@@ -918,7 +999,10 @@
     {
         return retVal;
     }
-    std::vector<uint8_t> pldmResp;
+	std::vector<uint8_t> pldmResp;
+	uint8_t* responseMsg = nullptr;
+	size_t responseMsgSize{};
+	std::optional<mctpw_eid_t> eid = tidMapper.getMappedEID(currentTid);
     if (!sendReceivePldmMessage(yield, currentTid, timeout, retryCount, pldmReq,
                                 pldmResp))
     {
@@ -927,9 +1011,20 @@
             phosphor::logging::entry("TID=%d", currentTid));
         return PLDM_ERROR;
     }
-    auto msgResp = reinterpret_cast<pldm_msg*>(pldmResp.data());
+
+	// Send PLDM request msg and wait for response
+	int rc = pldm_send_recv(eid.value(), fd, pldmReq.data(), pldmReq.size(),
+					&responseMsg, &responseMsgSize);
+	if (0 > rc)
+	{
+			std::cout << "Failed to send message/receive response. RC = " << rc
+					<< ", errno = " << errno << "\n";
+			return -1;
+	}
+	pldm_msg* response = reinterpret_cast<pldm_msg*>(responseMsg);
+    //auto msgResp = reinterpret_cast<pldm_msg*>(pldmResp.data());
     retVal = decode_update_component_resp(
-        msgResp, pldmResp.size() - hdrSize, &completionCode,
+        response, responseMsgSize - hdrSize, &completionCode,
         &compCompatabilityResp, &compCompatabilityRespCode,
         &updateOptFlagsEnabled, &estimatedTimeReqFd);
     if (!validatePLDMRespDecode(currentTid, retVal, completionCode,
@@ -1028,10 +1123,22 @@
     if (!sendPldmMessage(yield, currentTid, retryCount, msgTag, false,
                          pldmResp))
     {
+#if 0
         phosphor::logging::log<phosphor::logging::level::WARNING>(
             "TransferComplete: Failed to send PLDM message");
         return PLDM_ERROR;
+#endif
     }
+	std::optional<mctpw_eid_t> eid = tidMapper.getMappedEID(currentTid);
+
+	// Send PLDM request msg and wait for response
+	int rc = pldm_send(eid.value(), fd, pldmResp.data(), pldmResp.size());
+	if (0 > rc)
+	{
+			std::cout << "Failed to send message/receive response. RC = " << rc
+					<< ", errno = " << errno << "\n";
+			return -1;
+	}
     return PLDM_SUCCESS;
 }
 
@@ -1120,10 +1227,22 @@
     if (!sendPldmMessage(yield, currentTid, retryCount, msgTag, false,
                          pldmResp))
     {
+#if 0
         phosphor::logging::log<phosphor::logging::level::WARNING>(
             "verifyComplete: sendErrorCompletionCode failed.");
         return PLDM_ERROR;
+#endif
     }
+	std::optional<mctpw_eid_t> eid = tidMapper.getMappedEID(currentTid);
+
+	// Send PLDM request msg and wait for response
+	int rc = pldm_send(eid.value(), fd, pldmResp.data(), pldmResp.size());
+	if (0 > rc)
+	{
+			std::cout << "Failed to send message/receive response. RC = " << rc
+					<< ", errno = " << errno << "\n";
+			return -1;
+	}
     return PLDM_SUCCESS;
 }
 
@@ -1207,11 +1326,23 @@
     if (!sendPldmMessage(yield, currentTid, retryCount, msgTag, false,
                          pldmResp))
     {
+#if 0
         phosphor::logging::log<phosphor::logging::level::ERR>(
             "ApplyComplete: Failed to send PLDM message",
             phosphor::logging::entry("TID=%d", currentTid));
         return PLDM_ERROR;
+#endif
     }
+	std::optional<mctpw_eid_t> eid = tidMapper.getMappedEID(currentTid);
+
+	// Send PLDM request msg and wait for response
+	int rc = pldm_send(eid.value(), fd, pldmResp.data(), pldmResp.size());
+	if (0 > rc)
+	{
+			std::cout << "Failed to send message/receive response. RC = " << rc
+					<< ", errno = " << errno << "\n";
+			return -1;
+	}
 
     return PLDM_SUCCESS;
 }
@@ -1398,11 +1529,23 @@
     if (!sendPldmMessage(yield, currentTid, retryCount, msgTag, false,
                          pldmResp))
     {
+#if 0
         phosphor::logging::log<phosphor::logging::level::ERR>(
             "requestFirmwareData: Failed to send PLDM message",
             phosphor::logging::entry("TID=%d", currentTid));
         return PLDM_ERROR;
+#endif
     }
+	std::optional<mctpw_eid_t> eid = tidMapper.getMappedEID(currentTid);
+
+	// Send PLDM request msg and wait for response
+	int rc = pldm_send(eid.value(), fd, pldmResp.data(), pldmResp.size());
+	if (0 > rc)
+	{
+			std::cout << "Failed to send message/receive response. RC = " << rc
+					<< ", errno = " << errno << "\n";
+			return -1;
+	}
 
     return PLDM_SUCCESS;
 }
@@ -1622,10 +1765,22 @@
     if (!sendPldmMessage(yield, currentTid, retryCount, msgTag, false,
                          pldmResp))
     {
+#if 0
         phosphor::logging::log<phosphor::logging::level::ERR>(
             ("sendPackageData: Failed to send PLDM message"));
         return PLDM_ERROR;
+#endif
     }
+	std::optional<mctpw_eid_t> eid = tidMapper.getMappedEID(currentTid);
+
+	// Send PLDM request msg and wait for response
+	int rc = pldm_send(eid.value(), fd, pldmResp.data(), pldmResp.size());
+	if (0 > rc)
+	{
+			std::cout << "Failed to send message/receive response. RC = " << rc
+					<< ", errno = " << errno << "\n";
+			return -1;
+	}
 
     return PLDM_SUCCESS;
 }
@@ -1702,18 +1857,34 @@
         return retVal;
     }
     std::vector<uint8_t> pldmResp;
+	uint8_t* responseMsg = nullptr;
+	size_t responseMsgSize{};
+	std::optional<mctpw_eid_t> eid = tidMapper.getMappedEID(currentTid);
     if (!sendReceivePldmMessage(yield, currentTid, timeout, retryCount, pldmReq,
                                 pldmResp))
     {
+#if 0
         phosphor::logging::log<phosphor::logging::level::ERR>(
             "ActivateFirmware: Failed to send or receive PLDM message",
             phosphor::logging::entry("TID=%d", currentTid));
         return PLDM_ERROR;
+#endif
     }
-    auto msgResp = reinterpret_cast<pldm_msg*>(pldmResp.data());
-    size_t payloadLen = pldmResp.size() - hdrSize;
+
+	// Send PLDM request msg and wait for response
+	int rc = pldm_send_recv(eid.value(), fd, pldmReq.data(), pldmReq.size(),
+					&responseMsg, &responseMsgSize);
+	if (0 > rc)
+	{
+			std::cout << "Failed to send message/receive response. RC = " << rc
+					<< ", errno = " << errno << "\n";
+			return -1;
+	}
+	pldm_msg* response = reinterpret_cast<pldm_msg*>(responseMsg);
+    //auto msgResp = reinterpret_cast<pldm_msg*>(pldmResp.data());
+    size_t payloadLen = responseMsgSize - hdrSize;
     retVal =
-        decode_activate_firmware_resp(msgResp, payloadLen, &completionCode,
+        decode_activate_firmware_resp(response, payloadLen, &completionCode,
                                       &estimatedTimeForSelfContainedActivation);
 
     if (!validatePLDMRespDecode(currentTid, retVal, completionCode,
@@ -1735,17 +1906,35 @@
         return retVal;
     }
     std::vector<uint8_t> pldmResp;
+	uint8_t* responseMsg = nullptr;
+	size_t responseMsgSize{};
+	std::optional<mctpw_eid_t> eid = tidMapper.getMappedEID(currentTid);
+
     if (!sendReceivePldmMessage(yield, currentTid, timeout, retryCount, pldmReq,
                                 pldmResp))
     {
+#if 0
         phosphor::logging::log<phosphor::logging::level::ERR>(
             "GetStatus: Failed to send or receive PLDM message",
             phosphor::logging::entry("TID=%d", currentTid));
         return PLDM_ERROR;
+#endif
     }
-    auto msgResp = reinterpret_cast<pldm_msg*>(pldmResp.data());
-    size_t payloadLen = pldmResp.size() - hdrSize;
-    retVal = decode_get_status_resp(msgResp, payloadLen, &completionCode,
+
+
+	// Send PLDM request msg and wait for response
+	int rc = pldm_send_recv(eid.value(), fd, pldmReq.data(), pldmReq.size(),
+					&responseMsg, &responseMsgSize);
+	if (0 > rc)
+	{
+			std::cout << "Failed to send message/receive response. RC = " << rc
+					<< ", errno = " << errno << "\n";
+			return -1;
+	}
+	pldm_msg* response = reinterpret_cast<pldm_msg*>(responseMsg);
+    //auto msgResp = reinterpret_cast<pldm_msg*>(pldmResp.data());
+    size_t payloadLen = responseMsgSize - hdrSize;
+    retVal = decode_get_status_resp(response, payloadLen, &completionCode,
                                     &currentState, &previousState, &auxState,
                                     &auxStateStatus, &progressPercent,
                                     &reasonCode, &updateOptionFlagsEnabled);
@@ -1796,18 +1985,34 @@
                 .c_str());
         return retVal;
     }
-    std::vector<uint8_t> pldmResp;
+	std::vector<uint8_t> pldmResp;
+	uint8_t* responseMsg = nullptr;
+	size_t responseMsgSize{};
+	std::optional<mctpw_eid_t> eid = tidMapper.getMappedEID(currentTid);
     if (!sendReceivePldmMessage(yield, currentTid, timeout, retryCount, pldmReq,
                                 pldmResp))
     {
+#if 0
         phosphor::logging::log<phosphor::logging::level::ERR>(
             "CancelUpdateComponent: Failed to send or receive PLDM message",
             phosphor::logging::entry("TID=%d", currentTid));
         return PLDM_ERROR;
+#endif
     }
-    auto msgResp = reinterpret_cast<pldm_msg*>(pldmResp.data());
-    size_t payloadLen = pldmResp.size() - hdrSize;
-    retVal = decode_cancel_update_component_resp(msgResp, payloadLen,
+
+	// Send PLDM request msg and wait for response
+	int rc = pldm_send_recv(eid.value(), fd, pldmReq.data(), pldmReq.size(),
+					&responseMsg, &responseMsgSize);
+	if (0 > rc)
+	{
+			std::cout << "Failed to send message/receive response. RC = " << rc
+					<< ", errno = " << errno << "\n";
+			return -1;
+	}
+	pldm_msg* response = reinterpret_cast<pldm_msg*>(responseMsg);
+    //auto msgResp = reinterpret_cast<pldm_msg*>(pldmResp.data());
+    size_t payloadLen = responseMsgSize - hdrSize;
+    retVal = decode_cancel_update_component_resp(response, payloadLen,
                                                  &completionCode);
     if (!validatePLDMRespDecode(currentTid, retVal, completionCode,
                                 "CancelUpdateComponent"))
@@ -1861,18 +2066,34 @@
     {
         return retVal;
     }
-    std::vector<uint8_t> pldmResp;
+	std::vector<uint8_t> pldmResp;
+	uint8_t* responseMsg = nullptr;
+	size_t responseMsgSize{};
+	std::optional<mctpw_eid_t> eid = tidMapper.getMappedEID(currentTid);
     if (!sendReceivePldmMessage(yield, currentTid, timeout, retryCount, pldmReq,
                                 pldmResp))
     {
+#if 0
         phosphor::logging::log<phosphor::logging::level::ERR>(
             "CancelUpdate: Failed to send or receive PLDM message",
             phosphor::logging::entry("TID=%d", currentTid));
         return PLDM_ERROR;
+#endif
     }
-    auto msgResp = reinterpret_cast<pldm_msg*>(pldmResp.data());
-    size_t payloadLen = pldmResp.size() - hdrSize;
-    retVal = decode_cancel_update_resp(msgResp, payloadLen, &completionCode,
+
+	// Send PLDM request msg and wait for response
+	int rc = pldm_send_recv(eid.value(), fd, pldmReq.data(), pldmReq.size(),
+					&responseMsg, &responseMsgSize);
+	if (0 > rc)
+	{
+			std::cout << "Failed to send message/receive response. RC = " << rc
+					<< ", errno = " << errno << "\n";
+			return -1;
+	}
+	pldm_msg* response = reinterpret_cast<pldm_msg*>(responseMsg);
+    //auto msgResp = reinterpret_cast<pldm_msg*>(pldmResp.data());
+    size_t payloadLen = responseMsgSize - hdrSize;
+    retVal = decode_cancel_update_resp(response, payloadLen, &completionCode,
                                        &nonFunctioningComponentIndication,
                                        &nonFunctioningComponentBitmap);
     if (!validatePLDMRespDecode(currentTid, retVal, completionCode,
@@ -2567,7 +2788,7 @@
     associationsIntf->initialize();
 }
 
-bool fwuInit(boost::asio::yield_context yield, const pldm_tid_t tid)
+bool fwuInit(boost::asio::yield_context yield, const pldm_tid_t tid, const mctpw_eid_t eid)
 {
 
     if (!fwuBaseInitialized)
@@ -2576,7 +2797,7 @@
         registerAssociationsProperty();
         fwuBaseInitialized = true;
     }
-    FWInventoryInfo inventoryInfo(tid);
+    FWInventoryInfo inventoryInfo(tid, eid);
     std::optional<FDProperties> properties =
         inventoryInfo.runInventoryCommands(yield);
 
diff -Naur a/src/fru.cpp b/src/fru.cpp
--- a/src/fru.cpp	2024-09-13 12:59:32.154507868 +0800
+++ b/src/fru.cpp	2024-09-13 13:02:13.116489326 +0800
@@ -286,33 +286,47 @@
             return PLDM_ERROR;
         }
 
-        std::vector<uint8_t> responseMsg;
-
-        if (!sendReceivePldmMessage(yield, tid, timeout, retryCount, requestMsg,
-                                    responseMsg))
-        {
-            phosphor::logging::log<phosphor::logging::level::ERR>(
-                "GetFruRecordTable: Failed to send or receive PLDM message",
-                phosphor::logging::entry("TID=%d", tid));
-            return PLDM_ERROR;
-        }
+        std::vector<uint8_t> res;
+		uint8_t* responseMsg = nullptr;
+		size_t responseMsgSize{};
+
+		// Send PLDM request msg and wait for response
+		rc = pldm_send_recv(eid, fd, requestMsg.data(), requestMsg.size(),
+						&responseMsg, &responseMsgSize);
+		if (0 > rc)
+		{
+			std::cout << "Failed to send message/receive response. RC = " << rc
+					<< ", errno = " << errno << "\n";
+            return PLDM_ERROR;
+		}
+		pldm_msg* response = reinterpret_cast<pldm_msg*>(responseMsg);
+
+		if (!sendReceivePldmMessage(yield, tid, timeout, retryCount, requestMsg,
+								res))
+		{
+#if 0
+				phosphor::logging::log<phosphor::logging::level::ERR>(
+								"GetFruRecordTable: Failed to send or receive PLDM message",
+								phosphor::logging::entry("TID=%d", tid));
+				return PLDM_ERROR;
+#endif
+		}
 
-        auto responsePtr = reinterpret_cast<pldm_msg*>(responseMsg.data());
-        size_t payloadLen = responseMsg.size() - pldmHdrSize;
+		size_t payloadLen = responseMsgSize - pldmHdrSize;
 
-        if (payloadLen < PLDM_GET_FRU_RECORD_TABLE_MIN_RESP_BYTES)
-        {
-            phosphor::logging::log<phosphor::logging::level::ERR>(
-                "GetFruRecordTable: payloadLen cannot be less than 6",
-                phosphor::logging::entry("TID=%d", tid));
-            return PLDM_ERROR;
-        }
+		if (payloadLen < PLDM_GET_FRU_RECORD_TABLE_MIN_RESP_BYTES)
+		{
+				phosphor::logging::log<phosphor::logging::level::ERR>(
+								"GetFruRecordTable: payloadLen cannot be less than 6",
+								phosphor::logging::entry("TID=%d", tid));
+				return PLDM_ERROR;
+		}
 
         std::vector<uint8_t> partialFRURecordTableData(
             payloadLen - PLDM_GET_FRU_RECORD_TABLE_MIN_RESP_BYTES);
 
         rc = decode_get_fru_record_table_resp(
-            responsePtr, payloadLen, &cc, &nextDataTransferHandle,
+            response, payloadLen, &cc, &nextDataTransferHandle,
             &transferFlag, partialFRURecordTableData.data(),
             &fruRecordTableLen);
 
@@ -384,19 +398,32 @@
         return PLDM_ERROR;
     }
 
-    std::vector<uint8_t> responseMsg;
+	std::vector<uint8_t> res;
+	uint8_t* responseMsg = nullptr;
+	size_t responseMsgSize{};
 
     if (!sendReceivePldmMessage(yield, tid, timeout, retryCount, requestMsg,
-                                responseMsg))
+                                res))
     {
+#if 0
         phosphor::logging::log<phosphor::logging::level::ERR>(
             "GetFRURecordTableMetadata: Failed to send or receive PLDM message",
             phosphor::logging::entry("TID=%d", tid));
         return PLDM_ERROR;
+#endif
     }
 
-    auto responsePtr = reinterpret_cast<pldm_msg*>(responseMsg.data());
-    size_t payloadLen = responseMsg.size() - pldmHdrSize;
+	// Send PLDM request msg and wait for response
+	rc = pldm_send_recv(eid, fd, requestMsg.data(), requestMsg.size(),
+					&responseMsg, &responseMsgSize);
+	if (0 > rc)
+	{
+		std::cout << "Failed to send message/receive response. RC = " << rc
+					<< ", errno = " << errno << "\n";
+        return PLDM_ERROR;
+	}
+	pldm_msg* response = reinterpret_cast<pldm_msg*>(responseMsg);    
+	size_t payloadLen = responseMsgSize - pldmHdrSize;
 
     // parse response
     uint8_t cc = PLDM_ERROR;
@@ -409,7 +436,7 @@
     uint32_t checksum;
 
     rc = decode_get_fru_record_table_metadata_resp(
-        responsePtr, payloadLen, &cc, &fruDataMajorVersion,
+        response, payloadLen, &cc, &fruDataMajorVersion,
         &fruDataMinorVersion, &fruTableMaximumSize, &fruTableLength,
         &totalRecordSetIdentifiers, &totalTableRecords, &checksum);
 
@@ -425,10 +452,11 @@
     return PLDM_SUCCESS;
 }
 
-GetPLDMFRU::GetPLDMFRU(boost::asio::yield_context yieldVal,
-                       const pldm_tid_t tidVal) :
+GetPLDMFRU::GetPLDMFRU(boost::asio::yield_context yieldVal, 
+                       const pldm_tid_t tidVal, const mctpw_eid_t eidVal) :
     yield(yieldVal),
-    tid(tidVal)
+    tid(tidVal),
+	eid(eidVal)
 {
 }
 
@@ -462,7 +490,7 @@
     return true;
 }
 
-SetPLDMFRU::SetPLDMFRU(const pldm_tid_t tidVal) : tid(tidVal)
+SetPLDMFRU::SetPLDMFRU(const pldm_tid_t tidVal, const mctpw_eid_t eidVal) : eid(eidVal), tid(tidVal)
 {
 }
 
@@ -562,27 +590,42 @@
             return PLDM_ERROR;
         }
 
-        std::vector<uint8_t> responseMsg;
+		std::vector<uint8_t> res;
+		uint8_t* responseMsg = nullptr;
+		size_t responseMsgSize{};
+
         if (!sendReceivePldmMessage(yield, tid, timeout, retryCount, requestMsg,
-                                    responseMsg))
+                                    res))
         {
+#if 0
             phosphor::logging::log<phosphor::logging::level::ERR>(
                 "SetFruRecordTable: Failed to send or receive PLDM message",
                 phosphor::logging::entry("TID=%d", tid));
             return PLDM_ERROR;
+#endif
         }
 
-        auto responsePtr = reinterpret_cast<pldm_msg*>(responseMsg.data());
-        size_t payloadLen = responseMsg.size() - pldmHdrSize;
 
-        // parse response
-        uint8_t cc = PLDM_ERROR;
-        uint32_t nextDataTransferHandle = 0;
+		// Send PLDM request msg and wait for response
+		int rc = pldm_send_recv(eid, fd, requestMsg.data(), requestMsg.size(),
+						&responseMsg, &responseMsgSize);
+		if (0 > rc)
+		{
+			std::cout << "Failed to send message/receive response. RC = " << rc
+						<< ", errno = " << errno << "\n";
+            return PLDM_ERROR;
+		}
+		pldm_msg* response = reinterpret_cast<pldm_msg*>(responseMsg);
+		size_t payloadLen = responseMsgSize - pldmHdrSize;
+
+		// parse response
+		uint8_t cc = PLDM_ERROR;
+		uint32_t nextDataTransferHandle = 0;
 
-        int rc = decode_set_fru_record_table_resp(responsePtr, payloadLen, &cc,
-                                                  &nextDataTransferHandle);
+		rc = decode_set_fru_record_table_resp(response, payloadLen, &cc,
+						&nextDataTransferHandle);
 
-        if (!validatePLDMRespDecode(tid, rc, cc, "SetFruRecordTable"))
+		if (!validatePLDMRespDecode(tid, rc, cc, "SetFruRecordTable"))
         {
             phosphor::logging::log<phosphor::logging::level::ERR>(
                 "SetFruRecordTable: Invalid Response");
@@ -661,7 +704,7 @@
     }
 
     // Re-query FRU data via get commands and update the FRU fields accordingly.
-    GetPLDMFRU getFRUCommands(yield, tid);
+    GetPLDMFRU getFRUCommands(yield, tid, eid);
     if (!getFRUCommands.runGetFRUCommands())
     {
         phosphor::logging::log<phosphor::logging::level::ERR>(
@@ -782,8 +825,8 @@
                                            "xyz.openbmc_project.PLDM.GetFRU");
     getFRUIface->register_method(
         "GetPldmFRU",
-        [](boost::asio::yield_context yieldVal, const pldm_tid_t tidVal) {
-            GetPLDMFRU fruManager(yieldVal, tidVal);
+        [](boost::asio::yield_context yieldVal, const pldm_tid_t tidVal, const mctpw_eid_t eidVal) {
+            GetPLDMFRU fruManager(yieldVal, tidVal, eidVal);
             std::optional<std::vector<uint8_t>> retVal =
                 fruManager.getPLDMFruRecordData();
 
@@ -806,11 +849,11 @@
                                            "xyz.openbmc_project.PLDM.SetFRU");
     setFRUIface->register_method(
         "SetFRU",
-        [](boost::asio::yield_context yieldVal, const pldm_tid_t tidVal,
+        [](boost::asio::yield_context yieldVal, const pldm_tid_t tidVal, const mctpw_eid_t eidVal,
            const std::vector<uint8_t>& data) {
             phosphor::logging::log<phosphor::logging::level::INFO>(
                 "SetFRURecordTable is called");
-            SetPLDMFRU setFRU(tidVal);
+            SetPLDMFRU setFRU(tidVal, eidVal);
             int retVal = setFRU.setFruRecordTableCmd(yieldVal, data);
 
             if (retVal != PLDM_SUCCESS)
@@ -832,7 +875,7 @@
     }
 }
 
-bool fruInit(boost::asio::yield_context yield, const pldm_tid_t tid)
+bool fruInit(boost::asio::yield_context yield, const pldm_tid_t tid, const mctpw_eid_t eid)
 {
     bool retVal = true;
     if (!setFRUIface || !setFRUIface->is_initialized())
@@ -841,7 +884,7 @@
         ipmiFru.initializeFRUSupport();
     }
 
-    GetPLDMFRU getFRUCommands(yield, tid);
+    GetPLDMFRU getFRUCommands(yield, tid, eid);
     if (!getFRUCommands.runGetFRUCommands())
     {
         retVal = false;
diff -Naur a/src/fwu_inventory.cpp b/src/fwu_inventory.cpp
--- a/src/fwu_inventory.cpp	2024-09-13 12:59:32.154507868 +0800
+++ b/src/fwu_inventory.cpp	2024-09-13 13:02:13.116489326 +0800
@@ -29,8 +29,8 @@
 // map that holds the properties of all terminus
 std::map<pldm_tid_t, FDProperties> terminusFwuProperties;
 
-FWInventoryInfo::FWInventoryInfo(const pldm_tid_t _tid) :
-    tid(_tid), objServer(getObjServer())
+FWInventoryInfo::FWInventoryInfo(const pldm_tid_t _tid, const mctpw_eid_t _eid) :
+    tid(_tid), eid(_eid), objServer(getObjServer())
 {
 }
 
@@ -54,24 +54,41 @@
     }
 
     std::vector<uint8_t> pldmResp;
+	uint8_t* responseMsg = nullptr;
+	size_t responseMsgSize{};
 
     if (!sendReceivePldmMessage(yield, tid, timeout, retryCount, pldmReq,
                                 pldmResp))
     {
+#if 0
         phosphor::logging::log<phosphor::logging::level::ERR>(
             "QueryDeviceIdentifiers: Failed to send or receive PLDM message",
             phosphor::logging::entry("TID=%d", tid));
-        return PLDM_ERROR;
-    }
-    auto msgResp = reinterpret_cast<pldm_msg*>(pldmResp.data());
-    size_t payloadLen = pldmResp.size() - hdrSize;
+		return PLDM_ERROR;
+#endif
+	}
+
+	//    auto msgResp = reinterpret_cast<pldm_msg*>(pldmResp.data());
+
+	// Send PLDM request msg and wait for response
+	int rc = pldm_send_recv(eid, fd, pldmReq.data(), pldmReq.size(),
+					&responseMsg, &responseMsgSize);
+	if (0 > rc)
+	{
+		std::cout << "Failed to send message/receive response. RC = " << rc
+					<< ", errno = " << errno << "\n";
+		return -1;
+	}
+	pldm_msg* response = reinterpret_cast<pldm_msg*>(responseMsg);
+
+    size_t payloadLen = responseMsgSize - hdrSize;
     uint8_t completionCode = PLDM_SUCCESS;
     uint32_t deviceIdentifiersLen = 0;
     uint8_t descriptorCount = 0;
     uint8_t* descriptorData = NULL;
 
     retVal = decode_query_device_identifiers_resp(
-        msgResp, payloadLen, &completionCode, &deviceIdentifiersLen,
+        response, payloadLen, &completionCode, &deviceIdentifiersLen,
         &descriptorCount, &descriptorData);
 
     if (retVal != PLDM_SUCCESS)
@@ -239,24 +256,44 @@
     }
 
     std::vector<uint8_t> pldmResp;
+	uint8_t* responseMsg = nullptr;
+	size_t responseMsgSize{};
 
     if (!sendReceivePldmMessage(yield, tid, timeout, retryCount, pldmReq,
                                 pldmResp))
     {
+#if 0
         phosphor::logging::log<phosphor::logging::level::ERR>(
             "GetFirmwareParameters: Failed to send or receive PLDM message",
             phosphor::logging::entry("TID=%d", tid));
         return PLDM_ERROR;
+#endif
     }
-    auto respMsg = reinterpret_cast<pldm_msg*>(pldmResp.data());
-    size_t payloadLen = pldmResp.size() - hdrSize;
+
+	// Send PLDM request msg and wait for response
+
+	int rc = pldm_send_recv(eid, fd, pldmReq.data(), pldmReq.size(),
+					&responseMsg, &responseMsgSize);
+
+
+	if (0 > rc)
+	{
+		std::cout << "Failed to send message/receive response. RC = " << rc
+					<< ", errno = " << errno << "\n";
+		return PLDM_ERROR;
+	}
+
+	pldm_msg* response = reinterpret_cast<pldm_msg*>(responseMsg);
+	pldmResp.assign(responseMsg, responseMsg + responseMsgSize);
+   // auto respMsg = reinterpret_cast<pldm_msg*>(pldmResp.data());
+    size_t payloadLen = responseMsgSize - hdrSize;
 
     struct get_firmware_parameters_resp resp;
     struct variable_field activeCompImageSetVerStr;
     struct variable_field pendingCompImageSetVerStr;
 
     retVal = decode_get_firmware_parameters_comp_img_set_resp(
-        respMsg, payloadLen, &resp, &activeCompImageSetVerStr,
+        response, payloadLen, &resp, &activeCompImageSetVerStr,
         &pendingCompImageSetVerStr);
 
     if (retVal != PLDM_SUCCESS)
@@ -277,14 +314,15 @@
                             resp.active_comp_image_set_ver_str_len +
                             resp.pending_comp_image_set_ver_str_len;
 
-    if (pldmResp.size() < compDataOffset)
+    if (responseMsgSize < compDataOffset)
     {
         phosphor::logging::log<phosphor::logging::level::ERR>(
             "GetFirmwareParameters: Response length is invalid");
         return PLDM_ERROR_INVALID_LENGTH;
     }
 
-    std::vector<uint8_t> compData(pldmResp.begin() + compDataOffset,
+
+	std::vector<uint8_t> compData(pldmResp.begin() + compDataOffset,
                                   pldmResp.end());
 
     return unpackCompData(resp.comp_count, compData);
@@ -531,9 +569,11 @@
     if (retVal != PLDM_SUCCESS)
     {
         phosphor::logging::log<phosphor::logging::level::ERR>(
-            "Failed to run GetFirmwareParameters command");
+            "failed to run getfirmwareparameters command");
         return std::nullopt;
     }
+        phosphor::logging::log<phosphor::logging::level::INFO>(
+            "Inventory commands");
     FDProperties fdProperties =
         std::make_tuple(fwuProperties, descriptors, compPropertiesMap);
     return fdProperties;
diff -Naur a/src/numeric_effecter_handler.cpp b/src/numeric_effecter_handler.cpp
--- a/src/numeric_effecter_handler.cpp	2024-09-13 12:59:32.154507868 +0800
+++ b/src/numeric_effecter_handler.cpp	2024-09-13 13:02:13.116489326 +0800
@@ -29,9 +29,9 @@
 constexpr const uint8_t transitionIntervalSec = 3;
 
 NumericEffecterHandler::NumericEffecterHandler(
-    const pldm_tid_t tid, const EffecterID effecterID, const std::string& name,
+    const pldm_tid_t tid, const mctpw_eid_t eid, const EffecterID effecterID, const std::string& name,
     const std::shared_ptr<pldm_numeric_effecter_value_pdr>& pdr) :
-    _tid(tid),
+    _tid(tid), _eid(eid),
     _effecterID(effecterID), _name(name), _pdr(pdr)
 {
 }
@@ -49,6 +49,7 @@
 bool NumericEffecterHandler::enableNumericEffecter(
     boost::asio::yield_context yield)
 {
+	(void)yield;
     uint8_t effecterOpState;
     switch (_pdr->effecter_init)
     {
@@ -87,7 +88,10 @@
         return false;
     }
 
-    std::vector<uint8_t> resp;
+//    std::vector<uint8_t> resp;
+	uint8_t* responseMsg = nullptr;
+	size_t responseMsgSize{};
+#if 0
     if (!sendReceivePldmMessage(yield, _tid, commandTimeout, commandRetryCount,
                                 req, resp))
     {
@@ -97,11 +101,20 @@
             phosphor::logging::entry("TID=%d", _tid));
         return false;
     }
-
+#endif
+	rc = pldm_send_recv(_eid, fd, req.data(), req.size(),
+					&responseMsg, &responseMsgSize);
+	if (0 > rc)
+	{
+			std::cout << "Failed to send message/receive response. RC = " << rc
+					<< ", errno = " << errno << "\n";
+			return -1;
+	}
+	pldm_msg* response = reinterpret_cast<pldm_msg*>(responseMsg);
     uint8_t completionCode;
-    auto rspMsg = reinterpret_cast<pldm_msg*>(resp.data());
+    //auto rspMsg = reinterpret_cast<pldm_msg*>(resp.data());
 
-    rc = decode_cc_only_resp(rspMsg, resp.size() - pldmMsgHdrSize,
+    rc = decode_cc_only_resp(response, responseMsgSize - pldmMsgHdrSize,
                              &completionCode);
     if (!validatePLDMRespDecode(_tid, rc, completionCode,
                                 "SetNumericEffecterEnable"))
@@ -280,6 +293,7 @@
 bool NumericEffecterHandler::getEffecterReading(
     boost::asio::yield_context yield)
 {
+	(void)yield;
     int rc;
     std::vector<uint8_t> req(pldmMsgHdrSize +
                              sizeof(pldm_get_numeric_effecter_value_req));
@@ -292,6 +306,7 @@
         return false;
     }
 
+#if 0
     std::vector<uint8_t> resp;
     if (!sendReceivePldmMessage(yield, _tid, commandTimeout, commandRetryCount,
                                 req, resp))
@@ -302,16 +317,28 @@
             phosphor::logging::entry("EFFECTER_ID=0x%0X", _effecterID));
         return false;
     }
+#endif
 
+	uint8_t* responseMsg = nullptr;
+	size_t responseMsgSize{};
+	rc = pldm_send_recv(_eid, fd, req.data(), req.size(),
+					&responseMsg, &responseMsgSize);
+	if (0 > rc)
+	{
+			std::cout << "Failed to send message/receive response. RC = " << rc
+					<< ", errno = " << errno << "\n";
+			return -1;
+	}
+	pldm_msg* response = reinterpret_cast<pldm_msg*>(responseMsg);
     uint8_t completionCode;
     uint8_t effecterDataSize;
     uint8_t effecterOperationalState;
     union_effecter_data_size pendingValue;
     union_effecter_data_size presentValue;
-    auto rspMsg = reinterpret_cast<pldm_msg*>(resp.data());
+    //auto rspMsg = reinterpret_cast<pldm_msg*>(resp.data());
 
     rc = decode_get_numeric_effecter_value_resp(
-        rspMsg, resp.size() - pldmMsgHdrSize, &completionCode,
+        response, responseMsgSize - pldmMsgHdrSize, &completionCode,
         &effecterDataSize, &effecterOperationalState,
         reinterpret_cast<uint8_t*>(&pendingValue),
         reinterpret_cast<uint8_t*>(&presentValue));
@@ -359,6 +386,7 @@
 bool NumericEffecterHandler::setEffecter(boost::asio::yield_context yield,
                                          double& value)
 {
+	(void)yield;
     if (value < minSettable || value > maxSettable)
     {
         phosphor::logging::log<phosphor::logging::level::ERR>(
@@ -406,7 +434,7 @@
     {
         return false;
     }
-
+#if 0
     std::vector<uint8_t> resp;
     if (!sendReceivePldmMessage(yield, _tid, commandTimeout, commandRetryCount,
                                 req, resp))
@@ -417,11 +445,22 @@
             phosphor::logging::entry("TID=%d", _tid));
         return false;
     }
-
+#endif
+	uint8_t* responseMsg = nullptr;
+	size_t responseMsgSize{};
+	rc = pldm_send_recv(_eid, fd, req.data(), req.size(),
+					&responseMsg, &responseMsgSize);
+	if (0 > rc)
+	{
+			std::cout << "Failed to send message/receive response. RC = " << rc
+					<< ", errno = " << errno << "\n";
+			return -1;
+	}
+	pldm_msg* response = reinterpret_cast<pldm_msg*>(responseMsg);
     uint8_t completionCode;
-    auto rspMsg = reinterpret_cast<pldm_msg*>(resp.data());
+//    auto rspMsg = reinterpret_cast<pldm_msg*>(resp.data());
 
-    rc = decode_cc_only_resp(rspMsg, resp.size() - pldmMsgHdrSize,
+    rc = decode_cc_only_resp(response, responseMsgSize - pldmMsgHdrSize,
                              &completionCode);
     if (!validatePLDMRespDecode(_tid, rc, completionCode,
                                 "SetNumericEffecterValue"))
diff -Naur a/src/numeric_sensor_handler.cpp b/src/numeric_sensor_handler.cpp
--- a/src/numeric_sensor_handler.cpp	2024-09-13 12:59:32.154507868 +0800
+++ b/src/numeric_sensor_handler.cpp	2024-09-13 13:02:13.116489326 +0800
@@ -29,9 +29,9 @@
 {
 
 NumericSensorHandler::NumericSensorHandler(
-    const pldm_tid_t tid, const SensorID sensorID, const std::string& name,
+    const pldm_tid_t tid, const mctpw_eid_t eid, const SensorID sensorID, const std::string& name,
     const std::shared_ptr<pldm_numeric_sensor_value_pdr>& pdr) :
-    _tid(tid),
+    _tid(tid), _eid(eid),
     _sensorID(sensorID), _name(name), _pdr(pdr)
 {
 }
@@ -91,20 +91,35 @@
     }
 
     std::vector<uint8_t> resp;
+    uint8_t* responseMsg = nullptr;
+    size_t responseMsgSize{};
     if (!sendReceivePldmMessage(yield, _tid, commandTimeout, commandRetryCount,
                                 req, resp))
     {
+#if 0
         phosphor::logging::log<phosphor::logging::level::ERR>(
             "Failed to send or receive  SetNumericSensorEnable request",
             phosphor::logging::entry("TID=%d", _tid),
             phosphor::logging::entry("SENSOR_ID=%d", _sensorID));
-        return false;
+//        return false;
+#endif
     }
 
+
+	// Send PLDM request msg and wait for response
+	rc = pldm_send_recv(_eid, fd, req.data(), req.size(),
+					&responseMsg, &responseMsgSize);
+	if (0 > rc)
+	{
+		std::cout << "Failed to send message/receive response. RC = " << rc
+					<< ", errno = " << errno << "\n";
+		return false;
+	}
+	pldm_msg* response = reinterpret_cast<pldm_msg*>(responseMsg);
+
     uint8_t completionCode;
-    auto rspMsg = reinterpret_cast<pldm_msg*>(resp.data());
 
-    rc = decode_cc_only_resp(rspMsg, resp.size() - pldmMsgHdrSize,
+    rc = decode_cc_only_resp(response, responseMsgSize - pldmMsgHdrSize,
                              &completionCode);
     if (!validatePLDMRespDecode(_tid, rc, completionCode,
                                 "SetNumericSensorEnable"))
@@ -356,28 +371,42 @@
     }
 
     std::vector<uint8_t> resp;
+	uint8_t* responseMsg = nullptr;
+	size_t responseMsgSize{};
     if (!sendReceivePldmMessage(yield, _tid, commandTimeout, commandRetryCount,
                                 req, resp))
     {
+#if 0
         phosphor::logging::log<phosphor::logging::level::ERR>(
             "Failed to send or receive GetSensorReading request",
             phosphor::logging::entry("TID=%d", _tid),
             phosphor::logging::entry("SENSOR_ID=0x%0X", _sensorID));
-        return false;
+    //    return false;
+#endif
     }
 
-    uint8_t completionCode;
-    uint8_t sensorDataSize = _pdr->sensor_data_size;
-    uint8_t sensorOperationalState;
-    uint8_t sensorEventMessageEnable;
+	// Send PLDM request msg and wait for response
+	rc = pldm_send_recv(_eid, fd, req.data(), req.size(),
+					&responseMsg, &responseMsgSize);
+	if (0 > rc)
+	{
+		std::cout << "Failed to send message/receive response. RC = " << rc
+					<< ", errno = " << errno << "\n";
+		return -1;
+	}
+	pldm_msg* response = reinterpret_cast<pldm_msg*>(responseMsg);
+
+	uint8_t completionCode;
+	uint8_t sensorDataSize = _pdr->sensor_data_size;
+	uint8_t sensorOperationalState;
+	uint8_t sensorEventMessageEnable;
     uint8_t presentState;
     uint8_t previousState;
     uint8_t eventState;
     union_sensor_data_size presentReading;
-    auto rspMsg = reinterpret_cast<pldm_msg*>(resp.data());
-
-    rc = decode_get_sensor_reading_resp(
-        rspMsg, resp.size() - pldmMsgHdrSize, &completionCode, &sensorDataSize,
+    
+	rc = decode_get_sensor_reading_resp(
+        response, responseMsgSize - pldmMsgHdrSize, &completionCode, &sensorDataSize,
         &sensorOperationalState, &sensorEventMessageEnable, &presentState,
         &previousState, &eventState,
         reinterpret_cast<uint8_t*>(&presentReading));
diff -Naur a/src/pdr_manager.cpp b/src/pdr_manager.cpp
--- a/src/pdr_manager.cpp	2024-09-13 12:59:32.154507868 +0800
+++ b/src/pdr_manager.cpp	2024-09-13 13:02:13.116489326 +0800
@@ -34,7 +34,7 @@
 namespace platform
 {
 
-PDRManager::PDRManager(const pldm_tid_t tid) : _tid(tid)
+PDRManager::PDRManager(const mctpw_eid_t eid, const pldm_tid_t tid) : _eid(eid), _tid(tid)
 {
 }
 
@@ -130,20 +130,36 @@
     }
 
     std::vector<uint8_t> resp;
+    uint8_t* responseMsg = nullptr;
+    size_t responseMsgSize{};
     if (!sendReceivePldmMessage(yield, _tid, commandTimeout, commandRetryCount,
                                 req, resp))
     {
+#if 0
         phosphor::logging::log<phosphor::logging::level::ERR>(
             "Failed to send or receive GetPDRRepositoryInfo request",
             phosphor::logging::entry("TID=%d", _tid));
-        return std::nullopt;
+    //    return std::nullopt;
+#endif
     }
 
-    pldm_get_pdr_repository_info_resp pdrInfo;
-    auto rspMsg = reinterpret_cast<pldm_msg*>(resp.data());
+
+	// Send PLDM request msg and wait for response
+	rc = pldm_send_recv(_eid, fd, req.data(), req.size(),
+					&responseMsg, &responseMsgSize);
+	if (0 > rc)
+	{
+		std::cout << "Failed to send message/receive response. RC = " << rc
+					<< ", errno = " << errno << "\n";
+		return std::nullopt;
+	}
+	pldm_msg* response = reinterpret_cast<pldm_msg*>(responseMsg);
+
+	pldm_get_pdr_repository_info_resp pdrInfo;
 
     rc = decode_get_pdr_repository_info_resp(
-        rspMsg, resp.size() - pldmMsgHdrSize, &pdrInfo);
+        response, responseMsgSize - pldmMsgHdrSize, &pdrInfo);
+
     if (!validatePLDMRespDecode(_tid, rc, pdrInfo.completion_code,
                                 "GetPDRRepositoryInfo"))
     {
@@ -156,13 +172,15 @@
     return pdrInfo.pdr_repo_info;
 }
 
-static bool handleGetPDRResp(pldm_tid_t tid, std::vector<uint8_t>& resp,
+static bool handleGetPDRResp(pldm_tid_t tid,
                              RecordHandle& nextRecordHandle,
                              transfer_op_flag& transferOpFlag,
                              uint16_t& recordChangeNumber,
                              DataTransferHandle& dataTransferHandle,
                              bool& transferComplete,
-                             std::vector<uint8_t>& pdrRecord)
+                             std::vector<uint8_t>& pdrRecord,
+							 uint8_t* &responseMsg,
+							 size_t responseMsgSize)
 {
     int rc;
     uint8_t completionCode{};
@@ -170,10 +188,10 @@
     uint8_t transferCRC{};
     uint16_t recordDataLen{};
     DataTransferHandle nextDataTransferHandle{};
-    auto respMsgPtr = reinterpret_cast<struct pldm_msg*>(resp.data());
+    pldm_msg* response = reinterpret_cast<pldm_msg*>(responseMsg);
 
     // Get the number of recordData bytes in the response
-    rc = decode_get_pdr_resp(respMsgPtr, resp.size() - pldmMsgHdrSize,
+    rc = decode_get_pdr_resp(response, responseMsgSize - pldmMsgHdrSize,
                              &completionCode, &nextRecordHandle,
                              &nextDataTransferHandle, &transferFlag,
                              &recordDataLen, nullptr, 0, &transferCRC);
@@ -184,10 +202,9 @@
 
     std::vector<uint8_t> pdrData(recordDataLen, 0);
     rc = decode_get_pdr_resp(
-        respMsgPtr, resp.size() - pldmMsgHdrSize, &completionCode,
+        response, responseMsgSize - pldmMsgHdrSize, &completionCode,
         &nextRecordHandle, &nextDataTransferHandle, &transferFlag,
         &recordDataLen, pdrData.data(), pdrData.size(), &transferCRC);
-
     if (!validatePLDMRespDecode(tid, rc, completionCode, "GetPDR"))
     {
         return false;
@@ -250,19 +267,34 @@
             break;
         }
 
-        std::vector<uint8_t> resp;
+		std::vector<uint8_t> resp;
+		uint8_t* responseMsg = nullptr;
+		size_t responseMsgSize{};
+
         if (!sendReceivePldmMessage(yield, _tid, commandTimeout,
                                     commandRetryCount, req, resp))
         {
+#if 0
             phosphor::logging::log<phosphor::logging::level::ERR>(
                 "Failed to send or receive GetPDR request",
                 phosphor::logging::entry("TID=%d", _tid));
-            break;
+        //    break;
+#endif
         }
 
+		// Send PLDM request msg and wait for response
+		rc = pldm_send_recv(_eid, fd, req.data(), req.size(),
+						&responseMsg, &responseMsgSize);
+		if (0 > rc)
+		{
+			std::cout << "Failed to send message/receive response. RC = " << rc
+						<< ", errno = " << errno << "\n";
+				return -1;
+		}
+
         bool ret = handleGetPDRResp(
-            _tid, resp, nextRecordHandle, transferOpFlag, recordChangeNumber,
-            dataTransferHandle, transferComplete, pdrRecord);
+            _tid, nextRecordHandle, transferOpFlag, recordChangeNumber,
+            dataTransferHandle, transferComplete, pdrRecord, responseMsg, responseMsgSize);
         if (!ret)
         {
             // Discard the record if decode failed
diff -Naur a/src/platform.cpp b/src/platform.cpp
--- a/src/platform.cpp	2024-09-13 12:59:32.154507868 +0800
+++ b/src/platform.cpp	2024-09-13 13:02:13.116489326 +0800
@@ -199,12 +199,14 @@
 
 std::optional<UUID> getTerminusUID(boost::asio::yield_context yield,
                                    const pldm_tid_t tid,
-                                   std::optional<mctpw_eid_t> eid)
+                                   const mctpw_eid_t eid)
 {
     static constexpr size_t hdrSize = sizeof(PLDMEmptyRequest);
     uint8_t instanceID = createInstanceId(tid);
     std::vector<uint8_t> getUIDRequest(hdrSize, 0x00);
     auto msg = reinterpret_cast<pldm_msg*>(getUIDRequest.data());
+    uint8_t* responseMsg = nullptr;
+    size_t responseMsgSize{};
 
     int rc = encode_get_terminus_uid_req(instanceID, msg);
     if (!validatePLDMReqEncode(tid, rc, "GetTerminusUUID"))
@@ -213,19 +215,36 @@
     }
 
     std::vector<uint8_t> getUIDResponse;
+
     if (!sendReceivePldmMessage(yield, tid, commandTimeout, commandRetryCount,
                                 getUIDRequest, getUIDResponse, eid))
     {
+#if 0
         phosphor::logging::log<phosphor::logging::level::ERR>(
             "Send or receive error during GetTerminusUUID request");
-        return std::nullopt;
+    //    return std::nullopt;
+#endif
     }
 
+
+	// Send PLDM request msg and wait for response
+	rc = pldm_send_recv(eid, fd, getUIDRequest.data(), getUIDRequest.size(),
+					&responseMsg, &responseMsgSize);
+	if (0 > rc)
+	{
+		std::cout << "Failed to send message/receive response. RC = " << rc
+					<< ", errno = " << errno << "\n";
+		return std::nullopt;
+	}
+	pldm_msg* response = reinterpret_cast<pldm_msg*>(responseMsg);
+
     uint8_t completionCode;
     UUID uuid;
-    rc = decode_get_terminus_uid_resp(
-        reinterpret_cast<pldm_msg*>(getUIDResponse.data()),
-        getUIDResponse.size() - hdrSize, &completionCode, uuid.data());
+    
+	rc = decode_get_terminus_uid_resp(
+        response,
+        responseMsgSize - hdrSize, &completionCode, uuid.data());
+
     if (!validatePLDMRespDecode(tid, rc, completionCode, "GetTerminusUUID"))
     {
         return std::nullopt;
@@ -277,9 +296,9 @@
         addUniqueInterface(objPath, "xyz.openbmc_project.PLDM.Platform");
     platformInterface->register_method(
         "RefreshPDR",
-        [](boost::asio::yield_context yield, const pldm_tid_t tid) {
+        [](boost::asio::yield_context yield, const mctpw_eid_t eid, const pldm_tid_t tid) {
             pauseSensorPolling();
-            platformInit(yield, tid, {});
+            platformInit(yield, eid, tid, {});
             resumeSensorPolling();
         });
     platformInterface->initialize();
@@ -300,7 +319,7 @@
 }
 
 bool Platform::initTerminus(
-    boost::asio::yield_context yield, const pldm_tid_t tid,
+    boost::asio::yield_context yield, const mctpw_eid_t eid, const pldm_tid_t tid,
     const pldm::base::CommandSupportTable& /*commandTable*/)
 {
     phosphor::logging::log<phosphor::logging::level::INFO>(
@@ -320,7 +339,7 @@
     try
     {
         std::shared_ptr<PlatformTerminus> platformTerminus =
-            std::make_shared<PlatformTerminus>(yield, tid);
+            std::make_shared<PlatformTerminus>(yield, eid, tid);
         if (isTerminusRemoved(tid))
         {
             phosphor::logging::log<phosphor::logging::level::WARNING>(
@@ -382,10 +401,10 @@
     platform.startSensorPolling();
 }
 
-bool platformInit(boost::asio::yield_context yield, const pldm_tid_t tid,
+bool platformInit(boost::asio::yield_context yield, const mctpw_eid_t eid, const pldm_tid_t tid,
                   const pldm::base::CommandSupportTable& commandTable)
 {
-    return platform.initTerminus(yield, tid, commandTable);
+    return platform.initTerminus(yield, eid, tid, commandTable);
 }
 
 bool deleteMnCTerminus(const pldm_tid_t tid)
diff -Naur a/src/platform_terminus.cpp b/src/platform_terminus.cpp
--- a/src/platform_terminus.cpp	2024-09-13 12:59:32.154507868 +0800
+++ b/src/platform_terminus.cpp	2024-09-13 13:02:13.116489326 +0800
@@ -22,8 +22,9 @@
 namespace platform
 {
 
-PlatformTerminus::PlatformTerminus(boost::asio::yield_context yield,
+PlatformTerminus::PlatformTerminus(boost::asio::yield_context yield, const mctpw_eid_t eid,
                                    const pldm_tid_t tid) :
+    _eid(eid),	
     _tid(tid)
 {
     if (!initPDRs(yield))
@@ -46,7 +47,7 @@
         if (auto pdr = pdrManager->getNumericSensorPDR(sensorID))
         {
             std::unique_ptr<NumericSensorHandler> numericSensorHandler =
-                std::make_unique<NumericSensorHandler>(_tid, sensorID,
+                std::make_unique<NumericSensorHandler>(_tid, _eid, sensorID,
                                                        sensorName, *pdr);
             if (!numericSensorHandler->sensorHandlerInit(yield))
             {
@@ -66,7 +67,7 @@
             try
             {
                 stateSensorHandler = std::make_unique<StateSensorHandler>(
-                    _tid, sensorID, sensorName, *pdr);
+                    _tid, _eid, sensorID, sensorName, *pdr);
             }
             catch (const std::exception& e)
             {
@@ -101,7 +102,7 @@
         if (auto pdr = pdrManager->getNumericEffecterPDR(effecterID))
         {
             std::unique_ptr<NumericEffecterHandler> numericEffecterHandler =
-                std::make_unique<NumericEffecterHandler>(_tid, effecterID,
+                std::make_unique<NumericEffecterHandler>(_tid, _eid, effecterID,
                                                          effecterName, *pdr);
             if (!numericEffecterHandler->effecterHandlerInit(yield))
             {
@@ -119,7 +120,7 @@
         if (auto pdr = pdrManager->getStateEffecterPDR(effecterID))
         {
             std::unique_ptr<StateEffecterHandler> stateEffecterHandler =
-                std::make_unique<StateEffecterHandler>(_tid, effecterID,
+                std::make_unique<StateEffecterHandler>(_tid, _eid, effecterID,
                                                        effecterName, pdr);
             if (!stateEffecterHandler->effecterHandlerInit(yield))
             {
@@ -137,7 +138,7 @@
 
 bool PlatformTerminus::initPDRs(boost::asio::yield_context yield)
 {
-    pdrManager = std::make_unique<PDRManager>(_tid);
+    pdrManager = std::make_unique<PDRManager>(_eid, _tid);
     if (!pdrManager->pdrManagerInit(yield))
     {
         phosphor::logging::log<phosphor::logging::level::ERR>(
@@ -151,4 +152,4 @@
     return true;
 }
 } // namespace platform
-} // namespace pldm
\ No newline at end of file
+} // namespace pldm
diff -Naur a/src/pldmd.cpp b/src/pldmd.cpp
--- a/src/pldmd.cpp	2024-09-13 12:59:32.154507868 +0800
+++ b/src/pldmd.cpp	2024-09-13 13:02:13.116489326 +0800
@@ -44,6 +44,7 @@
 static constexpr uint8_t mctpTypePLDM = 1;
 
 bool debug = false;
+int fd = -1;
 
 namespace pldm
 {
@@ -559,19 +560,19 @@
     };
 
     if (isSupported(PLDM_PLATFORM) &&
-        !pldm::platform::platformInit(yield, assignedTID, {}))
+        !pldm::platform::platformInit(yield, eid, assignedTID, {}))
     {
         phosphor::logging::log<phosphor::logging::level::ERR>(
             "PLDM platform init failed",
             phosphor::logging::entry("TID=%d", assignedTID));
     }
-    if (isSupported(PLDM_FRU) && !pldm::fru::fruInit(yield, assignedTID))
+    if (isSupported(PLDM_FRU) && !pldm::fru::fruInit(yield, assignedTID, eid))
     {
         phosphor::logging::log<phosphor::logging::level::ERR>(
             "PLDM fru init failed",
             phosphor::logging::entry("TID=%d", assignedTID));
     }
-    if (isSupported(PLDM_FWUP) && !pldm::fwu::fwuInit(yield, assignedTID))
+    if (isSupported(PLDM_FWUP) && !pldm::fwu::fwuInit(yield, assignedTID, eid))
     {
         phosphor::logging::log<phosphor::logging::level::ERR>(
             "PLDM firmware update init failed",
@@ -710,6 +711,15 @@
 
     enableDebug();
 
+	// Open connection to MCTP socket
+	fd = pldm_open();
+	if (-1 == fd)
+	{
+			std::cout << "Failed to init mctp"
+					<< "\n";
+			return -1;
+	}
+
 	std::vector<int> pldm_eid;
 	std::set<pldm::dbus::Service> mctpCtrlServices;
 	pldm::MctpInfos mctpInfos;
diff -Naur a/src/state_effecter_handler.cpp b/src/state_effecter_handler.cpp
--- a/src/state_effecter_handler.cpp	2024-09-13 12:59:32.154507868 +0800
+++ b/src/state_effecter_handler.cpp	2024-09-13 13:02:13.116489326 +0800
@@ -30,9 +30,9 @@
 constexpr const uint8_t transitionIntervalSec = 3;
 
 StateEffecterHandler::StateEffecterHandler(
-    const pldm_tid_t tid, const EffecterID effecterID, const std::string& name,
+    const pldm_tid_t tid, const mctpw_eid_t eid, const EffecterID effecterID, const std::string& name,
     const std::shared_ptr<StateEffecterPDR>& pdr) :
-    _tid(tid),
+    _tid(tid), _eid(eid),
     _effecterID(effecterID), _name(name), _pdr(pdr)
 {
     if (_pdr->possibleStates.empty())
@@ -178,6 +178,7 @@
 
 bool StateEffecterHandler::enableStateEffecter(boost::asio::yield_context yield)
 {
+	(void)yield;
     uint8_t effecterOpState;
     switch (_pdr->stateEffecterData.effecter_init)
     {
@@ -221,6 +222,7 @@
         return false;
     }
 
+#if 0
     std::vector<uint8_t> resp;
     if (!sendReceivePldmMessage(yield, _tid, commandTimeout, commandRetryCount,
                                 req, resp))
@@ -231,11 +233,22 @@
             phosphor::logging::entry("TID=%d", _tid));
         return false;
     }
-
+#endif
+	uint8_t* responseMsg = nullptr;
+	size_t responseMsgSize{};
+	rc = pldm_send_recv(_eid, fd, req.data(), req.size(),
+					&responseMsg, &responseMsgSize);
+	if (0 > rc)
+	{
+			std::cout << "Failed to send message/receive response. RC = " << rc
+					<< ", errno = " << errno << "\n";
+			return -1;
+	}
+	pldm_msg* response = reinterpret_cast<pldm_msg*>(responseMsg);
     uint8_t completionCode;
-    auto rspMsg = reinterpret_cast<pldm_msg*>(resp.data());
+//    auto rspMsg = reinterpret_cast<pldm_msg*>(resp.data());
 
-    rc = decode_cc_only_resp(rspMsg, resp.size() - pldmMsgHdrSize,
+    rc = decode_cc_only_resp(response, responseMsgSize - pldmMsgHdrSize,
                              &completionCode);
     if (!validatePLDMRespDecode(_tid, rc, completionCode,
                                 "SetStateEffecterEnable"))
@@ -341,6 +354,7 @@
 bool StateEffecterHandler::getStateEffecterStates(
     boost::asio::yield_context yield)
 {
+	(void)yield;
     int rc;
     std::vector<uint8_t> req(pldmMsgHdrSize +
                              sizeof(pldm_get_state_effecter_states_req));
@@ -352,7 +366,7 @@
     {
         return false;
     }
-
+#if 0
     std::vector<uint8_t> resp;
     if (!sendReceivePldmMessage(yield, _tid, commandTimeout, commandRetryCount,
                                 req, resp))
@@ -363,17 +377,28 @@
             phosphor::logging::entry("TID=%d", _tid));
         return false;
     }
-
+#endif
+	uint8_t* responseMsg = nullptr;
+	size_t responseMsgSize{};
+	rc = pldm_send_recv(_eid, fd, req.data(), req.size(),
+					&responseMsg, &responseMsgSize);
+	if (0 > rc)
+	{
+			std::cout << "Failed to send message/receive response. RC = " << rc
+					<< ", errno = " << errno << "\n";
+			return -1;
+	}
+	pldm_msg* response = reinterpret_cast<pldm_msg*>(responseMsg);
     uint8_t completionCode;
     // Pass compositeEffecterCount as 1 to indicate that only one effecter
     // instance is supported
     uint8_t compositeEffecterCount = PLDM_COMPOSITE_EFFECTER_COUNT_MIN;
     std::array<get_effecter_state_field, PLDM_COMPOSITE_EFFECTER_COUNT_MAX>
         stateField{};
-    auto rspMsg = reinterpret_cast<pldm_msg*>(resp.data());
+//    auto rspMsg = reinterpret_cast<pldm_msg*>(resp.data());
 
     rc = decode_get_state_effecter_states_resp(
-        rspMsg, resp.size() - pldmMsgHdrSize, &completionCode,
+        response, responseMsgSize - pldmMsgHdrSize, &completionCode,
         &compositeEffecterCount, stateField.data());
     if (!validatePLDMRespDecode(_tid, rc, completionCode,
                                 "GetStateEffecterStates"))
@@ -425,6 +450,7 @@
 bool StateEffecterHandler::setEffecter(boost::asio::yield_context yield,
                                        const uint8_t state)
 {
+	(void)yield;
     int rc;
 
     // Composite effecters not spported
@@ -441,7 +467,7 @@
     {
         return false;
     }
-
+#if 0
     std::vector<uint8_t> resp;
     if (!sendReceivePldmMessage(yield, _tid, commandTimeout, commandRetryCount,
                                 req, resp))
@@ -452,11 +478,22 @@
             phosphor::logging::entry("TID=%d", _tid));
         return false;
     }
-
+#endif
+	uint8_t* responseMsg = nullptr;
+	size_t responseMsgSize{};
+	rc = pldm_send_recv(_eid, fd, req.data(), req.size(),
+					&responseMsg, &responseMsgSize);
+	if (0 > rc)
+	{
+			std::cout << "Failed to send message/receive response. RC = " << rc
+					<< ", errno = " << errno << "\n";
+			return -1;
+	}
+	pldm_msg* response = reinterpret_cast<pldm_msg*>(responseMsg);
     uint8_t completionCode;
-    auto rspMsg = reinterpret_cast<pldm_msg*>(resp.data());
+    //auto rspMsg = reinterpret_cast<pldm_msg*>(resp.data());
 
-    rc = decode_cc_only_resp(rspMsg, resp.size() - pldmMsgHdrSize,
+    rc = decode_cc_only_resp(response, responseMsgSize - pldmMsgHdrSize,
                              &completionCode);
     if (!validatePLDMRespDecode(_tid, rc, completionCode,
                                 "SetStateEffecterStates"))
diff -Naur a/src/state_sensor_handler.cpp b/src/state_sensor_handler.cpp
--- a/src/state_sensor_handler.cpp	2024-09-13 12:59:32.154507868 +0800
+++ b/src/state_sensor_handler.cpp	2024-09-13 13:02:13.116489326 +0800
@@ -30,9 +30,9 @@
 constexpr const size_t errorThreshold = 3;
 
 StateSensorHandler::StateSensorHandler(
-    const pldm_tid_t tid, const SensorID sensorID, const std::string& name,
+    const pldm_tid_t tid, const mctpw_eid_t eid, const SensorID sensorID, const std::string& name,
     const std::shared_ptr<StateSensorPDR>& pdr) :
-    _tid(tid),
+    _tid(tid), _eid(eid),
     _sensorID(sensorID), _name(name), _pdr(pdr)
 {
     if (_pdr->possibleStates.empty())
@@ -336,20 +336,37 @@
     }
 
     std::vector<uint8_t> resp;
+    uint8_t* responseMsg = nullptr;
+    size_t responseMsgSize{};
+
     if (!sendReceivePldmMessage(yield, _tid, commandTimeout, commandRetryCount,
                                 req, resp))
     {
+#if 0
         phosphor::logging::log<phosphor::logging::level::ERR>(
             "Failed to send or receive SetStateSensorEnables request",
             phosphor::logging::entry("SENSOR_ID=0x%0X", _sensorID),
             phosphor::logging::entry("TID=%d", _tid));
-        return false;
+    //    return false;
+#endif
     }
 
-    uint8_t completionCode;
-    auto rspMsg = reinterpret_cast<pldm_msg*>(resp.data());
 
-    rc = decode_cc_only_resp(rspMsg, resp.size() - pldmMsgHdrSize,
+	// Send PLDM request msg and wait for response
+	rc = pldm_send_recv(_eid, fd, req.data(), req.size(),
+					&responseMsg, &responseMsgSize);
+	if (0 > rc)
+	{
+		std::cout << "Failed to send message/receive response. RC = " << rc
+					<< ", errno = " << errno << "\n";
+		return false;
+	}
+	pldm_msg* response = reinterpret_cast<pldm_msg*>(responseMsg);
+
+
+	uint8_t completionCode;
+
+    rc = decode_cc_only_resp(response, responseMsgSize - pldmMsgHdrSize,
                              &completionCode);
     if (!validatePLDMRespDecode(_tid, rc, completionCode,
                                 "SetStateSensorEnables"))
@@ -383,15 +400,31 @@
     }
 
     std::vector<uint8_t> resp;
+    uint8_t* responseMsg = nullptr;
+    size_t responseMsgSize{};
     if (!sendReceivePldmMessage(yield, _tid, commandTimeout, commandRetryCount,
                                 req, resp))
     {
+#if 0
         phosphor::logging::log<phosphor::logging::level::ERR>(
             "Failed to send or receive GetStateSensorReadings request",
             phosphor::logging::entry("SENSOR_ID=0x%0X", _sensorID),
             phosphor::logging::entry("TID=%d", _tid));
-        return false;
+    //    return false;
+#endif
+    }
+
+
+    // Send PLDM request msg and wait for response
+    rc = pldm_send_recv(_eid, fd, req.data(), req.size(),
+		    &responseMsg, &responseMsgSize);
+    if (0 > rc)
+    {
+	    std::cout << "Failed to send message/receive response. RC = " << rc
+		    << ", errno = " << errno << "\n";
+	    return false;
     }
+    pldm_msg* response = reinterpret_cast<pldm_msg*>(responseMsg);
 
     uint8_t completionCode;
     // Pass compositeSensorCount as 1 to indicate that only one sensor instance
@@ -399,10 +432,9 @@
     uint8_t compositeSensorCount = 1;
     constexpr size_t maxCompositeSensorCount = 0x08;
     std::array<get_sensor_state_field, maxCompositeSensorCount> stateField{};
-    auto rspMsg = reinterpret_cast<pldm_msg*>(resp.data());
 
     rc = decode_get_state_sensor_readings_resp(
-        rspMsg, resp.size() - pldmMsgHdrSize, &completionCode,
+        response, responseMsgSize - pldmMsgHdrSize, &completionCode,
         &compositeSensorCount, stateField.data());
     if (!validatePLDMRespDecode(_tid, rc, completionCode,
                                 "GetStateSensorReadings"))
