diff -Naur a/src/firmware_update.cpp b/src/firmware_update.cpp
--- a/src/firmware_update.cpp	2024-11-06 13:09:09.026663401 +0800
+++ b/src/firmware_update.cpp	2024-11-14 17:01:36.389872289 +0800
@@ -39,6 +39,7 @@
 #include <utility>
 
 const uint8_t MCTP_MSG_TYPE_PLDM = 1;
+uint8_t msg_tag;
 using namespace sdeventplus;
 using namespace sdeventplus::source;
 using sdeventplus::Clock;
@@ -180,15 +181,16 @@
 										fd, static_cast<void*>(requestMsg.data()), peekedLength, 0);
 						if (recvDataLength == peekedLength)
 						{
-								if (MCTP_MSG_TYPE_PLDM != requestMsg[1])
+								if (MCTP_MSG_TYPE_PLDM != requestMsg[2])
 								{
+									std::cout << "Message type mismatch\n" << std::endl;
 								}
 								else
 								{
 										// process message and send response
-										std::vector<uint8_t> dataFromThirdByte(requestMsg.begin() + 2, requestMsg.begin() + recvDataLength);
-
-										processRxMsg(dataFromThirdByte);
+										std::vector<uint8_t> dataFromFourthByte(requestMsg.begin() + 3, requestMsg.begin() + recvDataLength);
+										processRxMsg(dataFromFourthByte);
+										msg_tag = requestMsg[0];
 								}
 						}
 						io.get_event().exit(0);
@@ -236,7 +238,7 @@
 	std::optional<mctpw_eid_t> eid = tidMapper.getMappedEID(currentTid);
 
 	// Send PLDM request msg and wait for response
-	int rc = pldm_send(eid.value(), fd, pldmResp.data(), pldmResp.size());
+	int rc = pldm_send_req(msg_tag, eid.value(), fd, pldmResp.data(), pldmResp.size());
 	if (0 > rc)
 	{
 			std::cout << "Failed to send message/receive response. RC = " << rc
@@ -838,7 +840,7 @@
 	std::optional<mctpw_eid_t> eid = tidMapper.getMappedEID(currentTid);
 
 	// Send PLDM request msg and wait for response
-	int rc = pldm_send(eid.value(), fd, pldmResp.data(), pldmResp.size());
+	int rc = pldm_send_req(msg_tag, eid.value(), fd, pldmResp.data(), pldmResp.size());
 	if (0 > rc)
 	{
 			std::cout << "Failed to send message/receive response. RC = " << rc
@@ -1130,7 +1132,7 @@
 	std::optional<mctpw_eid_t> eid = tidMapper.getMappedEID(currentTid);
 
 	// Send PLDM request msg and wait for response
-	int rc = pldm_send(eid.value(), fd, pldmResp.data(), pldmResp.size());
+	int rc = pldm_send_req(msg_tag, eid.value(), fd, pldmResp.data(), pldmResp.size());
 	if (0 > rc)
 	{
 			std::cout << "Failed to send message/receive response. RC = " << rc
@@ -1224,7 +1226,7 @@
 	std::optional<mctpw_eid_t> eid = tidMapper.getMappedEID(currentTid);
 
 	// Send PLDM request msg and wait for response
-	int rc = pldm_send(eid.value(), fd, pldmResp.data(), pldmResp.size());
+	int rc = pldm_send_req(msg_tag, eid.value(), fd, pldmResp.data(), pldmResp.size());
 	if (0 > rc)
 	{
 			std::cout << "Failed to send message/receive response. RC = " << rc
@@ -1277,11 +1279,13 @@
 {
 
     auto msgReq = reinterpret_cast<const pldm_msg*>(pldmReq.data());
+
     int retVal = decode_apply_complete_req(msgReq, pldmReq.size() - hdrSize,
                                            &applyResult,
                                            &compActivationMethodsModification);
     if (retVal != PLDM_SUCCESS)
     {
+		std::cout << "error code" << retVal << std::endl;
         phosphor::logging::log<phosphor::logging::level::ERR>(
             "ApplyComplete: decode request failed",
             phosphor::logging::entry("TID=%d", currentTid),
@@ -1313,7 +1317,7 @@
 	std::optional<mctpw_eid_t> eid = tidMapper.getMappedEID(currentTid);
 
 	// Send PLDM request msg and wait for response
-	int rc = pldm_send(eid.value(), fd, pldmResp.data(), pldmResp.size());
+	int rc = pldm_send_req(msg_tag, eid.value(), fd, pldmResp.data(), pldmResp.size());
 	if (0 > rc)
 	{
 			std::cout << "Failed to send message/receive response. RC = " << rc
@@ -1507,7 +1511,7 @@
 	std::optional<mctpw_eid_t> eid = tidMapper.getMappedEID(currentTid);
 
 	// Send PLDM request msg and wait for response
-	int rc = pldm_send(eid.value(), fd, pldmResp.data(), pldmResp.size());
+	int rc = pldm_send_req(msg_tag, eid.value(), fd, pldmResp.data(), pldmResp.size());
 	if (0 > rc)
 	{
 			std::cout << "Failed to send message/receive response. RC = " << rc
@@ -1732,9 +1736,9 @@
     }
 
 	std::optional<mctpw_eid_t> eid = tidMapper.getMappedEID(currentTid);
-
+	std::cout << "Message Tag sendPackageData" << static_cast<int>(msg_tag) << std::endl;
 	// Send PLDM request msg and wait for response
-	int rc = pldm_send(eid.value(), fd, pldmResp.data(), pldmResp.size());
+	int rc = pldm_send_req(msg_tag, eid.value(), fd, pldmResp.data(), pldmResp.size());
 	if (0 > rc)
 	{
 			std::cout << "Failed to send message/receive response. RC = " << rc
