diff -Naur a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2025-04-11 11:02:18.633373635 +0800
+++ b/CMakeLists.txt	2025-04-11 11:03:19.256858752 +0800
@@ -3,7 +3,7 @@
 project (pldmd CXX)
 
 option (BUILD_STANDALONE "Use outside of YOCTO depedencies system" OFF)
-option (EXPOSE_BASEBOARD_SENSOR "Expose PLDM sensors in baseboard Redfish Chassis interface" OFF)
+option (EXPOSE_BASEBOARD_SENSOR "Expose PLDM sensors in baseboard Redfish Chassis interface" ON)
 option (EXPOSE_CHASSIS "Expose PLDM device as a standalone chassis in Redfish Chassis interface" OFF)
 
 set (BUILD_SHARED_LIBRARIES OFF)
diff -Naur a/include/state_sensor_handler.hpp b/include/state_sensor_handler.hpp
--- a/include/state_sensor_handler.hpp	2025-04-11 11:02:18.633373635 +0800
+++ b/include/state_sensor_handler.hpp	2025-04-11 11:04:08.709254477 +0800
@@ -122,6 +122,8 @@
         nullptr;
     std::unique_ptr<sdbusplus::asio::dbus_interface> operationalInterface =
         nullptr;
+    std::unique_ptr<sdbusplus::asio::dbus_interface> associationInterface =
+        nullptr;
 };
 
 } // namespace platform
diff -Naur a/src/state_sensor_handler.cpp b/src/state_sensor_handler.cpp
--- a/src/state_sensor_handler.cpp	2025-04-11 11:02:18.636373659 +0800
+++ b/src/state_sensor_handler.cpp	2025-04-11 11:07:18.653774437 +0800
@@ -20,6 +20,7 @@
 #include "sensor.hpp"
 #include "state_set.hpp"
 
+#include "platform_association.hpp"
 #include <phosphor-logging/log.hpp>
 
 namespace pldm
@@ -61,7 +62,10 @@
         path, "xyz.openbmc_project.State.Decorator.Availability");
 
     operationalInterface = addUniqueInterface(
-        path, "xyz.openbmc_project.State.Decorator.OperationalStatus");
+    	path, "xyz.openbmc_project.State.Decorator.OperationalStatus");
+
+    associationInterface = addUniqueInterface(
+	path, "xyz.openbmc_project.Association.Definitions");
 }
 
 void StateSensorHandler::initializeInterface()
@@ -79,6 +83,15 @@
         operationalInterface->register_property("Functional",
                                                 isFuntionalReading);
         operationalInterface->initialize();
+
+	const std::string& associationPath = association::getPath(_tid);
+
+	std::vector<std::tuple<std::string, std::string, std::string>>
+		association = {{"chassis", "all_sensors", associationPath}};
+
+	associationInterface->register_property("Associations", association);
+	associationInterface->initialize();
+
         interfaceInitialized = true;
     }
 }
