From a8a7d708e687eb237d94b2732f570740aa4b9876 Mon Sep 17 00:00:00 2001
From: pavithra <pavithrat@ami.com>
Date: Tue, 10 Mar 2026 13:21:56 +0530
Subject: [PATCH] Fix for Resource leak

---
 src/pam_ipmisave/pam_ipmisave.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/pam_ipmisave/pam_ipmisave.c b/src/pam_ipmisave/pam_ipmisave.c
index f381429..4b7095f 100644
--- a/src/pam_ipmisave/pam_ipmisave.c
+++ b/src/pam_ipmisave/pam_ipmisave.c
@@ -432,6 +432,7 @@ int update_pass_special_file(const pam_handle_t *pamh, const char *keyfilename,
 				goto done;
 			}
 			if (isValidipmiPassFile(opwfilesize, opwmp) < 0) {
+			        free(opwptext);
 				free(opwfilebuff);
 				fclose(opwfile);
 				fclose(pwfile);
-- 
2.34.1

