From a386352e47ba140961a29f7bebf788250f743bcd Mon Sep 17 00:00:00 2001
From: madhankumars <madhankumars@ami.com>
Date: Wed, 4 Sep 2024 16:45:50 +0530
Subject: [PATCH 38/38] DefaultUser_admin_entry_added_ipmipass

---
 meson.build | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 4eb0050..48adac4 100644
--- a/meson.build
+++ b/meson.build
@@ -23,7 +23,8 @@ if (fam == 'aarch64' or fam == 'mips64' or fam == 'ppc64' or
 elif (fam == 'arc' or fam == 'arm' or fam == 'mips' or
          fam == 'ppc' or fam == 'riscv32' or fam == 'sparc' or
          fam == 'x86')
-  pw_file = 'ipmi_pass_32'
+  #pw_file = 'ipmi_pass_32'
+  pw_file = 'ipmi_pass_defaultusers'
 else
   error('unknown target architecture')
 endif
-- 
2.34.1

