diff --git a/common/include/software_update.hpp b/common/include/software_update.hpp
index 2e794af..e416d18 100644
--- a/common/include/software_update.hpp
+++ b/common/include/software_update.hpp
@@ -29,7 +29,7 @@ class SoftwareUpdate :
 
     ~SoftwareUpdate();
 
-    auto method_call(start_update_t su, auto image, auto applyTime)
+    auto method_call(start_update_t, sdbusplus::message::unix_fd image, RequestedApplyTimes applyTime, bool forceUpdate, std::vector<sdbusplus::message::object_path> targets)
         -> sdbusplus::async::task<start_update_t::return_type>;
 
     auto get_property(allowed_apply_times_t aat) const;
diff --git a/common/src/software_manager.cpp b/common/src/software_manager.cpp
index 3ef4a51..47993eb 100644
--- a/common/src/software_manager.cpp
+++ b/common/src/software_manager.cpp
@@ -32,8 +32,7 @@ SoftwareManager::SoftwareManager(sdbusplus::async::context& ctx,
     configIntfAddedMatch(ctx, RulesIntf::interfacesAdded() + matchRuleSender),
     configIntfRemovedMatch(ctx, RulesIntf::interfacesRemoved() + matchRulePath),
     serviceName("xyz.openbmc_project.Software." + serviceNameSuffix),
-    manager(ctx, sdbusplus::client::xyz::openbmc_project::software::Version<>::
-                     namespace_path)
+    manager(ctx, "/xyz/openbmc_project/software")
 {
     debug("requesting dbus name {BUSNAME}", "BUSNAME", serviceName);
 
diff --git a/common/src/software_update.cpp b/common/src/software_update.cpp
index 6ef8951..6483582 100644
--- a/common/src/software_update.cpp
+++ b/common/src/software_update.cpp
@@ -37,8 +37,7 @@ SoftwareUpdate::~SoftwareUpdate()
     emit_removed();
 }
 
-auto SoftwareUpdate::method_call(start_update_t /*unused*/, auto image,
-                                 auto applyTime)
+auto SoftwareUpdate::method_call(start_update_t, sdbusplus::message::unix_fd image, RequestedApplyTimes applyTime, [[maybe_unused]] bool forceUpdate, [[maybe_unused]] std::vector<sdbusplus::message::object_path> targets)
     -> sdbusplus::async::task<start_update_t::return_type>
 {
     debug("Requesting Image update with {FD}", "FD", image.fd);
