From 663ba1c15e96315fef59d5447953cc2c231c8f6e Mon Sep 17 00:00:00 2001
From: madhusrib <madhusrib@ami.com>
Date: Fri, 26 Apr 2024 19:45:46 +0800
Subject: [PATCH 1/4] Added nvidia related yaml files for pldm integration

Signed-off-by: madhusrib <madhusrib@ami.com>

diff --git a/gen/com/meson.build b/gen/com/meson.build
index 3e7a7b6..afca559 100644
--- a/gen/com/meson.build
+++ b/gen/com/meson.build
@@ -4,3 +4,4 @@ subdir('google')
 subdir('ibm')
 subdir('intel')
 subdir('meta')
+subdir('nvidia')
diff --git a/gen/com/nvidia/Common/CompleteReset/meson.build b/gen/com/nvidia/Common/CompleteReset/meson.build
new file mode 100644
index 0000000..2f3aff4
--- /dev/null
+++ b/gen/com/nvidia/Common/CompleteReset/meson.build
@@ -0,0 +1,15 @@
+# Generated file; do not modify.
+generated_sources += custom_target(
+    'com/nvidia/Common/CompleteReset__cpp'.underscorify(),
+    input: [ '../../../../../yaml/com/nvidia/Common/CompleteReset.interface.yaml',  ],
+    output: [ 'common.hpp', 'server.cpp', 'server.hpp', 'aserver.hpp', 'client.hpp',  ],
+    depend_files: sdbusplusplus_depfiles,
+    command: [
+        sdbuspp_gen_meson_prog, '--command', 'cpp',
+        '--output', meson.current_build_dir(),
+        '--tool', sdbusplusplus_prog,
+        '--directory', meson.current_source_dir() / '../../../../../yaml',
+        'com/nvidia/Common/CompleteReset',
+    ],
+)
+
diff --git a/gen/com/nvidia/Common/meson.build b/gen/com/nvidia/Common/meson.build
new file mode 100644
index 0000000..643016d
--- /dev/null
+++ b/gen/com/nvidia/Common/meson.build
@@ -0,0 +1,16 @@
+# Generated file; do not modify.
+subdir('CompleteReset')
+generated_others += custom_target(
+    'com/nvidia/Common/CompleteReset__markdown'.underscorify(),
+    input: [ '../../../../yaml/com/nvidia/Common/CompleteReset.interface.yaml',  ],
+    output: [ 'CompleteReset.md' ],
+    depend_files: sdbusplusplus_depfiles,
+    command: [
+        sdbuspp_gen_meson_prog, '--command', 'markdown',
+        '--output', meson.current_build_dir(),
+        '--tool', sdbusplusplus_prog,
+        '--directory', meson.current_source_dir() / '../../../../yaml',
+        'com/nvidia/Common/CompleteReset',
+    ],
+)
+
diff --git a/gen/com/nvidia/ComputeHash/meson.build b/gen/com/nvidia/ComputeHash/meson.build
new file mode 100644
index 0000000..00e4ecb
--- /dev/null
+++ b/gen/com/nvidia/ComputeHash/meson.build
@@ -0,0 +1,15 @@
+# Generated file; do not modify.
+generated_sources += custom_target(
+    'com/nvidia/ComputeHash__cpp'.underscorify(),
+    input: [ '../../../../yaml/com/nvidia/ComputeHash.interface.yaml',  ],
+    output: [ 'common.hpp', 'server.cpp', 'server.hpp', 'aserver.hpp', 'client.hpp',  ],
+    depend_files: sdbusplusplus_depfiles,
+    command: [
+        sdbuspp_gen_meson_prog, '--command', 'cpp',
+        '--output', meson.current_build_dir(),
+        '--tool', sdbusplusplus_prog,
+        '--directory', meson.current_source_dir() / '../../../../yaml',
+        'com/nvidia/ComputeHash',
+    ],
+)
+
diff --git a/gen/com/nvidia/Edpp/meson.build b/gen/com/nvidia/Edpp/meson.build
new file mode 100644
index 0000000..79c8062
--- /dev/null
+++ b/gen/com/nvidia/Edpp/meson.build
@@ -0,0 +1,15 @@
+# Generated file; do not modify.
+generated_sources += custom_target(
+    'com/nvidia/Edpp__cpp'.underscorify(),
+    input: [ '../../../../yaml/com/nvidia/Edpp.interface.yaml',  ],
+    output: [ 'common.hpp', 'server.cpp', 'server.hpp', 'aserver.hpp', 'client.hpp',  ],
+    depend_files: sdbusplusplus_depfiles,
+    command: [
+        sdbuspp_gen_meson_prog, '--command', 'cpp',
+        '--output', meson.current_build_dir(),
+        '--tool', sdbusplusplus_prog,
+        '--directory', meson.current_source_dir() / '../../../../yaml',
+        'com/nvidia/Edpp',
+    ],
+)
+
diff --git a/gen/com/nvidia/GPMMetrics/meson.build b/gen/com/nvidia/GPMMetrics/meson.build
new file mode 100644
index 0000000..74d79df
--- /dev/null
+++ b/gen/com/nvidia/GPMMetrics/meson.build
@@ -0,0 +1,15 @@
+# Generated file; do not modify.
+generated_sources += custom_target(
+    'com/nvidia/GPMMetrics__cpp'.underscorify(),
+    input: [ '../../../../yaml/com/nvidia/GPMMetrics.interface.yaml',  ],
+    output: [ 'common.hpp', 'server.cpp', 'server.hpp', 'aserver.hpp', 'client.hpp',  ],
+    depend_files: sdbusplusplus_depfiles,
+    command: [
+        sdbuspp_gen_meson_prog, '--command', 'cpp',
+        '--output', meson.current_build_dir(),
+        '--tool', sdbusplusplus_prog,
+        '--directory', meson.current_source_dir() / '../../../../yaml',
+        'com/nvidia/GPMMetrics',
+    ],
+)
+
diff --git a/gen/com/nvidia/MemoryPageRetirementCount/meson.build b/gen/com/nvidia/MemoryPageRetirementCount/meson.build
new file mode 100644
index 0000000..6154e33
--- /dev/null
+++ b/gen/com/nvidia/MemoryPageRetirementCount/meson.build
@@ -0,0 +1,15 @@
+# Generated file; do not modify.
+generated_sources += custom_target(
+    'com/nvidia/MemoryPageRetirementCount__cpp'.underscorify(),
+    input: [ '../../../../yaml/com/nvidia/MemoryPageRetirementCount.interface.yaml',  ],
+    output: [ 'common.hpp', 'server.cpp', 'server.hpp', 'aserver.hpp', 'client.hpp',  ],
+    depend_files: sdbusplusplus_depfiles,
+    command: [
+        sdbuspp_gen_meson_prog, '--command', 'cpp',
+        '--output', meson.current_build_dir(),
+        '--tool', sdbusplusplus_prog,
+        '--directory', meson.current_source_dir() / '../../../../yaml',
+        'com/nvidia/MemoryPageRetirementCount',
+    ],
+)
+
diff --git a/gen/com/nvidia/MemorySpareChannel/meson.build b/gen/com/nvidia/MemorySpareChannel/meson.build
new file mode 100644
index 0000000..a5e9954
--- /dev/null
+++ b/gen/com/nvidia/MemorySpareChannel/meson.build
@@ -0,0 +1,15 @@
+# Generated file; do not modify.
+generated_sources += custom_target(
+    'com/nvidia/MemorySpareChannel__cpp'.underscorify(),
+    input: [ '../../../../yaml/com/nvidia/MemorySpareChannel.interface.yaml',  ],
+    output: [ 'common.hpp', 'server.cpp', 'server.hpp', 'aserver.hpp', 'client.hpp',  ],
+    depend_files: sdbusplusplus_depfiles,
+    command: [
+        sdbuspp_gen_meson_prog, '--command', 'cpp',
+        '--output', meson.current_build_dir(),
+        '--tool', sdbusplusplus_prog,
+        '--directory', meson.current_source_dir() / '../../../../yaml',
+        'com/nvidia/MemorySpareChannel',
+    ],
+)
+
diff --git a/gen/com/nvidia/MigMode/meson.build b/gen/com/nvidia/MigMode/meson.build
new file mode 100644
index 0000000..052e3ce
--- /dev/null
+++ b/gen/com/nvidia/MigMode/meson.build
@@ -0,0 +1,15 @@
+# Generated file; do not modify.
+generated_sources += custom_target(
+    'com/nvidia/MigMode__cpp'.underscorify(),
+    input: [ '../../../../yaml/com/nvidia/MigMode.interface.yaml',  ],
+    output: [ 'common.hpp', 'server.cpp', 'server.hpp', 'aserver.hpp', 'client.hpp',  ],
+    depend_files: sdbusplusplus_depfiles,
+    command: [
+        sdbuspp_gen_meson_prog, '--command', 'cpp',
+        '--output', meson.current_build_dir(),
+        '--tool', sdbusplusplus_prog,
+        '--directory', meson.current_source_dir() / '../../../../yaml',
+        'com/nvidia/MigMode',
+    ],
+)
+
diff --git a/gen/com/nvidia/NVLink/MemoryRowRemapping/meson.build b/gen/com/nvidia/NVLink/MemoryRowRemapping/meson.build
new file mode 100644
index 0000000..dafebe5
--- /dev/null
+++ b/gen/com/nvidia/NVLink/MemoryRowRemapping/meson.build
@@ -0,0 +1,15 @@
+# Generated file; do not modify.
+generated_sources += custom_target(
+    'com/nvidia/NVLink/MemoryRowRemapping__cpp'.underscorify(),
+    input: [ '../../../../../yaml/com/nvidia/NVLink/MemoryRowRemapping.interface.yaml',  ],
+    output: [ 'common.hpp', 'server.cpp', 'server.hpp', 'aserver.hpp', 'client.hpp',  ],
+    depend_files: sdbusplusplus_depfiles,
+    command: [
+        sdbuspp_gen_meson_prog, '--command', 'cpp',
+        '--output', meson.current_build_dir(),
+        '--tool', sdbusplusplus_prog,
+        '--directory', meson.current_source_dir() / '../../../../../yaml',
+        'com/nvidia/NVLink/MemoryRowRemapping',
+    ],
+)
+
diff --git a/gen/com/nvidia/NVLink/NVLinkMetrics/meson.build b/gen/com/nvidia/NVLink/NVLinkMetrics/meson.build
new file mode 100644
index 0000000..d9cb5fb
--- /dev/null
+++ b/gen/com/nvidia/NVLink/NVLinkMetrics/meson.build
@@ -0,0 +1,15 @@
+# Generated file; do not modify.
+generated_sources += custom_target(
+    'com/nvidia/NVLink/NVLinkMetrics__cpp'.underscorify(),
+    input: [ '../../../../../yaml/com/nvidia/NVLink/NVLinkMetrics.interface.yaml',  ],
+    output: [ 'common.hpp', 'server.cpp', 'server.hpp', 'aserver.hpp', 'client.hpp',  ],
+    depend_files: sdbusplusplus_depfiles,
+    command: [
+        sdbuspp_gen_meson_prog, '--command', 'cpp',
+        '--output', meson.current_build_dir(),
+        '--tool', sdbusplusplus_prog,
+        '--directory', meson.current_source_dir() / '../../../../../yaml',
+        'com/nvidia/NVLink/NVLinkMetrics',
+    ],
+)
+
diff --git a/gen/com/nvidia/NVLink/NVLinkRefClock/meson.build b/gen/com/nvidia/NVLink/NVLinkRefClock/meson.build
new file mode 100644
index 0000000..c9de14e
--- /dev/null
+++ b/gen/com/nvidia/NVLink/NVLinkRefClock/meson.build
@@ -0,0 +1,15 @@
+# Generated file; do not modify.
+generated_sources += custom_target(
+    'com/nvidia/NVLink/NVLinkRefClock__cpp'.underscorify(),
+    input: [ '../../../../../yaml/com/nvidia/NVLink/NVLinkRefClock.interface.yaml',  ],
+    output: [ 'common.hpp', 'server.cpp', 'server.hpp', 'aserver.hpp', 'client.hpp',  ],
+    depend_files: sdbusplusplus_depfiles,
+    command: [
+        sdbuspp_gen_meson_prog, '--command', 'cpp',
+        '--output', meson.current_build_dir(),
+        '--tool', sdbusplusplus_prog,
+        '--directory', meson.current_source_dir() / '../../../../../yaml',
+        'com/nvidia/NVLink/NVLinkRefClock',
+    ],
+)
+
diff --git a/gen/com/nvidia/NVLink/meson.build b/gen/com/nvidia/NVLink/meson.build
new file mode 100644
index 0000000..18df09e
--- /dev/null
+++ b/gen/com/nvidia/NVLink/meson.build
@@ -0,0 +1,46 @@
+# Generated file; do not modify.
+subdir('MemoryRowRemapping')
+generated_others += custom_target(
+    'com/nvidia/NVLink/MemoryRowRemapping__markdown'.underscorify(),
+    input: [ '../../../../yaml/com/nvidia/NVLink/MemoryRowRemapping.interface.yaml',  ],
+    output: [ 'MemoryRowRemapping.md' ],
+    depend_files: sdbusplusplus_depfiles,
+    command: [
+        sdbuspp_gen_meson_prog, '--command', 'markdown',
+        '--output', meson.current_build_dir(),
+        '--tool', sdbusplusplus_prog,
+        '--directory', meson.current_source_dir() / '../../../../yaml',
+        'com/nvidia/NVLink/MemoryRowRemapping',
+    ],
+)
+
+subdir('NVLinkMetrics')
+generated_others += custom_target(
+    'com/nvidia/NVLink/NVLinkMetrics__markdown'.underscorify(),
+    input: [ '../../../../yaml/com/nvidia/NVLink/NVLinkMetrics.interface.yaml',  ],
+    output: [ 'NVLinkMetrics.md' ],
+    depend_files: sdbusplusplus_depfiles,
+    command: [
+        sdbuspp_gen_meson_prog, '--command', 'markdown',
+        '--output', meson.current_build_dir(),
+        '--tool', sdbusplusplus_prog,
+        '--directory', meson.current_source_dir() / '../../../../yaml',
+        'com/nvidia/NVLink/NVLinkMetrics',
+    ],
+)
+
+subdir('NVLinkRefClock')
+generated_others += custom_target(
+    'com/nvidia/NVLink/NVLinkRefClock__markdown'.underscorify(),
+    input: [ '../../../../yaml/com/nvidia/NVLink/NVLinkRefClock.interface.yaml',  ],
+    output: [ 'NVLinkRefClock.md' ],
+    depend_files: sdbusplusplus_depfiles,
+    command: [
+        sdbuspp_gen_meson_prog, '--command', 'markdown',
+        '--output', meson.current_build_dir(),
+        '--tool', sdbusplusplus_prog,
+        '--directory', meson.current_source_dir() / '../../../../yaml',
+        'com/nvidia/NVLink/NVLinkRefClock',
+    ],
+)
+
diff --git a/gen/com/nvidia/Protocol/SMBPBI/Raw/meson.build b/gen/com/nvidia/Protocol/SMBPBI/Raw/meson.build
new file mode 100644
index 0000000..34b94c5
--- /dev/null
+++ b/gen/com/nvidia/Protocol/SMBPBI/Raw/meson.build
@@ -0,0 +1,15 @@
+# Generated file; do not modify.
+generated_sources += custom_target(
+    'com/nvidia/Protocol/SMBPBI/Raw__cpp'.underscorify(),
+    input: [ '../../../../../../yaml/com/nvidia/Protocol/SMBPBI/Raw.interface.yaml',  ],
+    output: [ 'common.hpp', 'server.cpp', 'server.hpp', 'aserver.hpp', 'client.hpp',  ],
+    depend_files: sdbusplusplus_depfiles,
+    command: [
+        sdbuspp_gen_meson_prog, '--command', 'cpp',
+        '--output', meson.current_build_dir(),
+        '--tool', sdbusplusplus_prog,
+        '--directory', meson.current_source_dir() / '../../../../../../yaml',
+        'com/nvidia/Protocol/SMBPBI/Raw',
+    ],
+)
+
diff --git a/gen/com/nvidia/Protocol/SMBPBI/meson.build b/gen/com/nvidia/Protocol/SMBPBI/meson.build
new file mode 100644
index 0000000..2cc655f
--- /dev/null
+++ b/gen/com/nvidia/Protocol/SMBPBI/meson.build
@@ -0,0 +1,16 @@
+# Generated file; do not modify.
+subdir('Raw')
+generated_others += custom_target(
+    'com/nvidia/Protocol/SMBPBI/Raw__markdown'.underscorify(),
+    input: [ '../../../../../yaml/com/nvidia/Protocol/SMBPBI/Raw.interface.yaml',  ],
+    output: [ 'Raw.md' ],
+    depend_files: sdbusplusplus_depfiles,
+    command: [
+        sdbuspp_gen_meson_prog, '--command', 'markdown',
+        '--output', meson.current_build_dir(),
+        '--tool', sdbusplusplus_prog,
+        '--directory', meson.current_source_dir() / '../../../../../yaml',
+        'com/nvidia/Protocol/SMBPBI/Raw',
+    ],
+)
+
diff --git a/gen/com/nvidia/Protocol/meson.build b/gen/com/nvidia/Protocol/meson.build
new file mode 100644
index 0000000..694a4b1
--- /dev/null
+++ b/gen/com/nvidia/Protocol/meson.build
@@ -0,0 +1,2 @@
+# Generated file; do not modify.
+subdir('SMBPBI')
diff --git a/gen/com/nvidia/Secureboot/Cec/meson.build b/gen/com/nvidia/Secureboot/Cec/meson.build
new file mode 100644
index 0000000..4805a45
--- /dev/null
+++ b/gen/com/nvidia/Secureboot/Cec/meson.build
@@ -0,0 +1,15 @@
+# Generated file; do not modify.
+generated_sources += custom_target(
+    'com/nvidia/Secureboot/Cec__cpp'.underscorify(),
+    input: [ '../../../../../yaml/com/nvidia/Secureboot/Cec.interface.yaml',  ],
+    output: [ 'common.hpp', 'server.cpp', 'server.hpp', 'aserver.hpp', 'client.hpp',  ],
+    depend_files: sdbusplusplus_depfiles,
+    command: [
+        sdbuspp_gen_meson_prog, '--command', 'cpp',
+        '--output', meson.current_build_dir(),
+        '--tool', sdbusplusplus_prog,
+        '--directory', meson.current_source_dir() / '../../../../../yaml',
+        'com/nvidia/Secureboot/Cec',
+    ],
+)
+
diff --git a/gen/com/nvidia/Secureboot/meson.build b/gen/com/nvidia/Secureboot/meson.build
new file mode 100644
index 0000000..67503c0
--- /dev/null
+++ b/gen/com/nvidia/Secureboot/meson.build
@@ -0,0 +1,16 @@
+# Generated file; do not modify.
+subdir('Cec')
+generated_others += custom_target(
+    'com/nvidia/Secureboot/Cec__markdown'.underscorify(),
+    input: [ '../../../../yaml/com/nvidia/Secureboot/Cec.interface.yaml',  ],
+    output: [ 'Cec.md' ],
+    depend_files: sdbusplusplus_depfiles,
+    command: [
+        sdbuspp_gen_meson_prog, '--command', 'markdown',
+        '--output', meson.current_build_dir(),
+        '--tool', sdbusplusplus_prog,
+        '--directory', meson.current_source_dir() / '../../../../yaml',
+        'com/nvidia/Secureboot/Cec',
+    ],
+)
+
diff --git a/gen/com/nvidia/StaticPowerHint/meson.build b/gen/com/nvidia/StaticPowerHint/meson.build
new file mode 100644
index 0000000..ef27ec8
--- /dev/null
+++ b/gen/com/nvidia/StaticPowerHint/meson.build
@@ -0,0 +1,15 @@
+# Generated file; do not modify.
+generated_sources += custom_target(
+    'com/nvidia/StaticPowerHint__cpp'.underscorify(),
+    input: [ '../../../../yaml/com/nvidia/StaticPowerHint.interface.yaml',  ],
+    output: [ 'common.hpp', 'server.cpp', 'server.hpp', 'aserver.hpp', 'client.hpp',  ],
+    depend_files: sdbusplusplus_depfiles,
+    command: [
+        sdbuspp_gen_meson_prog, '--command', 'cpp',
+        '--output', meson.current_build_dir(),
+        '--tool', sdbusplusplus_prog,
+        '--directory', meson.current_source_dir() / '../../../../yaml',
+        'com/nvidia/StaticPowerHint',
+    ],
+)
+
diff --git a/gen/com/nvidia/meson.build b/gen/com/nvidia/meson.build
new file mode 100644
index 0000000..65612bb
--- /dev/null
+++ b/gen/com/nvidia/meson.build
@@ -0,0 +1,110 @@
+# Generated file; do not modify.
+subdir('Common')
+subdir('ComputeHash')
+generated_others += custom_target(
+    'com/nvidia/ComputeHash__markdown'.underscorify(),
+    input: [ '../../../yaml/com/nvidia/ComputeHash.interface.yaml',  ],
+    output: [ 'ComputeHash.md' ],
+    depend_files: sdbusplusplus_depfiles,
+    command: [
+        sdbuspp_gen_meson_prog, '--command', 'markdown',
+        '--output', meson.current_build_dir(),
+        '--tool', sdbusplusplus_prog,
+        '--directory', meson.current_source_dir() / '../../../yaml',
+        'com/nvidia/ComputeHash',
+    ],
+)
+
+subdir('Edpp')
+generated_others += custom_target(
+    'com/nvidia/Edpp__markdown'.underscorify(),
+    input: [ '../../../yaml/com/nvidia/Edpp.interface.yaml',  ],
+    output: [ 'Edpp.md' ],
+    depend_files: sdbusplusplus_depfiles,
+    command: [
+        sdbuspp_gen_meson_prog, '--command', 'markdown',
+        '--output', meson.current_build_dir(),
+        '--tool', sdbusplusplus_prog,
+        '--directory', meson.current_source_dir() / '../../../yaml',
+        'com/nvidia/Edpp',
+    ],
+)
+
+subdir('GPMMetrics')
+generated_others += custom_target(
+    'com/nvidia/GPMMetrics__markdown'.underscorify(),
+    input: [ '../../../yaml/com/nvidia/GPMMetrics.interface.yaml',  ],
+    output: [ 'GPMMetrics.md' ],
+    depend_files: sdbusplusplus_depfiles,
+    command: [
+        sdbuspp_gen_meson_prog, '--command', 'markdown',
+        '--output', meson.current_build_dir(),
+        '--tool', sdbusplusplus_prog,
+        '--directory', meson.current_source_dir() / '../../../yaml',
+        'com/nvidia/GPMMetrics',
+    ],
+)
+
+subdir('MemoryPageRetirementCount')
+generated_others += custom_target(
+    'com/nvidia/MemoryPageRetirementCount__markdown'.underscorify(),
+    input: [ '../../../yaml/com/nvidia/MemoryPageRetirementCount.interface.yaml',  ],
+    output: [ 'MemoryPageRetirementCount.md' ],
+    depend_files: sdbusplusplus_depfiles,
+    command: [
+        sdbuspp_gen_meson_prog, '--command', 'markdown',
+        '--output', meson.current_build_dir(),
+        '--tool', sdbusplusplus_prog,
+        '--directory', meson.current_source_dir() / '../../../yaml',
+        'com/nvidia/MemoryPageRetirementCount',
+    ],
+)
+
+subdir('MemorySpareChannel')
+generated_others += custom_target(
+    'com/nvidia/MemorySpareChannel__markdown'.underscorify(),
+    input: [ '../../../yaml/com/nvidia/MemorySpareChannel.interface.yaml',  ],
+    output: [ 'MemorySpareChannel.md' ],
+    depend_files: sdbusplusplus_depfiles,
+    command: [
+        sdbuspp_gen_meson_prog, '--command', 'markdown',
+        '--output', meson.current_build_dir(),
+        '--tool', sdbusplusplus_prog,
+        '--directory', meson.current_source_dir() / '../../../yaml',
+        'com/nvidia/MemorySpareChannel',
+    ],
+)
+
+subdir('MigMode')
+generated_others += custom_target(
+    'com/nvidia/MigMode__markdown'.underscorify(),
+    input: [ '../../../yaml/com/nvidia/MigMode.interface.yaml',  ],
+    output: [ 'MigMode.md' ],
+    depend_files: sdbusplusplus_depfiles,
+    command: [
+        sdbuspp_gen_meson_prog, '--command', 'markdown',
+        '--output', meson.current_build_dir(),
+        '--tool', sdbusplusplus_prog,
+        '--directory', meson.current_source_dir() / '../../../yaml',
+        'com/nvidia/MigMode',
+    ],
+)
+
+subdir('NVLink')
+subdir('Protocol')
+subdir('Secureboot')
+subdir('StaticPowerHint')
+generated_others += custom_target(
+    'com/nvidia/StaticPowerHint__markdown'.underscorify(),
+    input: [ '../../../yaml/com/nvidia/StaticPowerHint.interface.yaml',  ],
+    output: [ 'StaticPowerHint.md' ],
+    depend_files: sdbusplusplus_depfiles,
+    command: [
+        sdbuspp_gen_meson_prog, '--command', 'markdown',
+        '--output', meson.current_build_dir(),
+        '--tool', sdbusplusplus_prog,
+        '--directory', meson.current_source_dir() / '../../../yaml',
+        'com/nvidia/StaticPowerHint',
+    ],
+)
+
diff --git a/gen/xyz/openbmc_project/Control/Boot/ClearNonVolatileVariables/meson.build b/gen/xyz/openbmc_project/Control/Boot/ClearNonVolatileVariables/meson.build
new file mode 100644
index 0000000..05ad214
--- /dev/null
+++ b/gen/xyz/openbmc_project/Control/Boot/ClearNonVolatileVariables/meson.build
@@ -0,0 +1,15 @@
+# Generated file; do not modify.
+generated_sources += custom_target(
+    'xyz/openbmc_project/Control/Boot/ClearNonVolatileVariables__cpp'.underscorify(),
+    input: [ '../../../../../../yaml/xyz/openbmc_project/Control/Boot/ClearNonVolatileVariables.interface.yaml',  ],
+    output: [ 'common.hpp', 'server.cpp', 'server.hpp', 'aserver.hpp', 'client.hpp',  ],
+    depend_files: sdbusplusplus_depfiles,
+    command: [
+        sdbuspp_gen_meson_prog, '--command', 'cpp',
+        '--output', meson.current_build_dir(),
+        '--tool', sdbusplusplus_prog,
+        '--directory', meson.current_source_dir() / '../../../../../../yaml',
+        'xyz/openbmc_project/Control/Boot/ClearNonVolatileVariables',
+    ],
+)
+
diff --git a/gen/xyz/openbmc_project/Control/Boot/meson.build b/gen/xyz/openbmc_project/Control/Boot/meson.build
index 6b64b29..f450889 100644
--- a/gen/xyz/openbmc_project/Control/Boot/meson.build
+++ b/gen/xyz/openbmc_project/Control/Boot/meson.build
@@ -1,4 +1,19 @@
 # Generated file; do not modify.
+subdir('ClearNonVolatileVariables')
+generated_others += custom_target(
+    'xyz/openbmc_project/Control/Boot/ClearNonVolatileVariables__markdown'.underscorify(),
+    input: [ '../../../../../yaml/xyz/openbmc_project/Control/Boot/ClearNonVolatileVariables.interface.yaml',  ],
+    output: [ 'ClearNonVolatileVariables.md' ],
+    depend_files: sdbusplusplus_depfiles,
+    command: [
+        sdbuspp_gen_meson_prog, '--command', 'markdown',
+        '--output', meson.current_build_dir(),
+        '--tool', sdbusplusplus_prog,
+        '--directory', meson.current_source_dir() / '../../../../../yaml',
+        'xyz/openbmc_project/Control/Boot/ClearNonVolatileVariables',
+    ],
+)
+
 subdir('Mode')
 generated_others += custom_target(
     'xyz/openbmc_project/Control/Boot/Mode__markdown'.underscorify(),
diff --git a/gen/xyz/openbmc_project/Control/Processor/RemoteDebug/meson.build b/gen/xyz/openbmc_project/Control/Processor/RemoteDebug/meson.build
new file mode 100644
index 0000000..03d1081
--- /dev/null
+++ b/gen/xyz/openbmc_project/Control/Processor/RemoteDebug/meson.build
@@ -0,0 +1,15 @@
+# Generated file; do not modify.
+generated_sources += custom_target(
+    'xyz/openbmc_project/Control/Processor/RemoteDebug__cpp'.underscorify(),
+    input: [ '../../../../../../yaml/xyz/openbmc_project/Control/Processor/RemoteDebug.interface.yaml',  ],
+    output: [ 'common.hpp', 'server.cpp', 'server.hpp', 'aserver.hpp', 'client.hpp',  ],
+    depend_files: sdbusplusplus_depfiles,
+    command: [
+        sdbuspp_gen_meson_prog, '--command', 'cpp',
+        '--output', meson.current_build_dir(),
+        '--tool', sdbusplusplus_prog,
+        '--directory', meson.current_source_dir() / '../../../../../../yaml',
+        'xyz/openbmc_project/Control/Processor/RemoteDebug',
+    ],
+)
+
diff --git a/gen/xyz/openbmc_project/Control/Processor/meson.build b/gen/xyz/openbmc_project/Control/Processor/meson.build
index 6537431..d4f3180 100644
--- a/gen/xyz/openbmc_project/Control/Processor/meson.build
+++ b/gen/xyz/openbmc_project/Control/Processor/meson.build
@@ -29,3 +29,18 @@ generated_others += custom_target(
     ],
 )
 
+subdir('RemoteDebug')
+generated_others += custom_target(
+    'xyz/openbmc_project/Control/Processor/RemoteDebug__markdown'.underscorify(),
+    input: [ '../../../../../yaml/xyz/openbmc_project/Control/Processor/RemoteDebug.interface.yaml',  ],
+    output: [ 'RemoteDebug.md' ],
+    depend_files: sdbusplusplus_depfiles,
+    command: [
+        sdbuspp_gen_meson_prog, '--command', 'markdown',
+        '--output', meson.current_build_dir(),
+        '--tool', sdbusplusplus_prog,
+        '--directory', meson.current_source_dir() / '../../../../../yaml',
+        'xyz/openbmc_project/Control/Processor/RemoteDebug',
+    ],
+)
+
diff --git a/gen/xyz/openbmc_project/Inventory/Decorator/Area/meson.build b/gen/xyz/openbmc_project/Inventory/Decorator/Area/meson.build
new file mode 100644
index 0000000..1e406c8
--- /dev/null
+++ b/gen/xyz/openbmc_project/Inventory/Decorator/Area/meson.build
@@ -0,0 +1,15 @@
+# Generated file; do not modify.
+generated_sources += custom_target(
+    'xyz/openbmc_project/Inventory/Decorator/Area__cpp'.underscorify(),
+    input: [ '../../../../../../yaml/xyz/openbmc_project/Inventory/Decorator/Area.interface.yaml',  ],
+    output: [ 'common.hpp', 'server.cpp', 'server.hpp', 'aserver.hpp', 'client.hpp',  ],
+    depend_files: sdbusplusplus_depfiles,
+    command: [
+        sdbuspp_gen_meson_prog, '--command', 'cpp',
+        '--output', meson.current_build_dir(),
+        '--tool', sdbusplusplus_prog,
+        '--directory', meson.current_source_dir() / '../../../../../../yaml',
+        'xyz/openbmc_project/Inventory/Decorator/Area',
+    ],
+)
+
diff --git a/gen/xyz/openbmc_project/Inventory/Decorator/Instance/meson.build b/gen/xyz/openbmc_project/Inventory/Decorator/Instance/meson.build
new file mode 100644
index 0000000..bde52c9
--- /dev/null
+++ b/gen/xyz/openbmc_project/Inventory/Decorator/Instance/meson.build
@@ -0,0 +1,15 @@
+# Generated file; do not modify.
+generated_sources += custom_target(
+    'xyz/openbmc_project/Inventory/Decorator/Instance__cpp'.underscorify(),
+    input: [ '../../../../../../yaml/xyz/openbmc_project/Inventory/Decorator/Instance.interface.yaml',  ],
+    output: [ 'common.hpp', 'server.cpp', 'server.hpp', 'aserver.hpp', 'client.hpp',  ],
+    depend_files: sdbusplusplus_depfiles,
+    command: [
+        sdbuspp_gen_meson_prog, '--command', 'cpp',
+        '--output', meson.current_build_dir(),
+        '--tool', sdbusplusplus_prog,
+        '--directory', meson.current_source_dir() / '../../../../../../yaml',
+        'xyz/openbmc_project/Inventory/Decorator/Instance',
+    ],
+)
+
diff --git a/gen/xyz/openbmc_project/Inventory/Decorator/Location/meson.build b/gen/xyz/openbmc_project/Inventory/Decorator/Location/meson.build
new file mode 100644
index 0000000..140f737
--- /dev/null
+++ b/gen/xyz/openbmc_project/Inventory/Decorator/Location/meson.build
@@ -0,0 +1,15 @@
+# Generated file; do not modify.
+generated_sources += custom_target(
+    'xyz/openbmc_project/Inventory/Decorator/Location__cpp'.underscorify(),
+    input: [ '../../../../../../yaml/xyz/openbmc_project/Inventory/Decorator/Location.interface.yaml',  ],
+    output: [ 'common.hpp', 'server.cpp', 'server.hpp', 'aserver.hpp', 'client.hpp',  ],
+    depend_files: sdbusplusplus_depfiles,
+    command: [
+        sdbuspp_gen_meson_prog, '--command', 'cpp',
+        '--output', meson.current_build_dir(),
+        '--tool', sdbusplusplus_prog,
+        '--directory', meson.current_source_dir() / '../../../../../../yaml',
+        'xyz/openbmc_project/Inventory/Decorator/Location',
+    ],
+)
+
diff --git a/gen/xyz/openbmc_project/Inventory/Decorator/meson.build b/gen/xyz/openbmc_project/Inventory/Decorator/meson.build
index 7a1f970..c5fa1eb 100644
--- a/gen/xyz/openbmc_project/Inventory/Decorator/meson.build
+++ b/gen/xyz/openbmc_project/Inventory/Decorator/meson.build
@@ -1,4 +1,19 @@
 # Generated file; do not modify.
+subdir('Area')
+generated_others += custom_target(
+    'xyz/openbmc_project/Inventory/Decorator/Area__markdown'.underscorify(),
+    input: [ '../../../../../yaml/xyz/openbmc_project/Inventory/Decorator/Area.interface.yaml',  ],
+    output: [ 'Area.md' ],
+    depend_files: sdbusplusplus_depfiles,
+    command: [
+        sdbuspp_gen_meson_prog, '--command', 'markdown',
+        '--output', meson.current_build_dir(),
+        '--tool', sdbusplusplus_prog,
+        '--directory', meson.current_source_dir() / '../../../../../yaml',
+        'xyz/openbmc_project/Inventory/Decorator/Area',
+    ],
+)
+
 subdir('Asset')
 generated_others += custom_target(
     'xyz/openbmc_project/Inventory/Decorator/Asset__markdown'.underscorify(),
@@ -119,6 +134,36 @@ generated_others += custom_target(
     ],
 )
 
+subdir('Instance')
+generated_others += custom_target(
+    'xyz/openbmc_project/Inventory/Decorator/Instance__markdown'.underscorify(),
+    input: [ '../../../../../yaml/xyz/openbmc_project/Inventory/Decorator/Instance.interface.yaml',  ],
+    output: [ 'Instance.md' ],
+    depend_files: sdbusplusplus_depfiles,
+    command: [
+        sdbuspp_gen_meson_prog, '--command', 'markdown',
+        '--output', meson.current_build_dir(),
+        '--tool', sdbusplusplus_prog,
+        '--directory', meson.current_source_dir() / '../../../../../yaml',
+        'xyz/openbmc_project/Inventory/Decorator/Instance',
+    ],
+)
+
+subdir('Location')
+generated_others += custom_target(
+    'xyz/openbmc_project/Inventory/Decorator/Location__markdown'.underscorify(),
+    input: [ '../../../../../yaml/xyz/openbmc_project/Inventory/Decorator/Location.interface.yaml',  ],
+    output: [ 'Location.md' ],
+    depend_files: sdbusplusplus_depfiles,
+    command: [
+        sdbuspp_gen_meson_prog, '--command', 'markdown',
+        '--output', meson.current_build_dir(),
+        '--tool', sdbusplusplus_prog,
+        '--directory', meson.current_source_dir() / '../../../../../yaml',
+        'xyz/openbmc_project/Inventory/Decorator/Location',
+    ],
+)
+
 subdir('LocationCode')
 generated_others += custom_target(
     'xyz/openbmc_project/Inventory/Decorator/LocationCode__markdown'.underscorify(),
diff --git a/gen/xyz/openbmc_project/Inventory/Item/Endpoint/meson.build b/gen/xyz/openbmc_project/Inventory/Item/Endpoint/meson.build
new file mode 100644
index 0000000..d6012da
--- /dev/null
+++ b/gen/xyz/openbmc_project/Inventory/Item/Endpoint/meson.build
@@ -0,0 +1,15 @@
+# Generated file; do not modify.
+generated_sources += custom_target(
+    'xyz/openbmc_project/Inventory/Item/Endpoint__cpp'.underscorify(),
+    input: [ '../../../../../../yaml/xyz/openbmc_project/Inventory/Item/Endpoint.interface.yaml',  ],
+    output: [ 'common.hpp', 'server.cpp', 'server.hpp', 'aserver.hpp', 'client.hpp',  ],
+    depend_files: sdbusplusplus_depfiles,
+    command: [
+        sdbuspp_gen_meson_prog, '--command', 'cpp',
+        '--output', meson.current_build_dir(),
+        '--tool', sdbusplusplus_prog,
+        '--directory', meson.current_source_dir() / '../../../../../../yaml',
+        'xyz/openbmc_project/Inventory/Item/Endpoint',
+    ],
+)
+
diff --git a/gen/xyz/openbmc_project/Inventory/Item/Port/meson.build b/gen/xyz/openbmc_project/Inventory/Item/Port/meson.build
new file mode 100644
index 0000000..8061fc4
--- /dev/null
+++ b/gen/xyz/openbmc_project/Inventory/Item/Port/meson.build
@@ -0,0 +1,15 @@
+# Generated file; do not modify.
+generated_sources += custom_target(
+    'xyz/openbmc_project/Inventory/Item/Port__cpp'.underscorify(),
+    input: [ '../../../../../../yaml/xyz/openbmc_project/Inventory/Item/Port.interface.yaml',  ],
+    output: [ 'common.hpp', 'server.cpp', 'server.hpp', 'aserver.hpp', 'client.hpp',  ],
+    depend_files: sdbusplusplus_depfiles,
+    command: [
+        sdbuspp_gen_meson_prog, '--command', 'cpp',
+        '--output', meson.current_build_dir(),
+        '--tool', sdbusplusplus_prog,
+        '--directory', meson.current_source_dir() / '../../../../../../yaml',
+        'xyz/openbmc_project/Inventory/Item/Port',
+    ],
+)
+
diff --git a/gen/xyz/openbmc_project/Inventory/Item/SPDMResponder/meson.build b/gen/xyz/openbmc_project/Inventory/Item/SPDMResponder/meson.build
new file mode 100644
index 0000000..5027a52
--- /dev/null
+++ b/gen/xyz/openbmc_project/Inventory/Item/SPDMResponder/meson.build
@@ -0,0 +1,15 @@
+# Generated file; do not modify.
+generated_sources += custom_target(
+    'xyz/openbmc_project/Inventory/Item/SPDMResponder__cpp'.underscorify(),
+    input: [ '../../../../../../yaml/xyz/openbmc_project/Inventory/Item/SPDMResponder.interface.yaml',  ],
+    output: [ 'common.hpp', 'server.cpp', 'server.hpp', 'aserver.hpp', 'client.hpp',  ],
+    depend_files: sdbusplusplus_depfiles,
+    command: [
+        sdbuspp_gen_meson_prog, '--command', 'cpp',
+        '--output', meson.current_build_dir(),
+        '--tool', sdbusplusplus_prog,
+        '--directory', meson.current_source_dir() / '../../../../../../yaml',
+        'xyz/openbmc_project/Inventory/Item/SPDMResponder',
+    ],
+)
+
diff --git a/gen/xyz/openbmc_project/Inventory/Item/meson.build b/gen/xyz/openbmc_project/Inventory/Item/meson.build
index f08aa57..3f067e7 100644
--- a/gen/xyz/openbmc_project/Inventory/Item/meson.build
+++ b/gen/xyz/openbmc_project/Inventory/Item/meson.build
@@ -223,6 +223,21 @@ generated_others += custom_target(
     ],
 )
 
+subdir('Endpoint')
+generated_others += custom_target(
+    'xyz/openbmc_project/Inventory/Item/Endpoint__markdown'.underscorify(),
+    input: [ '../../../../../yaml/xyz/openbmc_project/Inventory/Item/Endpoint.interface.yaml',  ],
+    output: [ 'Endpoint.md' ],
+    depend_files: sdbusplusplus_depfiles,
+    command: [
+        sdbuspp_gen_meson_prog, '--command', 'markdown',
+        '--output', meson.current_build_dir(),
+        '--tool', sdbusplusplus_prog,
+        '--directory', meson.current_source_dir() / '../../../../../yaml',
+        'xyz/openbmc_project/Inventory/Item/Endpoint',
+    ],
+)
+
 subdir('Ethernet')
 generated_others += custom_target(
     'xyz/openbmc_project/Inventory/Item/Ethernet__markdown'.underscorify(),
@@ -373,6 +388,21 @@ generated_others += custom_target(
     ],
 )
 
+subdir('Port')
+generated_others += custom_target(
+    'xyz/openbmc_project/Inventory/Item/Port__markdown'.underscorify(),
+    input: [ '../../../../../yaml/xyz/openbmc_project/Inventory/Item/Port.interface.yaml',  ],
+    output: [ 'Port.md' ],
+    depend_files: sdbusplusplus_depfiles,
+    command: [
+        sdbuspp_gen_meson_prog, '--command', 'markdown',
+        '--output', meson.current_build_dir(),
+        '--tool', sdbusplusplus_prog,
+        '--directory', meson.current_source_dir() / '../../../../../yaml',
+        'xyz/openbmc_project/Inventory/Item/Port',
+    ],
+)
+
 subdir('PowerSupply')
 generated_others += custom_target(
     'xyz/openbmc_project/Inventory/Item/PowerSupply__markdown'.underscorify(),
@@ -403,6 +433,21 @@ generated_others += custom_target(
     ],
 )
 
+subdir('SPDMResponder')
+generated_others += custom_target(
+    'xyz/openbmc_project/Inventory/Item/SPDMResponder__markdown'.underscorify(),
+    input: [ '../../../../../yaml/xyz/openbmc_project/Inventory/Item/SPDMResponder.interface.yaml',  ],
+    output: [ 'SPDMResponder.md' ],
+    depend_files: sdbusplusplus_depfiles,
+    command: [
+        sdbuspp_gen_meson_prog, '--command', 'markdown',
+        '--output', meson.current_build_dir(),
+        '--tool', sdbusplusplus_prog,
+        '--directory', meson.current_source_dir() / '../../../../../yaml',
+        'xyz/openbmc_project/Inventory/Item/SPDMResponder',
+    ],
+)
+
 subdir('Storage')
 generated_others += custom_target(
     'xyz/openbmc_project/Inventory/Item/Storage__markdown'.underscorify(),
diff --git a/gen/xyz/openbmc_project/Sensor/Aggregation/meson.build b/gen/xyz/openbmc_project/Sensor/Aggregation/meson.build
new file mode 100644
index 0000000..0b6d6d9
--- /dev/null
+++ b/gen/xyz/openbmc_project/Sensor/Aggregation/meson.build
@@ -0,0 +1,15 @@
+# Generated file; do not modify.
+generated_sources += custom_target(
+    'xyz/openbmc_project/Sensor/Aggregation__cpp'.underscorify(),
+    input: [ '../../../../../yaml/xyz/openbmc_project/Sensor/Aggregation.interface.yaml',  ],
+    output: [ 'common.hpp', 'server.cpp', 'server.hpp', 'aserver.hpp', 'client.hpp',  ],
+    depend_files: sdbusplusplus_depfiles,
+    command: [
+        sdbuspp_gen_meson_prog, '--command', 'cpp',
+        '--output', meson.current_build_dir(),
+        '--tool', sdbusplusplus_prog,
+        '--directory', meson.current_source_dir() / '../../../../../yaml',
+        'xyz/openbmc_project/Sensor/Aggregation',
+    ],
+)
+
diff --git a/gen/xyz/openbmc_project/Sensor/meson.build b/gen/xyz/openbmc_project/Sensor/meson.build
index 7b9c7a0..59e6d56 100644
--- a/gen/xyz/openbmc_project/Sensor/meson.build
+++ b/gen/xyz/openbmc_project/Sensor/meson.build
@@ -14,6 +14,21 @@ generated_others += custom_target(
     ],
 )
 
+subdir('Aggregation')
+generated_others += custom_target(
+    'xyz/openbmc_project/Sensor/Aggregation__markdown'.underscorify(),
+    input: [ '../../../../yaml/xyz/openbmc_project/Sensor/Aggregation.interface.yaml',  ],
+    output: [ 'Aggregation.md' ],
+    depend_files: sdbusplusplus_depfiles,
+    command: [
+        sdbuspp_gen_meson_prog, '--command', 'markdown',
+        '--output', meson.current_build_dir(),
+        '--tool', sdbusplusplus_prog,
+        '--directory', meson.current_source_dir() / '../../../../yaml',
+        'xyz/openbmc_project/Sensor/Aggregation',
+    ],
+)
+
 subdir('Device')
 generated_others += custom_target(
     'xyz/openbmc_project/Sensor/Device__markdown'.underscorify(),
diff --git a/gen/xyz/openbmc_project/Software/PackageInformation/meson.build b/gen/xyz/openbmc_project/Software/PackageInformation/meson.build
new file mode 100644
index 0000000..9fa7ae5
--- /dev/null
+++ b/gen/xyz/openbmc_project/Software/PackageInformation/meson.build
@@ -0,0 +1,15 @@
+# Generated file; do not modify.
+generated_sources += custom_target(
+    'xyz/openbmc_project/Software/PackageInformation__cpp'.underscorify(),
+    input: [ '../../../../../yaml/xyz/openbmc_project/Software/PackageInformation.interface.yaml',  ],
+    output: [ 'common.hpp', 'server.cpp', 'server.hpp', 'aserver.hpp', 'client.hpp',  ],
+    depend_files: sdbusplusplus_depfiles,
+    command: [
+        sdbuspp_gen_meson_prog, '--command', 'cpp',
+        '--output', meson.current_build_dir(),
+        '--tool', sdbusplusplus_prog,
+        '--directory', meson.current_source_dir() / '../../../../../yaml',
+        'xyz/openbmc_project/Software/PackageInformation',
+    ],
+)
+
diff --git a/gen/xyz/openbmc_project/Software/UpdatePolicy/meson.build b/gen/xyz/openbmc_project/Software/UpdatePolicy/meson.build
new file mode 100644
index 0000000..2db3d06
--- /dev/null
+++ b/gen/xyz/openbmc_project/Software/UpdatePolicy/meson.build
@@ -0,0 +1,15 @@
+# Generated file; do not modify.
+generated_sources += custom_target(
+    'xyz/openbmc_project/Software/UpdatePolicy__cpp'.underscorify(),
+    input: [ '../../../../../yaml/xyz/openbmc_project/Software/UpdatePolicy.interface.yaml',  ],
+    output: [ 'common.hpp', 'server.cpp', 'server.hpp', 'aserver.hpp', 'client.hpp',  ],
+    depend_files: sdbusplusplus_depfiles,
+    command: [
+        sdbuspp_gen_meson_prog, '--command', 'cpp',
+        '--output', meson.current_build_dir(),
+        '--tool', sdbusplusplus_prog,
+        '--directory', meson.current_source_dir() / '../../../../../yaml',
+        'xyz/openbmc_project/Software/UpdatePolicy',
+    ],
+)
+
diff --git a/gen/xyz/openbmc_project/Software/meson.build b/gen/xyz/openbmc_project/Software/meson.build
old mode 100644
new mode 100755
index 04da7b7..5aaf552
--- a/gen/xyz/openbmc_project/Software/meson.build
+++ b/gen/xyz/openbmc_project/Software/meson.build
@@ -119,6 +119,21 @@ generated_others += custom_target(
     ],
 )
 
+subdir('PackageInformation')
+generated_others += custom_target(
+    'xyz/openbmc_project/Software/PackageInformation__markdown'.underscorify(),
+    input: [ '../../../../yaml/xyz/openbmc_project/Software/PackageInformation.interface.yaml',  ],
+    output: [ 'PackageInformation.md' ],
+    depend_files: sdbusplusplus_depfiles,
+    command: [
+        sdbuspp_gen_meson_prog, '--command', 'markdown',
+        '--output', meson.current_build_dir(),
+        '--tool', sdbusplusplus_prog,
+        '--directory', meson.current_source_dir() / '../../../../yaml',
+        'xyz/openbmc_project/Software/PackageInformation',
+    ],
+)
+
 subdir('RedundancyPriority')
 generated_others += custom_target(
     'xyz/openbmc_project/Software/RedundancyPriority__markdown'.underscorify(),
@@ -179,6 +194,21 @@ generated_others += custom_target(
     ],
 )
 
+subdir('UpdatePolicy')
+generated_others += custom_target(
+    'xyz/openbmc_project/Software/UpdatePolicy__markdown'.underscorify(),
+    input: [ '../../../../yaml/xyz/openbmc_project/Software/UpdatePolicy.interface.yaml',  ],
+    output: [ 'UpdatePolicy.md' ],
+    depend_files: sdbusplusplus_depfiles,
+    command: [
+        sdbuspp_gen_meson_prog, '--command', 'markdown',
+        '--output', meson.current_build_dir(),
+        '--tool', sdbusplusplus_prog,
+        '--directory', meson.current_source_dir() / '../../../../yaml',
+        'xyz/openbmc_project/Software/UpdatePolicy',
+    ],
+)
+
 subdir('Version')
 generated_others += custom_target(
     'xyz/openbmc_project/Software/Version__markdown'.underscorify(),
diff --git a/gen/xyz/openbmc_project/State/Decorator/Health/meson.build b/gen/xyz/openbmc_project/State/Decorator/Health/meson.build
new file mode 100644
index 0000000..6dda0a2
--- /dev/null
+++ b/gen/xyz/openbmc_project/State/Decorator/Health/meson.build
@@ -0,0 +1,15 @@
+# Generated file; do not modify.
+generated_sources += custom_target(
+    'xyz/openbmc_project/State/Decorator/Health__cpp'.underscorify(),
+    input: [ '../../../../../../yaml/xyz/openbmc_project/State/Decorator/Health.interface.yaml',  ],
+    output: [ 'common.hpp', 'server.cpp', 'server.hpp', 'aserver.hpp', 'client.hpp',  ],
+    depend_files: sdbusplusplus_depfiles,
+    command: [
+        sdbuspp_gen_meson_prog, '--command', 'cpp',
+        '--output', meson.current_build_dir(),
+        '--tool', sdbusplusplus_prog,
+        '--directory', meson.current_source_dir() / '../../../../../../yaml',
+        'xyz/openbmc_project/State/Decorator/Health',
+    ],
+)
+
diff --git a/gen/xyz/openbmc_project/State/Decorator/Persistence/meson.build b/gen/xyz/openbmc_project/State/Decorator/Persistence/meson.build
new file mode 100644
index 0000000..9f9a836
--- /dev/null
+++ b/gen/xyz/openbmc_project/State/Decorator/Persistence/meson.build
@@ -0,0 +1,15 @@
+# Generated file; do not modify.
+generated_sources += custom_target(
+    'xyz/openbmc_project/State/Decorator/Persistence__cpp'.underscorify(),
+    input: [ '../../../../../../yaml/xyz/openbmc_project/State/Decorator/Persistence.interface.yaml',  ],
+    output: [ 'common.hpp', 'server.cpp', 'server.hpp', 'aserver.hpp', 'client.hpp',  ],
+    depend_files: sdbusplusplus_depfiles,
+    command: [
+        sdbuspp_gen_meson_prog, '--command', 'cpp',
+        '--output', meson.current_build_dir(),
+        '--tool', sdbusplusplus_prog,
+        '--directory', meson.current_source_dir() / '../../../../../../yaml',
+        'xyz/openbmc_project/State/Decorator/Persistence',
+    ],
+)
+
diff --git a/gen/xyz/openbmc_project/State/Decorator/SecureState/meson.build b/gen/xyz/openbmc_project/State/Decorator/SecureState/meson.build
new file mode 100644
index 0000000..ad4c76c
--- /dev/null
+++ b/gen/xyz/openbmc_project/State/Decorator/SecureState/meson.build
@@ -0,0 +1,15 @@
+# Generated file; do not modify.
+generated_sources += custom_target(
+    'xyz/openbmc_project/State/Decorator/SecureState__cpp'.underscorify(),
+    input: [ '../../../../../../yaml/xyz/openbmc_project/State/Decorator/SecureState.interface.yaml',  ],
+    output: [ 'common.hpp', 'server.cpp', 'server.hpp', 'aserver.hpp', 'client.hpp',  ],
+    depend_files: sdbusplusplus_depfiles,
+    command: [
+        sdbuspp_gen_meson_prog, '--command', 'cpp',
+        '--output', meson.current_build_dir(),
+        '--tool', sdbusplusplus_prog,
+        '--directory', meson.current_source_dir() / '../../../../../../yaml',
+        'xyz/openbmc_project/State/Decorator/SecureState',
+    ],
+)
+
diff --git a/gen/xyz/openbmc_project/State/Decorator/meson.build b/gen/xyz/openbmc_project/State/Decorator/meson.build
index 10603d7..d98ad71 100644
--- a/gen/xyz/openbmc_project/State/Decorator/meson.build
+++ b/gen/xyz/openbmc_project/State/Decorator/meson.build
@@ -14,6 +14,21 @@ generated_others += custom_target(
     ],
 )
 
+subdir('Health')
+generated_others += custom_target(
+    'xyz/openbmc_project/State/Decorator/Health__markdown'.underscorify(),
+    input: [ '../../../../../yaml/xyz/openbmc_project/State/Decorator/Health.interface.yaml',  ],
+    output: [ 'Health.md' ],
+    depend_files: sdbusplusplus_depfiles,
+    command: [
+        sdbuspp_gen_meson_prog, '--command', 'markdown',
+        '--output', meson.current_build_dir(),
+        '--tool', sdbusplusplus_prog,
+        '--directory', meson.current_source_dir() / '../../../../../yaml',
+        'xyz/openbmc_project/State/Decorator/Health',
+    ],
+)
+
 subdir('OperationalStatus')
 generated_others += custom_target(
     'xyz/openbmc_project/State/Decorator/OperationalStatus__markdown'.underscorify(),
@@ -29,6 +44,21 @@ generated_others += custom_target(
     ],
 )
 
+subdir('Persistence')
+generated_others += custom_target(
+    'xyz/openbmc_project/State/Decorator/Persistence__markdown'.underscorify(),
+    input: [ '../../../../../yaml/xyz/openbmc_project/State/Decorator/Persistence.interface.yaml',  ],
+    output: [ 'Persistence.md' ],
+    depend_files: sdbusplusplus_depfiles,
+    command: [
+        sdbuspp_gen_meson_prog, '--command', 'markdown',
+        '--output', meson.current_build_dir(),
+        '--tool', sdbusplusplus_prog,
+        '--directory', meson.current_source_dir() / '../../../../../yaml',
+        'xyz/openbmc_project/State/Decorator/Persistence',
+    ],
+)
+
 subdir('PowerState')
 generated_others += custom_target(
     'xyz/openbmc_project/State/Decorator/PowerState__markdown'.underscorify(),
@@ -59,3 +89,18 @@ generated_others += custom_target(
     ],
 )
 
+subdir('SecureState')
+generated_others += custom_target(
+    'xyz/openbmc_project/State/Decorator/SecureState__markdown'.underscorify(),
+    input: [ '../../../../../yaml/xyz/openbmc_project/State/Decorator/SecureState.interface.yaml',  ],
+    output: [ 'SecureState.md' ],
+    depend_files: sdbusplusplus_depfiles,
+    command: [
+        sdbuspp_gen_meson_prog, '--command', 'markdown',
+        '--output', meson.current_build_dir(),
+        '--tool', sdbusplusplus_prog,
+        '--directory', meson.current_source_dir() / '../../../../../yaml',
+        'xyz/openbmc_project/State/Decorator/SecureState',
+    ],
+)
+
diff --git a/gen/xyz/openbmc_project/State/PresenceState/meson.build b/gen/xyz/openbmc_project/State/PresenceState/meson.build
new file mode 100644
index 0000000..6e5721f
--- /dev/null
+++ b/gen/xyz/openbmc_project/State/PresenceState/meson.build
@@ -0,0 +1,15 @@
+# Generated file; do not modify.
+generated_sources += custom_target(
+    'xyz/openbmc_project/State/PresenceState__cpp'.underscorify(),
+    input: [ '../../../../../yaml/xyz/openbmc_project/State/PresenceState.interface.yaml',  ],
+    output: [ 'common.hpp', 'server.cpp', 'server.hpp', 'aserver.hpp', 'client.hpp',  ],
+    depend_files: sdbusplusplus_depfiles,
+    command: [
+        sdbuspp_gen_meson_prog, '--command', 'cpp',
+        '--output', meson.current_build_dir(),
+        '--tool', sdbusplusplus_prog,
+        '--directory', meson.current_source_dir() / '../../../../../yaml',
+        'xyz/openbmc_project/State/PresenceState',
+    ],
+)
+
diff --git a/gen/xyz/openbmc_project/State/ProcessorPerformance/meson.build b/gen/xyz/openbmc_project/State/ProcessorPerformance/meson.build
new file mode 100644
index 0000000..d23e6a6
--- /dev/null
+++ b/gen/xyz/openbmc_project/State/ProcessorPerformance/meson.build
@@ -0,0 +1,15 @@
+# Generated file; do not modify.
+generated_sources += custom_target(
+    'xyz/openbmc_project/State/ProcessorPerformance__cpp'.underscorify(),
+    input: [ '../../../../../yaml/xyz/openbmc_project/State/ProcessorPerformance.interface.yaml',  ],
+    output: [ 'common.hpp', 'server.cpp', 'server.hpp', 'aserver.hpp', 'client.hpp',  ],
+    depend_files: sdbusplusplus_depfiles,
+    command: [
+        sdbuspp_gen_meson_prog, '--command', 'cpp',
+        '--output', meson.current_build_dir(),
+        '--tool', sdbusplusplus_prog,
+        '--directory', meson.current_source_dir() / '../../../../../yaml',
+        'xyz/openbmc_project/State/ProcessorPerformance',
+    ],
+)
+
diff --git a/gen/xyz/openbmc_project/State/meson.build b/gen/xyz/openbmc_project/State/meson.build
index a38d0b9..29005c7 100644
--- a/gen/xyz/openbmc_project/State/meson.build
+++ b/gen/xyz/openbmc_project/State/meson.build
@@ -92,6 +92,36 @@ generated_others += custom_target(
     ],
 )
 
+subdir('PresenceState')
+generated_others += custom_target(
+    'xyz/openbmc_project/State/PresenceState__markdown'.underscorify(),
+    input: [ '../../../../yaml/xyz/openbmc_project/State/PresenceState.interface.yaml',  ],
+    output: [ 'PresenceState.md' ],
+    depend_files: sdbusplusplus_depfiles,
+    command: [
+        sdbuspp_gen_meson_prog, '--command', 'markdown',
+        '--output', meson.current_build_dir(),
+        '--tool', sdbusplusplus_prog,
+        '--directory', meson.current_source_dir() / '../../../../yaml',
+        'xyz/openbmc_project/State/PresenceState',
+    ],
+)
+
+subdir('ProcessorPerformance')
+generated_others += custom_target(
+    'xyz/openbmc_project/State/ProcessorPerformance__markdown'.underscorify(),
+    input: [ '../../../../yaml/xyz/openbmc_project/State/ProcessorPerformance.interface.yaml',  ],
+    output: [ 'ProcessorPerformance.md' ],
+    depend_files: sdbusplusplus_depfiles,
+    command: [
+        sdbuspp_gen_meson_prog, '--command', 'markdown',
+        '--output', meson.current_build_dir(),
+        '--tool', sdbusplusplus_prog,
+        '--directory', meson.current_source_dir() / '../../../../yaml',
+        'xyz/openbmc_project/State/ProcessorPerformance',
+    ],
+)
+
 subdir('ScheduledHostTransition')
 generated_others += custom_target(
     'xyz/openbmc_project/State/ScheduledHostTransition__markdown'.underscorify(),
diff --git a/meson.build b/meson.build
index 100361c..a06b1ce 100644
--- a/meson.build
+++ b/meson.build
@@ -44,6 +44,9 @@ endif
 if get_option('data_com_ami')
     selected_subdirs += 'com/ami'
 endif
+if get_option('data_com_nvidia')
+    selected_subdirs += 'com/nvidia'
+endif
 
 # Install the selected YAML files.
 inst_yaml_dir = get_option('datadir') / 'phosphor-dbus-yaml/yaml'
diff --git a/meson.options b/meson.options
index dba89f3..a3a2385 100644
--- a/meson.options
+++ b/meson.options
@@ -8,3 +8,4 @@ option('data_org_freedesktop', type: 'boolean', value: true)
 option('data_org_open_power', type: 'boolean', value: true)
 option('data_xyz_openbmc_project', type: 'boolean', value: true)
 option('data_com_ami', type: 'boolean', value: true)
+option('data_com_nvidia', type: 'boolean', value: true)
diff --git a/yaml/com/nvidia/Common/CompleteReset.interface.yaml b/yaml/com/nvidia/Common/CompleteReset.interface.yaml
new file mode 100644
index 0000000..41cc4c1
--- /dev/null
+++ b/yaml/com/nvidia/Common/CompleteReset.interface.yaml
@@ -0,0 +1,9 @@
+description: >
+    An interface to reset the target service to its factory state
+    by clearing settings, logs and diagnostic data.
+methods:
+    - name: CompleteReset
+      description: >
+          Performs complete reset of the target service. 
+      errors:
+          - xyz.openbmc_project.Common.Error.InternalFailure
diff --git a/yaml/com/nvidia/ComputeHash.interface.yaml b/yaml/com/nvidia/ComputeHash.interface.yaml
new file mode 100644
index 0000000..06400a7
--- /dev/null
+++ b/yaml/com/nvidia/ComputeHash.interface.yaml
@@ -0,0 +1,32 @@
+description: >
+   Implement to provide hash computation of firmware for non-pldm devices.
+properties:
+    - name: Digest
+      type: string
+      description: >
+           This property shall contain the digest of this firmware using the
+           hash algorithm indicated by Algorithm property.
+      flags:
+          - readonly
+    - name: Algorithm
+      type: string
+      description: >
+           This property shall contain the hash algorithm used to compute the
+           digest of this firmware.
+      flags:
+          - readonly
+methods:
+    - name: GetHash
+      description: >
+        This method shall compute the digest of this firmware using a hash
+        algorithm specified in Algorithm property. Once hash compute is
+        done Digest property will be updated with the hash value.
+      parameters:
+          - name: DeviceId
+            type: uint16
+            description: >
+                Device Id for which hash needs to be computed. For example
+                retimer Id.
+      errors:
+          - xyz.openbmc_project.Common.Error.InternalFailure
+          - xyz.openbmc_project.Common.Error.InvalidArgument
\ No newline at end of file
diff --git a/yaml/com/nvidia/Edpp.interface.yaml b/yaml/com/nvidia/Edpp.interface.yaml
new file mode 100644
index 0000000..4a4c0fc
--- /dev/null
+++ b/yaml/com/nvidia/Edpp.interface.yaml
@@ -0,0 +1,32 @@
+description: >
+   Implement to provide Processor Environment Metrics EDPp scaling.
+properties:
+    - name: AllowableMax
+      type: size
+      description: >
+           Maximum allowable value of the SetPoint property.
+      flags:
+          - readonly
+
+    - name: AllowableMin
+      type: size
+      description: >
+           minimum allowable value of the SetPoint property.
+      flags:
+          - readonly
+
+    - name: SetPoint
+      type: struct[size,boolean]
+      description: >
+            set point control value of EDPp scaling percentage and persistency value across reboots.
+
+methods:
+    - name: Reset
+      description: >
+        Reset Edpp.
+      errors:
+        - xyz.openbmc_project.Common.Error.InternalFailure
+        - xyz.openbmc_project.Common.Error.NotAllowed
+        - xyz.openbmc_project.Common.Error.Timeout
+        - xyz.openbmc_project.Common.Error.Unavailable
+
diff --git a/yaml/com/nvidia/GPMMetrics.interface.yaml b/yaml/com/nvidia/GPMMetrics.interface.yaml
new file mode 100644
index 0000000..40fe61e
--- /dev/null
+++ b/yaml/com/nvidia/GPMMetrics.interface.yaml
@@ -0,0 +1,94 @@
+description: >
+    Implement to provide GPU metric properties.
+properties:
+    - name: GraphicsEngineActivityPercent 
+      type: double
+      description: Graphcis engine activity as a percentage, which is calculated by dividing the active cycles by the total cycles.
+      flags:
+        - readonly
+    - name: SMActivityPercent
+      type: double
+      description: SM activity as a percentage, which is calculated by dividing the active cycles by the total cycles.
+      flags:
+          - readonly
+    - name: SMOccupancyPercent 
+      type: double
+      description: SM occupancy as a percentage, which is calculated by dividing the warps by the maximum warps.
+      flags:
+          - readonly
+    - name: TensorCoreActivityPercent
+      type: double
+      description: Tensor core activity as a percentage, which is calculated by dividing the active cycles by the total cycles.
+      flags:
+        - readonly
+    - name: FP64ActivityPercent
+      type: double
+      description: FP64 activity as a percentage, which is calculated by dividing the active cycles by the total cycles.
+      flags:
+        - readonly
+    - name: FP32ActivityPercent
+      type: double
+      description: FP32 activity as a percentage, which is calculated by dividing the active cycles by the total cycles.
+      flags:
+        - readonly
+    - name: FP16ActivityPercent
+      type: double
+      description: FP16 activity as a percentage, which is calculated by dividing the active cycles by the total cycles.
+      flags:
+        - readonly
+    - name: PCIeRawTxBandwidthGbps
+      type: double
+      description: current bandwidth transferred on PCIe in gigabit per second (Gbit/s). 
+      flags:
+        - readonly
+    - name: PCIeRawRxBandwidthGbps
+      type: double
+      description: current bandwidth received on PCIe in gigabit per second (Gbit/s).
+      flags:
+        - readonly
+    - name: NVDecUtilizationPercent
+      type: double
+      description: NvDec utilization as a percentage, which is calculated by dividing the active cycles by the total cycles.
+      flags:
+        - readonly
+    - name: NVJpgUtilizationPercent
+      type: double
+      description: NvJPG utilization as a percentage, which is calculated by dividing the active cycles by the total cycles.
+      flags:
+        - readonly
+    - name: NVOfaUtilizationPercent
+      type: double
+      description:  NvOFA utilization as a percentage, which is calculated by dividing the active cycles by the total cycles.
+      flags:
+        - readonly
+    - name: IntergerActivityUtilizationPercent
+      type: double
+      description:  Integer activity utilization as a percentage, which is calculated by dividing the active cycles by the total cycles.
+      flags:
+        - readonly
+    - name: DMMAUtilizationPercent
+      type: double
+      description:  DMMA utilization as a percentage, which is calculated by dividing the active cycles by the total cycles.
+      flags:
+        - readonly
+    - name: HMMAUtilizationPercent
+      type: double
+      description:  HMMA utilization as a percentage, which is calculated by dividing the active cycles by the total cycles.
+      flags:
+        - readonly
+    - name: IMMAUtilizationPercent
+      type: double
+      description:  IMMA utilization as a percentage, which is calculated by dividing the active cycles by the total cycles.
+      flags:
+        - readonly
+    - name: NVDecInstanceUtilizationPercent
+      type: array[double]
+      description:  Array of NvDec instance utilization as a percentage, calculated by dividing the active cycles by the total cycles.
+      flags:
+        - readonly
+    - name: NVJpgInstanceUtilizationPercent
+      type: array[double]
+      description: Array of NvJPG instance utilization as a percentage, calculated by dividing the active cycles by the total cycles.
+      flags:
+        - readonly
+
diff --git a/yaml/com/nvidia/MemoryPageRetirementCount.interface.yaml b/yaml/com/nvidia/MemoryPageRetirementCount.interface.yaml
new file mode 100644
index 0000000..b1e9284
--- /dev/null
+++ b/yaml/com/nvidia/MemoryPageRetirementCount.interface.yaml
@@ -0,0 +1,10 @@
+description: >
+   Implement to provide Grace Memory Controller details, the page retirement count.
+
+properties:
+    - name: MemoryPageRetirementCount
+      type: uint32
+      description: >
+            Contains page retirement count.
+      flags:
+          - readonly
diff --git a/yaml/com/nvidia/MemorySpareChannel.interface.yaml b/yaml/com/nvidia/MemorySpareChannel.interface.yaml
new file mode 100644
index 0000000..787920e
--- /dev/null
+++ b/yaml/com/nvidia/MemorySpareChannel.interface.yaml
@@ -0,0 +1,11 @@
+description: >
+   Implement to provide Grace Memory Controller details, the memory spare channel presence state.
+
+properties:
+    - name: MemorySpareChannelPresence
+      type: boolean
+      description: >
+            Memory Spare Channel is presence for this processor.
+      flags:
+          - readonly
+
diff --git a/yaml/com/nvidia/MigMode.interface.yaml b/yaml/com/nvidia/MigMode.interface.yaml
new file mode 100644
index 0000000..37dd79a
--- /dev/null
+++ b/yaml/com/nvidia/MigMode.interface.yaml
@@ -0,0 +1,9 @@
+description: >
+   Implement to provide GPU MIGMode Details The new Multi-Instance GPU (MIG) feature allows GPUs based on the NVIDIA Ampere architecture.
+
+properties:
+    - name: MIGModeEnabled
+      type: boolean
+      description: >
+            multiple-Instance GPU (MIG) mode is enabled for this processor.
+
diff --git a/yaml/com/nvidia/NVLink/MemoryRowRemapping.interface.yaml b/yaml/com/nvidia/NVLink/MemoryRowRemapping.interface.yaml
new file mode 100644
index 0000000..f36face
--- /dev/null
+++ b/yaml/com/nvidia/NVLink/MemoryRowRemapping.interface.yaml
@@ -0,0 +1,28 @@
+description: >
+    Implement to provide memory Row ReMapping Counts.
+properties:
+    - name: ceRowRemappingCount
+      type: uint32
+      description: >
+          Contains correctable Row Mapping Count.
+      flags:
+          - readonly
+    - name: ueRowRemappingCount
+      type: uint32
+      description: >
+          Contains uncorrectable Row Mapping Count.
+      flags:
+          - readonly
+    - name: RowRemappingFailureState
+      type: boolean
+      description: >
+          Represents RowRemapping Failure state.
+      flags:
+          - readonly
+    - name: RowRemappingPendingState
+      type: boolean
+      description: >
+          Represents RowRemapping Pending state.
+      flags:
+          - readonly
+
diff --git a/yaml/com/nvidia/NVLink/NVLinkMetrics.interface.yaml b/yaml/com/nvidia/NVLink/NVLinkMetrics.interface.yaml
new file mode 100644
index 0000000..c0743fd
--- /dev/null
+++ b/yaml/com/nvidia/NVLink/NVLinkMetrics.interface.yaml
@@ -0,0 +1,23 @@
+description: >
+    Implement to provide nvlink Metrics.
+properties:
+    - name: NVLinkRawTxBandwidthGbps
+      type: double
+      description: current bandwidth transferred on NVLink with protocol overhead in gigabit per second (Gbit/s).
+      flags:
+        - readonly
+    - name: NVLinkRawRxBandwidthGbps
+      type: double
+      description: current bandwidth received on NVLink with protocol overhead in gigabit per second (Gbit/s).
+      flags:
+        - readonly
+    - name: NVLinkDataTxBandwidthGbps
+      type: double
+      description: current bandwidth transferred on NVLink without protocol overhead in gigabit per second (Gbit/s).
+      flags:
+        - readonly
+    - name: NVLinkDataRxBandwidthGbps
+      type: double
+      description: current bandwidth received on NVLink without protocol overhead in gigabit per second (Gbit/s).
+      flags:
+        - readonly
diff --git a/yaml/com/nvidia/NVLink/NVLinkRefClock.interface.yaml b/yaml/com/nvidia/NVLink/NVLinkRefClock.interface.yaml
new file mode 100644
index 0000000..cc45b13
--- /dev/null
+++ b/yaml/com/nvidia/NVLink/NVLinkRefClock.interface.yaml
@@ -0,0 +1,9 @@
+description: >
+    Implement to provide nvlink clock reference status.
+properties:
+    - name: NVLinkReferenceClockEnabled
+      type: boolean
+      description: >
+          Contains NVLink reference clock status.
+      flags:
+          - readonly
\ No newline at end of file
diff --git a/yaml/com/nvidia/Protocol/SMBPBI/Raw.interface.yaml b/yaml/com/nvidia/Protocol/SMBPBI/Raw.interface.yaml
new file mode 100644
index 0000000..58b94f2
--- /dev/null
+++ b/yaml/com/nvidia/Protocol/SMBPBI/Raw.interface.yaml
@@ -0,0 +1,67 @@
+description: >
+    Implement to provide D-Bus raw SMBPBI access to the GPU, Nvswitch and Baseboard.
+
+methods:
+    - name: SyncCmd
+      description: >
+          Send raw SMBPBI Sync command(s) to the GPU,Nvswitch and Baseboard.
+      parameters:
+          - name: Device
+            type: string
+            description: >
+                The device to use for the command.
+          - name: DeviceId
+            type: byte
+            description: >
+                The deviceId to use for the command.
+          - name: OpCode
+            type: byte
+            description: >
+                The Opcode to use for the command.
+          - name: Arg1
+            type: byte
+            description: >
+                The argument 1  to use for the command.
+          - name: Arg2
+            type: byte
+            description: >
+                The argument 2  to use for the command.
+          - name: DataIn
+            type: uint32
+            description: >
+                Input Data value for the raw command.
+          - name: ExtDataIn
+            type: uint32
+            description: >
+                Extended Input Data value for the raw command.
+      returns:
+          - name: Results
+            type: struct[int32,uint32,uint32,uint32]
+            description: >
+                rc value and status, dataout, extdataout for the sync command.
+
+    - name: AsyncCmd
+      description: >
+          Send raw SMBPBI Async command(s) to the GPU,Nvswitch and Baseboard.
+      parameters:
+          - name: Device
+            type: string
+            description: >
+                The device to use for the command.
+          - name: DeviceId
+            type: byte
+            description: >
+                The deviceId to use for the command.
+          - name: Arg
+            type: byte
+            description: >
+                The argument to use for the command.
+          - name: DataIn
+            type: array[uint32]
+            description: >
+      returns:
+          - name: Results
+            type: struct[int32,uint32,uint32,array[uint32]]
+            description: >
+                rc value and status, dataout, asyncdataout for the Async command.
+
diff --git a/yaml/com/nvidia/Secureboot/Cec.interface.yaml b/yaml/com/nvidia/Secureboot/Cec.interface.yaml
new file mode 100644
index 0000000..e948244
--- /dev/null
+++ b/yaml/com/nvidia/Secureboot/Cec.interface.yaml
@@ -0,0 +1,8 @@
+description: >
+    Interface which communicates about the interrupt from CEC regarding the secure firmware update.
+properties:
+    - name: Status
+      type: boolean
+      description: >
+          A boolean gets toggled when CEC triggers the interrupt.
+
diff --git a/yaml/com/nvidia/StaticPowerHint.interface.yaml b/yaml/com/nvidia/StaticPowerHint.interface.yaml
new file mode 100644
index 0000000..d6f91c0
--- /dev/null
+++ b/yaml/com/nvidia/StaticPowerHint.interface.yaml
@@ -0,0 +1,119 @@
+description: >
+  Implement to provide Static Power Hint calculator for querying TDP of CPU to setting it ahead of running the application.
+  Static Power Hint = function(workload factor, CPU Clock, Temperature)
+
+properties:
+  - name: PowerEstimate
+    type: double
+    description: >
+      Estimated Power consumption in Watts.
+    flags:
+      - readonly
+  - name: Valid
+    type: boolean
+    description: >
+      Indicating the Power value is valid
+    flags:
+      - readonly
+  - name: CpuClockFrequency
+    type: double
+    description: >
+      The CPU Clock Frequency parameter for this estimation.
+    flags:
+      - readonly
+  - name: MaxCpuClockFrequency
+    type: double
+    description: >
+      Maximum allowable value of CPU Clock Frequency parameter.
+    flags:
+      - readonly
+  - name: MinCpuClockFrequency
+    type: double
+    description: >
+      Minimum allowable value of CPU Clock Frequency parameter.
+    flags:
+      - readonly
+  - name: Temperature
+    type: double
+    description: >
+      The Temperature parameter for this estimation.
+    flags:
+      - readonly
+  - name: MaxTemperature
+    type: double
+    description: >
+      Maximum allowable value of Temperature parameter.
+    flags:
+      - readonly
+  - name: MinTemperature
+    type: double
+    description: >
+      minimum allowable value of Temperature parameter.
+    flags:
+      - readonly
+  - name: WorkloadFactor
+    type: double
+    description: >
+      The WorkloadFactor parameter for this estimation.
+    flags:
+      - readonly
+  - name: MaxWorkloadFactor
+    type: double
+    description: >
+      Maximum allowable value of WorkloadFactor parameter.
+    flags:
+      - readonly
+  - name: MinWorkloadFactor
+    type: double
+    description: >
+      minimum allowable value of WorkloadFactor parameter.
+    flags:
+      - readonly
+  - name: StateOfLastEstimatePower
+    type: enum[self.StateOfEstimatePower]
+    default: "Invalid"
+    description: >
+      The state for last time of EstimatePower method call.
+
+methods:
+  - name: EstimatePower
+    description: >
+      This method estimates the power consumption according the input CPU clock frequency,
+      Workload factor and temperature.
+    parameters:
+      - name: CpuClockFrequency
+        type: double
+        description: >
+          CPU Clock Frequency in Hertz
+      - name: WorkloadFactor
+        type: double
+        description: >
+          Workload Factor
+      - name: Temperature
+        type: double
+        description: >
+          Temperature in degrees Celsius
+    errors:
+      - xyz.openbmc_project.Common.Error.Unavailable
+      - xyz.openbmc_project.Common.Error.InvalidArgument
+
+enumerations:
+  - name: StateOfEstimatePower
+    description: >
+      The state of EstimatePower method
+    values:
+      - name: "Completed"
+        description: >
+          EstimatePower operation is completed. PowerEstimate property is updated with the calculated value.
+      - name: "InProgress"
+        description: >
+          EstimatePower operation is in progress. PowerEstimate property is not ready yet.
+      - name: "Failed"
+        description: >
+          EstimatePower operation failed. Retry the EstimatePower operation to calculate PowerEstimate.
+      - name: "InvalidArgument"
+        description: >
+          EstimatePower operation failed due to invalid argument. Retry the EstimatePower operation with valid arguments
+      - name: "Invalid"
+        description: >
+          EstimatePower operation has not called since service started. PowerEstimate property is invalid.
diff --git a/yaml/xyz/openbmc_project/Control/Boot/ClearNonVolatileVariables.interface.yaml b/yaml/xyz/openbmc_project/Control/Boot/ClearNonVolatileVariables.interface.yaml
new file mode 100644
index 0000000..0824b71
--- /dev/null
+++ b/yaml/xyz/openbmc_project/Control/Boot/ClearNonVolatileVariables.interface.yaml
@@ -0,0 +1,8 @@
+description: >
+    Implement to specify Clear Nonvolatile Variables request
+
+properties:
+    - name: Clear
+      type: boolean
+      description: >
+          Clear Nonvolatile Variables on next boot.
diff --git a/yaml/xyz/openbmc_project/Control/Processor/RemoteDebug.interface.yaml b/yaml/xyz/openbmc_project/Control/Processor/RemoteDebug.interface.yaml
new file mode 100644
index 0000000..edc5988
--- /dev/null
+++ b/yaml/xyz/openbmc_project/Control/Processor/RemoteDebug.interface.yaml
@@ -0,0 +1,121 @@
+description: >
+  Processor Remote debug control
+properties:
+    - name: JtagDebug
+      type: enum[self.DebugState]
+      description: >
+          JTAG_ENABLE, Jtag debug enable
+      flags:
+          - readonly
+
+    - name: DeviceDebug
+      type: enum[self.DebugState]
+      description: >
+          DEVICEEN, Device debug enabled
+      flags:
+          - readonly
+
+    - name: SecurePrivilegeNonInvasiveDebug
+      type: enum[self.DebugState]
+      description: >
+        SPNIDEN, Secure Privilege Non-Invasive Debug Enable
+      flags:
+          - readonly
+
+    - name: SecurePrivilegeInvasiveDebug
+      type: enum[self.DebugState]
+      description: >
+          SPIDEN, Secure Privilege Invasive Debug Enable
+      flags:
+          - readonly
+
+    - name: NonInvasiveDebug
+      type: enum[self.DebugState]
+      description: >
+          NIDEN, non-invasive debug enable
+      flags:
+          - readonly
+
+    - name: InvasiveDebug
+      type: enum[self.DebugState]
+      description: >
+          DBGEN, invasive debug enable
+      flags:
+          - readonly
+
+    - name: Timeout
+      type: uint32
+      default: 30
+      description: >
+          Set a timeout of the remote debug in minutes. It is needed to refresh the timeout to keep the remote debug enabled.
+          Note that a 30 mins of timeout would be set by default unless a new timeout is set by the BMC.
+
+methods:
+    - name: Enable
+      description: >
+          Request to enable the debug policy
+      parameters:
+        - name: DebugPolicy
+          type: array[enum[self.DebugPolicy]]
+      errors:
+        - xyz.openbmc_project.Common.Error.InvalidArgument
+        - xyz.openbmc_project.Common.Error.NotAllowed
+
+    - name: Disable
+      description: >
+          Request to disable debug interface
+      parameters:
+          - name: DebugPolicy
+            type: array[enum[self.DebugPolicy]]
+      errors:
+        - xyz.openbmc_project.Common.Error.InvalidArgument
+        - xyz.openbmc_project.Common.Error.NotAllowed
+
+enumerations:
+    - name: DebugState
+      description: >
+          Debug policy state
+      values:
+          - name: Enabled
+            description: >
+                Always enabled or toggled on
+          - name: Disabled
+            description: >
+                Disconnected but toggleable
+          - name: Offline
+            description: >
+                Offline, cannot be enabled
+          - name: Pending
+            description: >
+                State is pending update
+          - name: Unknown
+            description: >
+                State is unknown.
+
+    - name: DebugPolicy
+      description: >
+          Available Debug Policy
+      values:
+          - name: JtagDebug
+            description: >
+                JTAG_ENABLE, Jtag debug enable
+
+          - name: DeviceDebug
+            description: >
+                DEVICEEN, Device debug enabled
+
+          - name: SecurePrivilegeNonInvasiveDebug
+            description: >
+                SPNIDEN, Secure Privilege Non-Invasive Debug Enable
+
+          - name: SecurePrivilegeInvasiveDebug
+            description: >
+                SPIDEN, Secure Privilege Invasive Debug Enable
+
+          - name: NonInvasiveDebug
+            description: >
+                NIDEN, non-invasive debug enable
+
+          - name: InvasiveDebug
+            description: >
+                DBGEN, invasive debug enable
diff --git a/yaml/xyz/openbmc_project/Inventory/Decorator/Area.interface.yaml b/yaml/xyz/openbmc_project/Inventory/Decorator/Area.interface.yaml
new file mode 100644
index 0000000..1a98068
--- /dev/null
+++ b/yaml/xyz/openbmc_project/Inventory/Decorator/Area.interface.yaml
@@ -0,0 +1,55 @@
+description: >
+    Implement to provide resource area properties. It defines the area or device
+    to which the inventory item belongs too.
+    It can apply to non-inventory item, such as "Exhaust".
+
+properties:
+    - name: PhysicalContext
+      type: enum[self.PhysicalContextType]
+      description: >
+          The area or device to which the resource belongs too.
+
+enumerations:
+    - name: PhysicalContextType
+      description: >
+        Physical context types.
+      values:
+        - name: Back
+          description: >
+            The back of the chassis.
+        - name: Backplane
+          description: >
+            A backplane within the chassis.
+        - name: CPU
+          description: >
+            A Processor (CPU).
+        - name: Fan
+          description: >
+            A fan.
+        - name: Front
+          description: >
+            The front of the chassis.
+        - name: GPU
+          description: >
+            A Graphics Processor (GPU).
+        - name: GPUSubsystem
+          description: >
+            A Graphics Processor (GPU) subsystem.
+        - name: Memory
+          description: >
+            A memory device.
+        - name: NetworkingDevice
+          description: >
+            A networking device.
+        - name: PowerSupply
+          description: >
+            A power supply.
+        - name: StorageDevice
+          description: >
+            A storage device.
+        - name: SystemBoard
+          description: >
+            The system board (PCB).
+        - name: VoltageRegulator
+          description: >
+            A voltage regulator device.
diff --git a/yaml/xyz/openbmc_project/Inventory/Decorator/Instance.interface.yaml b/yaml/xyz/openbmc_project/Inventory/Decorator/Instance.interface.yaml
new file mode 100644
index 0000000..60b24c9
--- /dev/null
+++ b/yaml/xyz/openbmc_project/Inventory/Decorator/Instance.interface.yaml
@@ -0,0 +1,7 @@
+description: >
+    Implement to provide a instance number for an inventory item.
+properties:
+    - name: InstanceNumber
+      type: uint64
+      description: >
+          The instance number under the same inventory item type
diff --git a/yaml/xyz/openbmc_project/Inventory/Decorator/Location.interface.yaml b/yaml/xyz/openbmc_project/Inventory/Decorator/Location.interface.yaml
new file mode 100644
index 0000000..bb170f1
--- /dev/null
+++ b/yaml/xyz/openbmc_project/Inventory/Decorator/Location.interface.yaml
@@ -0,0 +1,30 @@
+description: >
+    Implement to provide physical part location of an inventory item.
+properties:
+    - name: LocationType
+      type: enum[self.LocationTypes]
+      default: Unknown
+      description: >
+          The types of locations for a part within an enclosure.
+
+enumerations:
+    - name: LocationTypes
+      description: >
+          Possible part location types
+      values:
+        - name: Slot
+          description: >
+              It indicates the part is located in a slot.
+        - name: Socket
+          description: >
+              It indicates the part is located in a socket.
+        - name: Backplane
+          description: >
+              It indicates the part is a backplane in an enclosure.
+        - name: Embedded
+          description: >
+              It indicates the part is embedded or otherwise permanently
+              incorporated into a larger part or device.
+        - name: Unknown
+          description: >
+              An unknown chassis type.
diff --git a/yaml/xyz/openbmc_project/Inventory/Item/Endpoint.interface.yaml b/yaml/xyz/openbmc_project/Inventory/Item/Endpoint.interface.yaml
new file mode 100644
index 0000000..549f4af
--- /dev/null
+++ b/yaml/xyz/openbmc_project/Inventory/Item/Endpoint.interface.yaml
@@ -0,0 +1,9 @@
+description: >
+    Implement to provide endpoint attributes.
+    An endpoint represents the properties of an entity that
+    sends or receives protocol-defined messages over a transport.
+    An port item can define inventory endpoint item using association
+    as `connected_port`.
+    An inventory endpoint can be a inventory item like Accelerator
+    using association `entity_link` to an accelerator item.
+
diff --git a/yaml/xyz/openbmc_project/Inventory/Item/Port.interface.yaml b/yaml/xyz/openbmc_project/Inventory/Item/Port.interface.yaml
new file mode 100644
index 0000000..e4ef6e7
--- /dev/null
+++ b/yaml/xyz/openbmc_project/Inventory/Item/Port.interface.yaml
@@ -0,0 +1,223 @@
+description: >
+    This defines a Port to be exposed for system management.
+    It contains properties that describe a port of a switch, controller,
+    chassis, or any other device that could be connected to another entity.
+
+properties:
+    - name: Type
+      type: enum[self.PortType]
+      default: UpstreamPort
+      description: >
+          Indicates the port type.
+
+    - name: Protocol
+      type: enum[self.PortProtocol]
+      default: Ethernet
+      description: >
+          The protocol being sent over this port.
+    
+    - name: TXWidth
+      type: uint16
+      description: >
+          This property shall contain the total number of lanes for transmitting packets.
+      flags:
+          - readonly 
+
+    - name: RXWidth
+      type: uint16
+      description: >
+          This property shall contain the total number of lanes for receiving packets.
+      flags:
+          - readonly
+
+    - name: TXNoProtocolBytes
+      type: uint64
+      description: >
+          Transmitted NVLink data throughput without 
+          protocol overhead.
+      flags:
+          - readonly
+
+    - name: RXNoProtocolBytes
+      type: uint64
+      description: >
+          Received NVLink data throughput without 
+          protocol overhead.
+      flags:
+          - readonly
+
+    - name: ReplayCount
+      type: uint32
+      description: >
+          Nvlink replay error count.
+      flags:
+          - readonly
+    
+    - name: RecoveryCount
+      type: uint32
+      description: >
+          Nvlink recovery error count.
+      flags:
+          - readonly
+
+    - name: FlitCRCCount
+      type: uint32
+      description: >
+          Nvlink flit CRC error count.
+      flags:
+          - readonly
+
+    - name: DataCRCCount
+      type: uint32
+      description: >
+          Nvlink data CRC error count.
+      flags:
+          - readonly
+
+    - name: TrainingError
+      type: uint16
+      description: >
+          This property shall indicate whether the NVLink 
+          training error was detected.
+      flags:
+          - readonly
+
+    - name: RuntimeError
+      type: uint16
+      description: >
+          This property shall indicate whether the NVLink
+           runtime error was detected.
+      flags:
+          - readonly
+
+    - name: ProtocolVersion
+      type: string
+      description: >
+          The protocol version being sent over this port.
+          Example: "4.0" or "1.1"
+
+    - name: CurrentSpeed
+      type: size
+      description: >
+          The current speed of this port in Gbps.
+
+    - name: MaxSpeed
+      type: size
+      description: >
+          The maximum speed of this port in Gbps.
+
+    - name: Width
+      type: size
+      description: >
+          The number of lanes, phys, or other physical transport links
+          that this port contains.
+
+    - name: ActiveWidth
+      type: size
+      description: >
+          The number of active lanes for this interface.
+
+    - name: LinkState
+      type: enum[self.LinkStates]
+      default: Enabled
+      description: >
+          The desired link state for this interface.
+
+    - name: LinkStatus
+      type: enum[self.LinkStatusType]
+      default: LinkUp
+      description: >
+          The desired link status for this interface.
+
+    - name: TXBytes
+      type: size
+      description: >
+          The total number of bytes transmitted on a device.
+
+    - name: RXBytes
+      type: size
+      description: >
+          The total number of bytes received on a device.
+
+enumerations:
+    - name: PortType
+      description: >
+          Possible port types.
+      values:
+        - name: BidirectionalPort
+          description: >
+              This port connects to any type of device.
+        - name: DownstreamPort
+          description: >
+              This port connects to a target device.
+        - name: InterswitchPort
+          description: >
+              This port connects to another switch.
+        - name: ManagementPort
+          description: >
+              This port connects to a switch manager.
+        - name: UnconfiguredPort
+          description: >
+              This port has not yet been configured.
+        - name: UpstreamPort
+          description: >
+              This port connects to a host device.
+
+    - name: PortProtocol
+      description: >
+          Possible protocols supported by port.
+      values:
+        - name: Ethernet
+          description: >
+              An Ethernet.
+        - name: FC
+          description: >
+              A Fibre Channel.
+        - name: NVLink
+          description: >
+              An NVLink.
+        - name: OEM
+          description: >
+              An OEM-specific.
+        - name: PCIe
+          description: >
+              An PCIe.
+
+    - name: LinkStates
+      description: >
+          Possible link states.
+      values:
+        - name: Disabled
+          description: >
+              This link is disabled.
+        - name: Enabled
+          description: >
+              This link is enabled.
+        - name: Error
+          description: >
+              This link is in error state.              
+        - name: Unknown
+          description: >
+              This link state is not known.  
+
+    - name: LinkStatusType
+      description: >
+          Possible link status.
+      values:
+        - name: LinkDown
+          description: >
+              The link on this interface is down.
+        - name: LinkUp
+          description: >
+              This link on this interface is up.
+        - name: NoLink
+          description: >
+              No physical link detected on this interface.
+        - name: Starting
+          description: >
+              This link on this interface is starting. A physical
+              link has been established, but the port is not able
+              to transfer data.
+        - name: Training
+          description: >
+              This physical link on this interface is training.
diff --git a/yaml/xyz/openbmc_project/Inventory/Item/SPDMResponder.interface.yaml b/yaml/xyz/openbmc_project/Inventory/Item/SPDMResponder.interface.yaml
new file mode 100644
index 0000000..80c5919
--- /dev/null
+++ b/yaml/xyz/openbmc_project/Inventory/Item/SPDMResponder.interface.yaml
@@ -0,0 +1,10 @@
+description: >
+    Implement to provide SPDM specific attributes.
+
+associations:
+    - name: inventory
+      description: >
+          Objects that implement this interface can optionally implement the
+          inventory_object association to provide a link to the inventory paths
+          for which this device acts as a root of trust.
+      reverse_name: associated_ROT
diff --git a/yaml/xyz/openbmc_project/Sensor/Aggregation.interface.yaml b/yaml/xyz/openbmc_project/Sensor/Aggregation.interface.yaml
new file mode 100644
index 0000000..816f06c
--- /dev/null
+++ b/yaml/xyz/openbmc_project/Sensor/Aggregation.interface.yaml
@@ -0,0 +1,13 @@
+description: >
+    Implement to provide aggregated metrics for all sensors.
+properties:
+    - name: SensorMetrics
+      type: dict[string, struct[variant[string, int32, int16, int64, uint16, uint32, uint64, double, boolean], uint64, object_path]]
+      description: >
+          A map that stores mapping from sensor name to its value,
+          chassis path and timestamp when the sensor reading got refreshed
+    - name: StaleSensorUpperLimitms
+      type: uint32
+      description: >
+          Time interval in miliseconds to calculate the staleness of a sensor. This metric 
+          will be common to all the sensor for a given platform
diff --git a/yaml/xyz/openbmc_project/Software/PackageInformation.interface.yaml b/yaml/xyz/openbmc_project/Software/PackageInformation.interface.yaml
new file mode 100644
index 0000000..b59b004
--- /dev/null
+++ b/yaml/xyz/openbmc_project/Software/PackageInformation.interface.yaml
@@ -0,0 +1,33 @@
+description: >
+    PackageInformation interface provides information about firmware package
+    properties such as package version, package verification status.
+    To provide other package properties, applications which implements this
+    interface are also expected to implement EpochTime interface for timestamp,
+    ComputeHash interface for package digest information. Additionally
+    implement Delete Interface to remove the firmware package and clear
+    firmware package information properties.
+
+properties:
+    - name: PackageVersion
+      type: string
+      description: >
+          Package version of firmware package. This value should be read from
+          package metadata.
+    - name: VerificationStatus
+      type: enum[self.PackageVerificationStatus]
+      description: >
+          Verification status of firmware package. Should indicate verification
+          status of the package by validating package metadata, signature
+          verification etc.
+enumerations:
+    - name: PackageVerificationStatus
+      description: >
+          The possible values of this property indicate package verification
+          status of firmware package.
+      values:
+          - name: Valid
+            description: >
+                Package is valid.
+          - name: Invalid
+            description: >
+                Package is invalid.
diff --git a/yaml/xyz/openbmc_project/Software/UpdatePolicy.interface.yaml b/yaml/xyz/openbmc_project/Software/UpdatePolicy.interface.yaml
new file mode 100644
index 0000000..5b205d1
--- /dev/null
+++ b/yaml/xyz/openbmc_project/Software/UpdatePolicy.interface.yaml
@@ -0,0 +1,31 @@
+description: >
+    Implement this interface for clients to configure the firmware update
+    policy. The properties override the default policy of the firmware package
+    and is applied during firmware update. The properties are modified only by
+    the client and firmware update has no side effect on the properties. If the
+    policy is no longer applicable, the client is expected to reset the policy
+    to the defaults.
+
+properties:
+    - name: Targets
+      type: array[object_path]
+      description: >
+          An array of D-Bus object paths that are in the
+          /xyz/openbmc_project/software/ namespace that indicate the firmware
+          targets to be updated. The targets for firmware update will be the
+          intersection of the Targets property and the firmware inventory
+          the package is intended to update. If the Targets property is empty,
+          the targets intended to be updated by package will be updated.
+
+    - name: ForceUpdate
+      type: boolean
+      default: false
+      description: >
+          If ForceUpdate is set to true, it is a request to the implementer to
+          bypass update policies, such as allowing a component be downgraded.
+          If ForceUpdate is set to false, the implementer will proceed with the
+          default policies of the system or the firmware package. ForceUpdate
+          policy applies for all the targets to be updated.
+
+
+
diff --git a/yaml/xyz/openbmc_project/State/Decorator/Health.interface.yaml b/yaml/xyz/openbmc_project/State/Decorator/Health.interface.yaml
new file mode 100644
index 0000000..af90023
--- /dev/null
+++ b/yaml/xyz/openbmc_project/State/Decorator/Health.interface.yaml
@@ -0,0 +1,23 @@
+description: >
+    Implement to indicate the health state of the object
+properties:
+    - name: Health
+      type: enum[self.HealthType]
+      default: OK
+      description: >
+        The current health status of the object.
+
+enumerations:
+  - name: HealthType
+    description: >
+      Health type enum.
+    values:
+      - name: OK
+        description: >
+         The state of the object is OK.
+      - name: Warning
+        description: >
+         The state of the object is Warning.
+      - name: Critical
+        description: >
+         The state of the object is Critical.
\ No newline at end of file
diff --git a/yaml/xyz/openbmc_project/State/Decorator/Persistence.interface.yaml b/yaml/xyz/openbmc_project/State/Decorator/Persistence.interface.yaml
new file mode 100644
index 0000000..c17c1ac
--- /dev/null
+++ b/yaml/xyz/openbmc_project/State/Decorator/Persistence.interface.yaml
@@ -0,0 +1,7 @@
+description: >
+    Implement to indicate the persistence of the object
+properties:
+    - name: persistent
+      type: boolean
+      description: >
+        The persistence status of the object.
\ No newline at end of file
diff --git a/yaml/xyz/openbmc_project/State/Decorator/SecureState.interface.yaml b/yaml/xyz/openbmc_project/State/Decorator/SecureState.interface.yaml
new file mode 100644
index 0000000..f0a21a0
--- /dev/null
+++ b/yaml/xyz/openbmc_project/State/Decorator/SecureState.interface.yaml
@@ -0,0 +1,10 @@
+description: >
+    Implement to indicate the secure state of the object. A object in secure
+    state only can be updated by the private key used to create the object.
+properties:
+    - name: secure
+      type: boolean
+      description: >
+        The secure state of the object.
+      flags:
+          - readonly
diff --git a/yaml/xyz/openbmc_project/State/PresenceState.interface.yaml b/yaml/xyz/openbmc_project/State/PresenceState.interface.yaml
new file mode 100644
index 0000000..b8be877
--- /dev/null
+++ b/yaml/xyz/openbmc_project/State/PresenceState.interface.yaml
@@ -0,0 +1,10 @@
+description: >
+   Implement to provide Alert caused by PRSNT state run time change.
+
+properties:
+    - name: Presence
+      type: boolean
+      description: >
+            represents PRSNT state run time change.
+      flags:
+          - readonly
diff --git a/yaml/xyz/openbmc_project/State/ProcessorPerformance.interface.yaml b/yaml/xyz/openbmc_project/State/ProcessorPerformance.interface.yaml
new file mode 100644
index 0000000..7046cd2
--- /dev/null
+++ b/yaml/xyz/openbmc_project/State/ProcessorPerformance.interface.yaml
@@ -0,0 +1,122 @@
+description: >
+    Implement to provide value of performance state of a processor (CPU, GPU, etc).
+
+properties:
+    - name: Value
+      type: enum[self.PerformanceStates]    
+      description: >
+          This is current discreate value of the sensor. 
+
+    - name: ThrottleReason
+      type: array[enum[self.ThrottleReasons]]
+      description: >
+        Throttle reasons for a processor.
+      flags:
+        - readonly
+
+    - name: ThermalTrip
+      type: boolean
+      description: Thermal trip status for processor
+      flags:
+        - readonly
+
+    - name: PowerLimitThrottleDuration
+      type: uint64
+      description: Total duration of throttling in nano seconds caused by a power limit of the processor since reset.
+      flags:
+        - readonly
+    - name: ThermalLimitThrottleDuration
+      type: uint64
+      description: Total duration of throttling in nano seconds caused by a thermal limit of the processor since reset.
+      flags:
+        - readonly
+    - name: PCIeTXBytes
+      type: uint32
+      description: Total number of bytes sent by the PCIe transmitter.
+      flags:
+        - readonly
+    - name: PCIeRXBytes
+      type: uint32
+      description: Total number of bytes received by the PCIe receiver.
+      flags:
+        - readonly
+    - name: HardwareViolationThrottleDuration
+      type: uint64
+      description: Total duration of throttling in nano seconds caused by hardware violation of the processor since reset.
+      flags:
+        - readonly
+    - name: GlobalSoftwareViolationThrottleDuration
+      type: uint64
+      description: Total duration of throttling in nano seconds caused by global software violation of the processor since reset.
+      flags:
+        - readonly
+    - name: AccumulatedGPUContextUtilizationDuration
+      type: uint32
+      description:  Total duration of the accumulated GPU context utilization of the processor since reset in nano seconds.
+      flags:
+        - readonly
+    - name: AccumulatedSMUtilizationDuration
+      type: uint32
+      description:  Total duration of the accumulated SM utilization of the processor since reset in nano seconds.
+      flags:
+        - readonly
+
+enumerations:
+    - name: PerformanceStates
+      description: >
+          Different value types for performance discreate sensor.
+      values:
+          - name: Normal
+            description: >
+                To represent the performance state is normal.
+          - name: Throttled
+            description: >
+                To represent the performance state is throttled.
+          - name: Degraded  
+            description: >
+                To represent the performance state is degraded.  
+          - name: Unknown
+            description: >
+                To represent the performance state as unknown.
+
+    - name: ThrottleReasons
+      description: >
+          Different value types for processor throttling reason.
+      values:
+          - name: Idle
+            description: >
+                Processor idle.
+          - name: UserDefined
+            description: >
+                User defined clocks.
+          - name: AppClock
+            description: >
+                Applicaiton Clock Settings.
+          - name: SWPowerCap
+            description: >
+                Software powercap.
+          - name: HWSlowdown
+            description: >
+                Hardware slowdown.
+          - name: HWThermalSlowdown
+            description: >
+                Hardware thermal slowdown.
+          - name: HWPowerBrakeSlowdown
+            description: >
+                Hardware power brake slowdown.
+          - name: SyncBoost
+            description: >
+                Sync boost.
+          - name: SWThermalSlowdownTavg
+            description: >
+                Software thermal slowdown Tavg.
+          - name: SWThermalSlowdownTLimit
+            description: >
+                Software thermal slowdown Tlimit.
+          - name: DisplayClock
+            description: >
+                Display Clock Settings.
+          - name: None
+            description: >
+                Not Throttled.
+
-- 
2.34.1

