diff -Naur a/gen/xyz/openbmc_project/Inventory/Decorator/meson.build b/gen/xyz/openbmc_project/Inventory/Decorator/meson.build
--- a/gen/xyz/openbmc_project/Inventory/Decorator/meson.build	2024-08-16 12:02:18.628684066 +0800
+++ b/gen/xyz/openbmc_project/Inventory/Decorator/meson.build	2024-08-16 12:07:09.549387827 +0800
@@ -224,6 +224,21 @@
     ],
 )
 
+subdir('PortState')
+generated_others += custom_target(
+    'xyz/openbmc_project/Inventory/Decorator/PortState__markdown'.underscorify(),
+    input: [ '../../../../../yaml/xyz/openbmc_project/Inventory/Decorator/PortState.interface.yaml',  ],
+    output: [ 'PortState.md' ],
+    depend_files: sdbusplusplus_depfiles,
+    command: [
+        sdbuspp_gen_meson_prog, '--command', 'markdown',
+        '--output', meson.current_build_dir(),
+        '--tool', sdbusplusplus_prog,
+        '--directory', meson.current_source_dir() / '../../../../../yaml',
+        'xyz/openbmc_project/Inventory/Decorator/PortState',
+    ],
+)
+
 subdir('Replaceable')
 generated_others += custom_target(
     'xyz/openbmc_project/Inventory/Decorator/Replaceable__markdown'.underscorify(),
diff -Naur a/gen/xyz/openbmc_project/Inventory/Decorator/PortState/meson.build b/gen/xyz/openbmc_project/Inventory/Decorator/PortState/meson.build
--- a/gen/xyz/openbmc_project/Inventory/Decorator/PortState/meson.build	1970-01-01 08:00:00.000000000 +0800
+++ b/gen/xyz/openbmc_project/Inventory/Decorator/PortState/meson.build	2024-08-16 12:08:05.386230228 +0800
@@ -0,0 +1,15 @@
+# Generated file; do not modify.
+generated_sources += custom_target(
+    'xyz/openbmc_project/Inventory/Decorator/PortState__cpp'.underscorify(),
+    input: [ '../../../../../../yaml/xyz/openbmc_project/Inventory/Decorator/PortState.interface.yaml',  ],
+    output: [ 'common.hpp', 'server.cpp', 'server.hpp', 'aserver.hpp', 'client.hpp',  ],
+    depend_files: sdbusplusplus_depfiles,
+    command: [
+        sdbuspp_gen_meson_prog, '--command', 'cpp',
+        '--output', meson.current_build_dir(),
+        '--tool', sdbusplusplus_prog,
+        '--directory', meson.current_source_dir() / '../../../../../../yaml',
+        'xyz/openbmc_project/Inventory/Decorator/PortState',
+    ],
+)
+
