diff -Naur a/gen/com/nvidia/meson.build b/gen/com/nvidia/meson.build
--- a/gen/com/nvidia/meson.build	2025-01-21 16:42:52.000000000 +0800
+++ b/gen/com/nvidia/meson.build	2025-01-21 16:49:38.168669965 +0800
@@ -205,6 +205,7 @@
 subdir('PowerSupply')
 subdir('Protocol')
 subdir('ResetCounters')
+subdir('RoT')
 subdir('SMUtilization')
 generated_others += custom_target(
     'com/nvidia/SMUtilization__markdown'.underscorify(),
diff -Naur a/gen/com/nvidia/RoT/BootStatus/meson.build b/gen/com/nvidia/RoT/BootStatus/meson.build
--- a/gen/com/nvidia/RoT/BootStatus/meson.build	1970-01-01 08:00:00.000000000 +0800
+++ b/gen/com/nvidia/RoT/BootStatus/meson.build	2025-01-21 16:49:37.985662171 +0800
@@ -0,0 +1,15 @@
+# Generated file; do not modify.
+generated_sources += custom_target(
+    'com/nvidia/RoT/BootStatus__cpp'.underscorify(),
+    input: [ '../../../../../yaml/com/nvidia/RoT/BootStatus.interface.yaml',  ],
+    output: [ 'common.hpp', 'server.cpp', 'server.hpp', 'aserver.hpp', 'client.hpp',  ],
+    depend_files: sdbusplusplus_depfiles,
+    command: [
+        sdbuspp_gen_meson_prog, '--command', 'cpp',
+        '--output', meson.current_build_dir(),
+        '--tool', sdbusplusplus_prog,
+        '--directory', meson.current_source_dir() / '../../../../../yaml',
+        'com/nvidia/RoT/BootStatus',
+    ],
+)
+
diff -Naur a/gen/com/nvidia/RoT/meson.build b/gen/com/nvidia/RoT/meson.build
--- a/gen/com/nvidia/RoT/meson.build	1970-01-01 08:00:00.000000000 +0800
+++ b/gen/com/nvidia/RoT/meson.build	2025-01-21 16:49:38.002662895 +0800
@@ -0,0 +1,16 @@
+# Generated file; do not modify.
+subdir('BootStatus')
+generated_others += custom_target(
+    'com/nvidia/RoT/BootStatus__markdown'.underscorify(),
+    input: [ '../../../../yaml/com/nvidia/RoT/BootStatus.interface.yaml',  ],
+    output: [ 'BootStatus.md' ],
+    depend_files: sdbusplusplus_depfiles,
+    command: [
+        sdbuspp_gen_meson_prog, '--command', 'markdown',
+        '--output', meson.current_build_dir(),
+        '--tool', sdbusplusplus_prog,
+        '--directory', meson.current_source_dir() / '../../../../yaml',
+        'com/nvidia/RoT/BootStatus',
+    ],
+)
+
