From b8a8c4673ff9effb4cc60c88d9e06b41dc3d83f6 Mon Sep 17 00:00:00 2001
From: Revathir <revathir@ami.com>
Date: Mon, 29 Jul 2024 15:20:56 +0530
Subject: [PATCH] Fix to get response for all supported lan parameters via
 ipmitool

---
 lib/ipmi_lanp.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/ipmi_lanp.c b/lib/ipmi_lanp.c
index fe0046f..577a8d0 100644
--- a/lib/ipmi_lanp.c
+++ b/lib/ipmi_lanp.c
@@ -237,7 +237,7 @@ get_lan_param_select(struct ipmi_intf *intf, uint8_t chan, int param, int select
 		p->data = NULL;
 		p->data_len = 0;
 		rc = p;
-		/* fall through */
+		return rc;
 	default:
 		/* other completion codes are treated as error */
 		lprintf(LOG_INFO, "Get LAN Parameter '%s' command failed: %s",
