From c169c0ac28a6610f103161e1a9c23e5a93c705ed Mon Sep 17 00:00:00 2001
From: saravananm <saravananm@ami.com>
Date: Tue, 10 Mar 2026 23:49:45 +0530
Subject: [PATCH] Fix boost/asio/io_service.hpp deprecated header

Signed-off-by: saravananm <saravananm@ami.com>
---
 include/error_monitors/base_monitor.hpp | 2 --
 1 file changed, 2 deletions(-)

diff --git a/include/error_monitors/base_monitor.hpp b/include/error_monitors/base_monitor.hpp
index 185bdb3..bc38a97 100644
--- a/include/error_monitors/base_monitor.hpp
+++ b/include/error_monitors/base_monitor.hpp
@@ -16,8 +16,6 @@
 #pragma once
 #include <systemd/sd-journal.h>
 
-#include <boost/asio/io_context.hpp>
-#include <boost/asio/io_service.hpp>
 #include <gpiod.hpp>
 #include <sdbusplus/asio/object_server.hpp>
 #include <xyz/openbmc_project/Logging/Entry/common.hpp>
-- 
2.34.1

