From 294a1c588fc9a561ee7b9ee96bd12cb012192d38 Mon Sep 17 00:00:00 2001
From: krishnar4 <krishnar@ami.com>
Date: Wed, 25 Mar 2026 16:23:07 +0530
Subject: [PATCH] UpGraded the Sdbusplus version

Signed-off-by: krishnar4 <krishnar@ami.com>
---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index db92894..9b03c13 100644
--- a/meson.build
+++ b/meson.build
@@ -6,7 +6,7 @@ project(
   default_options: [
     'b_ndebug=if-release',
     'cpp_rtti=false',
-    'cpp_std=c++20',
+    'cpp_std=c++23',
     'warning_level=3',
     'werror=true',
   ]
-- 
2.34.1

