From 12dfd074a43b5260c510d369800df7dd5b293eb4 Mon Sep 17 00:00:00 2001
From: mickhuang <mickhuang@ami.com>
Date: Wed, 3 Sep 2025 17:02:54 +0800
Subject: [PATCH] Use misc init to train ltpi

---
 board/aspeed/evb_ast2700/evb_ast2700.c | 39 ++++++++++++++++++++++++++
 1 file changed, 39 insertions(+)

diff --git a/board/aspeed/evb_ast2700/evb_ast2700.c b/board/aspeed/evb_ast2700/evb_ast2700.c
index b34aa6e168..73577bef48 100644
--- a/board/aspeed/evb_ast2700/evb_ast2700.c
+++ b/board/aspeed/evb_ast2700/evb_ast2700.c
@@ -3,3 +3,42 @@
  * Copyright (C) ASPEED Technology Inc.
  */
 
+#include <common.h>
+
+/* LTPI */
+#define LTPI_TRAIN_RETRY       10
+#define OP_TIMEOUT_US          "300000"       /* Operational timeout 300ms */
+#define ADVRT_TIMEOUT_US_1_1   "100000"     /* 100ms for LTPI spec 1.1 */
+
+void train_ltpi(int retry)
+{
+	int i=0;
+
+	/* start LTPI with operational and advertise timeouts */
+	if(run_command("ltpi -T " OP_TIMEOUT_US " -t " ADVRT_TIMEOUT_US_1_1, 0) != 0)
+	{
+		for (i=0; i<retry; i++)
+		{
+			if(run_command("ltpi -T " OP_TIMEOUT_US " -t "ADVRT_TIMEOUT_US_1_1 ,0) == 0)
+			{
+				printf("LTPI link configured, proceeding to boot...\n");
+				break;
+			}
+			else
+			{
+				printf("Retrying Link training...\n");
+			}
+		}
+		if (i >= retry)
+        {
+			printf("LTPI failed to init, collect register dump!!!\n");
+        }
+	}
+}
+
+int misc_init_r(void)
+{
+	train_ltpi(LTPI_TRAIN_RETRY);
+
+	return 0;
+}
-- 
2.34.1

