From 0285f00b45991daf14695bc1b59ecb782c02eff6 Mon Sep 17 00:00:00 2001
From: Abinaya L <abinayal@ami.com>
Date: Tue, 27 Jun 2023 20:44:34 +0530
Subject: [PATCH] adding support for non pfr dual image inventory population

---
 item_updater.cpp | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/item_updater.cpp b/item_updater.cpp
index 983d98a..3884a58 100644
--- a/item_updater.cpp
+++ b/item_updater.cpp
@@ -205,7 +205,8 @@ void ItemUpdater::processBMCImage()
         if (0 ==
             iter.path().native().compare(0, BMC_RO_PREFIX_LEN, BMC_ROFS_PREFIX))
         {
-            std::string id = "bmc_active";
+            auto id = iter.path().native().substr(BMC_RO_PREFIX_LEN);
+
             // upstream changed this to relative_path ... is that right?
             fs::path releaseFile(OS_RELEASE_FILE);
             auto osRelease = iter.path() / releaseFile.relative_path();
@@ -236,6 +237,7 @@ void ItemUpdater::processBMCImage()
 
                 // Try to delete the version, same as above if the
                 // OS_RELEASE_FILE does not exist.
+                auto id = iter.path().native().substr(BMC_RO_PREFIX_LEN);
                 ItemUpdater::erase(id);
 
                 continue;
@@ -259,6 +261,7 @@ void ItemUpdater::processBMCImage()
                 // Set functional to true and remove the functional suffix
                 functional = true;
                 flashId.erase(flashId.length() - strlen(functionalSuffix));
+                id.erase(id.length() - strlen(functionalSuffix));
                 functionalFound = true;
             }
 
