From e3809396836cb2294fafb7dcc2b4552e2153b5bd Mon Sep 17 00:00:00 2001
From: kirubagharanas <kirubagharanas@ami.com>
Date: Tue, 18 Jul 2023 23:23:15 +0530
Subject: [PATCH] adding support for non pfr dual image inventory population

---
 item_updater.cpp | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/item_updater.cpp b/item_updater.cpp
index f598bf1..3d3730e 100644
--- a/item_updater.cpp
+++ b/item_updater.cpp
@@ -206,7 +206,8 @@ void ItemUpdater::processBMCImage()
         if (0 ==
             iter.path().native().compare(0, BMC_RO_PREFIX_LEN, BMC_ROFS_PREFIX))
         {
-            std::string id = "bmc_active";
+            auto id = iter.path().native().substr(BMC_RO_PREFIX_LEN);
+
             // upstream changed this to relative_path ... is that right?
             fs::path releaseFile(OS_RELEASE_FILE);
             auto osRelease = iter.path() / releaseFile.relative_path();
@@ -239,6 +240,7 @@ void ItemUpdater::processBMCImage()
 
                 // Try to delete the version, same as above if the
                 // OS_RELEASE_FILE does not exist.
+                auto id = iter.path().native().substr(BMC_RO_PREFIX_LEN);
                 ItemUpdater::erase(id);
 
                 continue;
@@ -262,6 +264,7 @@ void ItemUpdater::processBMCImage()
                 // Set functional to true and remove the functional suffix
                 functional = true;
                 flashId.erase(flashId.length() - strlen(functionalSuffix));
+                id.erase(id.length() - strlen(functionalSuffix));
                 functionalFound = true;
             }
 
