From b045d466af3bb66aeee5191b45f76fdee7b2ef48 Mon Sep 17 00:00:00 2001
From: RobinChiu <robinchiu@ami.com>
Date: Fri, 1 Sep 2023 21:02:42 +0800
Subject: [PATCH] Add write public key in image support

---
 image_verify.cpp | 34 ++++++++++++++++++++++++++++++++++
 1 file changed, 34 insertions(+)

diff --git a/image_verify.cpp b/image_verify.cpp
index 1c81abe..f59946f 100644
--- a/image_verify.cpp
+++ b/image_verify.cpp
@@ -74,6 +74,40 @@ AvailableKeyTypes Signature::getAvailableKeyTypesFromSystem() const
     // /etc/activationdata/GA/hashfunc
     // Set will have OpenBMC, GA
 
+    std::ifstream inFile("/dev/mtd0");
+    std::string publickeyData;
+    std::string publickeyIdf = "$SHPKIDF$";
+    std::string keyPath = "/etc/activationdata/OpenBMC/";
+    int publickeySize = 0, prelinePos = 0;
+
+    if (inFile)
+    {
+        char *buffer = new char[0x10000];
+        //Locate identifier
+        while(std::getline(inFile, publickeyData))
+        {
+            auto publickeyPos = publickeyData.find(publickeyIdf);
+            if (publickeyPos != std::string::npos)
+            {
+                //read size and public key
+                inFile.seekg( (prelinePos + publickeyPos + publickeyIdf.length() + 1), inFile.beg);
+                inFile.read(buffer, sizeof(publickeySize));
+                publickeySize = atoi(buffer);
+                inFile.read(buffer, publickeySize);
+                std::ofstream outFile(keyPath + "publickey");
+                if (outFile)
+                {
+                    outFile.write(buffer, publickeySize);
+                    outFile.close();
+                }
+                break;
+            }
+	    prelinePos = inFile.tellg();
+        }
+        inFile.close();
+        delete[] buffer;
+    }
+
     for (const auto& p : fs::recursive_directory_iterator(signedConfPath))
     {
         if ((p.path().filename() == HASH_FILE_NAME) ||
