Upstream-Status: Pending

diff --git a/bmc/update_manager.cpp b/bmc/update_manager.cpp
index 7881634..e0867f5 100644
--- a/bmc/update_manager.cpp
+++ b/bmc/update_manager.cpp
@@ -218,7 +218,11 @@ auto Manager::processImage(sdbusplus::message::unix_fd image,
 
 sdbusplus::message::object_path Manager::startUpdate(
     sdbusplus::message::unix_fd image,
-    ApplyTimeIntf::RequestedApplyTimes applyTime)
+    ApplyTimeIntf::RequestedApplyTimes applyTime,
+    bool overwrite,
+    std::vector<sdbusplus::message::details::string_path_wrapper> associations)
 {
+    (void)overwrite;
+    (void)associations;
     info("Starting update for image {FD}", "FD", static_cast<int>(image));
     using sdbusplus::xyz::openbmc_project::Common::Error::Unavailable;
     if (updateInProgress)
diff --git a/bmc/update_manager.hpp b/bmc/update_manager.hpp
index c76364c..2a8fda7 100644
--- a/bmc/update_manager.hpp
+++ b/bmc/update_manager.hpp
@@ -50,7 +50,9 @@ class Manager : public UpdateIntf
     sdbusplus::message::object_path startUpdate(
         sdbusplus::message::unix_fd image,
-        ApplyTimeIntf::RequestedApplyTimes applyTime) override;
+        ApplyTimeIntf::RequestedApplyTimes applyTime,
+        bool overwrite,
+        std::vector<sdbusplus::message::details::string_path_wrapper> associations) override;
 
     /* @brief Process the image supplied via image fd */
     auto processImage(sdbusplus::message::unix_fd image,
