From ab07d06e0f07e325c3bfda7b41bd38dfa69c0e36 Mon Sep 17 00:00:00 2001
From: kirubagharanas <kirubagharanas@ami.com>
Date: Tue, 6 Feb 2024 01:39:14 +0530
Subject: [PATCH] Patch to remove the image when verification fails

---
 activation.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/activation.cpp b/activation.cpp
index 1e5a40c..9a1f72d 100644
--- a/activation.cpp
+++ b/activation.cpp
@@ -114,6 +114,9 @@ auto Activation::activation(Activations value) -> Activations
                 // Stop the activation process, if fieldMode is enabled.
                 if (parent.control::FieldMode::fieldModeEnabled())
                 {
+                    info("Verification failed so cleaning up the uploaded image - {IMAGE}",
+                          "IMAGE",  (uploadDir / versionId).c_str());;
+                    fs::remove_all(uploadDir / versionId);
                     return softwareServer::Activation::activation(
                         softwareServer::Activation::Activations::Failed);
                 }
-- 
2.34.1

