From c914ffa405dfaa845ffb4271d0d47aef848f9d20 Mon Sep 17 00:00:00 2001
From: kirubagharanas <kirubagharanas@ami.com>
Date: Tue, 28 May 2024 16:48:14 +0530
Subject: [PATCH] fixed core-dump issue while verifying unsigned image

---
 activation.cpp | 19 ++++++++++---------
 1 file changed, 10 insertions(+), 9 deletions(-)

diff --git a/activation.cpp b/activation.cpp
index 8a28d79..27cce43 100644
--- a/activation.cpp
+++ b/activation.cpp
@@ -320,7 +320,8 @@ void Activation::onFlashWriteFailure()
     }
     if (!parent.activeUpdatesCount)
     {
-        activationBlocksTransition->disableRebootGuard();
+        if (activationBlocksTransition)
+            activationBlocksTransition->disableRebootGuard();
         if(parent.performReboot)
         {
             Activation::rebootBmc();
@@ -449,14 +450,14 @@ void Activation::unitStateChange(sdbusplus::message_t& msg)
         return;
     }
 
-#ifdef HOST_BIOS_UPGRADE
-    auto purpose = parent.versions.find(versionId)->second->purpose();
-    if (purpose == VersionPurpose::Host)
-    {
-        onStateChangesBios(msg);
-        return;
-    }
-#endif
+// #ifdef HOST_BIOS_UPGRADE
+//     auto purpose = parent.versions.find(versionId)->second->purpose();
+//     if (purpose == VersionPurpose::Host)
+//     {
+//         onStateChangesBios(msg);
+//         return;
+//     }
+// #endif
 
     onStateChanges(msg);
 
