From ebc465b7a8b8535381cdab6f650a13a5551f07e3 Mon Sep 17 00:00:00 2001
From: kirubagharanas <kirubagharanas@ami.com>
Date: Tue, 17 Dec 2024 15:02:46 +0530
Subject: [PATCH] BMC active and backup image are part of firmware inventory not image objects they should be never removed from the inventory

---
 item_updater.cpp | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/item_updater.cpp b/item_updater.cpp
index 3e6be87..d35d931 100644
--- a/item_updater.cpp
+++ b/item_updater.cpp
@@ -760,10 +760,12 @@ void ItemUpdater::freeSpace([[maybe_unused]] const Activation& caller)
 #ifdef BMC_STATIC_DUAL_IMAGE
     // For the golden image case, always remove the version on the primary side
     std::string versionIDtoErase;
+    std::string defaultBMC[] = {"bmc_active", "bmc_bkup"};
     for (const auto& iter : activations)
     {
         if (iter.second->redundancyPriority &&
-            iter.second->redundancyPriority.get()->priority() == 0)
+            iter.second->redundancyPriority.get()->priority() == 0 &&
+            find(begin(defaultBMC), end(defaultBMC), iter.second->versionId ) == end(defaultBMC))
         {
             versionIDtoErase = iter.second->versionId;
             break;
-- 
2.34.1

