From d1792e1b3bd3942fb7beef0ba9570387c1f6d6c5 Mon Sep 17 00:00:00 2001
From: hemanthkumarm <hemanthkumarm@ami.com>
Date: Tue, 29 Apr 2025 18:25:02 +0000
Subject: [PATCH] Add supoort for image-runtime signing for intel platforms

---
 image_verify.cpp | 1 +
 images.hpp       | 2 +-
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/image_verify.cpp b/image_verify.cpp
index 662a849..856be18 100644
--- a/image_verify.cpp
+++ b/image_verify.cpp
@@ -158,6 +158,7 @@ bool Signature::verifyFullImage()
         fs::path(imageDirPath) / "image-bmc.sig",
         fs::path(imageDirPath) / "image-hostfw.sig",
         fs::path(imageDirPath) / "image-kernel.sig",
+        fs::path(imageDirPath) / "image-runtime.sig",
         fs::path(imageDirPath) / "image-rofs.sig",
         fs::path(imageDirPath) / "image-rwfs.sig",
         fs::path(imageDirPath) / "image-u-boot.sig",
diff --git a/images.hpp b/images.hpp
index a1b52bc..713ebf7 100644
--- a/images.hpp
+++ b/images.hpp
@@ -11,7 +11,7 @@ namespace image
 {
 
 // BMC flash image file name list.
-const std::vector<std::string> bmcImages = {"image-kernel", "image-rofs",
+const std::vector<std::string> bmcImages = {"image-kernel", "image-runtime", "image-rofs",
                                             "image-rwfs", "image-u-boot"};
 // BMC flash image file name list for full flash image (image-bmc)
 const std::string bmcFullImages = {"image-bmc"};
-- 
2.43.0

