From e7a6860820ffc13898a4bede688c10d5dc0af005 Mon Sep 17 00:00:00 2001
From: dhanusha <dhanusha@ami.com>
Date: Fri, 20 Feb 2026 15:28:52 +0530
Subject: [PATCH 13/13] Add-Purpose-For-PSU-Component

Upstream-Status: Inappropriate [AMI-specific customizations]

---
 item_updater.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/item_updater.cpp b/item_updater.cpp
index 4b1dc3f9..52bd6315 100644
--- a/item_updater.cpp
+++ b/item_updater.cpp
@@ -74,7 +74,8 @@ void ItemUpdater::createActivation(sdbusplus::message_t& msg)
                         value == VersionPurpose::Host ||
 #endif
                         value == VersionPurpose::System ||
-                        value == VersionPurpose::Other )
+                        value == VersionPurpose::Other ||
+			value == VersionPurpose::PSU)
                     {
                         purpose = value;
                     }
-- 
2.34.1

