From 84b878bb7411fab531c5761166b3651863603f62 Mon Sep 17 00:00:00 2001
From: pandiarajt <pandiarajt@ami.com>
Date: Sat, 13 Dec 2025 18:34:02 +0530
Subject: [PATCH 129/129] Fix for NTPsec persistent across reboots

---
 manager.cpp        |  6 +++---
 ntpsec_manager.cpp | 16 ++++++++++++++++
 2 files changed, 19 insertions(+), 3 deletions(-)

diff --git a/manager.cpp b/manager.cpp
index b6d7cd2..094ac36 100644
--- a/manager.cpp
+++ b/manager.cpp
@@ -79,7 +79,7 @@ void Manager::onPropertyChanged(const std::string& key,
                         ntpSecInterface, propertyNtpSec ,
                         newNtpSecMode);
 
-                info("Updated NTPSec setting: {ENABLED}", "ENABLED", newNtpMode);
+                info("Updated NTPSec setting: {ENABLED}", "ENABLED", newNtpSecMode);
             }
             catch (const sdbusplus::exception_t& ex)
             {
@@ -108,7 +108,7 @@ void Manager::onPropertyChanged(const std::string& key,
                             ntpSecInterface, propertyNtpSec ,
                             newNtpSecMode);
 
-                    info("Updated NTPSec setting: {ENABLED}", "ENABLED", newNtpMode);
+                    info("Updated NTPSec setting: {ENABLED}", "ENABLED", newNtpSecMode);
                 }
                 catch (const sdbusplus::exception_t& ex)
                 {
@@ -132,7 +132,7 @@ void Manager::onPropertyChanged(const std::string& key,
                         ntpSecInterface, propertyNtpSec ,
                         newNtpSecMode);
 
-                info("Updated NTPSec setting: {ENABLED}", "ENABLED", newNtpMode);
+                info("Updated NTPSec setting: {ENABLED}", "ENABLED", newNtpSecMode);
             }
             catch (const sdbusplus::exception_t& ex)
             {
diff --git a/ntpsec_manager.cpp b/ntpsec_manager.cpp
index 75a2edf..842cdd8 100644
--- a/ntpsec_manager.cpp
+++ b/ntpsec_manager.cpp
@@ -114,6 +114,17 @@ void controlSystemdService(const std::string& serviceName, ServiceAction action)
     }
 }
 
+void setServiceEnabled(const std::string& serviceName, bool enable) {
+    std::string command = "systemctl ";
+    command += (enable ? "enable " : "disable ");
+    command += serviceName;
+    int ret = std::system(command.c_str());
+    if(ret != 0)
+    {
+	std::cerr << "Failed to update " << serviceName << " Enbled status: " << enable << std::endl;
+    }
+}
+
 void registerNTPSecDbus(std::shared_ptr<sdbusplus::asio::dbus_interface> iface) {
     std::vector<std::string> servers = {};
     bool isActive = isServiceActive();
@@ -127,6 +138,7 @@ void registerNTPSecDbus(std::shared_ptr<sdbusplus::asio::dbus_interface> iface)
             }
             bool isActive = isServiceActive();
             configureNtpServers(newServers);
+
             if(isActive)
             {
                 controlSystemdService("ntpd.service", ServiceAction::Restart);
@@ -144,9 +156,13 @@ void registerNTPSecDbus(std::shared_ptr<sdbusplus::asio::dbus_interface> iface)
         [](const bool& newStatus, bool& current) {
             if (newStatus == current) return true;
             controlSystemdService("ntpd.service", newStatus ? ServiceAction::Start : ServiceAction::Stop);
+	    setServiceEnabled("systemd-timesyncd.service", !newStatus);
             std::this_thread::sleep_for(std::chrono::seconds(2));
             current = isServiceActive();
             return true;
+        },
+	[](const auto&) {
+            return isServiceActive();
         }
     );
 
-- 
2.34.1

