From 8718e90204f503d2b2193230346f3d62bf842716 Mon Sep 17 00:00:00 2001
From: pandiarajt <pandiarajt@ami.com>
Date: Tue, 27 Jan 2026 17:45:08 +0530
Subject: [PATCH 130/130] Handle Error for NTPSec configuration

---
 ntpsec_manager.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/ntpsec_manager.cpp b/ntpsec_manager.cpp
index 842cdd8..b8f7b12 100644
--- a/ntpsec_manager.cpp
+++ b/ntpsec_manager.cpp
@@ -137,6 +137,11 @@ void registerNTPSecDbus(std::shared_ptr<sdbusplus::asio::dbus_interface> iface)
                 return true;
             }
             bool isActive = isServiceActive();
+
+	    if (newServers.size() > 3) 
+	    {
+	        return false;
+	    }
             configureNtpServers(newServers);
 
             if(isActive)
-- 
2.34.1

