--- /dev/null	2023-05-03 14:25:33.599526689 -0400
+++ a/dist/env/asd	2023-07-11 17:52:14.022928151 -0400
@@ -0,0 +1,12 @@
+#D-Bus object path
+#example: /xyz/openbmc_project/certs/server/asd
+ENDPOINT=asd
+
+#Path for the certificate file
+CERTPATH=/etc/ssl/certs/asd/server.pem
+
+#Units to restart
+UNIT=bmcweb.service
+
+#Type of the service client/server
+TYPE=server
--- /dev/null	2023-05-03 14:25:33.599526689 -0400
+++ a/dist/busconfig/phosphor-asd-cert-config.conf	2023-07-12 15:54:52.067453993 -0400
@@ -0,0 +1,8 @@
+<!DOCTYPE busconfig PUBLIC "-//freedesktop//DTD D-BUS Bus Configuration 1.0//EN"
+ "http://www.freedesktop.org/standards/dbus/1.0/busconfig.dtd">
+<busconfig>
+  <policy user="root">
+    <allow own="xyz.openbmc_project.Certs.Manager.Server.Asd"/>
+    <allow send_destination="xyz.openbmc_project.Certs.Manager.Server.Asd"/>
+  </policy>
+</busconfig>
--- a/meson_options.txt	2023-07-12 15:07:26.057020782 -0400
+++ b/meson_options.txt	2023-07-12 15:06:44.500658301 -0400
@@ -21,6 +21,11 @@
     description: 'Install nslcd authority cert configs',
 )
 
+option('config-asd',
+    type: 'feature',
+    description: 'Install asd cert configs',
+)
+
 option('authorities-list-name',
     type: 'string',
     value: 'trust_bundle',
--- a/dist/meson.build	2023-07-12 18:02:42.284584131 -0400
+++ b/dist/meson.build	2023-07-12 18:02:00.568151889 -0400
@@ -32,6 +32,15 @@
     ]]
 endif
 
+if not get_option('config-asd').disabled()
+    busconfig += 'busconfig/phosphor-asd-cert-config.conf'
+    certs += 'env/asd'
+    systemd_alias += [[
+        '../phosphor-certificate-manager@.service',
+        'multi-user.target.wants/phosphor-certificate-manager@asd.service'
+    ]]
+endif
+
 install_data(
     service_files,
     install_dir: systemd_system_unit_dir,
