From 1fabd46a116c550e91180a137970d5646a6472aa Mon Sep 17 00:00:00 2001
From: Minal Upadhye <minalsunilu@ami.com>
Date: Mon, 23 Feb 2026 18:23:37 +0530
Subject: [PATCH 5/5] throw-CertificateExists-error LF

Signed-off-by: Minal Upadhye <minalsunilu@ami.com>
---
 certs_manager.cpp | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/certs_manager.cpp b/certs_manager.cpp
index cfccd60..35f65fe 100644
--- a/certs_manager.cpp
+++ b/certs_manager.cpp
@@ -53,10 +53,13 @@ using namespace phosphor::logging;
 using ::sdbusplus::xyz::openbmc_project::Certs::Error::InvalidCertificate;
 using ::sdbusplus::xyz::openbmc_project::Common::Error::InternalFailure;
 using ::sdbusplus::xyz::openbmc_project::Common::Error::NotAllowed;
+using ::sdbusplus::xyz::openbmc_project::Certs::Error::CertificateExists;
 using NotAllowedReason =
     ::phosphor::logging::xyz::openbmc_project::Common::NotAllowed::REASON;
 using InvalidCertificateReason = ::phosphor::logging::xyz::openbmc_project::
     Certs::InvalidCertificate::REASON;
+using CertificateExistsReason = ::phosphor::logging::xyz::openbmc_project::
+    Certs::CertificateExists::REASON;
 using ::sdbusplus::xyz::openbmc_project::Common::Error::InvalidArgument;
 using Argument =
     ::phosphor::logging::xyz::openbmc_project::Common::InvalidArgument;
@@ -260,7 +263,7 @@ std::string Manager::install(const std::string filePath)
 {
     if (certType != CertificateType::authority && !installedCerts.empty())
     {
-        elog<NotAllowed>(NotAllowedReason("Certificate already exist"));
+        elog<CertificateExists>(CertificateExistsReason("Certificate already exist"));
     }
     else if (certType == CertificateType::authority &&
              installedCerts.size() >= maxNumAuthorityCertificates)
@@ -280,7 +283,7 @@ std::string Manager::install(const std::string filePath)
     }
     else
     {
-        elog<NotAllowed>(NotAllowedReason("Certificate already exist"));
+        elog<CertificateExists>(CertificateExistsReason("Certificate already exist"));
     }
 
     return certObjectPath;
-- 
2.34.1

