From 458bf958ec3c4e5f94b356b687f90edbcfe1342e Mon Sep 17 00:00:00 2001
From: RythonCai <rythoncai@ami.com>
Date: Tue, 27 Feb 2024 11:53:58 +0800
Subject: [PATCH] throw CertificateExists error

---
 certs_manager.cpp | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/certs_manager.cpp b/certs_manager.cpp
index 67a3ed9..ea13a4d 100644
--- a/certs_manager.cpp
+++ b/certs_manager.cpp
@@ -53,10 +53,13 @@ using namespace phosphor::logging;
 using ::sdbusplus::xyz::openbmc_project::Certs::Error::InvalidCertificate;
 using ::sdbusplus::xyz::openbmc_project::Common::Error::InternalFailure;
 using ::sdbusplus::xyz::openbmc_project::Common::Error::NotAllowed;
+using ::sdbusplus::xyz::openbmc_project::Certs::Error::CertificateExists;
 using NotAllowedReason =
     ::phosphor::logging::xyz::openbmc_project::Common::NotAllowed::REASON;
 using InvalidCertificateReason = ::phosphor::logging::xyz::openbmc_project::
     Certs::InvalidCertificate::REASON;
+using CertificateExistsReason = ::phosphor::logging::xyz::openbmc_project::
+    Certs::CertificateExists::REASON;
 using ::sdbusplus::xyz::openbmc_project::Common::Error::InvalidArgument;
 using Argument =
     ::phosphor::logging::xyz::openbmc_project::Common::InvalidArgument;
@@ -260,7 +263,7 @@ std::string Manager::install(const std::string filePath)
 {
     if (certType != CertificateType::authority && !installedCerts.empty())
     {
-        elog<NotAllowed>(NotAllowedReason("Certificate already exist"));
+        elog<CertificateExists>(CertificateExistsReason("Certificate already exist"));
     }
     else if (certType == CertificateType::authority &&
              installedCerts.size() >= maxNumAuthorityCertificates)
@@ -280,7 +283,7 @@ std::string Manager::install(const std::string filePath)
     }
     else
     {
-        elog<NotAllowed>(NotAllowedReason("Certificate already exist"));
+        elog<CertificateExists>(CertificateExistsReason("Certificate already exist"));
     }
 
     return certObjectPath;
-- 
2.34.1

