From 2fde35f0a334c7de6112d37a2995a44c93440962 Mon Sep 17 00:00:00 2001
From: RythonCai <rythoncai@ami.com>
Date: Wed, 10 Apr 2024 17:28:26 +0800
Subject: [PATCH] Throws an error if the private key file is not found

---
 certificate.cpp | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/certificate.cpp b/certificate.cpp
index 5fbb04e..941da64 100644
--- a/certificate.cpp
+++ b/certificate.cpp
@@ -44,6 +44,8 @@ using InvalidCertificateError =
     ::sdbusplus::xyz::openbmc_project::Certs::Error::InvalidCertificate;
 using ::phosphor::logging::xyz::openbmc_project::Certs::InvalidCertificate;
 using ::sdbusplus::xyz::openbmc_project::Common::Error::InternalFailure;
+using ::sdbusplus::xyz::openbmc_project::Certs::Error::PrivateKeyNotFound;
+using PrivateKeyNotFoundReason = ::phosphor::logging::xyz::openbmc_project::Certs::PrivateKeyNotFound::REASON;
 
 // RAII support for openSSL functions.
 using BIOMemPtr = std::unique_ptr<BIO, decltype(&::BIO_free)>;
@@ -587,7 +589,8 @@ void Certificate::checkAndAppendPrivateKey(const std::string& filePath)
         {
             lg2::error("Private key file is not found, FILE:{FILE}", "FILE",
                        privateKeyFile);
-            elog<InternalFailure>();
+            elog<PrivateKeyNotFound>(
+                PrivateKeyNotFoundReason("Private key file is not found"));
         }
 
         std::ifstream privKeyFileStream;
-- 
2.34.1

