From 914bff487e7347d7cb164cde3220dc1de37ba1df Mon Sep 17 00:00:00 2001
From: Minal Upadhye <minalsunilu@ami.com>
Date: Thu, 26 Feb 2026 16:19:45 +0530
Subject: [PATCH] Fix-For-rekey-rekey LF

Signed-off-by: Minal Upadhye <minalsunilu@ami.com>
---
 certs_manager.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/certs_manager.cpp b/certs_manager.cpp
index 42cbe6d..0e5cac9 100644
--- a/certs_manager.cpp
+++ b/certs_manager.cpp
@@ -700,7 +700,7 @@ void Manager::generateCSRHelper(
       }
     // Used EC algorithm as default if user did not give algorithm type.
     if (keyPairAlgorithm == "RSA")
-        pKey = getRSAKeyPair(keyBitLength);
+        pKey = generateRSAKeyPair(keyBitLength);
     else if ((keyPairAlgorithm == "EC") || (keyPairAlgorithm.empty()))
         pKey = generateECKeyPair(keyCurveId);
     else
-- 
2.34.1

