From dee95cb1f58a22cbc2a6de8e33212aa41a2f5ba1 Mon Sep 17 00:00:00 2001
From: viswanathc <viswanathc@ami.com>
Date: Thu, 13 Nov 2025 01:35:21 -0500
Subject: [PATCH] Fix for high Coverity issue in Core Manageability-4

Signed-off-by: viswanathc <viswanathc@ami.com>
---
 certs_manager.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/certs_manager.cpp b/certs_manager.cpp
index cbb1e69..fab5df2 100644
--- a/certs_manager.cpp
+++ b/certs_manager.cpp
@@ -140,8 +140,8 @@ Manager::Manager(sdbusplus::bus_t& bus, sdeventplus::Event& event,
                  const char* path, CertificateType type,
                  const std::string& unit, const std::string& installPath) :
     internal::ManagerInterface(bus, path), bus(bus), event(event),
-    objectPath(path), certType(type), unitToRestart(std::move(unit)),
-    certInstallPath(std::move(installPath)),
+    objectPath(path), certType(type), unitToRestart(unit),
+    certInstallPath(installPath),
     certParentInstallPath(fs::path(certInstallPath).parent_path())
 {
     try
-- 
2.34.1

