From 46da56dfd1fb7023d8de2f274808bad28abce456 Mon Sep 17 00:00:00 2001
From: viswanathc <viswanathc@ami.com>
Date: Tue, 2 Dec 2025 06:45:06 -0500
Subject: [PATCH] Response Time Error

Signed-off-by: viswanathc <viswanathc@ami.com>
---
 certs_manager.cpp | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/certs_manager.cpp b/certs_manager.cpp
index cbb1e69..e53f2da 100644
--- a/certs_manager.cpp
+++ b/certs_manager.cpp
@@ -273,6 +273,12 @@ std::string Manager::install(const std::string filePath)
     std::string certObjectPath;
     if (isCertificateUnique(filePath))
     {
+        X509StorePtr x509Store = getX509Store(filePath);
+        // Load Certificate file into the X509 structure.
+        internal::X509Ptr cert = loadCert(filePath);
+        // Perform validation
+        validateCertificateAgainstStore(*x509Store, *cert);
+        validateCertificateStartDate(*cert);
         certObjectPath = objectPath + '/' + std::to_string(certIdCounter);
         installedCerts.emplace_back(std::make_unique<Certificate>(
             bus, certObjectPath, certType, certInstallPath, filePath,
-- 
2.34.1

