From 017770e3db94b2cc4bace9c461bd036f884e6423 Mon Sep 17 00:00:00 2001
From: pandiarajt <pandiarajt@ami.com>
Date: Wed, 13 Nov 2024 15:21:39 +0530
Subject: [PATCH 196/196] fix for solssh issue

---
 config.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/config.c b/config.c
index f36af8c..855ef38 100644
--- a/config.c
+++ b/config.c
@@ -43,7 +43,7 @@ const char *config_get_value(struct config *config, const char *name)
 	char buf[CONFIG_MAX_KEY_LENGTH];
 	int rc;
 
-	if (!config->dict) {
+	if (!config || !config->dict) {
 		return NULL;
 	}
 
-- 
2.34.1

