From 618828c000cb949ecd8d090cc6b6fc488eaabec2 Mon Sep 17 00:00:00 2001
From: pandiarajt <pandiarajt@ami.com>
Date: Fri, 13 Dec 2024 15:13:16 +0530
Subject: [PATCH 198/198] Added support to enable/disable obmc-console log with respect to phosphor-hostlogger

---
 meson.build   | 8 +++++++-
 meson.options | 1 +
 2 files changed, 8 insertions(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 46f7e71..0891d45 100644
--- a/meson.build
+++ b/meson.build
@@ -54,13 +54,19 @@ endif
 
 iniparser_dep = dependency('iniparser')
 
+log_handler = []
+
+if get_option('obmc-console-log')
+  log_handler += 'log-handler.c'
+endif
+
 server = executable('obmc-console-server',
            'config.c',
            'console-dbus.c',
            'console-server.c',
            'console-socket.c',
            'console-mux.c',
-           'log-handler.c',
+           log_handler,
            'ringbuffer.c',
            'socket-handler.c',
            'tty-handler.c',
diff --git a/meson.options b/meson.options
index d8837f2..8e4f20a 100644
--- a/meson.options
+++ b/meson.options
@@ -2,3 +2,4 @@ option('udev', type: 'feature', description: 'Install console udev rules')
 option('concurrent-servers', type: 'boolean', value: false, description: 'Support multiple concurrent obmc-console instances')
 option('ssh', type: 'feature', description: 'Support obmc-console-ssh and obmc-console-ssh-socket')
 option('tests', type: 'boolean', description: 'Enable the test suite')
+option('obmc-console-log', type: 'boolean', description: 'Enable log support in obmc-console')
-- 
2.34.1

