From 21531c12a27d19fa3412ad8d9af77ebc2bfaa7d4 Mon Sep 17 00:00:00 2001
From: pandiarajt <pandiarajt@ami.com>
Date: Mon, 16 Jun 2025 17:18:44 +0530
Subject: [PATCH 200/200] Added changes to save baudrate in config file

---
 config.c       |  8 ++++++++
 config.h       |  5 +++++
 console-dbus.c | 54 ++++++++++++++++++++++++++++++++++++++++++++++++++
 3 files changed, 67 insertions(+)

diff --git a/config.c b/config.c
index 855ef38..12809da 100644
--- a/config.c
+++ b/config.c
@@ -37,6 +37,7 @@
 #include "util.h"
 
 static const char *config_default_filename = SYSCONFDIR "/obmc-console.conf";
+char configFilePath[MAX_LINE_LEN] = "";
 
 const char *config_get_value(struct config *config, const char *name)
 {
@@ -72,6 +73,13 @@ struct config *config_init(const char *filename)
 	if (!filename) {
 		filename = config_default_filename;
 	}
+    
+    int ret = snprintf(configFilePath, sizeof(configFilePath), "%s", filename);
+    if (ret < 0 || ((size_t)ret >= sizeof(configFilePath)))
+    {
+        warn("Buffer Overflow \n");
+        return NULL;
+    }
 
 	if (access(filename, R_OK) == 0) {
 		dict = iniparser_load(filename);
diff --git a/config.h b/config.h
index 0a8a4cd..9757412 100644
--- a/config.h
+++ b/config.h
@@ -20,6 +20,9 @@
 #include <stdint.h>
 #include <termios.h>
 
+#define MAX_LINES 64
+#define MAX_LINE_LEN 256
+
 struct config;
 
 const char *config_get_section_value(struct config *config, const char *secname,
@@ -37,3 +40,5 @@ int config_parse_bytesize(const char *size_str, size_t *size);
 
 int config_count_sections(struct config *config);
 const char *config_get_section_name(struct config *config, int i);
+
+extern char configFilePath[MAX_LINE_LEN];
diff --git a/console-dbus.c b/console-dbus.c
index 778f1cd..1cc0ca2 100644
--- a/console-dbus.c
+++ b/console-dbus.c
@@ -33,6 +33,9 @@ const size_t dbus_obj_path_len = 1024;
 #define UART_INTF   "xyz.openbmc_project.Console.UART"
 #define ACCESS_INTF "xyz.openbmc_project.Console.Access"
 
+#define BAUD_PREFIX     "baud = "
+#define BAUD_PREFIX_LEN (sizeof(BAUD_PREFIX) - 1)
+
 static void tty_change_baudrate(struct console *console)
 {
 	int i;
@@ -58,6 +61,50 @@ static void tty_change_baudrate(struct console *console)
 	}
 }
 
+int update_baud_in_config(const char *filepath, uint64_t baud) {
+    char lines[MAX_LINES][MAX_LINE_LEN];
+    int ret, n = 0;
+    int baud_found = 0;
+    FILE *fp = fopen(filepath, "r");
+    if (fp) {
+        while (n < MAX_LINES && fgets(lines[n], sizeof(lines[n]), fp)) {
+            if (strncmp(lines[n], BAUD_PREFIX, BAUD_PREFIX_LEN) == 0) {
+                ret = snprintf(lines[n], sizeof(lines[n]), "baud = %" PRIu64 "\n", baud);
+                if(ret < 0 || ret >= (int)sizeof(lines[n]))
+                {
+                     warnx("Buffer Overflow");
+                     fclose(fp);
+                     return -1;
+                }
+                baud_found = 1;
+            }
+            n++;
+        }
+        fclose(fp);
+    }
+
+    if (!baud_found && n < MAX_LINES) {
+        ret = snprintf(lines[n], sizeof(lines[n]), "baud = %" PRIu64 "\n", baud);
+        if(ret < 0 || ret >= (int)sizeof(lines[n]))
+        {
+            warnx("Buffer Overflow");
+            return -1;
+        }
+        n++;
+    }
+
+    fp = fopen(filepath, "w");
+    if (!fp) {
+        return -1;
+    }
+    for (int i = 0; i < n; ++i) {
+        fputs(lines[i], fp);
+    }
+    fclose(fp);
+    return 0;
+}
+
+
 static int set_baud_handler(sd_bus *bus, const char *path,
 			    const char *interface, const char *property,
 			    sd_bus_message *msg, void *userdata,
@@ -87,6 +134,13 @@ static int set_baud_handler(sd_bus *bus, const char *path,
 	console->server->tty.uart.baud = speed;
 	tty_change_baudrate(console);
 
+
+    r = update_baud_in_config(configFilePath, baudrate);
+    if (r != 0)
+    {
+        warnx("Fail to write baud rate in config file");
+        return -EINVAL;
+    }
 	sd_bus_emit_properties_changed(bus, path, interface, property, NULL);
 
 	return 1;
-- 
2.34.1

