From f39e80d84264110179712838885f51e942052b70 Mon Sep 17 00:00:00 2001
From: vmani <manikandanv@ami.com>
Date: Thu, 23 Nov 2023 15:08:51 +0530
Subject: [PATCH 143/143] Even if the log file exists in BMC, SOL log are store
 permanently

---
 log-handler.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/log-handler.c b/log-handler.c
index ab30827..61ce337 100644
--- a/log-handler.c
+++ b/log-handler.c
@@ -62,7 +62,7 @@ static int log_trim(struct log_handler *lh)
 		/* don't return, as we need to re-open the logfile */
 	}
 
-	lh->fd = open(lh->log_filename, O_RDWR | O_CREAT | O_TRUNC, 0644);
+	lh->fd = open(lh->log_filename, O_RDWR | O_CREAT , 0644);
 	if (lh->fd < 0) {
 		warn("Can't open log buffer file %s", lh->log_filename);
 		return -1;
@@ -155,7 +155,7 @@ static int log_init(struct handler *handler, struct console *console,
 		filename = default_filename;
 	}
 
-	lh->fd = open(filename, O_RDWR | O_CREAT | O_TRUNC, 0644);
+	lh->fd = open(filename, O_RDWR | O_CREAT , 0644);
 	if (lh->fd < 0) {
 		warn("Can't open log buffer file %s", filename);
 		return -1;
-- 
2.41.0

