From 624591a48ca21fae73305c93c4c2aee04c180536 Mon Sep 17 00:00:00 2001
From: Sivaprabu Ganesan <sivaprabug@ami.com>
Date: Fri, 19 May 2023 11:04:02 +0530
Subject: [PATCH] Firmware version not showing

Signed-off-by: Sivaprabu Ganesan <sivaprabug@ami.com>
---
 src/views/Overview/OverviewFirmware.vue | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/src/views/Overview/OverviewFirmware.vue b/src/views/Overview/OverviewFirmware.vue
index c2c7f4a..fb210dd 100644
--- a/src/views/Overview/OverviewFirmware.vue
+++ b/src/views/Overview/OverviewFirmware.vue
@@ -15,7 +15,7 @@
       <b-col sm="6">
         <dl>
           <dt>{{ $t('pageOverview.firmwareVersion') }}</dt>
-          <dd>{{ dataFormatter(firmwareVersion) }}</dd>
+          <dd>{{ dataFormatter(runningVersion) }}</dd>
         </dl>
       </b-col>
     </b-row>
@@ -35,7 +35,6 @@ export default {
   mixins: [DataFormatterMixin],
   computed: {
     ...mapState({
-      server: (state) => state.system.systems[0],
       backupBmcFirmware() {
         return this.$store.getters['firmware/backupBmcFirmware'];
       },
@@ -45,9 +44,6 @@ export default {
       activeBmcFirmware() {
         return this.$store.getters[`firmware/activeBmcFirmware`];
       },
-      firmwareVersion() {
-        return this.server?.firmwareVersion;
-      },
       runningVersion() {
         return this.activeBmcFirmware?.version;
       },
-- 
2.25.1

