From be9a8a125d98c9839ea49951b0fd91029cce1be9 Mon Sep 17 00:00:00 2001
From: pandiarajt <pandiarajt@ami.com>
Date: Fri, 15 Nov 2024 17:56:23 +0530
Subject: [PATCH 196/196] Fix for lauching sol session with SOL loop test

---
 console-server.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/console-server.c b/console-server.c
index 60d2d2e..54d7ce1 100644
--- a/console-server.c
+++ b/console-server.c
@@ -65,7 +65,7 @@ static void usage(const char *progname)
 
 static bool console_server_pollfd_reclaimable(struct pollfd *p)
 {
-	return p->fd == -1 && p->events == 0 && p->revents == ~0;
+	return p->fd == -1 && p->events == 0 && p->revents == 0;
 }
 
 static ssize_t
@@ -128,7 +128,7 @@ int console_server_release_pollfd(struct console_server *server,
 	// https://www.man7.org/linux/man-pages/man2/poll.2.html
 	pfd->fd = -1;
 	pfd->events = 0;
-	pfd->revents = ~0;
+
 
 	return 0;
 }
-- 
2.34.1

