From 95d36e78e7632fccf7c2fd0e2afffc52ae89d5de Mon Sep 17 00:00:00 2001
From: pandiarajt <pandiarajt@ami.com>
Date: Tue, 14 Oct 2025 12:50:45 +0530
Subject: [PATCH 212/212] Added support to enable/disable obmc-console log

---
 meson.build   | 8 +++++++-
 meson.options | 1 +
 2 files changed, 8 insertions(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index db987d4..03e5592 100644
--- a/meson.build
+++ b/meson.build
@@ -63,6 +63,12 @@ endif
 
 iniparser_dep = dependency('iniparser')
 
+log_handler = []
+
+if get_option('obmc-console-log')
+    log_handler += 'log-handler.c'
+endif
+
 server = executable(
     'obmc-console-server',
     'config.c',
@@ -70,7 +76,7 @@ server = executable(
     'console-server.c',
     'console-socket.c',
     'console-mux.c',
-    'log-handler.c',
+    log_handler,
     'ringbuffer.c',
     'socket-handler.c',
     'tty-handler.c',
diff --git a/meson.options b/meson.options
index 3dcc6ee..977a966 100644
--- a/meson.options
+++ b/meson.options
@@ -11,3 +11,4 @@ option(
     description: 'Support obmc-console-ssh and obmc-console-ssh-socket',
 )
 option('tests', type: 'boolean', description: 'Enable the test suite')
+option('obmc-console-log', type: 'boolean', description: 'Enable log support in obmc-console')
-- 
2.34.1

