From 87bc32bd270f1793cd7003d7ee89c68decec7751 Mon Sep 17 00:00:00 2001
From: mickhuang <mickhuang@ami.com>
Date: Tue, 15 Oct 2024 11:47:15 +0800
Subject: [PATCH 2/2] Fix HID report descriptor allocation

Signed-off-by: mickhuang <mickhuang@ami.com>
---
 drivers/usb/gadget/function/f_hid.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/usb/gadget/function/f_hid.c b/drivers/usb/gadget/function/f_hid.c
index 6be6009f911e..11188ed869c6 100644
--- a/drivers/usb/gadget/function/f_hid.c
+++ b/drivers/usb/gadget/function/f_hid.c
@@ -88,6 +88,7 @@ static void hidg_release(struct device *dev)
 {
 	struct f_hidg *hidg = container_of(dev, struct f_hidg, dev);
 
+	kfree(hidg->report_desc);
 	kfree(hidg->set_report_buf);
 	kfree(hidg);
 }
@@ -1287,7 +1288,7 @@ static struct usb_function *hidg_alloc(struct usb_function_instance *fi)
 	hidg->report_length = opts->report_length;
 	hidg->report_desc_length = opts->report_desc_length;
 	if (opts->report_desc) {
-		hidg->report_desc = devm_kmemdup(&hidg->dev, opts->report_desc,
+		hidg->report_desc = kmemdup(opts->report_desc,
 						 opts->report_desc_length,
 						 GFP_KERNEL);
 		if (!hidg->report_desc) {
-- 
2.34.1

