From e2bca20aff8eda8d26906fbfa79d993979a64225 Mon Sep 17 00:00:00 2001
From: gayathrid <gayathrid@ami.com>
Date: Tue, 21 May 2024 12:27:22 -0400
Subject: [PATCH] Remove TPMI Support From Ras

Signed-off-by: gayathrid <gayathrid@ami.com>
---
 src/ras_manager.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/ras_manager.cpp b/src/ras_manager.cpp
index 3e66f1e..819bfd5 100644
--- a/src/ras_manager.cpp
+++ b/src/ras_manager.cpp
@@ -983,7 +983,7 @@ static void signalWhea()
 
 static void err0RasHandler()
 {
-    if (!discoverTpmi()) return;
+    //if (!discoverTpmi()) return;
     bool signalOs = false;
 
     PRINT_INFO("ERR0 Asserted\n");
@@ -1468,7 +1468,7 @@ static void MirrorFailOverRasHandler()
 {
     RAS_STATUS status;
 
-    if (!discoverTpmi()) return;
+    //if (!discoverTpmi()) return;
 
     if (rasFeatureInitializedCheck(RAS_FEATURE_MEMORY_CE_ID) != RAS_CC_SUCCESS)
     {
-- 
2.34.1

