From 71c2136004692d227f3245d8cdfcffb4c2426ac7 Mon Sep 17 00:00:00 2001
From: Dillibabu <dillibabug@ami.com>
Date: Fri, 22 Mar 2024 17:16:12 +0530
Subject: [PATCH] 1. Modified authenticate to quit on state machine.

Signed-off-by: Dillibabu <dillibabug@ami.com>
---
 smtp-auth.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/smtp-auth.c b/smtp-auth.c
index 9b7edac..76661cb 100644
--- a/smtp-auth.c
+++ b/smtp-auth.c
@@ -223,7 +223,7 @@ rsp_auth (siobuf_t conn, smtp_session_t session)
 	     mail command since the MTA is required to accept mail for
 	     its own domain. */
 	  if (next_auth_mechanism (session))
-	    session->rsp_state = S_auth;
+	    session->rsp_state = S_quit;
 #ifdef USE_ETRN
 	  else if (check_etrn (session))
 	    session->rsp_state = S_etrn;
-- 
2.34.1

