From 95c79a168b3fc937b4ee8342930d2239338f2454 Mon Sep 17 00:00:00 2001
From: pavithra <pavithrat@ami.com>
Date: Tue, 11 Nov 2025 15:06:34 +0530
Subject: [PATCH] libesmtp_coverity_fix

---
 smtp-tls.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/smtp-tls.c b/smtp-tls.c
index 8ed056f..779c9c4 100644
--- a/smtp-tls.c
+++ b/smtp-tls.c
@@ -420,6 +420,7 @@ starttls_create_ssl (smtp_session_t session)
       if (!SSL_use_certificate_file (ssl, keyfile, SSL_FILETYPE_PEM))
 	{
 	  /* FIXME: set an error code */
+	  SSL_free(ssl);
 	  return NULL;
 	}
       if (!SSL_use_PrivateKey_file (ssl, keyfile, SSL_FILETYPE_PEM))
@@ -429,7 +430,10 @@ starttls_create_ssl (smtp_session_t session)
 	    (*session->event_cb) (session, SMTP_EV_NO_CLIENT_CERTIFICATE,
 				  session->event_cb_arg, &ok);
 	  if (!ok)
+	  {
+	    SSL_free(ssl);
 	    return NULL;
+	  }
 	}
     }
   else if (status == FILE_PROBLEM)
-- 
2.34.1

