From 7f3dd1ba433836bb7822cb5bda7727f88a827791 Mon Sep 17 00:00:00 2001
From: Vijaysankar R <vijaysankarr@ami.com>
Date: Tue, 2 Sep 2025 15:50:54 +0530
Subject: [PATCH 110/110] Coverity-Fix

---
 message-callbacks.c |  8 ++++++--
 smtp-tls.c          | 21 +++++++++++++++++++--
 2 files changed, 25 insertions(+), 4 deletions(-)

diff --git a/message-callbacks.c b/message-callbacks.c
index cac13ed..59aaf7a 100644
--- a/message-callbacks.c
+++ b/message-callbacks.c
@@ -93,8 +93,12 @@ _smtp_message_str_cb (void **ctx, int *len, void *arg)
   const char *string = arg;
   struct state *state;
 
-  if (*ctx == NULL)
-    *ctx = malloc (sizeof (struct state));
+  if (*ctx == NULL) {
+    *ctx = calloc(1, sizeof(struct state));
+    if (*ctx == NULL) {
+      return NULL;
+    }
+  }
   state = *ctx;
 
   if (len == NULL)
diff --git a/smtp-tls.c b/smtp-tls.c
index 146e072..8ed056f 100644
--- a/smtp-tls.c
+++ b/smtp-tls.c
@@ -325,9 +325,25 @@ starttls_init_ctx (smtp_session_t session, SSL_CTX *ctx)
 
   /* Load the CAs we trust */
   if (cafile != NULL || capath != NULL)
-    SSL_CTX_load_verify_locations (ctx, cafile, capath);
+  {
+	  if(SSL_CTX_load_verify_locations (ctx, cafile, capath)!=1)
+	  {
+		  if (session->event_cb != NULL)
+			  (*session->event_cb) (session, SMTP_EV_UNUSABLE_CA_LIST,
+					  session->event_cb_arg, NULL);
+		  return 0;
+	  }
+  }
   else
-    SSL_CTX_set_default_verify_paths (ctx);
+  {
+	  if(SSL_CTX_set_default_verify_paths (ctx)!=1)
+	  {
+		  if (session->event_cb != NULL)
+			  (*session->event_cb) (session, SMTP_EV_UNUSABLE_CA_LIST,
+					  session->event_cb_arg, NULL);
+		  return 0;
+	  }
+  }
 
   return 1;
 }
@@ -421,6 +437,7 @@ starttls_create_ssl (smtp_session_t session)
       if (session->event_cb != NULL)
 	(*session->event_cb) (session, SMTP_EV_UNUSABLE_CLIENT_CERTIFICATE,
 			      session->event_cb_arg, NULL);
+      SSL_free(ssl);
       return NULL;
     }
 
-- 
2.34.1

