From 7628bd91d58f00caf0109512a8792f07d7b2aa0b Mon Sep 17 00:00:00 2001
From: Yaswanth <yaswanthsai@ami.com>
Date: Mon, 15 Apr 2024 11:12:30 +0530
Subject: [PATCH 1/1] Fix for sensorlist timeout

---
 include/ipmid/utils.hpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/include/ipmid/utils.hpp b/include/ipmid/utils.hpp
index 1e2005f..62de855 100644
--- a/include/ipmid/utils.hpp
+++ b/include/ipmid/utils.hpp
@@ -28,9 +28,9 @@ constexpr auto METHOD_GET = "Get";
 constexpr auto METHOD_GET_ALL = "GetAll";
 constexpr auto METHOD_SET = "Set";
 
-/* Use a value of 5s which aligns with BT/KCS bridged timeouts, rather
+/* Use a value of 10s which aligns with BT/KCS bridged timeouts, rather
  * than the default 25s D-Bus timeout. */
-constexpr std::chrono::microseconds IPMI_DBUS_TIMEOUT = 5s;
+constexpr std::chrono::microseconds IPMI_DBUS_TIMEOUT = 10s;
 
 /** @class ServiceCache
  *  @brief Caches lookups of service names from the object mapper.
-- 
2.34.1

