From 49012007c31daf4760bf0e759210067b6ecf92ad Mon Sep 17 00:00:00 2001
From: shamim ali <ashamim@ami.com>
Date: Fri, 23 May 2025 02:33:59 -0400
Subject: [PATCH] Add remote mac address

Signed-off-by: shamim ali <ashamim@ami.com>
---
 apphandler.cpp | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/apphandler.cpp b/apphandler.cpp
index 81faf72..e19816b 100644
--- a/apphandler.cpp
+++ b/apphandler.cpp
@@ -1146,8 +1146,7 @@ static constexpr uint8_t macAddrLen = 6;
  */
 using SessionDetails =
     std::tuple<uint2_t, uint6_t, uint4_t, uint4_t, uint4_t, uint4_t, uint32_t,
-               std::array<uint8_t, macAddrLen>, uint16_t>;
-
+		std::vector<uint8_t>, uint16_t>;
 /** @brief get session details for a given session
  *
  *  @param[in] ctx - ipmi::Context pointer for accessing D-Bus
@@ -1195,6 +1194,8 @@ ipmi::Cc getSessionDetails(ipmi::Context::ptr ctx, const std::string& service,
         std::get<5>(details) = rmcpPlusProtocol;
         std::get<6>(details) =
             ipmi::mappedVariant<uint32_t>(sessionProps, "RemoteIPAddr", 0);
+	std::get<7>(details) = ipmi::mappedVariant<std::vector<uint8_t>>(
+            sessionProps, "RemoteMACAddress", std::vector<uint8_t>(6, 0));
         // std::get<7>(details) = {{0}}; // default constructed to all 0
         std::get<8>(details) =
             ipmi::mappedVariant<uint16_t>(sessionProps, "RemotePort", 0);
-- 
2.34.1

