diff --git a/transporthandler.cpp b/transporthandler.cpp
old mode 100755
new mode 100644
index e3a95b3..b425537
--- a/transporthandler.cpp
+++ b/transporthandler.cpp
@@ -398,14 +398,7 @@ void reconfigureIfAddr4(sdbusplus::bus_t& bus, const ChannelParams& params,
         deleteObjectIfExists(bus, params.service, ifaddr->path);
     }

-    if (struct in_addr nullIPv4{0};
-        (address == std::nullopt && prefix != std::nullopt) ||
-        (address != std::nullopt &&
-         (address.value().s_addr != nullIPv4.s_addr)))
-    {
-        createIfAddr<AF_INET>(bus, params, address.value_or(ifaddr->address),
-                              prefix.value_or(fallbackPrefix));
-    }
+    createIfAddr<AF_INET>(bus, params, address.value_or(ifaddr->address), prefix.value_or(fallbackPrefix));
 }

 template <int family>
@@ -1031,6 +1031,7 @@ RspType<> setLan(Context::ptr ctx, uint4_t channelBits, uint4_t reserved1,
                 case IPSrc::Unspecified:
                 case IPSrc::Static:
                 {
+                    channelCall<reconfigureIfAddr4>(channel, std::nullopt, std::nullopt);
                     channelCall<setDHCPv4Property>(
                         channel, EthernetInterface::DHCPConf::none);
                     return responseSuccess();
