diff -Naur a/transporthandler.cpp b/transporthandler.cpp
--- a/transporthandler.cpp	2023-09-05 22:39:47.804481547 -0700
+++ b/transporthandler.cpp	2023-09-05 01:07:52.000000000 -0700
@@ -1248,7 +1248,7 @@
         case LanParam::Gateway1MAC:
         {
             log<level::ERR>("Set Lan - Not allow to set gateway MAC Address");
-            return responseInvalidCommand();
+            return response(ipmiCCWriteReadParameter);
         }
         case LanParam::VLANId:
         {
