From 90b243490cfcf4488a36bdf63c0a49392b195672 Mon Sep 17 00:00:00 2001
From: pandiarajt <pandiarajt@ami.com>
Date: Fri, 8 Sep 2023 13:41:41 +0530
Subject: [PATCH 1153/1153] Fix for setting SOL Conf Parameter 0, 3 and 4

---
 transporthandler.cpp | 25 +++++++++++++++++++++----
 1 file changed, 21 insertions(+), 4 deletions(-)

diff --git a/transporthandler.cpp b/transporthandler.cpp
index 530b1dd..a123d39 100644
--- a/transporthandler.cpp
+++ b/transporthandler.cpp
@@ -2506,10 +2506,16 @@ ipmi::RspType<> setSOLConfParams(ipmi::Context::ptr ctx, uint4_t chNum,
     {
         case sol::Parameter::progress:
         {
-            if (configParamData2 || !(configParamData1 <= progressMask))
+            if (configParamData2)
             {
                 return ipmi::responseReqDataLenInvalid();
             }
+
+            if ( !(configParamData1 < progressMask) )
+            {
+                return ipmi::responseInvalidFieldRequest();
+            }
+
             uint8_t progress = configParamData1 & progressMask;
             ipmi::Value currentProgress = 0;
             if (getSOLParameter("Progress", currentProgress, channelNum) < 0)
@@ -2522,6 +2528,11 @@ ipmi::RspType<> setSOLConfParams(ipmi::Context::ptr ctx, uint4_t chNum,
                 return ipmi::responseSetInProgressActive();
             }
 
+            if (progress == 2)
+            {
+                return ipmi::responseParmNotSupported();
+            }
+
             if (setSOLParameter("Progress", progress, channelNum) < 0)
             {
                 return ipmi::responseUnspecifiedError();
@@ -2572,11 +2583,11 @@ ipmi::RspType<> setSOLConfParams(ipmi::Context::ptr ctx, uint4_t chNum,
         }
         case sol::Parameter::accumulate:
         {
-            if (!(configParamData1 == commonMask && configParamData2 == commonMask))
+            if (!configParamData2)
             {
                 return ipmi::responseReqDataLenInvalid();
             }
-            if (*configParamData2 == 0)
+            if (configParamData1 == 0 || *configParamData2 == 0)
             {
                 return ipmi::responseInvalidFieldRequest();
             }
@@ -2593,10 +2604,16 @@ ipmi::RspType<> setSOLConfParams(ipmi::Context::ptr ctx, uint4_t chNum,
         }
         case sol::Parameter::retry:
         {
-            if (!(configParamData1 <= retryMask && configParamData2 == commonMask))
+            if(!(configParamData2))
             {
                 return ipmi::responseReqDataLenInvalid();
             }
+
+            if (configParamData1 > retryMask)
+            {
+                return ipmi::responseInvalidFieldRequest();
+            }
+
             if ((setSOLParameter(
                      "RetryCount",
                      static_cast<uint8_t>(configParamData1 & retryMask),
-- 
2.25.1

