From cae208015972f7c2e852640b6669a6c245f61867 Mon Sep 17 00:00:00 2001
From: ramsankarr <ramsnkarr@ami.com>
Date: Thu, 7 Sep 2023 17:21:09 +0530
Subject: [PATCH 1151/1151] userNameValidationBugFixOnFirstChar

---
 user_channel/user_mgmt.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/user_channel/user_mgmt.cpp b/user_channel/user_mgmt.cpp
index 2e6e841..5a291cb 100644
--- a/user_channel/user_mgmt.cpp
+++ b/user_channel/user_mgmt.cpp
@@ -565,7 +565,7 @@ bool UserAccess::isValidUserName(const std::string& userName)
         return false;
     }
     if (!std::regex_match(userName.c_str(),
-                          std::regex("[a-zA-z_][a-zA-Z_0-9]*")))
+                          std::regex("^[A-Za-z][A-Za-z0-9_]*$")))
     {
         log<level::ERR>("Unsupported characters in user name");
         return false;
-- 
2.25.1

