From ae9debdbecdc1b7bfbb9850fd02d747c237cf756 Mon Sep 17 00:00:00 2001
From: mvinod <mvinod@ami.com>
Date: Fri, 8 Sep 2023 12:58:07 +0530
Subject: [PATCH 1151/1151] Fix for setting SOL Conf Parameter 2

---
 transporthandler.cpp | 30 ++++++++++++++++++++----------
 1 file changed, 20 insertions(+), 10 deletions(-)

diff --git a/transporthandler.cpp b/transporthandler.cpp
index 37223e1..0214638 100644
--- a/transporthandler.cpp
+++ b/transporthandler.cpp
@@ -2460,33 +2460,43 @@ ipmi::RspType<> setSOLConfParams(ipmi::Context::ptr ctx, uint4_t chNum,
         }
         case sol::Parameter::authentication:
         {
-            if (configParamData2 || (configParamData1 <= commonMask) || !(configParamData1 <= privileageMask))
+            if (configParamData2)
             {
                 return ipmi::responseReqDataLenInvalid();
             }
-            uint8_t encrypt = (configParamData1 & encryptMask) >> encryptShift;
-            uint8_t auth = (configParamData1 & authMask) >> authShift;
+            if ((configParamData1 & 0x30) != 0)
+            {
+                return ipmi::responseInvalidFieldRequest();
+            }
+            bool encrypt = (configParamData1 & encryptMask) >> encryptShift;
+            bool auth = (configParamData1 & authMask) >> authShift;
             uint8_t privilege = configParamData1 & privilegeMask;
             // For security considering encryption and authentication must be
             // true.
-            if (!encrypt || !auth)
+            /*if (!encrypt || !auth)
             {
                 return ipmi::responseSystemInfoParameterSetReadOnly();
-            }
-            else if (privilege <
+            }*/
+            if (privilege <
                          static_cast<uint8_t>(sol::Privilege::userPriv) ||
                      privilege > static_cast<uint8_t>(sol::Privilege::oemPriv))
             {
                 return ipmi::responseInvalidFieldRequest();
             }
-
+            if (setSOLParameter("ForceEncryption", encrypt, channelNum) < 0)
+            {
+                return ipmi::responseUnspecifiedError();
+            }
+            if (setSOLParameter("ForceAuthentication", auth, channelNum) < 0)
+            {
+                return ipmi::responseUnspecifiedError();
+            }
             if (setSOLParameter("Privilege", privilege, channelNum) < 0)
             {
                 return ipmi::responseUnspecifiedError();
             }
-
-            break;
-        }
+            break;  
+  	}
         case sol::Parameter::accumulate:
         {
             if (!(configParamData1 == commonMask && configParamData2 == commonMask))
-- 
2.42.0

