From 12a039362684db40a3329f73c48d57b586f22121 Mon Sep 17 00:00:00 2001
From: Ethan Wei <ethanwei@ami.com>
Date: Tue, 3 Oct 2023 19:24:54 -0700
Subject: [PATCH] Fix the Info of IPv6 Static Route Missing after changing to Dynamic Route

---
 transporthandler.cpp | 14 --------------
 1 file changed, 14 deletions(-)

diff --git a/transporthandler.cpp b/transporthandler.cpp
index 530b1dd..f45f27e 100644
--- a/transporthandler.cpp
+++ b/transporthandler.cpp
@@ -883,20 +883,6 @@ void setIPv6StaticRtr(sdbusplus::bus_t& bus, const ChannelParams& params,
 {
     setDbusProperty(bus, params.service, params.logicalPath, INTF_ETHERNET,
                     "IPv6EnableStaticRtr", enabled);
-
-    if (!enabled) {
-        auto oldStaticAddr = getStaticRtrAddr<AF_INET6>(bus, params);
-        std::optional<IfNeigh<AF_INET6>> neighbor;
-        if (!oldStaticAddr.empty())
-        {
-            ObjectLookupCache neighbors(bus, params, INTF_NEIGHBOR);
-            neighbor = findStaticNeighbor<AF_INET6>(bus, params, maybeStringToAddr<AF_INET6>(oldStaticAddr.c_str()).value(), neighbors);
-            if (neighbor)
-            {
-                deleteObjectIfExists(bus, params.service, neighbor->path);
-            }
-        }
-    }
 }
 
 /**
-- 
2.25.1

