From f9fb266db9eaef4258a7728cc28bedf0994006fc Mon Sep 17 00:00:00 2001
From: Ethan Wei <ethanwei@ami.com>
Date: Fri, 17 Nov 2023 00:30:21 -0800
Subject: [PATCH] Fix IPMI Shows Wrong IPv4 Source 	modified:  
 transporthandler.cpp

---
 transporthandler.cpp | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/transporthandler.cpp b/transporthandler.cpp
index f745e59..ac6e01b 100644
--- a/transporthandler.cpp
+++ b/transporthandler.cpp
@@ -199,6 +199,9 @@ void setDHCPv4Property(sdbusplus::bus_t& bus, const ChannelParams& params,
         if ((currentDhcp == EthernetInterface::DHCPConf::v6) ||
             (currentDhcp == EthernetInterface::DHCPConf::both))
             nextDhcp = EthernetInterface::DHCPConf::both;
+        else if ((currentDhcp == EthernetInterface::DHCPConf::v4v6stateless) ||
+                 ((currentDhcp == EthernetInterface::DHCPConf::v6stateless)))
+            nextDhcp = EthernetInterface::DHCPConf::v4v6stateless;
         else if ((currentDhcp == EthernetInterface::DHCPConf::v4) ||
                  (currentDhcp == EthernetInterface::DHCPConf::none))
             nextDhcp = EthernetInterface::DHCPConf::v4;
@@ -208,6 +211,9 @@ void setDHCPv4Property(sdbusplus::bus_t& bus, const ChannelParams& params,
         if ((currentDhcp == EthernetInterface::DHCPConf::v6) ||
             (currentDhcp == EthernetInterface::DHCPConf::both))
             nextDhcp = EthernetInterface::DHCPConf::v6;
+        else if ((currentDhcp == EthernetInterface::DHCPConf::v4v6stateless) ||
+                 ((currentDhcp == EthernetInterface::DHCPConf::v6stateless)))
+            nextDhcp = EthernetInterface::DHCPConf::v6stateless;
         else if ((currentDhcp == EthernetInterface::DHCPConf::v4) ||
                  (currentDhcp == EthernetInterface::DHCPConf::none))
             nextDhcp = EthernetInterface::DHCPConf::none;
@@ -1725,6 +1731,7 @@ RspType<message::Payload> getLan(Context::ptr ctx, uint4_t channelBits,
             EthernetInterface::DHCPConf dhcp =
                 channelCall<getDHCPProperty>(channel);
             if ((dhcp == EthernetInterface::DHCPConf::v4) ||
+                (dhcp == EthernetInterface::DHCPConf::v4v6stateless) ||
                 (dhcp == EthernetInterface::DHCPConf::both))
             {
                 src = IPSrc::DHCP;
-- 
2.25.1

