From 4e258b0ee92d46fa1e54d7e50dcbd8a2718f6694 Mon Sep 17 00:00:00 2001
From: Pranoy Goru <gorumanip@ami.com>
Date: Thu, 1 Aug 2024 11:54:26 +0530
Subject: [PATCH] Added routing table for NVME BRCM MSCC

Signed-off-by: Pranoy Goru <gorumanip@ami.com>
---
 redfish-core/src/redfish.cpp | 95 +++++++++++++++++++++++++++++++++++-
 1 file changed, 93 insertions(+), 2 deletions(-)

diff --git a/redfish-core/src/redfish.cpp b/redfish-core/src/redfish.cpp
index 0794f4a2..c0d2ec4a 100644
--- a/redfish-core/src/redfish.cpp
+++ b/redfish-core/src/redfish.cpp
@@ -62,6 +62,22 @@
 #include "update_service.hpp"
 #include "virtual_media.hpp"
 
+#if BMCWEB_AMI_RAIDBRCM_MACRO
+#include "redfish-core/lib/ext/brcm/storage_brcm.hpp"
+#endif
+
+#if BMCWEB_AMI_RAIDMSCC_MACRO
+#include "redfish-core/lib/ext/mscc/storage_mscc.hpp"
+#endif
+
+#if BMCWEB_AMI_NVME_MACRO
+#include "redfish-core/lib/ext/nvme/storage_nvme.hpp"
+#endif
+
+#if (BMCWEB_AMI_RAIDBRCM_MACRO) || (BMCWEB_AMI_RAIDMSCC_MACRO) || (BMCWEB_AMI_NVME_MACRO)
+#include "redfish-core/lib/ext/storage_ext.hpp"
+#endif
+
 namespace redfish
 {
 
@@ -120,10 +136,85 @@ RedfishService::RedfishService(App& app)
     requestRoutesChassisDrive(app);
     requestRoutesChassisDriveName(app);
     requestRoutesUpdateService(app);
-    requestRoutesStorageCollection(app);
-    requestRoutesStorage(app);
+//    requestRoutesStorageCollection(app);
+//    requestRoutesStorage(app);
     requestRoutesStorageControllerCollection(app);
     requestRoutesStorageController(app);
+
+    #if (BMCWEB_AMI_NVME_MACRO) || (BMCWEB_AMI_RAIDMSCC_MACRO) || (BMCWEB_AMI_RAIDBRCM_MACRO)
+    {
+         redfish::ext::core::resource::requestStorageCollectionRoutes(app);
+         redfish::ext::core::resource::requestRoutesStorage(app);
+    }
+    #else
+    {
+        requestRoutesStorageCollection(app);
+        requestRoutesStorage(app);
+    }
+    #endif
+    #if BMCWEB_AMI_NVME_MACRO
+    {
+        requestRoutesNvmeControllers(app);
+        requestRoutesNvmeControllersInstance(app);
+        requestRoutesNvmePorts(app);
+        requestRoutesNvmePortsInstance(app);
+        requestRoutesNvmeDrive(app);
+    }
+    #endif 
+    #if BMCWEB_AMI_RAIDMSCC_MACRO
+    {
+        requestRoutesRaidLogicalMSCC(app);
+        requestRoutesRaidLogicalDriveMSCC(app);
+        requestRoutesPhysicalDriveMSCC(app);
+        requestRoutesMSCCCreateLogicalDriveAction(app);
+        requestRoutesMSCCDeleteLogicalDriveAction(app);
+        requestRoutesMSCCCreateLogicalDriveOnArrayDriveAction(app);
+        requestRoutesMSCCCreateLuCacheDriveAction(app);
+        requestRoutesMSCCDeleteArrayDriveAction(app);
+        requestRoutesMSCCSetControllerPropertiesAction(app);
+        requestRoutesMSCCAddPhysicalDriveToArrayDriveAction(app);
+        requestRoutesMSCCRemovePhysicalDriveFromArrayDriveAction(app);
+        requestRoutesMSCCAddSpareDriveToArrayDriveAction(app);
+        requestRoutesMSCCRemoveSpareDriveFromArrayDriveAction(app);
+        requestRoutesMSCCStartLocatePhysicalDriveAction(app);
+        requestRoutesMSCCStartLocateArrayDriveAction(app);
+        requestRoutesMSCCStartLocateLogicalDriveAction(app);
+        requestRoutesMSCCStopLocatePhysicalDriveAction(app);
+        requestRoutesMSCCStopLocateArrayDriveAction(app);
+        requestRoutesMSCCStopLocateLogicalDriveAction(app);
+        requestRoutesArrayDriveInstance(app);
+	 requestRoutesMSCCImportConfigFileAction(app);
+	 requestRoutesMSCCExportConfigFileAction(app);
+    }
+    #endif
+    #if BMCWEB_AMI_RAIDBRCM_MACRO
+    {
+        // redfish::brcm::core::resource::requestStorageCollectionRoutes(app);
+        // redfish::brcm::core::resource::requestRoutesStorage(app);
+        requestRoutesDeleteLogicalDriveAction(app);
+        requestRoutesCreateLogicalDriveAction(app);
+        requestRoutesAddHotSparePhysicalDriveAction(app);
+        requestRoutesDeleteHotSparePhysicalDriveAction(app);
+        requestRoutesPreparePhysicalDriveForRemovalAction(app);
+        requestRoutesUndoPreparePhysicalDriveForRemovalAction(app);
+        requestRoutesSetCopybackConfigurationAction(app);
+        requestRoutesRaidLogicalBRCM(app);
+        requestRoutesRaidLogicalDriveBRCM(app);
+        requestRoutesPhysicalDriveBRCM(app);requestRoutesPhysicalDriveBRCMHBA(app);
+        requestRoutesStartCopybackAction(app);
+        requestRoutesStopCopybackAction(app);
+        requestRoutesStartLocatePhysicalDriveAction(app);
+        requestRoutesStopLocatePhysicalDriveAction(app);
+        requestRoutesStartLocateLogicalDriveAction(app);
+        requestRoutesStopLocateLogicalDriveAction(app);
+        requestRoutesSetPhysicalDriveStateAction(app);
+        requestRoutesSetControllerPersonalityAction(app);
+        requestRoutesSetLogicalDrivePropertiesAction(app);
+        requestRoutesSetControllerPropertiesAction(app);
+	 requestRoutesBRCMImportConfigFileAction(app);
+	 requestRoutesBRCMExportConfigFileAction(app);
+    }
+    #endif
     requestRoutesDrive(app);
     requestRoutesCable(app);
     requestRoutesCableCollection(app);
-- 
2.34.1

