From b1138a3713ee8236c682b2751be73a1f9900187e Mon Sep 17 00:00:00 2001
From: saikumaru <saikumaru@ami.com>
Date: Tue, 29 Apr 2025 11:26:59 +0530
Subject: [PATCH] Resolved boot spi error

Signed-off-by: saikumaru <saikumaru@ami.com>
---
 include/configs/evb_ast2600.h | 21 +++++++++++++++++++++
 1 file changed, 21 insertions(+)

diff --git a/include/configs/evb_ast2600.h b/include/configs/evb_ast2600.h
index 4592c85d52..803ab9391b 100644
--- a/include/configs/evb_ast2600.h
+++ b/include/configs/evb_ast2600.h
@@ -16,4 +16,25 @@
 /* Memory Info */
 #define CONFIG_SYS_LOAD_ADDR		0x83000000
 
+/* SPL */
+#define CONFIG_SPL_TEXT_BASE            0x00000000
+#define CONFIG_SPL_MAX_SIZE             0x00010000
+#define CONFIG_SPL_STACK                0x10016000
+#define CONFIG_SPL_BSS_START_ADDR       0x90000000
+#define CONFIG_SPL_BSS_MAX_SIZE         0x00100000
+#define CONFIG_SPL_LOAD_FIT_ADDRESS     0x00010000
+
+/* Extra ENV for Boot Command */
+#define STR_HELPER(n)   #n
+#define STR(n)          STR_HELPER(n)
+
+#undef CONFIG_EXTRA_ENV_SETTINGS
+#define CONFIG_EXTRA_ENV_SETTINGS       \
+        "loadaddr=" STR(CONFIG_SYS_LOAD_ADDR) "\0"      \
+        "bootspi=fdt addr 20100000 && fdt header get fitsize totalsize && cp.b 20100000 ${loadaddr} ${fitsize} && bootm; echo Error loading kernel FIT image\0" \
+        "verify=yes\0"  \
+        "fieldmode=true\0" \
+        ""
+
+
 #endif	/* __CONFIG_H */
-- 
2.47.1

