From 2957c235cd74fbf0a6bfcea240d3ac823c5be0b8 Mon Sep 17 00:00:00 2001
From: sonan <sonan@ami.com>
Date: Wed, 19 Feb 2025 22:48:33 +0530
Subject: [PATCH] Increasing the Poll Rate for CPU Usage

Signed-off-by: sonan <sonan@ami.com>
---
 conf.hpp | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/conf.hpp b/conf.hpp
index 4e024dc..63178e2 100644
--- a/conf.hpp
+++ b/conf.hpp
@@ -62,11 +62,11 @@ struct ControllerInfo
 
 struct CycleTime
 {
-    /* The time interval every cycle. 0.1 seconds by default */
-    uint64_t cycleIntervalTimeMS = 100; // milliseconds
+    /* The time interval every cycle. 0.2 seconds by default */
+    uint64_t cycleIntervalTimeMS = 200; // milliseconds
 
-    /* The interval of updating thermals. 1 second by default */
-    uint64_t updateThermalsTimeMS = 1000; // milliseconds
+    /* The interval of updating thermals. 2 second by default */
+    uint64_t updateThermalsTimeMS = 2000; // milliseconds
 };
 
 /*
-- 
2.43.0

