From 76d396c8751811d712116802cb043b90a164e196 Mon Sep 17 00:00:00 2001
From: pavithra <pavithrat@ami.com>
Date: Wed, 9 Jul 2025 11:36:36 +0530
Subject: [PATCH] Coverity Uninitialized scalar variable

---
 pid/buildjson.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pid/buildjson.cpp b/pid/buildjson.cpp
index feb3bb8..63a1734 100644
--- a/pid/buildjson.cpp
+++ b/pid/buildjson.cpp
@@ -208,7 +208,7 @@ std::pair<std::map<int64_t, conf::PIDConf>, std::map<int64_t, conf::ZoneConfig>>
     {
         int64_t id;
         conf::PIDConf thisZone;
-        conf::ZoneConfig thisZoneConfig;
+        conf::ZoneConfig thisZoneConfig{};
 
         /* TODO: using at() throws a specific exception we can catch */
         id = zone["id"];
-- 
2.34.1

