From 221d74fee122575e88ee0e6e4b6fd0ccec4bae93 Mon Sep 17 00:00:00 2001
From: Manoj Ashok <amanoj@ami.com>
Date: Wed, 28 Feb 2024 11:49:31 +0530
Subject: [PATCH] initialize unintialized var

Signed-off-by: Manoj Ashok <amanoj@ami.com>
---
 peci_cmds.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/peci_cmds.c b/peci_cmds.c
index 797d2b107..7109630fc 100644
--- a/peci_cmds.c
+++ b/peci_cmds.c
@@ -101,7 +101,7 @@ int main(int argc, char* argv[])
     int c;
     int i = 0;
     char* cmd = NULL;
-    EPECIStatus ret;
+    EPECIStatus ret = PECI_CC_INVALID_REQ;
     uint8_t address = 0x30; // use default address of 48d
     uint8_t domainId = 0;   // use default domain ID of 0
     uint8_t u8Size = 4;     // default to a DWORD
-- 
2.34.1

