From 758db8a177802dd22a37c0582af257aea85013a2 Mon Sep 17 00:00:00 2001
From: praveenrajv <praveenrajv@ami.com>
Date: Wed, 13 May 2026 12:53:36 +0530
Subject: [PATCH] Fix-boost-io_service-include

---
 src/sensor_reader.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/sensor_reader.cpp b/src/sensor_reader.cpp
index 4828a29..e7e4e6d 100644
--- a/src/sensor_reader.cpp
+++ b/src/sensor_reader.cpp
@@ -2,7 +2,7 @@
 
 #include <boost/archive/text_iarchive.hpp>
 #include <boost/archive/text_oarchive.hpp>
-#include <boost/asio/io_service.hpp>
+#include <boost/asio/io_context.hpp>
 #include <boost/serialization/list.hpp>
 #include <boost/serialization/map.hpp>
 #include <boost/shared_ptr.hpp>
-- 
2.34.1

