From 9587f82a0d1978a5df932cb7ad07e41cf981459e Mon Sep 17 00:00:00 2001
From: shenglo <shenglo@ami.com>
Date: Mon, 5 Aug 2024 04:38:05 +0800
Subject: [PATCH] Support variant type properties

---
 gen_serialization.mako.hpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/gen_serialization.mako.hpp b/gen_serialization.mako.hpp
index f7e19fc..91fe462 100644
--- a/gen_serialization.mako.hpp
+++ b/gen_serialization.mako.hpp
@@ -8,6 +8,7 @@
 #include <cereal/types/set.hpp>
 #include <cereal/types/string.hpp>
 #include <cereal/types/tuple.hpp>
+#include <cereal/types/variant.hpp>
 #include <cereal/types/vector.hpp>
 % for iface in interfaces:
 #include <${iface.header()}>
-- 
2.45.2

