From 20ee3133464698bd1a2a366fa237af65016b8a28 Mon Sep 17 00:00:00 2001
From: vaibhavs <vaibhavs@ami.com>
Date: Fri, 26 May 2023 01:29:57 +0000
Subject: [PATCH] 0008-Allow-empty-gateway6-when-ipv6-source-is-static

---
 src/ethernet_interface.cpp | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/src/ethernet_interface.cpp b/src/ethernet_interface.cpp
index ab8b815..8c32f06 100644
--- a/src/ethernet_interface.cpp
+++ b/src/ethernet_interface.cpp
@@ -245,9 +245,12 @@ ObjectPath EthernetInterface::ip(IP::Protocol protType, std::string ipaddress,
                 break;
             case IP::Protocol::IPv6:
                 addr = ToAddr<in6_addr>{}(ipaddress);
-                gateway = ToAddr<in6_addr>{}(ipgateway);
                 ip_address::isValidIPv6Addr((in6_addr*)&addr, ip_address::Type::IP6_ADDRESS);
-                ip_address::isValidIPv6Addr((in6_addr*)&gateway, ip_address::Type::GATEWAY6_ADDRESS);
+                if(!ipgateway.empty())
+                {
+                        gateway = ToAddr<in6_addr>{}(ipgateway);
+                        ip_address::isValidIPv6Addr((in6_addr*)&gateway, ip_address::Type::GATEWAY6_ADDRESS);
+                }
                 EthernetInterfaceIntf::defaultGateway6(ipgateway);
                 break;
             default:
-- 
2.25.1
