diff --git a/src/ethernet_interface.cpp b/src/ethernet_interface.cpp
index eb31b46..4ce4346 100644
--- a/src/ethernet_interface.cpp
+++ b/src/ethernet_interface.cpp
@@ -236,10 +236,13 @@ ObjectPath EthernetInterface::ip(IP::Protocol protType, std::string ipaddress,
         {
             case IP::Protocol::IPv4:
                 addr = ToAddr<in_addr>{}(ipaddress);
-                gateway = ToAddr<in_addr>{}(ipgateway);
                 ip_address::isValidIPv4Addr((in_addr*)&addr, ip_address::Type::IP4_ADDRESS);
-                ip_address::isValidIPv4Addr((in_addr*)&gateway, ip_address::Type::GATEWAY4_ADDRESS);
-                EthernetInterfaceIntf::defaultGateway(ipgateway);
+                if(!ipgateway.empty()) {
+                    gateway = ToAddr<in_addr>{}(ipgateway);
+                    ip_address::isValidIPv4Addr((in_addr*)&gateway, ip_address::Type::GATEWAY4_ADDRESS);
+                    EthernetInterfaceIntf::defaultGateway(ipgateway);
+                }
+
                 break;
             case IP::Protocol::IPv6:
                 addr = ToAddr<in6_addr>{}(ipaddress);
