From 9cb85b94552dcc44c318aa9a0307c6359fe050f8 Mon Sep 17 00:00:00 2001
From: vaibhavs <vaibhavs@ami.com>
Date: Mon, 6 Nov 2023 07:20:21 +0000
Subject: [PATCH] Update Bond active slave when all active slaves are down

---
 src/rtnetlink.cpp | 20 ++++++++++++++++++++
 1 file changed, 20 insertions(+)

diff --git a/src/rtnetlink.cpp b/src/rtnetlink.cpp
index a0ee8ea..5994aeb 100644
--- a/src/rtnetlink.cpp
+++ b/src/rtnetlink.cpp
@@ -4,6 +4,7 @@
 #include "util.hpp"
 
 #include <linux/rtnetlink.h>
+#include <fcntl.h>
 
 namespace phosphor::network::netlink
 {
@@ -32,6 +33,7 @@ static void parseBondInfo(InterfaceInfo& info, std::string_view msg)
 {
 
     uint32_t ifindex=0;
+    bool activeSlaveFlag=true;
 
     if (msg.data() == nullptr)
     {
@@ -44,6 +46,7 @@ static void parseBondInfo(InterfaceInfo& info, std::string_view msg)
         {
             case IFLA_BOND_ACTIVE_SLAVE:
 
+		activeSlaveFlag=false;
                 if(!data.empty()){
                     ifindex = stdplus::raw::copyFrom<uint32_t>(data);
                 }
@@ -74,6 +77,23 @@ static void parseBondInfo(InterfaceInfo& info, std::string_view msg)
                 break;
         }
     }
+
+    if(activeSlaveFlag == true)
+    {
+        char buf[1]={0};
+        int fd=open("/sys/class/net/bond0/bonding/active_slave",O_RDONLY);
+        if(fd < 0)
+        {
+	    throw std::runtime_error("Failed to open active_slave file");
+        }
+
+        if(read(fd,(char*)&buf,1) == 0)
+        {
+	    info.bondInfo.emplace("");
+        }
+
+        close(fd);
+   }
 }
 
 static void parseLinkInfo(InterfaceInfo& info, std::string_view msg)
