From 9cb85b94552dcc44c318aa9a0307c6359fe050f8 Mon Sep 17 00:00:00 2001
From: vaibhavs <vaibhavs@ami.com>
Date: Mon, 6 Nov 2023 07:20:21 +0000
Subject: [PATCH] Update Bond active slave when all active slaves are down

---
 src/rtnetlink.cpp | 20 ++++++++++++++++++++
 1 file changed, 20 insertions(+)

diff --git a/src/rtnetlink.cpp b/src/rtnetlink.cpp
index 109091b..1c84a02 100644
--- a/src/rtnetlink.cpp
+++ b/src/rtnetlink.cpp
@@ -3,6 +3,7 @@
 #include "netlink.hpp"
 #include "util.hpp"
 
+#include <fcntl.h>
 #include <linux/rtnetlink.h>
 
 namespace phosphor::network::netlink
@@ -31,6 +32,7 @@ static void parseVlanInfo(InterfaceInfo& info, std::string_view msg)
 static void parseBondInfo(InterfaceInfo& info, std::string_view msg)
 {
     uint32_t ifindex = 0;
+    bool activeSlaveFlag = true;
 
     if (msg.data() == nullptr)
     {
@@ -42,7 +44,7 @@ static void parseBondInfo(InterfaceInfo& info, std::string_view msg)
         switch (hdr.rta_type)
         {
             case IFLA_BOND_ACTIVE_SLAVE:
-
+                activeSlaveFlag = false;
                 if (!data.empty())
                 {
                     ifindex = stdplus::raw::copyFrom<uint32_t>(data);
@@ -75,6 +77,23 @@ static void parseBondInfo(InterfaceInfo& info, std::string_view msg)
                 break;
         }
     }
+
+    if (activeSlaveFlag == true)
+    {
+        char buf[1] = {0};
+        int fd = open("/sys/class/net/bond0/bonding/active_slave", O_RDONLY);
+        if (fd < 0)
+        {
+            throw std::runtime_error("Failed to open active_slave file");
+        }
+
+        if (read(fd, (char*)&buf, 1) == 0)
+        {
+            info.bondInfo.emplace("");
+        }
+
+        close(fd);
+    }
 }
 
 static void parseLinkInfo(InterfaceInfo& info, std::string_view msg)
-- 
2.43.0

