From 70f378353ac4719afb42911952914fe0c3f758cb Mon Sep 17 00:00:00 2001
From: Ethan Wei <ethanwei@ami.com>
Date: Fri, 10 Nov 2023 00:24:04 -0800
Subject: [PATCH] Write VLAN Interface Configuration File when VLAN Interface
 Created

---
 src/ethernet_interface.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/ethernet_interface.cpp b/src/ethernet_interface.cpp
index 037f509..0233fa7 100644
--- a/src/ethernet_interface.cpp
+++ b/src/ethernet_interface.cpp
@@ -1067,7 +1067,7 @@ ObjectPath EthernetInterface::createVLAN(uint16_t id)
     auto vlanIntf = std::make_unique<EthernetInterface>(
         bus, manager, info, objRoot, config::Parser(), nicEnabled());
     ObjectPath ret = vlanIntf->objPath;
-
+    vlanIntf->writeConfigurationFile();
     manager.get().interfaces.emplace(intfName, std::move(vlanIntf));
 
     // write the device file for the vlan interface.
-- 
2.25.1

