From e69943433f732d14e291b88e92c4a56632529680 Mon Sep 17 00:00:00 2001
From: Ethan Wei <ethanwei@ami.com>
Date: Thu, 23 Nov 2023 22:31:48 -0800
Subject: [PATCH] Write Configuration File after Setting Default Gateway

---
 src/ethernet_interface.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/ethernet_interface.cpp b/src/ethernet_interface.cpp
index 0233fa7..77a2aac 100644
--- a/src/ethernet_interface.cpp
+++ b/src/ethernet_interface.cpp
@@ -1625,6 +1625,8 @@ std::string EthernetInterface::defaultGateway(std::string gateway)
                 .prefixLength=prefixLength
             }
         );
+
+        writeConfigurationFile();
         manager.get().reloadConfigs();
     }
     return gateway;
@@ -1641,6 +1643,7 @@ std::string EthernetInterface::defaultGateway6(std::string gateway)
     if (gateway != defaultGateway6())
     {
         gateway = EthernetInterfaceIntf::defaultGateway6(std::move(gateway));
+        writeConfigurationFile();
         manager.get().reloadConfigs();
     }
     return gateway;
-- 
2.25.1

