From 2b6865b395e06e9351cc6bdb23d676a857a98b84 Mon Sep 17 00:00:00 2001
From: vaibhavs <vaibhavs@ami.com>
Date: Wed, 10 Jan 2024 06:11:55 +0000
Subject: [PATCH] Do not allow invalid DNS Server IP Address

---
 src/dns_updater.cpp | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/dns_updater.cpp b/src/dns_updater.cpp
index 99eb6a8..878c7a5 100644
--- a/src/dns_updater.cpp
+++ b/src/dns_updater.cpp
@@ -589,6 +589,13 @@ int16_t Configuration::setDNSServer(std::string interface, std::vector<std::stri
     }
 
     for (auto& ipaddress:servers) {
+        /* DNS Server IP "0.0.0.0" is invalid and hence should be blocked */
+        if(ipaddress.compare("0.0.0.0") == 0)
+        {
+            lg2::error("Invalid IP `{IPADDRESS}`: invalid_argument\n", "IPADDRESS", ipaddress);
+            return -1;
+        }
+
         try {
             stdplus::fromStr<stdplus::InAnyAddr>(ipaddress);
         } catch (std::invalid_argument e) {
