From 80e656c0e86ac593dec21551f781c2940ff5ee4a Mon Sep 17 00:00:00 2001
From: vaibhavs <vaibhavs@ami.com>
Date: Tue, 26 Mar 2024 15:26:20 +0530
Subject: [PATCH] Fix to update static gateway6 when ipv6 source is static

---
 src/ethernet_interface.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/ethernet_interface.cpp b/src/ethernet_interface.cpp
index e851250..44a85dc 100644
--- a/src/ethernet_interface.cpp
+++ b/src/ethernet_interface.cpp
@@ -1441,7 +1441,7 @@ void EthernetInterface::writeConfigurationFile()
                 }
             }
 
-            if (!ipv6AcceptRA() && EthernetInterfaceIntf::ipv6Enable())
+            if (!dhcp6() && EthernetInterfaceIntf::ipv6Enable())
             {
                 auto gateway6 = EthernetInterfaceIntf::defaultGateway6();
                 if (!gateway6.empty())
