diff --git a/src/firewall_configuration.cpp b/src/firewall_configuration.cpp
index 78bdb4f..a46b22b 100644
--- a/src/firewall_configuration.cpp
+++ b/src/firewall_configuration.cpp
@@ -100,15 +100,30 @@ int16_t Configuration::addRule(FirewallIface::Target target, uint8_t control,
         "-A INPUT -j {}",
         target == FirewallIface::Target::ACCEPT ? "ACCEPT" : "DROP");
 
-    if ((control & (uint8_t)ControlBit::PROTOCOL) ==
-        (uint8_t)ControlBit::PROTOCOL)
+    if (startIPAddress.find(":") == std::string::npos)
     {
-        params = params + " -p " +
-                 (protocol == FirewallIface::Protocol::TCP    ? "tcp"
-                  : protocol == FirewallIface::Protocol::UDP  ? "udp"
-                  : protocol == FirewallIface::Protocol::ICMP ? "icmp"
-                                                              : "all");
-    } // if
+        if ((control & (uint8_t)ControlBit::PROTOCOL) ==
+            (uint8_t)ControlBit::PROTOCOL)
+        {
+            params = params + " -p " +
+                     (protocol == FirewallIface::Protocol::TCP    ? "tcp"
+                      : protocol == FirewallIface::Protocol::UDP  ? "udp"
+                      : protocol == FirewallIface::Protocol::ICMP ? "icmp"
+                                                                  : "all");
+        } // if
+    }     // if
+    else if (startIPAddress.find(":") != std::string::npos)
+    {
+        if ((control & (uint8_t)ControlBit::PROTOCOL) ==
+            (uint8_t)ControlBit::PROTOCOL)
+        {
+            params = params + " -p " +
+                     (protocol == FirewallIface::Protocol::TCP    ? "tcp"
+                      : protocol == FirewallIface::Protocol::UDP  ? "udp"
+                      : protocol == FirewallIface::Protocol::ICMP ? "icmpv6"
+                                                                  : "all");
+        } // if
+    }
 
     if ((control & (uint8_t)ControlBit::IP) == (uint8_t)ControlBit::IP)
     {
@@ -282,15 +297,30 @@ int16_t Configuration::delRule(FirewallIface::Target target, uint8_t control,
         "-D INPUT -j {}",
         target == FirewallIface::Target::ACCEPT ? "ACCEPT" : "DROP");
 
-    if ((control & (uint8_t)ControlBit::PROTOCOL) ==
-        (uint8_t)ControlBit::PROTOCOL)
+    if (startIPAddress.find(":") == std::string::npos)
     {
-        params = params + " -p " +
-                 (protocol == FirewallIface::Protocol::TCP    ? "tcp"
-                  : protocol == FirewallIface::Protocol::UDP  ? "udp"
-                  : protocol == FirewallIface::Protocol::ICMP ? "icmp"
-                                                              : "all");
+        if ((control & (uint8_t)ControlBit::PROTOCOL) ==
+            (uint8_t)ControlBit::PROTOCOL)
+        {
+            params = params + " -p " +
+                     (protocol == FirewallIface::Protocol::TCP    ? "tcp"
+                      : protocol == FirewallIface::Protocol::UDP  ? "udp"
+                      : protocol == FirewallIface::Protocol::ICMP ? "icmp"
+                                                                  : "all");
+        }
     } // if
+    else if (startIPAddress.find(":") != std::string::npos)
+    {
+        if ((control & (uint8_t)ControlBit::PROTOCOL) ==
+            (uint8_t)ControlBit::PROTOCOL)
+        {
+            params = params + " -p " +
+                     (protocol == FirewallIface::Protocol::TCP    ? "tcp"
+                      : protocol == FirewallIface::Protocol::UDP  ? "udp"
+                      : protocol == FirewallIface::Protocol::ICMP ? "icmpv6"
+                                                                  : "all");
+        } // if
+    }
 
     if ((control & (uint8_t)ControlBit::IP) == (uint8_t)ControlBit::IP)
     {
-- 
2.43.0