diff --git a/src/firewall_configuration.cpp b/src/firewall_configuration.cpp
index 1f4c1dc..2ed3a3c 100644
--- a/src/firewall_configuration.cpp
+++ b/src/firewall_configuration.cpp
@@ -67,11 +67,20 @@ int16_t Configuration::addRule(FirewallIface::Target target, uint8_t control, Fi
         return -1;
     } // if
     std::string params = fmt::format("-A INPUT -j {}", target == FirewallIface::Target::ACCEPT ? "ACCEPT" : "DROP");
-    if ((control & (uint8_t)ControlBit::PROTOCOL) == (uint8_t)ControlBit::PROTOCOL) {
-        params = params + " -p " + (protocol == FirewallIface::Protocol::TCP ? "tcp" :
-                                    protocol == FirewallIface::Protocol::UDP ? "udp" :
-                                    protocol == FirewallIface::Protocol::ICMP ? "icmp" : "all");
+    if (startIPAddress.find(":") == std::string::npos ) {
+	    if ((control & (uint8_t)ControlBit::PROTOCOL) == (uint8_t)ControlBit::PROTOCOL) {
+		params = params + " -p " + (protocol == FirewallIface::Protocol::TCP ? "tcp" :
+		                            protocol == FirewallIface::Protocol::UDP ? "udp" :
+		                            protocol == FirewallIface::Protocol::ICMP ? "icmp" : "all");
+	    } // if
     } // if
+    else if (startIPAddress.find(":") != std::string::npos ) {
+	    if ((control & (uint8_t)ControlBit::PROTOCOL) == (uint8_t)ControlBit::PROTOCOL) {
+		params = params + " -p " + (protocol == FirewallIface::Protocol::TCP ? "tcp" :
+		                            protocol == FirewallIface::Protocol::UDP ? "udp" :
+		                            protocol == FirewallIface::Protocol::ICMP ? "icmpv6" : "all");
+	    } // if    
+    }
 
     if ((control & (uint8_t)ControlBit::IP) == (uint8_t)ControlBit::IP) {
         std::variant<in_addr, in6_addr> addr1, addr2;
@@ -183,11 +192,20 @@ int16_t Configuration::delRule(FirewallIface::Target target, uint8_t control, Fi
 
     std::string params = fmt::format("-D INPUT -j {}", target == FirewallIface::Target::ACCEPT ? "ACCEPT" : "DROP");
 
-    if ((control & (uint8_t)ControlBit::PROTOCOL) == (uint8_t)ControlBit::PROTOCOL) {
-        params = params + " -p " + (protocol == FirewallIface::Protocol::TCP ? "tcp" :
-                                    protocol == FirewallIface::Protocol::UDP ? "udp" :
-                                    protocol == FirewallIface::Protocol::ICMP ? "icmp" : "all");
+    if (startIPAddress.find(":") == std::string::npos ) {
+	    if ((control & (uint8_t)ControlBit::PROTOCOL) == (uint8_t)ControlBit::PROTOCOL) {
+		params = params + " -p " + (protocol == FirewallIface::Protocol::TCP ? "tcp" :
+		                            protocol == FirewallIface::Protocol::UDP ? "udp" :
+		                            protocol == FirewallIface::Protocol::ICMP ? "icmp" : "all");
+	    } // if
     } // if
+    else if (startIPAddress.find(":") != std::string::npos ) {
+	    if ((control & (uint8_t)ControlBit::PROTOCOL) == (uint8_t)ControlBit::PROTOCOL) {
+		params = params + " -p " + (protocol == FirewallIface::Protocol::TCP ? "tcp" :
+		                            protocol == FirewallIface::Protocol::UDP ? "udp" :
+		                            protocol == FirewallIface::Protocol::ICMP ? "icmpv6" : "all");
+	    } // if  
+    }
 
     if ((control & (uint8_t)ControlBit::IP) == (uint8_t)ControlBit::IP) {
         std::variant<in_addr, in6_addr> addr1, addr2;
@@ -422,4 +440,4 @@ void Configuration::restoreConfigurationFile(){
 
 } // namespace firewall
 } // namespace network
-} // namespace phosphor
\ No newline at end of file
+} // namespace phosphor
