From ca360054b79203ff730663c1d1fc324ea41f5a05 Mon Sep 17 00:00:00 2001
From: Revathir <revathir@ami.com>
Date: Fri, 31 May 2024 12:54:31 +0530
Subject: [PATCH] Clear IPv6 Static Router Address when Static control disabled

---
 src/ethernet_interface.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/ethernet_interface.cpp b/src/ethernet_interface.cpp
index a467b08..3bf87e4 100644
--- a/src/ethernet_interface.cpp
+++ b/src/ethernet_interface.cpp
@@ -2265,6 +2265,11 @@ bool EthernetInterface::ipv6EnableStaticRtr(bool value) {
         return value;
     }
 
+    if (value == false)
+    {
+        EthernetInterfaceIntf::ipv6StaticRtrAddr({});
+    }
+
     EthernetInterfaceIntf::ipv6EnableStaticRtr(value);
     writeConfigurationFile();
     manager.get().reloadConfigs();
