From efc25917de4be1bfe1a65a5a19be4e9f39b7e0cb Mon Sep 17 00:00:00 2001
From: Ethan Wei <ethanwei@ami.com>
Date: Mon, 22 Jul 2024 09:52:32 +0800
Subject: [PATCH] Fix IP & Gateway Missing in Static Source

---
 src/ethernet_interface.cpp | 2 --
 1 file changed, 2 deletions(-)

diff --git a/src/ethernet_interface.cpp b/src/ethernet_interface.cpp
index 9e51ab7..1a4389e 100644
--- a/src/ethernet_interface.cpp
+++ b/src/ethernet_interface.cpp
@@ -268,8 +268,6 @@ EthernetInterface::EthernetInterface(stdplus::PinnedRef<sdbusplus::bus_t> bus,
     registerSignal(bus);
     manager.get().getDNSConf().addInterfaceConf(interfaceName());

-    writeConfigurationFile();
-    manager.get().reloadConfigs();
 }

 void EthernetInterface::updateInfo(const InterfaceInfo& info, bool skipSignal)
--
2.34.1
