From 3a992a0055c268e146fe46f9f50c59983427df80 Mon Sep 17 00:00:00 2001
From: vaibhavs <vaibhavs@ami.com>
Date: Tue, 23 Jul 2024 09:05:05 +0530
Subject: [PATCH] Fix Static Router2 IP When Static Router is Enabled or
 Disabled

---
 src/ethernet_interface.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/ethernet_interface.cpp b/src/ethernet_interface.cpp
index 9e51ab7..66b739f 100644
--- a/src/ethernet_interface.cpp
+++ b/src/ethernet_interface.cpp
@@ -2566,6 +2566,7 @@ bool EthernetInterface::ipv6EnableStaticRtr(bool value)
     if (value == false)
     {
         EthernetInterfaceIntf::ipv6StaticRtrAddr({});
+        EthernetInterfaceIntf::ipv6StaticRtr2Addr({});
     }
 
     EthernetInterfaceIntf::ipv6EnableStaticRtr(value);
