From 05da749e319fc02cebe7f0b72043306623f5b0d8 Mon Sep 17 00:00:00 2001
From: Ethan Wei <ethanwei@ami.com>
Date: Wed, 17 Jul 2024 09:38:59 +0800
Subject: [PATCH] Fix Index of VLAN Interface Not Got

---
 src/ethernet_interface.cpp | 12 +-----------
 1 file changed, 1 insertion(+), 11 deletions(-)

diff --git a/src/ethernet_interface.cpp b/src/ethernet_interface.cpp
index 81b2aa8..fb4dda5 100644
--- a/src/ethernet_interface.cpp
+++ b/src/ethernet_interface.cpp
@@ -2963,18 +2963,8 @@ std::string EthernetInterface::dhcpv6DUID() const
             lg2::error("Failure to create socket");
             elog<InternalFailure>();
         }
-        struct ifreq ifr;
-        std::memset(&ifr, 0, sizeof(ifr));
-        std::strncpy(ifr.ifr_name, interfaceName().c_str(), IFNAMSIZ - 1);
-        if (ioctl(sockfd, SIOCGIFINDEX, &ifr) < 0)
-        {
-            close(sockfd);
-            lg2::error("ioctl call failed to get ifindex");
-            elog<InternalFailure>();
-        }
-        close(sockfd);
-        filename += std::to_string(ifr.ifr_ifindex);

+        filename += std::to_string(ifIdx);
         std::ifstream infile;
         infile.open(filename.c_str(), std::ios::in);
         if (infile.is_open())
--
2.34.1
