From 986ab1a6a91cc9a9ad3466ed7bff06673f03a6cd Mon Sep 17 00:00:00 2001
From: Ethan Wei <ethanwei@ami.com>
Date: Thu, 8 Aug 2024 16:27:18 +0800
Subject: [PATCH] Flush Static Nameserver Addresses when Changing to Dynamic

---
 src/dhcp_configuration.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/dhcp_configuration.cpp b/src/dhcp_configuration.cpp
index 977ec12..e1e4b85 100644
--- a/src/dhcp_configuration.cpp
+++ b/src/dhcp_configuration.cpp
@@ -137,6 +137,10 @@ bool Configuration::dnsEnabled(bool value)
     {
         ConfigIntf::domainEnabled(value);
     }
+    else
+    {
+        parent.get().staticNameServers({});
+    }
     parent.get().writeConfigurationFile();
     parent.get().reloadConfigs();
 
-- 
2.34.1


