From 068c174ba00ec07b7f8dee96cc19358302b39901 Mon Sep 17 00:00:00 2001
From: Ethan Wei <ethanwei@ami.com>
Date: Thu, 1 Aug 2024 17:39:08 +0800
Subject: [PATCH] ICMPv6 Not Work

---
 src/firewall_configuration.cpp | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/src/firewall_configuration.cpp b/src/firewall_configuration.cpp
index b03b390..90a15bc 100644
--- a/src/firewall_configuration.cpp
+++ b/src/firewall_configuration.cpp
@@ -71,7 +71,7 @@ int16_t Configuration::addRule(FirewallIface::Target target, uint8_t control,
         return -1;
     } // if
 
-    if (startIPAddress.find(":") == std::string::npos &&
+    if (startIPAddress.find(".") != std::string::npos &&
         (getRules(FirewallIface::IP::IPV4)).size() >= MAX_RULE_NUM)
     {
         return -1;
@@ -81,6 +81,11 @@ int16_t Configuration::addRule(FirewallIface::Target target, uint8_t control,
     {
         return -1;
     } // else if
+    else if (getRules(FirewallIface::IP::IPV4).size() >= MAX_RULE_NUM ||
+             getRules(FirewallIface::IP::IPV6).size() >= MAX_RULE_NUM)
+    {
+        return -1;
+    } // else if
 
     if (!endIPAddress.empty() &&
         !((startIPAddress.find(":") == std::string::npos &&
@@ -227,6 +232,8 @@ int16_t Configuration::addRule(FirewallIface::Target target, uint8_t control,
     if ((control & (uint8_t)ControlBit::IP) != (uint8_t)ControlBit::IP)
     {
         ret = std::system(fmt::format("iptables {}", params).c_str());
+        if (auto index = params.find("icmp"); index != std::string::npos)
+            params.replace(index, 4, "icmpv6");
         ret |= std::system(fmt::format("ip6tables {}", params).c_str());
     } // if
     else
-- 
2.43.0


