From b34a5ac63126bf91169f602e2debf2ecc74b02d3 Mon Sep 17 00:00:00 2001
From: Ethan Wei <ethanwei@ami.com>
Date: Thu, 1 Aug 2024 15:35:40 +0800
Subject: [PATCH] Remove-Number-Check-When-Deleting-Firewall-Rules

---
 src/firewall_configuration.cpp | 11 -----------
 1 file changed, 11 deletions(-)

diff --git a/src/firewall_configuration.cpp b/src/firewall_configuration.cpp
index a46b22b..b03b390 100644
--- a/src/firewall_configuration.cpp
+++ b/src/firewall_configuration.cpp
@@ -268,17 +268,6 @@ int16_t Configuration::delRule(FirewallIface::Target target, uint8_t control,
         return -1;
     } // if
 
-    if (startIPAddress.find(":") == std::string::npos &&
-        (getRules(FirewallIface::IP::IPV4)).size() >= MAX_RULE_NUM)
-    {
-        return -1;
-    } // if
-    else if (startIPAddress.find(":") != std::string::npos &&
-             (getRules(FirewallIface::IP::IPV6).size()) >= MAX_RULE_NUM)
-    {
-        return -1;
-    } // else if
-
     if (!endIPAddress.empty() &&
         !(startIPAddress.find(":") == std::string::npos &&
               endIPAddress.find(":") == std::string::npos ||
-- 
2.43.0
