From 760988db327a5d3de96569ccca74b60ca427bce5 Mon Sep 17 00:00:00 2001
From: Ethan Wei <ethanwei@ami.com>
Date: Thu, 8 Aug 2024 16:08:40 +0800
Subject: [PATCH] Fix Interface Count Issue

---
 src/system_configuration.cpp | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/system_configuration.cpp b/src/system_configuration.cpp
index fa02278..4893319 100644
--- a/src/system_configuration.cpp
+++ b/src/system_configuration.cpp
@@ -111,9 +111,10 @@ uint8_t SystemConfiguration::interfaceCount() const
          it != manager.get().interfaces.end(); it++)
     {
         if (it->second->interfaceName().find(".") == std::string::npos &&
-            it->second->interfaceName().find_first_of("eth") !=
-                std::string::npos)
+            it->second->interfaceName().find("eth") != std::string::npos)
+        {
             count++;
+        }
     }
 
     return count;
-- 
2.34.1
