From da447b73cc0856f340d3f028111e5c9473bbbe17 Mon Sep 17 00:00:00 2001
From: Ethan Wei <ethanwei@ami.com>
Date: Tue, 13 Aug 2024 19:51:33 +0800
Subject: [PATCH] Clear NCSI Settings before Changing Package and Channel

---
 src/ncsi_configuration.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/ncsi_configuration.cpp b/src/ncsi_configuration.cpp
index 010f0fa..b13b6fd 100644
--- a/src/ncsi_configuration.cpp
+++ b/src/ncsi_configuration.cpp
@@ -72,6 +72,8 @@ int16_t Configuration::setPackageChannel(uint8_t package, uint8_t channel)
 
     if (package != NCSIIface::package() || channel != NCSIIface::channel())
     {
+        clearInterface(eth.getIfIdx());
+        std::this_thread::sleep_for(std::chrono::seconds(1));
         if (setChannel(eth.getIfIdx(), package, channel) != 0)
             return -1;
 
-- 
2.34.1

