From b003c7ce7fe37aaa45276b302254a283998cbe30 Mon Sep 17 00:00:00 2001
From: Ethan Wei <ethanwei@ami.com>
Date: Mon, 26 Aug 2024 14:34:22 +0800
Subject: [PATCH] Change the range the VLAN ID

---
 src/network_manager.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/network_manager.cpp b/src/network_manager.cpp
index fa3f7de..a02a3fb 100644
--- a/src/network_manager.cpp
+++ b/src/network_manager.cpp
@@ -510,7 +510,7 @@ void Manager::removeDefGw(unsigned ifidx, stdplus::InAnyAddr addr)
 
 ObjectPath Manager::vlan(std::string interfaceName, uint32_t id)
 {
-    if (id == 0 || id >= 4095)
+    if (id <= 1 || id >= 4095)
     {
         lg2::error("VLAN ID {NET_VLAN} is not valid", "NET_VLAN", id);
         elog<InvalidArgument>(
-- 
2.34.1
