From 6331459f16f48a996af8dc4ea59753276152364d Mon Sep 17 00:00:00 2001
From: Ethan Wei <ethanwei@ami.com>
Date: Thu, 26 Sep 2024 16:04:30 +0800
Subject: [PATCH] Reconfigure Interface after LinkUp

---
 src/ethernet_interface.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/ethernet_interface.cpp b/src/ethernet_interface.cpp
index 731f4c0..ec005f1 100644
--- a/src/ethernet_interface.cpp
+++ b/src/ethernet_interface.cpp
@@ -3364,6 +3364,11 @@ void EthernetInterface::registerSignal(sdbusplus::bus::bus& bus)
                             }
                         }
                     }
+                    else if (t.first == "OnlineState")
+                    {
+                        if (std::get<std::string>(t.second) == "online")
+                            manager.get().reconfigLink(ifIdx);
+                    }
                 }
             });
         }
-- 
2.43.0


