From 61610cc10efb8a2c9cb8406148df96e012390f05 Mon Sep 17 00:00:00 2001
From: Ethan Wei <ethanwei@ami.com>
Date: Wed, 9 Oct 2024 17:24:58 +0800
Subject: [PATCH] Fix Bond0 Create Failed

---
 src/ethernet_interface.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/ethernet_interface.cpp b/src/ethernet_interface.cpp
index 731f4c0..0f4e29e 100644
--- a/src/ethernet_interface.cpp
+++ b/src/ethernet_interface.cpp
@@ -280,7 +280,7 @@ EthernetInterface::EthernetInterface(stdplus::PinnedRef<sdbusplus::bus_t> bus,
     {
         addStaticNeigh(neigh);
     }
-    if (std::string mac = getMAC(config); !mac.empty())
+    if (std::string mac = getMAC(config); (!mac.empty() && !info.intf.bondInfo))
     {
         system::setNICUp(interfaceName(), false);
         MacAddressIntf::macAddress(mac, true);
--
2.34.1
