From f9d72e170ec941e1f66682189007c751028c0a3e Mon Sep 17 00:00:00 2001
From: sonan <sonan@ami.com>
Date: Wed, 6 Nov 2024 14:23:59 +0530
Subject: [PATCH] Fix for the default NTP Server

Signed-off-by: sonan <sonan@ami.com>
---
 src/ethernet_interface.cpp | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/ethernet_interface.cpp b/src/ethernet_interface.cpp
index 247400b..1612b93 100644
--- a/src/ethernet_interface.cpp
+++ b/src/ethernet_interface.cpp
@@ -1227,8 +1227,7 @@ ServerList EthernetInterface::staticNameServers(ServerList value)
 void EthernetInterface::loadNTPServers(const config::Parser& config)
 {
     EthernetInterfaceIntf::ntpServers(getNTPServerFromTimeSyncd());
-    EthernetInterfaceIntf::staticNTPServers(
-        config.map.getValueStrings("Network", "NTP"));
+    EthernetInterfaceIntf::staticNTPServers({"in.pool.ntp.org"});
 }
 
 void EthernetInterface::loadNameServers(const config::Parser& config)
-- 
2.43.0

