From fff7f7d50ec75b939f6ab365d55dc9db2e27ee09 Mon Sep 17 00:00:00 2001
From: Ethan Wei <ethanwei@ami.com>
Date: Mon, 18 Nov 2024 21:24:24 +0800
Subject: [PATCH] Fix Write Configuration if Bond Enable

---
 src/ethernet_interface.cpp | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/ethernet_interface.cpp b/src/ethernet_interface.cpp
index 70a6a58..693fac5 100644
--- a/src/ethernet_interface.cpp
+++ b/src/ethernet_interface.cpp
@@ -1897,7 +1897,9 @@ void EthernetInterface::writeConfigurationFile()
 
     if ((it != manager.get().interfaces.end()) &&
         (interfaceName().compare(bondIfcName) != 0) &&
-        (interfaceName().compare("usb0") != 0))
+        (interfaceName().compare("usb0") != 0) &&
+        (it->second->linkUp())
+        )
     {
         std::error_code ec{};
         if (fs::exists(config::pathForIntfConf(manager.get().getConfDir(),
-- 
2.34.1


