From 843df5c3378262ee9d85dc032fac4390b8e2880b Mon Sep 17 00:00:00 2001
From: dineshkumar <dineshkumar@ami.com>
Date: Thu, 12 Dec 2024 18:33:07 +0530
Subject: [PATCH] Fix to skip reconfiguration for hostusb when host is powered
 on

---
 src/ethernet_interface.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/ethernet_interface.cpp b/src/ethernet_interface.cpp
index a95ec65..b8f6d1f 100644
--- a/src/ethernet_interface.cpp
+++ b/src/ethernet_interface.cpp
@@ -3375,7 +3375,7 @@ void EthernetInterface::registerSignal(sdbusplus::bus::bus& bus)
                     }
                     else if (t.first == "OnlineState")
                     {
-                        if (std::get<std::string>(t.second) == "online")
+                        if ((std::get<std::string>(t.second) == "online") && (ifIdx != 4))
                             manager.get().reconfigLink(ifIdx);
                     }
                 }
