From 09fe59cf47e48b1f7e7a02ea315d2484ea4a1d42 Mon Sep 17 00:00:00 2001
From: Ethan Wei <ethanwei@ami.com>
Date: Tue, 26 Nov 2024 14:02:39 +0800
Subject: [PATCH] Fix IPv4 Unreachable when IPv6 Disables

---
 src/ethernet_interface.cpp | 18 ++----------------
 1 file changed, 2 insertions(+), 16 deletions(-)

diff --git a/src/ethernet_interface.cpp b/src/ethernet_interface.cpp
index 8432bed..a52d51f 100644
--- a/src/ethernet_interface.cpp
+++ b/src/ethernet_interface.cpp
@@ -2706,22 +2706,8 @@ bool EthernetInterface::ipv6Enable(bool value)
         return value;
     }
 
-    if (value)
-    {
-        EthernetInterfaceIntf::linkLocalAutoConf(LinkLocalConf::v6);
-        EthernetInterfaceIntf::dhcp6(true);
-        EthernetInterfaceIntf::ipv6AcceptRA(true);
-    }
-    else
-    {
-        if (EthernetInterfaceIntf::dhcp6())
-        {
-            EthernetInterfaceIntf::dhcp6(false);
-        } // if
-
-        EthernetInterfaceIntf::linkLocalAutoConf(LinkLocalConf::none);
-        EthernetInterfaceIntf::ipv6AcceptRA(false);
-    }
+    std::system(fmt::format("/bin/echo {} > /proc/sys/net/ipv6/conf/{}/disable_ipv6", 
+                            value ? 1 : 0, interfaceName()).c_str());
 
     EthernetInterfaceIntf::ipv6Enable(value);
     writeConfigurationFile();
-- 
2.34.1

