From c4d03a95ccedd9745d2d3869cf29c6a5a080033b Mon Sep 17 00:00:00 2001
From: Ethan Wei <ethanwei@ami.com>
Date: Thu, 21 Nov 2024 15:25:02 +0800
Subject: [PATCH] Interface Not Up when Nic is Enabled

---
 src/ethernet_interface.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/ethernet_interface.cpp b/src/ethernet_interface.cpp
index 2965dbf..ed6f53e 100644
--- a/src/ethernet_interface.cpp
+++ b/src/ethernet_interface.cpp
@@ -1195,6 +1195,11 @@ bool EthernetInterface::nicEnabled(bool value)
         manager.get().addReloadPreHook(
             [ifname = interfaceName()]() { system::setNICUp(ifname, false); });
     }
+    else
+    {
+        manager.get().addReloadPreHook(
+            [ifname = interfaceName()]() { system::setNICUp(ifname, true); });
+    }
     manager.get().reloadConfigs();

     return value;
--
2.34.1

