From adfe12d3f8901129c6c7e014808db75904933475 Mon Sep 17 00:00:00 2001
From: Ethan Wei <ethanwei@ami.com>
Date: Tue, 26 Nov 2024 14:02:39 +0800
Subject: [PATCH] Fix IPv4 Unreachable when IPv6 Disables

---
 src/ethernet_interface.cpp | 10 ++--------
 1 file changed, 2 insertions(+), 8 deletions(-)

diff --git a/src/ethernet_interface.cpp b/src/ethernet_interface.cpp
index a31aa0f..197743d 100644
--- a/src/ethernet_interface.cpp
+++ b/src/ethernet_interface.cpp
@@ -2705,19 +2705,13 @@ bool EthernetInterface::ipv6Enable(bool value)
     if (value)
     {
         EthernetInterfaceIntf::linkLocalAutoConf(LinkLocalConf::v6);
-        EthernetInterfaceIntf::dhcp6(true);
-        EthernetInterfaceIntf::ipv6AcceptRA(true);
     }
     else
     {
-        if (EthernetInterfaceIntf::dhcp6())
-        {
-            EthernetInterfaceIntf::dhcp6(false);
-        } // if
-
         EthernetInterfaceIntf::linkLocalAutoConf(LinkLocalConf::none);
-        EthernetInterfaceIntf::ipv6AcceptRA(false);
     }
+    std::system(fmt::format("/bin/echo {} > /proc/sys/net/ipv6/conf/{}/disable_ipv6",
+                            value ? 0 : 1, interfaceName()).c_str());
 
     EthernetInterfaceIntf::ipv6Enable(value);
     writeConfigurationFile();
-- 
2.34.1

