diff --git a/src/firewall_configuration.cpp b/src/firewall_configuration.cpp
index 9fb71e4..37bc89a 100644
--- a/src/firewall_configuration.cpp
+++ b/src/firewall_configuration.cpp
@@ -71,6 +71,7 @@ int16_t Configuration::addRule(FirewallIface::Target target, uint8_t control,
                                std::string startTime, std::string stopTime)
 {
     int16_t ret = 0;
+    auto customIPv4rules = 0, customIPv6rules = 0;
     if (control == (uint8_t)ControlBit::TIMEOUT ||
         (control > (uint8_t)ControlBit::TIMEOUT &&
          (control & (uint8_t)ControlBit::IP) == 0 &&
@@ -81,18 +82,22 @@ int16_t Configuration::addRule(FirewallIface::Target target, uint8_t control,
         return -1;
     } // if
 
+    for(const auto& elementTuple : getRules(FirewallIface::IP::IPV4))
+    {
+        if(!std::get<0>(elementTuple)) customIPv4rules++;
+    }
+    for(const auto& elementTuple : getRules(FirewallIface::IP::IPV6))
+    {
+        if(!std::get<0>(elementTuple)) customIPv6rules++;
+    }
+
     if (startIPAddress.find(".") != std::string::npos &&
-        (getRules(FirewallIface::IP::IPV4)).size() >= MAX_RULE_NUM)
+         customIPv4rules >= MAX_RULE_NUM)
     {
         return -1;
     } // if
     else if (startIPAddress.find(":") != std::string::npos &&
-             (getRules(FirewallIface::IP::IPV6).size()) >= MAX_RULE_NUM)
-    {
-        return -1;
-    } // else if
-    else if (getRules(FirewallIface::IP::IPV4).size() >= MAX_RULE_NUM ||
-             getRules(FirewallIface::IP::IPV6).size() >= MAX_RULE_NUM)
+         customIPv6rules >= MAX_RULE_NUM)
     {
         return -1;
     } // else if
