From 44f5be0ac1a0d90fd4e69366d4ea08dc62b25b3e Mon Sep 17 00:00:00 2001
From: krishnar4 <krishnar@ami.com>
Date: Wed, 11 Oct 2023 11:24:19 +0530
Subject: [PATCH] fix bootcount 

---
 src/post_code.cpp | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/src/post_code.cpp b/src/post_code.cpp
index ddabf30..464c051 100644
--- a/src/post_code.cpp
+++ b/src/post_code.cpp
@@ -90,7 +90,14 @@ void PostCode::savePostCodes(postcode_t code)
     {
         firstPostCodeTimeSteady = postCodeTimeSteady;
         firstPostCodeUsSinceEpoch = tsUS; // uS since epoch for 1st post code
-        incrBootCycle();
+        uint16_t count = currentBootCycleCount();
+        std::ifstream isEmpty(postCodeListPath / std::to_string(count),
+                              std::ios::binary);
+        isEmpty.seekg(0, std::ios::end);
+        if (!(isEmpty.tellg() == 8))
+        {
+            incrBootCycle();
+        }
     }
     else
     {
-- 
2.41.0

