From 686c98a9ee3f59536c49e0c2df3ee6903a432827 Mon Sep 17 00:00:00 2001
From: viswanathc <viswanathc@ami.com>
Date: Wed, 9 Jul 2025 13:10:31 -0400
Subject: [PATCH] Ensure proper sequencing of PostCode logs

Signed-off-by: viswanathc <viswanathc@ami.com>
---
 src/post_code.cpp | 15 +--------------
 1 file changed, 1 insertion(+), 14 deletions(-)

diff --git a/src/post_code.cpp b/src/post_code.cpp
index 4151367..3b1f9c2 100644
--- a/src/post_code.cpp
+++ b/src/post_code.cpp
@@ -46,31 +46,18 @@ std::vector<postcode_t> PostCode::getPostCodes(uint16_t index)
                                   std::to_string(maxBootCycleNum()).c_str()));
     }
     std::vector<postcode_t> codesVec;
-    if (1 == index && !postCodes.empty())
-    {
-        std::transform(postCodes.begin(), postCodes.end(),
-                       std::back_inserter(codesVec),
-                       [](const auto& kv) { return kv.second; });
-    }
-    else
-    {
         uint16_t bootNum = getBootNum(index);
 
         decltype(postCodes) codes;
         deserializePostCodes(postCodeListPath / std::to_string(bootNum), codes);
         std::transform(codes.begin(), codes.end(), std::back_inserter(codesVec),
                        [](const auto& kv) { return kv.second; });
-    }
     return codesVec;
 }
 
 std::map<uint64_t, postcode_t> PostCode::getPostCodesWithTimeStamp(
     uint16_t index)
 {
-    if (1 == index && !postCodes.empty())
-    {
-        return postCodes;
-    }
 
     uint16_t bootNum = getBootNum(index);
     decltype(postCodes) codes;
@@ -271,7 +258,7 @@ uint16_t PostCode::getBootNum(const uint16_t index) const
     }
     else
     {
-        bootNum = currentBootCycleIndex - index + 1;
+        bootNum = index ;
     }
     return bootNum;
 }
-- 
2.34.1

