From e3568eb3d4fa8a0394df54f02ac7a563a90d2556 Mon Sep 17 00:00:00 2001
From: Arun R <rarun@ami.com>
Date: Thu, 26 Feb 2026 16:39:29 +0530
Subject: [PATCH] Fix spi driver issue

Signed-off-by: Arun R <rarun@ami.com>
---
 drivers/spi/spi-aspeed-smc.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/spi/spi-aspeed-smc.c b/drivers/spi/spi-aspeed-smc.c
index d6d8fc9032c8..b3ff7d48243f 100644
--- a/drivers/spi/spi-aspeed-smc.c
+++ b/drivers/spi/spi-aspeed-smc.c
@@ -587,6 +587,9 @@ static int do_aspeed_spi_exec_op(struct spi_mem *mem, const struct spi_mem_op *o
 	     op->cmd.opcode == SPINOR_OP_SRST))
 		return ret;
 
+	if (aspi->data->safs_stop)
+                aspi->data->safs_stop(aspi);
+
 	if (op->data.dir == SPI_MEM_DATA_IN) {
 		if (!op->addr.nbytes)
 			ret = aspeed_spi_read_reg(chip, op);
-- 
2.34.1

