From 3bc614038f3a07bded9ec18deea3bcf2ddeb6959 Mon Sep 17 00:00:00 2001
From: saikumaru <saikumaru@ami.com>
Date: Wed, 22 Oct 2025 15:13:19 +0530
Subject: [PATCH] Added the sysfs file creation func for Dual Image support
 2700

Signed-off-by: saikumaru <saikumaru@ami.com>
---
 drivers/spi/spi-aspeed-smc.c | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/drivers/spi/spi-aspeed-smc.c b/drivers/spi/spi-aspeed-smc.c
index 5f63813d7d36..8a2c28ee43a4 100644
--- a/drivers/spi/spi-aspeed-smc.c
+++ b/drivers/spi/spi-aspeed-smc.c
@@ -92,12 +92,12 @@
 
 #define ASPEED_SPI_OP_BUF_LEN		0x10000
 
-#define OFFSET_ABR_CTRL_STATUS 0x64
-#define OFFSET_ABR_TIMER_RELOAD 0x68
-#define OFFSET_ABR_TIMER_RESTART 0x6c
+#define OFFSET_ABR_CTRL_STATUS 0x0c
+#define OFFSET_ABR_TIMER_RELOAD 0x10
+#define OFFSET_ABR_TIMER_RESTART 0x14
 
-#define ABR_WDT_ENABLE BIT(0)
-#define ABR_BOOT_SRC_INDICATE BIT(4)
+#define ABR_WDT_ENABLE BIT(1)
+#define ABR_BOOT_SRC_INDICATE BIT(1)
 #define ABR_RESTART_MAGIC 0x4755
 #define ABR_CLEAR_BOOT_SRC_MAGIC (0xEA << 16)
 #define ABR_RELOAD_MAX_VALUE 0x3ff
@@ -1892,7 +1892,7 @@ static int aspeed_spi_probe(struct platform_device *pdev)
 		goto end;
 	}
 
-	if (aspi->data == &ast2600_fmc_data) {
+	if (aspi->data == &ast2600_fmc_data|| aspi->data == &ast2700_fmc_data) {
 	if (devm_device_add_groups(dev, bcurrent_boot_groups))
 			dev_warn(dev, "Could not add current_boot\n");
 	/* if boot from alt source, show access_primary, otherwise show access_backup */
-- 
2.47.1

