From 190226e44c94fefc957d626ff5cfa7c1428490a8 Mon Sep 17 00:00:00 2001
From: saikumaru <saikumaru@ami.com>
Date: Wed, 22 Oct 2025 17:58:46 +0530
Subject: [PATCH] spi-aspeed-smc-add-ast2700-fmc-forward-declaration

Signed-off-by: saikumaru <saikumaru@ami.com>
---
 drivers/spi/spi-aspeed-smc.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/spi/spi-aspeed-smc.c b/drivers/spi/spi-aspeed-smc.c
index 8a2c28ee43a4..212d1304c33b 100644
--- a/drivers/spi/spi-aspeed-smc.c
+++ b/drivers/spi/spi-aspeed-smc.c
@@ -1144,6 +1144,7 @@ static const struct aspeed_spi_data ast2500_fmc_data;
 static const struct aspeed_spi_data ast2500_spi_data;
 static const struct aspeed_spi_data ast2600_spi_data;
 static const struct aspeed_spi_data ast2600_fmc_data;
+static const struct aspeed_spi_data ast2700_fmc_data;
 
 static int aspeed_spi_set_window(struct aspeed_spi *aspi)
 {
-- 
2.47.1

