From 416175f0f0056b0707ec27bc0ff1027def31c19b Mon Sep 17 00:00:00 2001
From: Ethan Wei <ethanwei@ami.com>
Date: Tue, 18 Feb 2025 15:48:32 +0800
Subject: [PATCH] Fix OT 11492. Remove WARN_ON_ONCE in NCSI

---
 net/ncsi/ncsi-manage.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/net/ncsi/ncsi-manage.c b/net/ncsi/ncsi-manage.c
index 9ae749383160..9f09b8b93459 100644
--- a/net/ncsi/ncsi-manage.c
+++ b/net/ncsi/ncsi-manage.c
@@ -1847,8 +1847,8 @@ void ncsi_stop_dev(struct ncsi_dev *nd)
 			old_state = nc->state;
 			spin_unlock_irqrestore(&nc->lock, flags);
 
-			WARN_ON_ONCE(chained ||
-				     old_state == NCSI_CHANNEL_INVISIBLE);
+			// WARN_ON_ONCE(chained ||
+			// 	     old_state == NCSI_CHANNEL_INVISIBLE);
 		}
 	}
 
-- 
2.34.1

