From 4acb0c35994dd38acb0a670bd194964d26d699cb Mon Sep 17 00:00:00 2001
From: Vijay K S <vijayks@ami.com>
Date: Tue, 17 Mar 2026 10:57:30 +0530
Subject: [PATCH] Fix for dual image hardware failsafe in ast2700

Signed-off-by: Vijay K S <vijayks@ami.com>
---
 drivers/spi/spi-aspeed-smc.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/spi/spi-aspeed-smc.c b/drivers/spi/spi-aspeed-smc.c
index 8c88235c3c36..5cf2ccd007c0 100644
--- a/drivers/spi/spi-aspeed-smc.c
+++ b/drivers/spi/spi-aspeed-smc.c
@@ -100,7 +100,7 @@
 #define ABR_BOOT_SRC_INDICATE BIT(1)
 #define ABR_RESTART_MAGIC 0x4755
 #define ABR_CLEAR_BOOT_SRC_MAGIC (0xEA << 16)
-#define ABR_RELOAD_MAX_VALUE 0x3ff
+#define ABR_RELOAD_MAX_VALUE 22
 #define ABR_WDT_TIMEOUT_STATUS 0x10
 
 #define ABR_MODE_FILE_PATH "/run/media/abr_bootmode"
@@ -338,7 +338,9 @@ static inline ssize_t _access_backup_store(void __iomem *regs,
 	 * 0: Immediately timeout
 	 */
 	count = count < ABR_RELOAD_MAX_VALUE ? count : ABR_RELOAD_MAX_VALUE;
-	
+
+	count = count * 1000000; //convert to microsec for timeout value
+
 	writel(0, wdt_base + OFFSET_ABR_CTRL_STATUS);
 	writel(count, wdt_base + OFFSET_ABR_TIMER_RELOAD);
 
-- 
2.34.1

