From a3b0b76bb8b65becf3f7a9b77ebfa243cbdb4b34 Mon Sep 17 00:00:00 2001
From: santhoshs <santhoshs@ami.com>
Date: Tue, 21 Oct 2025 17:06:47 +0000
Subject: [PATCH] Resolved the undefined reference to Fts5TombstoneArray

Upstream-Status: Pending

Signed-off-by: santhoshs <santhoshs@ami.com>
---
 sqlite3.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/sqlite3.c b/sqlite3.c
index 6abb4ac..1a039aa 100644
--- a/sqlite3.c
+++ b/sqlite3.c
@@ -239101,6 +239101,10 @@ struct Fts5TombstoneArray {
   Fts5Data *apTombstone[1];       /* Array of tombstone pages */
 };
 
+/* Size (in bytes) of an Fts5TombstoneArray holding up to N tombstones */
+#define SZ_FTS5TOMBSTONEARRAY(N) \
+  (offsetof(Fts5TombstoneArray,apTombstone)+(N)*sizeof(Fts5Data*))
+
 /*
 ** Argument is a pointer to an Fts5Data structure that contains a
 ** leaf page.
-- 
2.34.1

