diff -Naur a/mkfru.cpp b/mkfru.cpp
--- a/mkfru.cpp	2022-10-03 10:14:36.139655619 +0530
+++ b/mkfru.cpp	2022-10-03 10:24:52.450621136 +0530
@@ -28,6 +28,10 @@
 constexpr uint8_t eof = 0xc1;
 const std::string intel = "Intel Corporation";
 
+const std::string board_serial = "Intel_3245";
+const std::string product_serial = "Intel_3245";
+
+
 // round up to nearest block size (power of 2)
 constexpr size_t blockRound(size_t len, size_t blk)
 {
@@ -117,7 +121,8 @@
     constexpr size_t pnSize = 10;
     const std::string version = "FRU Ver 0.01";
     size_t contentSize = headerSize + 1 + name.size() + 1 + intel.size() + 1 +
-                         snSize + 1 + pnSize + 1 + version.size() + sizeof(eof);
+                        board_serial.size() + 1 + pnSize + 1 + version.size() + sizeof(eof);
+
     std::vector<uint8_t> data(contentSize);
     size_t offset = 0;
     // chassis type (main server chassis)
@@ -130,7 +135,7 @@
     // product name
     offset += fillStr(data.begin() + offset, name);
     // board sn
-    offset += fillDots(data.begin() + offset, snSize);
+    offset += fillStr(data.begin() + offset, board_serial);
     // board pn
     offset += fillDots(data.begin() + offset, pnSize);
     // fru version string
@@ -150,8 +155,8 @@
     constexpr size_t idSize = 0;
     const std::string version = "FRU Ver 0.01";
     size_t contentSize = headerSize + 1 + intel.size() + 1 + name.size() + 1 +
-                         pnSize + 1 + pvSize + 1 + snSize + 1 + atSize + 1 +
-                         idSize + sizeof(eof);
+                         pnSize + 1 + pvSize + 1 + product_serial.size() + 1 + atSize + 1 +
+                          idSize + sizeof(eof);
     std::vector<uint8_t> data(contentSize);
     size_t offset = 0;
     // chassis type (main server chassis)
@@ -165,7 +170,7 @@
     // product version
     offset += fillDots(data.begin() + offset, pvSize);
     // product serial number
-    offset += fillDots(data.begin() + offset, snSize);
+    offset += fillStr(data.begin() + offset, product_serial);
     // product asset tag
     offset += fillDots(data.begin() + offset, atSize);
     // empty fru file id
