From d74ae53eec0c1c33d8b138c196e3a33ad5033084 Mon Sep 17 00:00:00 2001
From: nirmalkumarv <nirmalkumarv@ami.com>
Date: Wed, 12 Nov 2025 21:10:54 -0800
Subject: [PATCH] Coverity-fix

---
 phosphor-power-supply/util.hpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/phosphor-power-supply/util.hpp b/phosphor-power-supply/util.hpp
index 22e192e..f16cfba 100644
--- a/phosphor-power-supply/util.hpp
+++ b/phosphor-power-supply/util.hpp
@@ -64,7 +64,7 @@ class Util : public UtilBase
         invIntf.emplace(extraIface, PropertyMap());
 
         ObjectMap invObj;
-        invObj.emplace(std::move(invpath), std::move(invIntf));
+        invObj.emplace(invpath, std::move(invIntf));
 
         try
         {
@@ -99,7 +99,7 @@ class Util : public UtilBase
         invProp.emplace(AVAILABLE_PROP, available);
         invIntf.emplace(AVAILABILITY_IFACE, std::move(invProp));
 
-        invObj.emplace(std::move(invpath), std::move(invIntf));
+        invObj.emplace(invpath, std::move(invIntf));
 
         try
         {
-- 
2.43.0

