From 3c2afde0c70a1f509c967c69e835eeaee6373a34 Mon Sep 17 00:00:00 2001
From: nirmalkumarv <nirmalkumarv@ami.com>
Date: Mon, 2 Mar 2026 02:57:47 -0800
Subject: [PATCH] recreated-Coverity-fix

---
 phosphor-power-supply/util.hpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/phosphor-power-supply/util.hpp b/phosphor-power-supply/util.hpp
index 192b9d1..b286b75 100644
--- a/phosphor-power-supply/util.hpp
+++ b/phosphor-power-supply/util.hpp
@@ -61,7 +61,7 @@ class Util : public UtilBase
         invIntf.emplace(extraIface, PropertyMap());
 
         ObjectMap invObj;
-        invObj.emplace(std::move(invpath), std::move(invIntf));
+        invObj.emplace(invpath, std::move(invIntf));
 
         try
         {
@@ -94,7 +94,7 @@ class Util : public UtilBase
         invProp.emplace(AVAILABLE_PROP, available);
         invIntf.emplace(AVAILABILITY_IFACE, std::move(invProp));
 
-        invObj.emplace(std::move(invpath), std::move(invIntf));
+        invObj.emplace(invpath, std::move(invIntf));
 
         try
         {
-- 
2.43.0

