From 0f666243e5822113027259a7be3a4d9283b40a69 Mon Sep 17 00:00:00 2001
From: sanjays <sanjays@ami.com>
Date: Thu, 25 Jun 2026 10:38:04 +0000
Subject: [PATCH] High-Coverity-fixes

---
 json_parser_utils.cpp | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/json_parser_utils.cpp b/json_parser_utils.cpp
index 78064fd..11147f9 100644
--- a/json_parser_utils.cpp
+++ b/json_parser_utils.cpp
@@ -86,8 +86,8 @@ double parseDouble(const nlohmann::json& element,
         const char* first = strValue.data();
         const char* last = strValue.data() + strValue.size();
         double value;
-        auto [ptr, ec] = std::from_chars(first, last, value);
-        if ((ptr == last) && (ec == std::errc()))
+        auto result = std::from_chars(first, last, value);
+        if ((result.ptr == last) && (result.ec == std::errc()))
         {
             return value;
         }
@@ -149,8 +149,8 @@ int parseInteger(const nlohmann::json& element,
         const char* first = strValue.data();
         const char* last = strValue.data() + strValue.size();
         int value;
-        auto [ptr, ec] = std::from_chars(first, last, value);
-        if ((ptr == last) && (ec == std::errc()))
+        auto result = std::from_chars(first, last, value);
+        if ((result.ptr == last) && (result.ec == std::errc()))
         {
             return value;
         }
-- 
2.43.0

