From 1c0254979d7116585083be12ce3730b12e367921 Mon Sep 17 00:00:00 2001
From: desmondwu <desmondwu@ami.com>
Date: Thu, 2 May 2024 16:35:07 +0800
Subject: [PATCH] fix incompatible pointer type of i2c-slave-mqueue

---
 drivers/i2c/i2c-slave-mqueue.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/i2c/i2c-slave-mqueue.c b/drivers/i2c/i2c-slave-mqueue.c
index cf31ca63f1b8..f2356546b4ca 100644
--- a/drivers/i2c/i2c-slave-mqueue.c
+++ b/drivers/i2c/i2c-slave-mqueue.c
@@ -117,8 +117,7 @@ static ssize_t i2c_slave_mqueue_bin_read(struct file *filp,
 	return ret;
 }
 
-static int i2c_slave_mqueue_probe(struct i2c_client *client,
-				  const struct i2c_device_id *id)
+static int i2c_slave_mqueue_probe(struct i2c_client *client)
 {
 	struct device *dev = &client->dev;
 	struct mq_queue *mq;
