From 87f1e747be9099f2a6d4bdc96a2e41a02757b2cf Mon Sep 17 00:00:00 2001
From: madhankumars <madhankumars@ami.com>
Date: Thu, 31 Oct 2024 00:04:27 +0530
Subject: [PATCH 308/308] CheckLockStatus For IpmiUsers

---
 command/rakp12.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/command/rakp12.cpp b/command/rakp12.cpp
index f167339..d777e5e 100644
--- a/command/rakp12.cpp
+++ b/command/rakp12.cpp
@@ -208,7 +208,7 @@ std::vector<uint8_t> RAKP12(const std::vector<uint8_t>& inPayload,
     }
 #ifdef PAM_AUTHENTICATE
     // Check whether user is already locked for failed attempts
-    if (!ipmi::ipmiUserPamAuthenticate(userName, passwd))
+    if ( (ipmi::ipmiUserLockStatus(userName)) || (!ipmi::ipmiUserPamAuthenticate(userName, passwd)))
     {
         lg2::error(
             "Authentication failed - user already locked out, user id: {ID}",
-- 
2.34.1

