From 36d9600b13871eed6579f001772850e699f07d77 Mon Sep 17 00:00:00 2001
From: pandiarajt <pandiarajt@ami.com>
Date: Wed, 4 Feb 2026 12:27:21 +0530
Subject: [PATCH 312/312] Correct SOL console name for obmc-console Dbus objpath

---
 sol/sol_manager.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sol/sol_manager.cpp b/sol/sol_manager.cpp
index 76e1a15..14e9a21 100644
--- a/sol/sol_manager.cpp
+++ b/sol/sol_manager.cpp
@@ -238,7 +238,7 @@ void registerSOLServiceChangeCallback()
     {
         auto servicePath = ipmi::getDbusObject(
             bus, "xyz.openbmc_project.Control.Service.Attributes",
-            "/xyz/openbmc_project/control/service", "_6fbmc_2dconsole");
+            "/xyz/openbmc_project/control/service", "obmc_2dconsole");
 
         if (!std::empty(servicePath.first))
         {
-- 
2.34.1

