
diff --git a/src/state/activating_state.cpp b/src/state/activating_state.cpp
index 7143545..7baf56a 100644
--- a/src/state/activating_state.cpp
+++ b/src/state/activating_state.cpp
@@ -266,6 +266,7 @@ std::unique_ptr<resource::Process>
         // Use curl plugin ...
         "curl",
         // ... to mount http resource at url
+       "sslverify=false",
         "url=" + url,
         // custom OpenBMC path for CA
         "cainfo=", "capath=/etc/ssl/certs/authority", "ssl-version=tlsv1.2",
