From 92e411f28ef2af031865d775188034592394cef3 Mon Sep 17 00:00:00 2001
From: Mohammed Javith Akthar M <mohammedjavitham@ami.com>
Date: Mon, 24 Jul 2023 11:52:23 +0545
Subject: [PATCH 1/2] virtual-media https support

Signed-off-by: Mohammed Javith Akthar M <mohammedjavitham@ami.com>
---
 src/state/activating_state.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/state/activating_state.cpp b/src/state/activating_state.cpp
index 7143545..2e40fc4 100644
--- a/src/state/activating_state.cpp
+++ b/src/state/activating_state.cpp
@@ -266,6 +266,7 @@ std::unique_ptr<resource::Process>
         // Use curl plugin ...
         "curl",
         // ... to mount http resource at url
+       "sslverify=false",
         "url=" + url,
         // custom OpenBMC path for CA
         "cainfo=", "capath=/etc/ssl/certs/authority", "ssl-version=tlsv1.2",
-- 
2.40.1

