From 6129e0eda78880571b060b470bdef386e1dd6379 Mon Sep 17 00:00:00 2001
From: thrishalig <thrishalig@ami.com>
Date: Fri, 28 Jul 2023 11:31:45 +0530
Subject: [PATCH] AST2600 EVB Power Control

Signed-off-by: thrishalig <thrishalig@ami.com>
---
 src/power_control.cpp | 21 ++++++++++++---------
 1 file changed, 12 insertions(+), 9 deletions(-)

diff --git a/src/power_control.cpp b/src/power_control.cpp
index 8be8303..71c172a 100644
--- a/src/power_control.cpp
+++ b/src/power_control.cpp
@@ -2871,8 +2871,8 @@ int main(int argc, char* argv[])
             // if power button is masked, ignore this
             if (!powerButtonMask)
             {
-                sendPowerControlEvent(Event::powerOnRequest);
-                addRestartCause(RestartCause::command);
+                powerOn();
+                 addRestartCause(RestartCause::command);
             }
             else
             {
@@ -2887,8 +2887,10 @@ int main(int argc, char* argv[])
             // if power button is masked, ignore this
             if (!powerButtonMask)
             {
-                sendPowerControlEvent(Event::powerCycleRequest);
-                addRestartCause(RestartCause::command);
+                 setPowerState(PowerState::gracefulTransitionToCycleOff);
+                 gracefulPowerOffTimerStart();
+                 gracefulPowerOff();
+                     addRestartCause(RestartCause::command);
             }
             else
             {
@@ -2903,8 +2905,7 @@ int main(int argc, char* argv[])
             // if reset button is masked, ignore this
             if (!resetButtonMask)
             {
-                sendPowerControlEvent(Event::gracefulPowerCycleRequest);
-                addRestartCause(RestartCause::command);
+              lg2::info("GracefulWarmReboot support not available");
             }
             else
             {
@@ -2919,7 +2920,9 @@ int main(int argc, char* argv[])
             // if reset button is masked, ignore this
             if (!resetButtonMask)
             {
-                sendPowerControlEvent(Event::resetRequest);
+                gracefulPowerOffTimer.cancel();
+                setPowerState(PowerState::on);
+                reset();
                 addRestartCause(RestartCause::command);
             }
             else
@@ -2961,8 +2964,8 @@ int main(int argc, char* argv[])
             // if power button is masked, ignore this
             if (!powerButtonMask)
             {
-                sendPowerControlEvent(Event::powerOffRequest);
-                addRestartCause(RestartCause::command);
+                gracefulPowerOff();
+                 addRestartCause(RestartCause::command);
             }
             else
             {
-- 
2.25.1

