From a47e9fd4d83a62b13c18d0ed483da8a49d46ee6c Mon Sep 17 00:00:00 2001
From: sanjays <sanjays@ami.com>
Date: Thu, 14 May 2026 13:00:08 +0530
Subject: [PATCH] power-status-workaround

---
 src/power_control.cpp | 13 ++++++++++++-
 1 file changed, 12 insertions(+), 1 deletion(-)

diff --git a/src/power_control.cpp b/src/power_control.cpp
index b35a91e..f00a2d7 100644
--- a/src/power_control.cpp
+++ b/src/power_control.cpp
@@ -2047,7 +2047,18 @@ static void powerStateWaitForPowerOK(const Event event)
         }
         case Event::powerOKWatchdogTimerExpired:
             setPowerState(PowerState::off);
-            powerOKFailedLog();
+	    if (powerOkConfig.type == ConfigType::GPIO &&
+                powerOKLine.get_value() ==
+                    static_cast<int>(powerOkConfig.polarity))
+            {
+                lg2::info("powerStateWaitForPowerOK: PS_PWROK already asserted,"
+                          " recovering from missed edge event");
+                sendPowerControlEvent(Event::powerOKAssert);
+            }
+	    else
+	    {
+	      powerOKFailedLog();	    
+	    }
             break;
         case Event::sioPowerGoodAssert:
             powerOKWatchdogTimer.cancel();
-- 
2.34.1

