From 316656f34edd40e1719d0c44f019d8436d3b7a34 Mon Sep 17 00:00:00 2001
From: sanjays <sanjays@ami.com>
Date: Wed, 8 Apr 2026 15:57:31 +0530
Subject: [PATCH] Added try catch

---
 bmc_state_manager.cpp | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/bmc_state_manager.cpp b/bmc_state_manager.cpp
index 12b3bf3..af69ffc 100644
--- a/bmc_state_manager.cpp
+++ b/bmc_state_manager.cpp
@@ -270,7 +270,17 @@ BMC::BMCState BMC::currentBMCState(BMCState value)
     {
         using StateChanged =
             sdbusplus::event::xyz::openbmc_project::state::BMC::StateChanged;
-        lg2::commit(StateChanged("STATE", value));
+	try
+	{
+           lg2::commit(StateChanged("STATE", value));
+	}
+	catch (const std::exception& e)
+        {
+           auto rc = errno;
+           error("Failed to Set BMCState  with errno {ERRNO}",
+           "ERRNO", rc);
+        }
+
     }
 
     return server::BMC::currentBMCState(value);
-- 
2.43.0

