From 4258c41435c0fcbd914f494507417418e8662615 Mon Sep 17 00:00:00 2001
From: shamim-ali <shamima@ami.com>
Date: Thu, 29 Sep 2022 16:14:00 +0530
Subject: [PATCH 10/10] Add warm reset config

Signed-off-by: shamim-ali <shamima@ami.com>
---
 ipmi-whitelist.conf | 1 +
 1 file changed, 1 insertion(+)

diff --git a/ipmi-whitelist.conf b/ipmi-whitelist.conf
index 5a3f154b..d92bbf49 100644
--- a/ipmi-whitelist.conf
+++ b/ipmi-whitelist.conf
@@ -39,6 +39,7 @@
 0x04:0x30:0x7f7f   //<Sensor/Event>:<Set Sensor Reading and Event Status>
 0x06:0x01:0xffff   //<App>:<Get Device ID>
 0x06:0x02:0x7f7f   //<App>:<Cold Reset>
+0x06:0x03:0x7f7f   //<App>:<Warm Reset>
 0x06:0x04:0xff7f   //<App>:<Get Self Test>
 0x06:0x05:0x7f7f   //<App>:<Manufacturing Test Mode>
 0x06:0x06:0xffff   //<App>:<Set ACPI Power State>
-- 
2.25.1

