diff --git a/src/oemcommands.cpp b/src/oemcommands.cpp
index 5196b62..eeedfbe 100644
--- a/src/oemcommands.cpp
+++ b/src/oemcommands.cpp
@@ -97,7 +97,7 @@ static constexpr const char* mediumTypeString = "medium_type";
 static constexpr const char* protocolTypeString = "protocol_type";
 static constexpr const char* sessionSupportedString = "session_supported";
 static constexpr const char* isIpmiString = "is_ipmi";
-static constexpr const char* redfishHostInterfaceChannel = "usb0";
+static constexpr const char* redfishHostInterfaceChannel = "usb1";
 
 // User Manager object in dbus
 static constexpr const char* userMgrObjBasePath = "/xyz/openbmc_project/user";
