From d8f70b93cd7577aee52d403d1cbf16e94585e416 Mon Sep 17 00:00:00 2001
From: robertlin <robertlin@ami.com>
Date: Mon, 14 Aug 2023 17:57:18 +0800
Subject: [PATCH] Support Inband Firmware Update

---
 ipmi-allowlist.conf | 1 +
 1 file changed, 1 insertion(+)

diff --git a/ipmi-allowlist.conf b/ipmi-allowlist.conf
index 8c6b7f1..fa632b6 100644
--- a/ipmi-allowlist.conf
+++ b/ipmi-allowlist.conf
@@ -175,6 +175,7 @@
 0x2c:0x30:0x7f7f   //<Group Extension>:<Set TPM Configuration>
 0x2c:0x31:0x7f7f   //<Group Extension>:<TPM Configuration Update Complete>
 0x2c:0x37:0xff7f   //<Group Extension>:<Read PCIe Cable EEPROM Data>
+0x2e:0x80:0xffff   //<Intel OEM>:<Support Inband firmware update>
 0x2e:0xC0:0x7f7f   //<Intel OEM>:<Enable/Disable Node Manager Policy Control>
 0x2e:0xC1:0x7f7f   //<Intel OEM>:<Set Node Manager Policy>
 0x2e:0xC2:0xff7f   //<Intel OEM>:<Get Node Manager Policy>
