From d4bf596c3910351dba2011d20f5a7f1d0680febc Mon Sep 17 00:00:00 2001
From: krishna <krishnar@ami.com>
Date: Fri, 3 Nov 2023 00:21:14 +0530
Subject: [PATCH 7/7] battery discrete

---
 include/sdrutils.hpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/include/sdrutils.hpp b/include/sdrutils.hpp
index a71c128..e6cfd04 100644
--- a/include/sdrutils.hpp
+++ b/include/sdrutils.hpp
@@ -390,6 +390,7 @@ enum class SensorTypeCodes : uint8_t
     os = 0x20,
     acpisystem = 0x22,
     watchdog2 = 0x23,
+    battery = 0x29,
     other = 0xB,
 };
 
@@ -405,6 +406,7 @@ const static boost::container::flat_map<const char*, SensorTypeCodes, CmpStr>
                  {"os", SensorTypeCodes::os},
                  {"acpisystem", SensorTypeCodes::acpisystem},
                  {"watchdog", SensorTypeCodes::watchdog2},
+                 {"battery", SensorTypeCodes::battery},
                  {"power", SensorTypeCodes::other}}};
 
 inline static std::string getSensorTypeStringFromPath(const std::string& path)
@@ -464,6 +466,7 @@ inline static uint8_t getSensorEventTypeFromPath(const std::string& path)
     uint8_t sensorType = (getSensorTypeFromPath(path));
     if (sensorType == static_cast<uint8_t>(SensorTypeCodes::processor) ||
         sensorType == static_cast<uint8_t>(SensorTypeCodes::watchdog2) ||
+        sensorType == static_cast<uint8_t>(SensorTypeCodes::battery) ||
         sensorType == static_cast<uint8_t>(SensorTypeCodes::powerunit) ||
         sensorType == static_cast<uint8_t>(SensorTypeCodes::acpisystem) ||
         sensorType == static_cast<uint8_t>(SensorTypeCodes::os) ||
-- 
2.34.1

