From bb7bcff3a0dc99900ab35c76552354cade9f390a Mon Sep 17 00:00:00 2001
From: dineshsv <dineshsv@ami.com>
Date: Thu, 28 Dec 2023 13:56:29 +0530
Subject: [PATCH] Setting a sensor upper critical value, affecting other
 threshold values

Signed-off-by: dineshsv <dineshsv@ami.com>
---
 src/sensorcommands.cpp | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/src/sensorcommands.cpp b/src/sensorcommands.cpp
index 20ef45c..9ba3251 100644
--- a/src/sensorcommands.cpp
+++ b/src/sensorcommands.cpp
@@ -844,6 +844,21 @@ ipmi::RspType<> ipmiSenSetSensorThresholds(
     double min = 0;
     getSensorMaxMin(sensorMap, max, min);
 
+    if (lowerNonCriticalThreshMask || lowerCriticalThreshMask)
+    {
+      if(lowerNonCritical < min || lowerCritical < min)
+       {
+          return ipmi::responseInvalidFieldRequest();
+       }
+    }
+    if (upperNonCriticalThreshMask || upperCriticalThreshMask )
+    {
+      if(upperNonCritical > max || upperCritical > max )
+       {
+          return ipmi::responseInvalidFieldRequest();
+       }
+    }
+
     int16_t mValue = 0;
     int16_t bValue = 0;
     int8_t rExp = 0;
-- 
2.25.1

