From 86f00ccb097cacdba9e1e946957f8f2b0fe211c4 Mon Sep 17 00:00:00 2001
From: dineshsv <dineshsv@ami.com>
Date: Mon, 21 Oct 2024 13:22:37 +0530
Subject: [PATCH] Add Error and info limit.

Signed-off-by: dineshsv <dineshsv@ami.com>
---
 log_manager.cpp | 32 ++++++++++++++++++++++++++++++++
 log_manager.hpp |  7 +++++++
 2 files changed, 39 insertions(+)

diff --git a/log_manager.cpp b/log_manager.cpp
index 776c932..58ad52a 100644
--- a/log_manager.cpp
+++ b/log_manager.cpp
@@ -261,6 +261,35 @@ std::string Manager::getSelPolicy()
     return std::get<std::string>(property);
 }
 
+void Manager::updateEntryLimits()
+{
+  bool oldErrorFlag = errorFlag;
+     bool oldInfoFlag = infoFlag;
+ 
+     errorFlag = realErrors.size() >= ERROR_CAP;
+     infoFlag = infoErrors.size() >= ERROR_INFO_CAP;
+ 
+     // Helper function to handle the DBus method call
+     auto sendFlagUpdate = [this](const std::string& flagName, bool flagValue) {
+         auto methodCall = this->busLog.new_method_call(
+             "xyz.openbmc_project.Settings", 
+             "/xyz/openbmc_project/logging/settings",
+             "org.freedesktop.DBus.Properties", 
+             "Set"
+         );
+         methodCall.append("xyz.openbmc_project.Logging.Settings", flagName, std::variant<bool>(flagValue));
+         this->busLog.call(methodCall);
+     };
+ 
+     // Send updates only if there are changes in flags
+     if (oldInfoFlag != infoFlag) {
+         sendFlagUpdate("InfoFlag", infoFlag);
+     }
+     if (oldErrorFlag != errorFlag) {
+         sendFlagUpdate("ErrorFlag", errorFlag);
+     }
+ }
+
 void Manager::createEntry(std::string errMsg, Entry::Level errLvl,
                           std::vector<std::string> additionalData,
                           const FFDCEntries& ffdc)
@@ -372,6 +401,7 @@ void Manager::createEntry(std::string errMsg, Entry::Level errLvl,
     doExtensionLogCreate(*entries.find(entryId)->second, ffdc);
 
     // Note: No need to close the file descriptors in the FFDC.
+    updateEntryLimits();
 }
 
 bool Manager::isQuiesceOnErrorEnabled()
@@ -673,6 +703,8 @@ void Manager::erase(uint32_t entryId)
     {
         lg2::error("Invalid entry ID ({ID}) to delete", "ID", entryId);
     }
+
+    updateEntryLimits();
 }
 
 void Manager::restore()
diff --git a/log_manager.hpp b/log_manager.hpp
index 2ca7b94..55797f5 100644
--- a/log_manager.hpp
+++ b/log_manager.hpp
@@ -249,6 +249,8 @@ class Manager : public details::ServerObject<details::ManagerIface>
     /** @brief Persistent map of Entry dbus objects and their ID */
     std::map<uint32_t, std::unique_ptr<Entry>> entries;
 
+    /** @brief update error and info limit is reached**/
+     virtual void updateEntryLimits();
   private:
     /*
      * @fn _commit()
@@ -341,6 +343,11 @@ class Manager : public details::ServerObject<details::ManagerIface>
     /** @brief Map of entry id to call back object on properties changed */
     std::map<uint32_t, std::unique_ptr<sdbusplus::bus::match_t>>
         propChangedEntryCallback;
+    /** @brief Flag maintained to inform error limit is reached **/
+     bool errorFlag;
+     
+    /** @brief Flag maintained to inform error limit is reached **/
+     bool infoFlag;
 };
 
 } // namespace internal
-- 
2.25.1

