From 7a72d95686159fef3586d91ed6abddf0975469bf Mon Sep 17 00:00:00 2001
From: shamim ali <ashamim@ami.com>
Date: Fri, 30 May 2025 08:25:47 -0400
Subject: [PATCH] changed logs storage path

Signed-off-by: shamim ali <ashamim@ami.com>
---
 config_main.h | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/config_main.h b/config_main.h
index 313b163..b297264 100644
--- a/config_main.h
+++ b/config_main.h
@@ -1,8 +1,8 @@
 #pragma once
 
-const char* ERRLOG_PERSIST_PATH = "/var/lib/phosphor-logging/errors";
-const char* EXTENSION_PERSIST_DIR = "/var/lib/phosphor-logging/extensions";
-const char* ERRLOG_PERSIST_PATH_SEL = "/var/lib/phosphor-logging/ipmi/errors";
-const char* ERRLOG_PERSIST_PATH_RAID = "/var/lib/phosphor-logging/raid/errors";
+const char* ERRLOG_PERSIST_PATH = "/etc/extlog/phosphor-logging/errors";
+const char* EXTENSION_PERSIST_DIR = "/etc/extlog/phosphor-logging/extensions";
+const char* ERRLOG_PERSIST_PATH_SEL = "/etc/extlog/phosphor-logging/ipmi/errors";
+const char* ERRLOG_PERSIST_PATH_RAID = "/etc/extlog/phosphor-logging/raid/errors";
 
 const bool IS_UNIT_TEST = false;
-- 
2.34.1

