From dc978b9ee076375238816b46f9792733db94600f Mon Sep 17 00:00:00 2001
From: shamim ali <ashamim@ami.com>
Date: Sun, 8 Jun 2025 12:47:57 -0400
Subject: [PATCH] Fixed clang format and entryId issues.

Signed-off-by: shamim ali <ashamim@ami.com>
---
 log_manager.cpp | 305 ++++++++++++++++++++++++------------------------
 1 file changed, 152 insertions(+), 153 deletions(-)

diff --git a/log_manager.cpp b/log_manager.cpp
index 64f9090..d4b715d 100644
--- a/log_manager.cpp
+++ b/log_manager.cpp
@@ -220,7 +220,6 @@ void Manager::_commit(uint64_t transactionId [[maybe_unused]],
 
 std::string Manager::getSelPolicy()
 {
-
     DBusSubTree subtree;
 
     auto method = this->busLog.new_method_call(mapperBusName, mapperObjPath,
@@ -281,41 +280,40 @@ void Manager::updateEntryLimits(LogType logType)
     std::string logTypeStr = toString(logType);
 
     bool newErrorFlag = realErrorsMap[logType].size() >= ERROR_CAP;
-    bool newInfoFlag  = infoErrorsMap[logType].size() >= ERROR_INFO_CAP;
+    bool newInfoFlag = infoErrorsMap[logType].size() >= ERROR_INFO_CAP;
 
     // Update internal state
     bool oldErrorFlag = errorFlags[logTypeStr];
-    bool oldInfoFlag  = infoFlags[logTypeStr];
+    bool oldInfoFlag = infoFlags[logTypeStr];
 
     errorFlags[logTypeStr] = newErrorFlag;
-    infoFlags[logTypeStr]  = newInfoFlag;
+    infoFlags[logTypeStr] = newInfoFlag;
 
-    auto updateFlagIfChanged = [this](const std::string& flagName,
-                                      const std::map<std::string, bool>& fullFlagMap,
-                                      bool changed) {
-        if (!changed)
-            return;
+    auto updateFlagIfChanged =
+        [this](const std::string& flagName,
+               const std::map<std::string, bool>& fullFlagMap, bool changed) {
+            if (!changed)
+                return;
 
-        auto methodCall = busLog.new_method_call(
-            "xyz.openbmc_project.Settings",
-            "/xyz/openbmc_project/logging/settings",
-            "org.freedesktop.DBus.Properties",
-            "Set");
+            auto methodCall = busLog.new_method_call(
+                "xyz.openbmc_project.Settings",
+                "/xyz/openbmc_project/logging/settings",
+                "org.freedesktop.DBus.Properties", "Set");
 
-        methodCall.append("xyz.openbmc_project.Logging.Settings",
-                          flagName,
-                          std::variant<std::map<std::string, bool>>(fullFlagMap));
+            methodCall.append(
+                "xyz.openbmc_project.Logging.Settings", flagName,
+                std::variant<std::map<std::string, bool>>(fullFlagMap));
 
-        try
-        {
-            busLog.call(methodCall);
-        }
-        catch (const sdbusplus::exception_t& e)
-        {
-            lg2::error("Failed to update flag {FLAG} via D-Bus: {ERROR}",
-                       "FLAG", flagName, "ERROR", e);
-        }
-    };
+            try
+            {
+                busLog.call(methodCall);
+            }
+            catch (const sdbusplus::exception_t& e)
+            {
+                lg2::error("Failed to update flag {FLAG} via D-Bus: {ERROR}",
+                           "FLAG", flagName, "ERROR", e);
+            }
+        };
 
     updateFlagIfChanged("ErrorFlags", errorFlags, oldErrorFlag != newErrorFlag);
     updateFlagIfChanged("InfoFlags", infoFlags, oldInfoFlag != newInfoFlag);
@@ -325,7 +323,6 @@ void Manager::createEntry(std::string errMsg, Entry::Level errLvl,
                           std::vector<std::string> additionalData,
                           const FFDCEntries& ffdc)
 {
-
     LogType logType = LogType::DEFAULT;
 
     for (const auto& item : additionalData)
@@ -340,100 +337,99 @@ void Manager::createEntry(std::string errMsg, Entry::Level errLvl,
         }
     }
 
-
     // SEL policy check only for IPMI
     if (logType == LogType::IPMI)
     {
+        bool selEnabled;
+        sdbusplus::bus::bus bus = sdbusplus::bus::new_default();
+        auto method =
+            bus.new_method_call(settingService, globalEnblObjpath,
+                                "org.freedesktop.DBus.Properties", "Get");
+        // Append the interface and property name to the method call
+        method.append(globalEnblInterface, "Sel");
 
-    bool selEnabled;
-    sdbusplus::bus::bus bus = sdbusplus::bus::new_default();
-    auto method = bus.new_method_call(settingService, globalEnblObjpath,
-                                      "org.freedesktop.DBus.Properties", "Get");
-    // Append the interface and property name to the method call
-    method.append(globalEnblInterface, "Sel");
-
-    try
-    {
-        auto reply = bus.call(method);
-        // Extract the value from the response
-        std::variant<bool> value;
-        reply.read(value);
-        selEnabled = std::get<bool>(value);
-    }
-    catch (const sdbusplus::exception_t& e)
-    {
-        lg2::error("Failed to get D-Bus property:{ERROR}", "ERROR", e);
-        selEnabled = false;
-    }
-    if (!selEnabled)
-    {
-        lg2::info("SEL is disabled");
-        return;
-    }
-    if (!Extensions::disableDefaultLogCaps())
-    {
-        std::string currentPolicy = getSelPolicy();
-        if (currentPolicy == policyLinear)
+        try
+        {
+            auto reply = bus.call(method);
+            // Extract the value from the response
+            std::variant<bool> value;
+            reply.read(value);
+            selEnabled = std::get<bool>(value);
+        }
+        catch (const sdbusplus::exception_t& e)
+        {
+            lg2::error("Failed to get D-Bus property:{ERROR}", "ERROR", e);
+            selEnabled = false;
+        }
+        if (!selEnabled)
         {
-            if (errLvl < Entry::sevLowerLimit)
+            lg2::info("SEL is disabled");
+            return;
+        }
+        if (!Extensions::disableDefaultLogCaps())
+        {
+            std::string currentPolicy = getSelPolicy();
+            if (currentPolicy == policyLinear)
             {
-                if (realErrorsMap[logType].size() >= ERROR_CAP)
+                if (errLvl < Entry::sevLowerLimit)
                 {
-                    lg2::info(
-                        "Linear SEL: Error Capacity limit reached {ERROR_CAP}",
-                        "ERROR_CAP", ERROR_CAP);
-                    return;
+                    if (realErrorsMap[logType].size() >= ERROR_CAP)
+                    {
+                        lg2::info(
+                            "Linear SEL: Error Capacity limit reached {ERROR_CAP}",
+                            "ERROR_CAP", ERROR_CAP);
+                        return;
+                    }
                 }
-            }
-            else
-            {
-                if (infoErrorsMap[logType].size() >= ERROR_INFO_CAP)
+                else
                 {
-                    lg2::info("Linear SEL: Information Error Capacity limit "
-                              "reached {ERROR_CAP}",
-                              "ERROR_CAP", ERROR_INFO_CAP);
-                    return;
+                    if (infoErrorsMap[logType].size() >= ERROR_INFO_CAP)
+                    {
+                        lg2::info(
+                            "Linear SEL: Information Error Capacity limit "
+                            "reached {ERROR_CAP}",
+                            "ERROR_CAP", ERROR_INFO_CAP);
+                        return;
+                    }
                 }
             }
-        }
-        else
-        {
-            if (errLvl < Entry::sevLowerLimit)
+            else
             {
-                if (realErrorsMap[logType].size() >= ERROR_CAP)
+                if (errLvl < Entry::sevLowerLimit)
                 {
-                    erase(logType, realErrorsMap[logType].front());
+                    if (realErrorsMap[logType].size() >= ERROR_CAP)
+                    {
+                        erase(logType, realErrorsMap[logType].front());
+                    }
                 }
-            }
-            else
-            {
-                if (infoErrorsMap[logType].size() >= ERROR_INFO_CAP)
+                else
                 {
-                    erase(logType, infoErrorsMap[logType].front());
+                    if (infoErrorsMap[logType].size() >= ERROR_INFO_CAP)
+                    {
+                        erase(logType, infoErrorsMap[logType].front());
+                    }
                 }
             }
         }
     }
-
-    }
-     else
+    else
+    {
+        if (errLvl < Entry::sevLowerLimit)
         {
-            if (errLvl < Entry::sevLowerLimit)
+            if (realErrorsMap[logType].size() >= ERROR_CAP)
             {
-                if (realErrorsMap[logType].size() >= ERROR_CAP)
-                {
-                    erase(logType, realErrorsMap[logType].front());
-                }
+                erase(logType, realErrorsMap[logType].front());
             }
-            else
+        }
+        else
+        {
+            if (infoErrorsMap[logType].size() >= ERROR_INFO_CAP)
             {
-                if (infoErrorsMap[logType].size() >= ERROR_INFO_CAP)
-                {
-                    erase(logType, infoErrorsMap[logType].front());
-                }
+                erase(logType, infoErrorsMap[logType].front());
             }
         }
-  
+    }
+
     // Generate new entry ID and update entryIdCounterMap
     entryId = ++entryIdCounterMap[logType];
 
@@ -443,17 +439,17 @@ void Manager::createEntry(std::string errMsg, Entry::Level errLvl,
     if (logType == LogType::RAID)
     {
         objPath = std::string(OBJ_ENTRY_RAID) + "/" + std::to_string(entryId);
-        entryPath = getEntrySerializePath(entryId,ERRLOG_PERSIST_PATH_RAID);
+        entryPath = getEntrySerializePath(entryId, ERRLOG_PERSIST_PATH_RAID);
     }
     else if (logType == LogType::IPMI)
     {
         objPath = std::string(OBJ_ENTRY_SEL) + "/" + std::to_string(entryId);
-        entryPath = getEntrySerializePath(entryId,ERRLOG_PERSIST_PATH_SEL);
+        entryPath = getEntrySerializePath(entryId, ERRLOG_PERSIST_PATH_SEL);
     }
     else
     {
         objPath = std::string(OBJ_ENTRY) + "/" + std::to_string(entryId);
-        entryPath = getEntrySerializePath(entryId,ERRLOG_PERSIST_PATH);
+        entryPath = getEntrySerializePath(entryId, ERRLOG_PERSIST_PATH);
     }
 
     if (errLvl >= Entry::sevLowerLimit)
@@ -463,13 +459,12 @@ void Manager::createEntry(std::string errMsg, Entry::Level errLvl,
     else
     {
         realErrorsMap[logType].push_back(entryId);
-
     }
 
     auto ms = std::chrono::duration_cast<std::chrono::milliseconds>(
                   std::chrono::system_clock::now().time_since_epoch())
                   .count();
-    //auto objPath = std::string(OBJ_ENTRY) + '/' + std::to_string(entryId);
+    // auto objPath = std::string(OBJ_ENTRY) + '/' + std::to_string(entryId);
 
     AssociationList objects{};
     processMetadata(errMsg, additionalData, objects);
@@ -480,34 +475,37 @@ void Manager::createEntry(std::string errMsg, Entry::Level errLvl,
         errLvl, std::move(errMsg), std::move(additionalData),
         std::move(objects), fwVersion, entryPath, *this);
 
-     if (logType == LogType::RAID)
+    if (logType == LogType::RAID)
     {
-        serialize(*e,ERRLOG_PERSIST_PATH_RAID);
+        serialize(*e, ERRLOG_PERSIST_PATH_RAID);
     }
     else if (logType == LogType::IPMI)
     {
-        serialize(*e,ERRLOG_PERSIST_PATH_SEL);
+        serialize(*e, ERRLOG_PERSIST_PATH_SEL);
     }
     else
     {
         serialize(*e);
     }
 
-    entries.insert(std::make_pair(std::make_pair(logType, entryId), std::move(e)));
+    entries.insert(
+        std::make_pair(std::make_pair(logType, entryId), std::move(e)));
 
     if (logType == LogType::DEFAULT)
     {
-
-	// TODO: Currently, `quiesceOnError` and `Pels` are supported for default logging events.
-        // Support for different types of logs can be added in the future.
+        // TODO: Currently, `quiesceOnError` and `Pels` are supported for
+        // default logging events. Support for different types of logs can be
+        // added in the future.
         if (isQuiesceOnErrorEnabled() && (errLvl < Entry::sevLowerLimit) &&
             isCalloutPresent(*e))
         {
             quiesceOnError(entryId);
         }
 
-        // Add entry before calling the extensions so that they have access to it
-        doExtensionLogCreate(*(entries.find(std::make_pair(logType, entryId))->second), ffdc);
+        // Add entry before calling the extensions so that they have access to
+        // it
+        doExtensionLogCreate(
+            *(entries.find(std::make_pair(logType, entryId))->second), ffdc);
     }
 
     // Note: No need to close the file descriptors in the FFDC.
@@ -563,8 +561,8 @@ void Manager::findAndRemoveResolvedBlocks()
 {
     for (auto& entry : entries)
     {
-	const auto& [logType, entryId] = entry.first;
-	if (logType != LogType::DEFAULT)
+        const auto& [logType, entryId] = entry.first;
+        if (logType != LogType::DEFAULT)
         {
             continue;
         }
@@ -751,9 +749,9 @@ void Manager::checkAndRemoveBlockingError(uint32_t entryId)
     return;
 }
 
-void Manager::erase(LogType logType,  uint32_t entryId)
+void Manager::erase(LogType logType, uint32_t entryId)
 {
-//    auto entryFound = entries.find(entryId);
+    //    auto entryFound = entries.find(entryId);
     auto entryFound = entries.find(std::make_pair(logType, entryId));
     if (entries.end() != entryFound)
     {
@@ -778,7 +776,7 @@ void Manager::erase(LogType logType,  uint32_t entryId)
         }
 
         // Delete the persistent representation of these error.
-	fs::path errorPath;
+        fs::path errorPath;
         if (logType == LogType::IPMI)
         {
             errorPath = ERRLOG_PERSIST_PATH_SEL;
@@ -813,23 +811,24 @@ void Manager::erase(LogType logType,  uint32_t entryId)
         entries.erase(entryFound);
 
         if (logType == LogType::DEFAULT)
-	{
-             checkAndRemoveBlockingError(entryId);
-
-             for (auto& remove : Extensions::getDeleteFunctions())
-             {
-                 try
-                 {
-                     remove(entryId);
-                 }
-                 catch (const std::exception& e)
-                 {
-                     lg2::error("An extension's delete function threw an exception: "
-                                "{ERROR}",
-                                "ERROR", e);
-                 }
-             }
-	}
+        {
+            checkAndRemoveBlockingError(entryId);
+
+            for (auto& remove : Extensions::getDeleteFunctions())
+            {
+                try
+                {
+                    remove(entryId);
+                }
+                catch (const std::exception& e)
+                {
+                    lg2::error(
+                        "An extension's delete function threw an exception: "
+                        "{ERROR}",
+                        "ERROR", e);
+                }
+            }
+        }
     }
     else
     {
@@ -876,8 +875,8 @@ void Manager::restore()
         {
             auto id = file.path().filename().c_str();
             auto idNum = std::stol(id);
-            auto e = std::make_unique<Entry>(
-                busLog, objEntry + '/' + id, idNum, *this);
+            auto e = std::make_unique<Entry>(busLog, objEntry + '/' + id, idNum,
+                                             *this);
 
             if (deserialize(file.path(), *e))
             {
@@ -895,7 +894,8 @@ void Manager::restore()
                         realErrorsMap[logType].push_back(idNum);
                     }
 
-                    entries.emplace(std::make_pair(logType, idNum), std::move(e));
+                    entries.emplace(std::make_pair(logType, idNum),
+                                    std::move(e));
                 }
                 else
                 {
@@ -906,24 +906,22 @@ void Manager::restore()
                 }
             }
         }
-    }
-
-    // Set last entryId from DEFAULT only
-    auto it = entries.lower_bound(std::make_pair(LogType::DEFAULT, 0));
-    if (it != entries.end())
-    {
-        // Go to last entry with DEFAULT logType
-        auto lastIt = it;
-        for (; lastIt != entries.end() && lastIt->first.first == LogType::DEFAULT; ++lastIt) {}
-        if (lastIt != it)
+        auto it = entries.lower_bound(std::make_pair(logType, 0));
+        if (it != entries.end())
         {
-            --lastIt;
-            entryId = lastIt->first.second;
+            auto lastIt = it;
+            for (; lastIt != entries.end() && lastIt->first.first == logType;
+                 ++lastIt)
+            {}
+            if (lastIt != it)
+            {
+                --lastIt;
+                entryIdCounterMap[logType] = lastIt->first.second;
+            }
         }
     }
 }
 
-
 std::string Manager::readFWVersion()
 {
     auto version = util::getOSReleaseValue("VERSION_ID");
@@ -981,7 +979,7 @@ uint16_t Manager::eraseLogTypeEntries(std::string& logTypeStr, uint32_t entryId)
     if (entryId == 0)
     {
         // Case 2: Delete all entries of the specified log type
-        for (auto iter = entries.begin(); iter != entries.end(); )
+        for (auto iter = entries.begin(); iter != entries.end();)
         {
             if (iter->first.first == type)
             {
@@ -1011,13 +1009,14 @@ uint16_t Manager::eraseLogTypeEntries(std::string& logTypeStr, uint32_t entryId)
         else
         {
             // Entry not found
-            throw sdbusplus::xyz::openbmc_project::Common::Error::InvalidArgument();
+            throw sdbusplus::xyz::openbmc_project::Common::Error::
+                InvalidArgument();
         }
     }
 
-    if(!entryId)
+    if (!entryId)
     {
-	updateEntryLimits(type);
+        updateEntryLimits(type);
     }
 
     return erasedCount;
-- 
2.34.1

