From fd126d72d300f44f2fe1c6f3a5940276b6050614 Mon Sep 17 00:00:00 2001
From: shamim ali <ashamim@ami.com>
Date: Fri, 25 Jul 2025 05:29:32 -0400
Subject: [PATCH] Add tcp modules enable-disable using  transmission protocol

Signed-off-by: shamim ali <ashamim@ami.com>
---
 phosphor-rsyslog-config/server-conf.cpp | 68 +++++++++++++++++++++++++
 1 file changed, 68 insertions(+)

diff --git a/phosphor-rsyslog-config/server-conf.cpp b/phosphor-rsyslog-config/server-conf.cpp
index 526988a..9a28690 100644
--- a/phosphor-rsyslog-config/server-conf.cpp
+++ b/phosphor-rsyslog-config/server-conf.cpp
@@ -194,6 +194,10 @@ NetworkClient::TransportProtocol
     Server::transportProtocol(NetworkClient::TransportProtocol value)
 {
     TransportProtocol result{};
+    const std::string filePath = "/etc/rsyslog.conf";
+    const std::string tempPath = "/etc/rsyslog.conf.tmp";
+    std::ifstream inFile(filePath);
+    std::ofstream outFile(tempPath);
 
     try
     {
@@ -205,6 +209,70 @@ NetworkClient::TransportProtocol
 
         writeConfig(address(), port(), value, configFilePath.c_str());
         result = NetworkClient::transportProtocol(value);
+
+	if(value == NetworkClient::TransportProtocol::TCP)
+        {
+	   std::string line;
+           while (std::getline(inFile, line))
+           {
+               std::string trimmed = line;
+               trimmed.erase(0, trimmed.find_first_not_of(" \t"));
+
+               if (trimmed == "#module(load=\"imtcp\")" ||
+               trimmed == "#module(load=\"lmnsd_ossl\")")
+              {
+                 auto pos = line.find('#');
+                 if (pos != std::string::npos)
+                 {
+                     line.erase(pos, 1);
+                 }
+              }
+              outFile << line << '\n';
+              if (!outFile)
+              {
+                 log<level::ERR> ("Error writing to temp file");
+		 return result;
+              }
+          }
+
+          inFile.close();
+          outFile.close();
+
+         if (std::rename(tempPath.c_str(), filePath.c_str()) != 0)
+         {
+	     log<level::ERR>("Error replacing original file");
+             std::remove(tempPath.c_str());
+         }
+        }
+        else if (value == NetworkClient::TransportProtocol::UDP)
+        {
+	    std::string line;
+            while (std::getline(inFile, line))
+            {
+                std::string trimmed = line;
+                trimmed.erase(0, trimmed.find_first_not_of(" \t"));
+
+                if (trimmed == "module(load=\"imtcp\")" ||
+                    trimmed == "module(load=\"lmnsd_ossl\")")
+                {
+                    if (trimmed[0] != '#')
+                    {
+                        outFile << "#" << line << "\n";
+                        continue;
+                    }
+                }
+                outFile << line << "\n";
+            }
+
+            inFile.close();
+            outFile.close();
+
+            if (std::rename(tempPath.c_str(), filePath.c_str()) != 0)
+            {
+                log<level::ERR>( "Failed to replace original config file.");
+		return result;
+            }
+        }
     }
     catch (const InternalFailure& e)
     {
-- 
2.34.1

