From 70b04257ebb4890e2eceada19f5e4f56efe0d606 Mon Sep 17 00:00:00 2001
From: vinothkumars <vinothkumars@ami.com>
Date: Tue, 5 Aug 2025 19:26:02 +0530
Subject: [PATCH] Added-ObjectManager-Interface-For-EventEntry

---
 log_manager_main.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/log_manager_main.cpp b/log_manager_main.cpp
index 62c03bb..fbfd50f 100644
--- a/log_manager_main.cpp
+++ b/log_manager_main.cpp
@@ -22,12 +22,14 @@ int main(int /*argc*/, char* /*argv*/[])
 
     // Add sdbusplus ObjectManager for the 'root' path of the logging manager.
     sdbusplus::server::manager_t objManager(bus, OBJ_LOGGING);
+    sdbusplus::server::manager_t objEventManager(bus, OBJ_ENTRY);
     sdbusplus::server::manager_t objSelManager(bus, OBJ_SEL_LOGGING);
     sdbusplus::server::manager_t objRaidManager(bus, OBJ_RAID_LOGGING);
 
 
     phosphor::logging::internal::Manager iMgr(bus, OBJ_INTERNAL);
     phosphor::logging::Manager mgr(bus, OBJ_LOGGING, iMgr);
+    phosphor::logging::Manager eventMgr(bus, OBJ_ENTRY, iMgr);
     phosphor::logging::Manager selMgr(bus, OBJ_SEL_LOGGING, iMgr);
     phosphor::logging::Manager raidMgr(bus, OBJ_RAID_LOGGING, iMgr);
 
-- 
2.34.1

