From d6828b8cb7b62b912091f1e670d5c13c244552ce Mon Sep 17 00:00:00 2001
From: shamim ali <ashamim@ami.com>
Date: Thu, 30 Oct 2025 06:14:51 -0400
Subject: [PATCH] Select storage type for error log storage

Signed-off-by: shamim ali <ashamim@ami.com>
---
 config_main.h | 10 ++++++++++
 meson.build   | 13 +++++++++++++
 meson.options |  9 +++++++++
 3 files changed, 32 insertions(+)

diff --git a/config_main.h b/config_main.h
index fd0ce62..92d0955 100644
--- a/config_main.h
+++ b/config_main.h
@@ -1,9 +1,19 @@
 #pragma once
 
+#ifdef STORAGE_EMMC_SDCARD
 const char* ERRLOG_PERSIST_PATH = "/etc/extlog/phosphor-logging/errors";
 const char* EXTENSION_PERSIST_DIR = "/etc/extlog/phosphor-logging/extensions";
 const char* ERRLOG_PERSIST_PATH_SEL = "/etc/extlog/phosphor-logging/ipmi/errors";
 const char* ERRLOG_PERSIST_PATH_RAID = "/etc/extlog/phosphor-logging/raid/errors";
 const char* IPMI_BACKUP_PATH = "/etc/extlog/phosphor-logging/ipmi_rollover_backup";
+#else
+const char* ERRLOG_PERSIST_PATH = "/var/lib/phosphor-logging/errors";
+const char* EXTENSION_PERSIST_DIR = "/var/lib/phosphor-logging/extensions";
+const char* ERRLOG_PERSIST_PATH_SEL = "/var/lib/phosphor-logging/ipmi/errors";
+const char* ERRLOG_PERSIST_PATH_RAID = "/var/lib/phosphor-logging/raid/errors";
+const char* IPMI_BACKUP_PATH = "/var/lib/phosphor-logging/ipmi_rollover_backup";
+
+
+#endif
 
 const bool IS_UNIT_TEST = false;
diff --git a/meson.build b/meson.build
index 2832504..41ae6f7 100644
--- a/meson.build
+++ b/meson.build
@@ -10,6 +10,19 @@ project('phosphor-logging', 'cpp',
     ],
     version: '1.0.0',
 )
+
+# Read the 'storage-select' option
+storage_type = get_option('storage-select')
+
+# Define compile-time macro based on the selected storage type
+if storage_type == 'emmc_sdcard'
+    message('Building with emmc/sdcard storage support')
+    add_project_arguments('-DSTORAGE_EMMC_SDCARD', language: ['c', 'cpp'])
+else
+    message('Building with SPI (/var/lib) storage support')
+    add_project_arguments('-DSTORAGE_SPI', language: ['c', 'cpp'])
+endif
+
 cpp = meson.get_compiler('cpp')
 
 python_prog = find_program('python3', native: true)
diff --git a/meson.options b/meson.options
index 0dd7c3e..3d96631 100644
--- a/meson.options
+++ b/meson.options
@@ -41,3 +41,12 @@ option(
     value: '/etc/rsyslog.d/server.conf',
     description: 'Path to rsyslog server conf file',
 )
+
+option(
+    'storage-select',
+    type: 'combo',
+    choices: ['emmc_sdcard', 'spi'],
+    value: 'emmc_sdcard',
+    description: 'Select storage type for error log storage: emmc_sdcard or spi',
+)
+
-- 
2.34.1

