From 2121bb072cd6dcd3fc28dd2a56d85dd173480b8a Mon Sep 17 00:00:00 2001
From: Sharath <sharathkumark@ami.com>
Date: Tue, 26 Aug 2025 04:09:04 +0000
Subject: [PATCH] OT-11473 coverity fix

Signed-off-by: Sharath <sharathkumark@ami.com>
---
 include/threshold_event_monitor.hpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/include/threshold_event_monitor.hpp b/include/threshold_event_monitor.hpp
index dd8147c..e9cd37e 100644
--- a/include/threshold_event_monitor.hpp
+++ b/include/threshold_event_monitor.hpp
@@ -51,8 +51,8 @@ inline static sdbusplus::bus::match_t startThresholdAssertMonitor(
         std::string sensorName;
         std::string thresholdInterface;
         std::string event;
-        bool assert;
-        double assertValue;
+        bool assert = false;
+        double assertValue = 0.0;
         try
         {
             msg.read(sensorName, thresholdInterface, event, assert,
@@ -356,7 +356,7 @@ inline static sdbusplus::bus::match_t startThresholdAssertMonitor(
         }
         if (eventType != eventNone)
         {
-            unsigned int recordId;
+            unsigned int recordId = 0;
             std::string selDataStr;
             toHexStr(eventData, selDataStr);
             sdbusplus::message_t AddToLog = conn->new_method_call(
-- 
2.43.0

