From cf3dca9e75b66280c4be149fd1e22ecf9da23cde Mon Sep 17 00:00:00 2001
From: karunaiarasu <karunaiarasu@ami.com>
Date: Wed, 3 Sep 2025 12:31:34 +0530
Subject: [PATCH] crashErrorEventMonitor-coredump-fixed

---
 include/crash_event_monitor.hpp | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/include/crash_event_monitor.hpp b/include/crash_event_monitor.hpp
index 44272fe..722a73c 100644
--- a/include/crash_event_monitor.hpp
+++ b/include/crash_event_monitor.hpp
@@ -37,7 +37,14 @@ inline static sdbusplus::bus::match_t
         sdbusplus::message::object_path jobPath;
         std::string jobUnit{};
         std::string jobResult{};
-        msg.read(jobID, jobPath, jobUnit, jobResult);
+        try
+        {
+           msg.read(jobID, jobPath, jobUnit, jobResult);
+        }
+        catch (const sdbusplus::exception_t& e)
+        {
+              std::cerr << "Failed to read value from " << msg.get_path() << " e= " << e.what() << "\n";
+        }
         std::string test = jobPath.str;
 
         if (jobResult == "failed")
-- 
2.34.1

