From d38ff09e8d77d03018ee9c458387cf9fed9dbe22 Mon Sep 17 00:00:00 2001
From: viswanathc <viswanathc@ami.com>
Date: Wed, 12 Nov 2025 12:14:19 -0500
Subject: [PATCH] High-Coverity-Fix-CM-4-phosphor-sel-logger

Signed-off-by: viswanathc <viswanathc@ami.com>
---
 src/sel_logger.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/sel_logger.cpp b/src/sel_logger.cpp
index 5754fae..03bf821 100644
--- a/src/sel_logger.cpp
+++ b/src/sel_logger.cpp
@@ -450,7 +450,7 @@ static uint16_t selAddSystemRecord(
     {
         throw std::invalid_argument("Event data too large");
     }
-    unsigned int recordId;
+    unsigned int recordId = 0;
     uint8_t sensorType = 0xFF;
     std::string selDataStr;
     toHexStr(selData, selDataStr);
-- 
2.34.1

