From fead272a8fbd54f7f5a1e15d4f79893aea5b2061 Mon Sep 17 00:00:00 2001
From: pavithra <pavithrat@ami.com>
Date: Thu, 21 May 2026 19:32:26 +0530
Subject: [PATCH] Fixed SNMP trap deassert event message to align with SEL list

---
 include/threshold_event_monitor.hpp | 60 ++++++-----------------------
 1 file changed, 12 insertions(+), 48 deletions(-)

diff --git a/include/threshold_event_monitor.hpp b/include/threshold_event_monitor.hpp
index a7a1e70..99c8694 100644
--- a/include/threshold_event_monitor.hpp
+++ b/include/threshold_event_monitor.hpp
@@ -228,66 +228,30 @@ inline static sdbusplus::bus::match_t startThresholdAssertMonitor(
         if (event == "CriticalLow")
         {
             threshold = "critical low";
-            if (assert)
-            {
-                eventType = eventErr;
-                direction = "low";
-                redfishMessageID += ".SensorThresholdCriticalLowGoingLow";
-            }
-            else
-            {
-                eventType = eventInfo;
-                direction = "high";
-                redfishMessageID += ".SensorThresholdCriticalLowGoingHigh";
-            }
+            eventType = eventErr;
+            direction = "low";
+            redfishMessageID += ".SensorThresholdCriticalLowGoingLow";
         }
         else if (event == "WarningLow")
         {
             threshold = "warning low";
-            if (assert)
-            {
-                eventType = eventWarn;
-                direction = "low";
-                redfishMessageID += ".SensorThresholdWarningLowGoingLow";
-            }
-            else
-            {
-                eventType = eventInfo;
-                direction = "high";
-                redfishMessageID += ".SensorThresholdWarningLowGoingHigh";
-            }
+            eventType = eventWarn;
+            direction = "low";
+            redfishMessageID += ".SensorThresholdWarningLowGoingLow";
         }
         else if (event == "WarningHigh")
         {
             threshold = "warning high";
-            if (assert)
-            {
-                eventType = eventWarn;
-                direction = "high";
-                redfishMessageID += ".SensorThresholdWarningHighGoingHigh";
-            }
-            else
-            {
-                eventType = eventInfo;
-                direction = "low";
-                redfishMessageID += ".SensorThresholdWarningHighGoingLow";
-            }
+            eventType = eventWarn;
+            direction = "high";
+            redfishMessageID += ".SensorThresholdWarningHighGoingHigh";
         }
         else if (event == "CriticalHigh")
         {
             threshold = "critical high";
-            if (assert)
-            {
-                eventType = eventErr;
-                direction = "high";
-                redfishMessageID += ".SensorThresholdCriticalHighGoingHigh";
-            }
-            else
-            {
-                eventType = eventInfo;
-                direction = "low";
-                redfishMessageID += ".SensorThresholdCriticalHighGoingLow";
-            }
+            eventType = eventErr;
+            direction = "high";
+            redfishMessageID += ".SensorThresholdCriticalHighGoingHigh";
         }
         else if (event == "NonRecoverableHigh")
         {
-- 
2.34.1

