diff --git a/user_mgr.cpp b/user_mgr.cpp
index 8a079b1..d8c50e0 100644
--- a/user_mgr.cpp
+++ b/user_mgr.cpp
@@ -916,6 +916,26 @@ size_t UserMgr::getIpmiUsersCount()
     return userList.size();
 }
 
+std::vector<std::string> UserMgr::allGroups() const
+{
+    std::vector<std::string> userGroups;
+    /*The "redfish-hostiface" group can only be used by BIOS/HOST
+     * with Get BootStrap Credentials IPMI command to create a HI user.
+     * And it must not be used for creating local user either with
+     * IPMITOOL set user command or Redfish AccountService schema.
+     * Therefore, block the "redfish-hostiface" group from
+     * "AllGroups" DBus property.
+     */
+    for (const auto& group : groupsMgr)
+    {
+        if (group.compare("redfish-hostiface"))
+        {
+            userGroups.push_back(group);
+        }
+    }
+    return userGroups;
+}
+
 bool UserMgr::isUserEnabled(const std::string& userName)
 {
     // Call The User Service to verify if user is enabled
diff --git a/user_mgr.hpp b/user_mgr.hpp
index 8592cef..1dc43d2 100644
--- a/user_mgr.hpp
+++ b/user_mgr.hpp
@@ -210,6 +210,14 @@ class UserMgr : public Ifaces
      **/
     UserInfoMap getUserInfo(std::string userName) override;
 
+    /** @brief returns groups info
+     * Checks for the available user groups, by which new user can be created.
+     *
+     * @param[in] - none
+     * @return -  vector of strings of user groups
+     **/
+    std::vector<std::string> allGroups() const override;
+
   private:
     /** @brief sdbusplus handler */
     sdbusplus::bus_t& bus;
@@ -224,7 +232,7 @@ class UserMgr : public Ifaces
                                         "priv-user", "priv-noaccess"};
 
     /** @brief groups manager container */
-    std::vector<std::string> groupsMgr = {"web", "redfish", "ipmi", "ssh"};
+    std::vector<std::string> groupsMgr = {"web", "redfish", "ipmi", "ssh", "redfish-hostiface"};
 
     /** @brief map container to hold users object */
     using UserName = std::string;
