From f7f1165eefa643e3f06295063d5e3f4e2a03cb7e Mon Sep 17 00:00:00 2001
From: vamsikrishnak <vamsikrishnak@ami.com>
Date: Mon, 12 Jun 2023 13:14:11 +0530
Subject: [PATCH] Adding code for setting Root unlock Timeout

Signed-off-by: vamsikrishnak <vamsikrishnak@ami.com>
---
 user_mgr.cpp | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/user_mgr.cpp b/user_mgr.cpp
index 204f27c..6f9fba0 100644
--- a/user_mgr.cpp
+++ b/user_mgr.cpp
@@ -73,6 +73,7 @@ static constexpr const char* maxSequence = "maxsequence=";
 static constexpr const char* remOldPasswdCount = "remember=";
 static constexpr const char* maxFailedAttempt = "deny=";
 static constexpr const char* unlockTimeout = "unlock_time=";
+static constexpr const char* rootUnlockTimeout = "root_unlock_time";
 static constexpr const char* maxAlphaSequence = "maxalphasequence=";
 static constexpr const char* maxAlphaRepeat = "maxalpharepeat=";
 static constexpr const char* multipleAlpha = "multiplealpha=";
@@ -579,6 +580,14 @@ uint32_t UserMgr::accountUnlockTimeout(uint32_t value)
         log<level::ERR>("Unable to set accountUnlockTimeout");
         elog<InternalFailure>();
     }
+    if (setPamModuleArgValue(pamTally2, rootUnlockTimeout,std::to_string(value)) != success)
+    {
+
+        log<level::ERR>("Unable to set root accountUnlockTimeout");
+
+        elog<InternalFailure>();
+
+    }
     return AccountPolicyIface::accountUnlockTimeout(value);
 }
 
-- 
2.17.1

