--- a/user_service.cpp	2022-11-27 22:46:51.000000000 -0500
+++ b/user_service.cpp	2023-06-22 15:13:28.519353489 -0400
@@ -216,10 +216,12 @@
         // TODO  phosphor-user-manager#10 phosphor::user::shadow::Lock lock{};
         try
         {
+	    std::string oldHomeDir = "/home/" + userName;
             std::string newHomeDir = "/home/" + newUserName;
+	    phosphor::user::executeCmd("/bin/mv", oldHomeDir.c_str(), newHomeDir.c_str());
             phosphor::user::executeCmd("/usr/sbin/usermod", "-l",
                                        newUserName.c_str(), userName.c_str(),
-                                       "-d", newHomeDir.c_str(), "-m");
+                                       "-d", newHomeDir.c_str());
         }
         catch (const phosphor::user::InternalFailure& e)
         {
