From b43a30b4dc22ae89079835ad728e10981c4c4edc Mon Sep 17 00:00:00 2001
From: vamsikrishnak <vamsikrishnak@ami.com>
Date: Fri, 8 Mar 2024 11:14:09 +0530
Subject: [PATCH] Restricting the asd user under redfish

Signed-off-by: vamsikrishnak <vamsikrishnak@ami.com>
---
 user_mgr.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/user_mgr.cpp b/user_mgr.cpp
index cb2f636..5101bde 100644
--- a/user_mgr.cpp
+++ b/user_mgr.cpp
@@ -1667,6 +1667,11 @@ void UserMgr::initUserObjects(void)
                 }
             }
             // Add user objects to the Users path.
+	    if(userPriv.empty())
+	    {
+		    //skip asd user
+		    continue;
+	    }
             sdbusplus::message::object_path tempObjPath(usersObjPath);
             tempObjPath /= user;
             std::string objPath(tempObjPath);
-- 
2.43.0

