From a16347ba80dbe07a259a1efea274829b76ac2647 Mon Sep 17 00:00:00 2001
From: vamsikrishnak <vamsikrishnak@ami.com>
Date: Thu, 31 Aug 2023 14:46:27 +0530
Subject: [PATCH] SSH Active User Delete Fix

Signed-off-by: vamsikrishnak <vamsikrishnak@ami.com>
---
 user_mgr.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/user_mgr.cpp b/user_mgr.cpp
index b2caa1f..557d847 100644
--- a/user_mgr.cpp
+++ b/user_mgr.cpp
@@ -1797,7 +1797,7 @@ void UserMgr::executeUserAdd(const char* userName, const char* groups,
 
 void UserMgr::executeUserDelete(const char* userName)
 {
-    executeCmd("/usr/sbin/userdel", userName, "-r");
+    executeCmd("/usr/sbin/userdel", userName, "-rf");
 }
 
 void UserMgr::executeUserRename(const char* userName, const char* newUserName)
-- 
2.9.5

