From 60e784d968967f31a911380e7f153b1acb6704db Mon Sep 17 00:00:00 2001
From: Abinaya L <abinayal@ami.com>
Date: Fri, 8 Sep 2023 18:13:15 +0530
Subject: [PATCH 18/18] 0018-Added-group-user-for-host-interface

---
 user_mgr.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/user_mgr.cpp b/user_mgr.cpp
index 557d847..7854add 100644
--- a/user_mgr.cpp
+++ b/user_mgr.cpp
@@ -123,8 +123,8 @@ namespace
 {
 
 // The hardcoded groups in OpenBMC projects
-constexpr std::array<const char*, 5> predefinedGroups = {
-    "web", "redfish", "ipmi", "ssh", "hostconsole"};
+constexpr std::array<const char*, 6> predefinedGroups = {
+    "web", "redfish", "ipmi", "ssh", "hostconsole", "redfish-hostiface"};
 
 // These prefixes are for Dynamic Redfish authorization. See
 // https://github.com/openbmc/docs/blob/master/designs/redfish-authorization.md
-- 
2.25.1

