From 54d2b16b4a667c53667e758289736d45857b0811 Mon Sep 17 00:00:00 2001
From: vamsikrishnak <vamsikrishnak@ami.com>
Date: Wed, 5 Jun 2024 10:34:47 +0530
Subject: [PATCH] Added group user for host interface

Signed-off-by: vamsikrishnak <vamsikrishnak@ami.com>
---
 user_mgr.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/user_mgr.cpp b/user_mgr.cpp
index dc75b57..ed099b7 100644
--- a/user_mgr.cpp
+++ b/user_mgr.cpp
@@ -123,8 +123,8 @@ namespace
 {
 
 // The hardcoded groups in OpenBMC projects
-constexpr std::array<const char*, 4> predefinedGroups = {"redfish", "ipmi",
-                                                         "ssh", "hostconsole"};
+constexpr std::array<const char*, 6> predefinedGroups = {
+    "web", "redfish", "ipmi", "ssh", "hostconsole", "redfish-hostiface"};
 
 // These prefixes are for Dynamic Redfish authorization. See
 // https://github.com/openbmc/docs/blob/master/designs/redfish-authorization.md
-- 
2.34.1

