--- a/user_mgr.cpp	2023-10-02 19:17:15.476733543 -0400
+++ b/user_mgr.cpp	2023-10-02 19:24:38.981728098 -0400
@@ -1031,14 +1031,16 @@
         lastFailedAttempt = std::max(failTimestamp, lastFailedAttempt);
     }
 
+    // check if failed attempts have reached max login attempts yet
     if (failAttempts < AccountPolicyIface::maxLoginAttemptBeforeLockout())
     {
         return false;
     }
-
-    if (lastFailedAttempt +
+    // check if NOT in Manual mode && time has expired
+    if ((AccountPolicyIface::accountUnlockTimeout() != 0) &&
+	(lastFailedAttempt +
             static_cast<time_t>(AccountPolicyIface::accountUnlockTimeout()) <=
-        std::time(NULL))
+	 std::time(NULL)))
     {
         return false;
     }
