diff -Naur a/meson.build b/meson.build
--- a/meson.build	2024-01-19 22:23:33.585662021 -0500
+++ b/meson.build	2024-01-19 22:23:41.404844213 -0500
@@ -13,6 +13,10 @@
     add_project_arguments('-DENABLE_ROOT_USER_MGMT', language:'cpp')
 endif
 
+if get_option('oem-privilege').enabled()
+    add_project_arguments('-DFEATURE_OEM_PRIV', language:'cpp')
+endif
+
 conf_data = configuration_data()
 
 conf_data.set_quoted('USER_MANAGER_BUSNAME', 'xyz.openbmc_project.User.Manager',
diff -Naur a/meson.options b/meson.options
--- a/meson.options	2024-01-19 22:22:12.206912690 -0500
+++ b/meson.options	2024-01-19 22:23:58.250069808 -0500
@@ -1,2 +1,4 @@
 option('tests', type: 'feature', description: 'Build tests', value: 'enabled')
 option('root_user_mgmt', type: 'feature', description: 'Enable management of the root user', value: 'enabled')
+# OEM Privilege support
+option('oem-privilege', type: 'feature', value: 'disabled', description: 'oem-privilege is disabled by default; offer a way to enable it')
diff -Naur a/user_mgr.cpp b/user_mgr.cpp
--- a/user_mgr.cpp	2024-01-19 22:22:36.199025426 -0500
+++ b/user_mgr.cpp	2024-01-19 22:24:30.115166437 -0500
@@ -123,8 +123,13 @@
 {
 
 // The hardcoded groups in OpenBMC projects
+#ifdef FEATURE_OEM_PRIV
+constexpr std::array<const char*, 7> predefinedGroups = {
+    "web", "redfish", "ipmi", "ssh", "hostconsole", "redfish-hostiface", "media"};
+#else
 constexpr std::array<const char*, 6> predefinedGroups = {
     "web", "redfish", "ipmi", "ssh", "hostconsole", "redfish-hostiface"};
+#endif
 
 // These prefixes are for Dynamic Redfish authorization. See
 // https://github.com/openbmc/docs/blob/master/designs/redfish-authorization.md
