--- a/user_mgr.cpp	2024-02-12 20:13:02.748636592 -0500
+++ b/user_mgr.cpp	2024-02-12 20:18:29.179158263 -0500
@@ -123,9 +123,15 @@
 {
 
 // The hardcoded groups in OpenBMC projects
-#ifdef FEATURE_OEM_PRIV
+#if defined FEATURE_OEM_PRIV && defined FEATURE_SNMP_TRAPV3
+constexpr std::array<const char*, 8> predefinedGroups = {
+    "web", "redfish", "ipmi", "ssh", "hostconsole", "redfish-hostiface", "media", "snmp"};
+#elif defined FEATURE_OEM_PRIV
 constexpr std::array<const char*, 7> predefinedGroups = {
     "web", "redfish", "ipmi", "ssh", "hostconsole", "redfish-hostiface", "media"};
+#elif defined FEATURE_SNMP_TRAPV3
+constexpr std::array<const char*, 7> predefinedGroups = {
+    "web", "redfish", "ipmi", "ssh", "hostconsole", "redfish-hostiface", "snmp"};
 #else
 constexpr std::array<const char*, 6> predefinedGroups = {
     "web", "redfish", "ipmi", "ssh", "hostconsole", "redfish-hostiface"};

diff --git a/meson.build b/meson.build
index b7e7acc..c02d85d 100644
--- a/meson.build
+++ b/meson.build
@@ -17,6 +17,8 @@ if get_option('oem-privilege').enabled()
     add_project_arguments('-DFEATURE_OEM_PRIV', language:'cpp')
 endif

+add_project_arguments('-DFEATURE_SNMP_TRAPV3', language:'cpp')
+
 conf_data = configuration_data()

 conf_data.set_quoted('USER_MANAGER_BUSNAME', 'xyz.openbmc_project.User.Manager',


