--- a/user_mgr.cpp	2024-02-12 20:13:02.748636592 -0500
+++ b/user_mgr.cpp	2024-02-12 20:18:29.179158263 -0500
@@ -123,9 +123,15 @@
 {
 
 // The hardcoded groups in OpenBMC projects
-#ifdef FEATURE_OEM_PRIV
+#if defined FEATURE_OEM_PRIV && defined FEATURE_SNMP_TRAPV3
+constexpr std::array<const char*, 8> predefinedGroups = {
+    "web", "redfish", "ipmi", "ssh", "hostconsole", "redfish-hostiface", "media", "snmp"};
+#elif defined FEATURE_OEM_PRIV
 constexpr std::array<const char*, 7> predefinedGroups = {
     "web", "redfish", "ipmi", "ssh", "hostconsole", "redfish-hostiface", "media"};
+#elif defined FEATURE_SNMP_TRAPV3
+constexpr std::array<const char*, 7> predefinedGroups = {
+    "web", "redfish", "ipmi", "ssh", "hostconsole", "redfish-hostiface", "snmp"};
 #else
 constexpr std::array<const char*, 6> predefinedGroups = {
     "web", "redfish", "ipmi", "ssh", "hostconsole", "redfish-hostiface"};
