From 3a6ac90585d1ea25868480d6b8a19724f3c3d4b0 Mon Sep 17 00:00:00 2001
From: pandiarajt <pandiarajt@ami.com>
Date: Tue, 13 Aug 2024 13:07:59 +0530
Subject: [PATCH 191/191] Fix to add support to include dot(.) in username

---
 user_mgr.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/user_mgr.cpp b/user_mgr.cpp
index 1c7ad21..2528b1f 100644
--- a/user_mgr.cpp
+++ b/user_mgr.cpp
@@ -282,7 +282,7 @@ void UserMgr::throwForUserNameConstraints(
                               Argument::ARGUMENT_VALUE("Invalid length"));
     }
     if (!std::regex_match(userName.c_str(),
-                          std::regex("[a-zA-Z_][a-zA-Z_0-9]*")))
+                          std::regex("[a-zA-Z_][a-zA-Z_0-9.]*")))
     {
         lg2::error("Invalid username '{USERNAME}'", "USERNAME", userName);
         elog<InvalidArgument>(Argument::ARGUMENT_NAME("User name"),
-- 
2.34.1

