From 072749f23b73442f734eb645e2f7b09e7f670a91 Mon Sep 17 00:00:00 2001
From: Sharath <sharathkumark@ami.com>
Date: Tue, 17 Jun 2025 07:12:32 +0000
Subject: [PATCH] OT 14429 Updated username validation check

Signed-off-by: Sharath <sharathkumark@ami.com>
---
 user_mgr.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/user_mgr.cpp b/user_mgr.cpp
index a6300aa..d14e971 100644
--- a/user_mgr.cpp
+++ b/user_mgr.cpp
@@ -310,7 +310,7 @@ void UserMgr::throwForUserNameConstraints(
                               Argument::ARGUMENT_VALUE("Invalid length"));
     }
     if (!std::regex_match(userName.c_str(),
-                          std::regex("[a-zA-Z_][a-zA-Z_0-9.]*")))
+                          std::regex("^[a-zA-Z0-9_.*]{1,16}$")))
     {
         lg2::error("Invalid username '{USERNAME}'", "USERNAME", userName);
         elog<InvalidArgument>(Argument::ARGUMENT_NAME("User name"),
-- 
2.43.0

