From 047b13fc0c7b6d0b105a9c057e436f17776b6759 Mon Sep 17 00:00:00 2001
From: Vijaysankar R <vijaysankarr@ami.com>
Date: Fri, 23 May 2025 19:39:12 +0530
Subject: [PATCH 205/205] Removed ipmi Group Check Internal Users

---
 user_mgr.cpp | 12 ++----------
 1 file changed, 2 insertions(+), 10 deletions(-)

diff --git a/user_mgr.cpp b/user_mgr.cpp
index 0d7f3c2..24d15ea 100644
--- a/user_mgr.cpp
+++ b/user_mgr.cpp
@@ -1421,18 +1421,12 @@ UserInfoMap UserMgr::getUserInfo(std::string userName)
     struct privilageStruct privBuf[3] = {
         {1, "priv-admin"}, {2, "priv-operator"}, {3, "priv-user"}};
     UserInfoMap userInfo;
-    int userID = 0, retval = 0, userType = 1;
+    int userID = 0, retval = 0;
     std::vector<uint8_t> currentPamOrder;
     currentPamOrder = getPamOrder();
     for (uint8_t value : currentPamOrder){
     if(value == 1){
-    struct group *grp = getgrnam("ipmi");
     if (isUserExist(userName)){
-    for (char** member = grp->gr_mem; *member != nullptr; member++)
-    {
-    if (strcmp(*member, userName.c_str()) == 0)
-    {
-	userType = 0;
         const auto& user = usersList[userName];
         userInfo.emplace("UserPrivilege", user.get()->userPrivilege());
         userInfo.emplace("UserGroups", user.get()->userGroups());
@@ -1444,9 +1438,7 @@ UserInfoMap UserMgr::getUserInfo(std::string userName)
         userInfo.emplace("RemoteUser", false);
 	return userInfo;
     }
-    }
-    }
-    if(userType)
+    else
     continue;
     }
     if(value == 2)
-- 
2.34.1

