diff --git a/user_mgr.cpp b/user_mgr.cpp
index 413043e..e6c17ac 100644
--- a/user_mgr.cpp
+++ b/user_mgr.cpp
@@ -2201,6 +2201,7 @@ UserInfoMap UserMgr::getUserInfo(std::string userName,  std::string ipAddress)
 			    userInfo.emplace("UserPasswordExpired",
 					    user.get()->userPasswordExpired());
 			    userInfo.emplace("RemoteUser", false);
+                userInfo.emplace("UserType", std::string("internal"));
 			    return userInfo;
 		    }
 	    }
@@ -2236,6 +2237,15 @@ UserInfoMap UserMgr::getUserInfo(std::string userName,  std::string ipAddress)
                 continue;
             }
 
+            std::string userType;
+            if (ldapConfigPath.find("active_directory") != std::string::npos)
+            {
+                userType = "ad";
+            }
+            else if (ldapConfigPath.find("openldap") != std::string::npos)
+            {
+                userType = "ldap";
+            }
             for (const auto& [path, interfaces] : objects)
             {
                 if (!path.str.starts_with(ldapConfigPath))
@@ -2279,6 +2289,7 @@ UserInfoMap UserMgr::getUserInfo(std::string userName,  std::string ipAddress)
 	    {
 		    userInfo.emplace("UserPrivilege", userPrivilege);
 		    userInfo.emplace("RemoteUser", true);
+            userInfo.emplace("UserType", userType);
 		    break;
 	    }
 	    else
@@ -2289,6 +2300,7 @@ UserInfoMap UserMgr::getUserInfo(std::string userName,  std::string ipAddress)
 					    "default \"priv-user\" is used");
 			    userInfo.emplace("UserPrivilege", "priv-user");
 			    userInfo.emplace("RemoteUser", true);
+                userInfo.emplace("UserType", userType);
 		    }
 		    else
 			    continue;
@@ -2327,7 +2339,6 @@ UserInfoMap UserMgr::getUserInfo(std::string userName,  std::string ipAddress)
                     if (userID >= 9000)
                     {
                         int iter = 0;
-                        userInfo.emplace("RemoteUser", true);
                         for (iter = 0;
                              iter <
                              ((signed)(sizeof(privBuf) / sizeof(privBuf[0])));
@@ -2338,6 +2349,7 @@ UserInfoMap UserMgr::getUserInfo(std::string userName,  std::string ipAddress)
                                 userInfo.emplace("UserPrivilege",
                                                  privBuf[iter].privStr);
                                 userInfo.emplace("RemoteUser", true);
+                                userInfo.emplace("UserType", std::string("radius"));
                                 return userInfo;
                             }
 			    else
