From ec44c4fdb4e1c218e11110beef0c5cb82930033a Mon Sep 17 00:00:00 2001
From: madhankumars <madhankumars@ami.com>
Date: Wed, 1 Apr 2026 15:50:55 +0530
Subject: [PATCH 242/242] getchannelfromip-loop-index-userinfo-alignment

---
 user_mgr.cpp | 21 +++++++++++----------
 1 file changed, 11 insertions(+), 10 deletions(-)

diff --git a/user_mgr.cpp b/user_mgr.cpp
index 96e1da8..1ce6aec 100644
--- a/user_mgr.cpp
+++ b/user_mgr.cpp
@@ -2147,15 +2147,16 @@ uint8_t UserMgr::getChannelFromIP(const std::string& targetIp)
         }
 
         bool found = false;
-        std::optional<uint8_t> channel;
-        for (const auto& [ch, name] : channelMap)
+        uint8_t loopValue = 0;
+        for (const auto& [channelId, name] : channelMap)
         {
+            (void)channelId;
             if (name == baseIface)
             {
-                channel = ch;
                 found = true;
                 break;
             }
+            ++loopValue;
         }
 
         if (!found)
@@ -2171,7 +2172,7 @@ uint8_t UserMgr::getChannelFromIP(const std::string& targetIp)
             if (targetIp == ipStr)
             {
                 freeifaddrs(ifaddr);
-                return static_cast<int>(channel.value());
+                return loopValue;
             }
         }
         else if (ifa->ifa_addr->sa_family == AF_INET6)
@@ -2181,7 +2182,7 @@ uint8_t UserMgr::getChannelFromIP(const std::string& targetIp)
             if (targetIp == ipStr)
             {
                 freeifaddrs(ifaddr);
-                return static_cast<int>(channel.value());
+                return loopValue;
             }
         }
     }
@@ -2203,18 +2204,18 @@ UserInfoMap UserMgr::getUserInfo(std::string userName,  std::string ipAddress)
     for (uint8_t value : currentPamOrder){
     if(value == 1){
 	    if (isUserExist(userName) &&
-            channelNo != std::numeric_limits<uint8_t>::max() && channelNo > 0)
+			    channelNo != std::numeric_limits<uint8_t>::max())
 	    {
 		    const auto& user = usersList[userName];
 		    userChannelAccess.resize(std::max(userChannelAccess.size(),
-					    static_cast<size_t>(channelNo)));
+					    static_cast<size_t>(channelNo + 1)));
 		    userChannelAccess = user.get()->userChannelAccess();
-		    if (userChannelAccess[channelNo - 1])
+		    if (userChannelAccess[channelNo])
 		    {
 			    std::vector<std::string> priv = user.get()->userPrivilege();
 			    priv.resize(
-					    std::max(priv.size(), static_cast<size_t>(channelNo)));
-			    userInfo.emplace("UserPrivilege", priv[channelNo - 1]);
+					    std::max(priv.size(), static_cast<size_t>(channelNo + 1)));
+			    userInfo.emplace("UserPrivilege", priv[channelNo]);
 			    userInfo.emplace("UserGroups", user.get()->userGroups());
 			    userInfo.emplace("UserEnabled", user.get()->userEnabled());
 			    userInfo.emplace("UserLockedForFailedAttempt",
-- 
2.34.1

