From 7d4c9d312d14700e3313d16c804d9633b6259359 Mon Sep 17 00:00:00 2001
From: lakshmanakumarv <lakshmanakumarv@ami.com>
Date: Wed, 22 Apr 2026 20:39:19 +0530
Subject: [PATCH 244/244] Fixed-Rename-User-LF

---
 user_mgr.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/user_mgr.cpp b/user_mgr.cpp
index ad1c7c9..b9dd448 100644
--- a/user_mgr.cpp
+++ b/user_mgr.cpp
@@ -1024,6 +1024,7 @@ void UserMgr::renameUser(std::string userName, std::string newUserName)
     bool snmpAccessEnableStatus = user.get()->snmpAccessEnableStatus();
     std::string mailid = user.get()->smtpMailID();
     uint64_t passwordExpiration = user.get()->passwordExpiration();
+    std::vector<uint8_t> channelAccess = user.get()->userChannelAccess();
     sdbusplus::message::object_path tempObjPath(usersObjPath);
     tempObjPath /= newUserName;
     std::string newUserObj(tempObjPath);
@@ -1034,9 +1035,8 @@ void UserMgr::renameUser(std::string userName, std::string newUserName)
     usersList.erase(userName);
     usersList.emplace(newUserName,
                       std::make_unique<phosphor::user::Users>(
-                          bus, newUserObj.c_str(), groupNames, priv, 
-			  user.get()->userChannelAccess(),enabled,
-                          passwordExpiration, *this));
+                          bus, newUserObj.c_str(), groupNames, priv,
+                          channelAccess, enabled, passwordExpiration, *this));
 
     auto& newUserRef = usersList.at(newUserName);
     newUserRef->snmpAccessEnableStatus(snmpAccessEnableStatus);
-- 
2.43.0

