diff --git a/user_mgr.cpp b/user_mgr.cpp
index 7611cec..413043e 100644
--- a/user_mgr.cpp
+++ b/user_mgr.cpp
@@ -508,7 +508,7 @@ void UserMgr::throwForUserNameConstraints(
                               Argument::ARGUMENT_VALUE("Invalid length"));
     }
     if (!std::regex_match(userName.c_str(),
-                          std::regex("^[a-zA-Z0-9_.*]{1,16}$")))
+                          std::regex("^[a-zA-Z0-9_*]{1,16}$")))
     {
         lg2::error("Invalid username '{USERNAME}'", "USERNAME", userName);
         elog<InvalidArgument>(Argument::ARGUMENT_NAME("User name"),
