From 8e958495fcc63d404ae9177ca3f41cfd9142e2e8 Mon Sep 17 00:00:00 2001
From: madhankumars <madhankumars@ami.com>
Date: Tue, 7 Jul 2026 12:55:53 +0530
Subject: [PATCH 248/248] Cordump Fix UserManagement

---
 user_mgr.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/user_mgr.cpp b/user_mgr.cpp
index e6c17ac..ab08815 100644
--- a/user_mgr.cpp
+++ b/user_mgr.cpp
@@ -2585,7 +2585,7 @@ void UserMgr::initUserObjects(void)
 					    "PATH", ipmiUserDataFile);
 		    }
 	    }
-            nlohmann::json userJson = *userJsonOpt;
+            nlohmann::json userJson = userJsonOpt.value_or(nlohmann::json::array());
             for (const auto& entry : userJson)
             {
                 if (!entry.contains("user_name") || entry["user_name"] != user)
-- 
2.34.1

