From 413c7c6f10a1491f88e47881f493b5e222938d51 Mon Sep 17 00:00:00 2001
From: Minal Upadhye <minalsunilu@ami.com>
Date: Tue, 30 Sep 2025 02:53:06 -0400
Subject: [PATCH] migrate journal events to dbus hsbp

Signed-off-by: Minal Upadhye <minalsunilu@ami.com>
---
 hsbp-manager/include/utils.hpp | 81 +++++++++++++++++++++++++++++-----
 1 file changed, 70 insertions(+), 11 deletions(-)

diff --git a/include/utils.hpp b/include/utils.hpp
index bfe549c..692e2d7 100644
--- a/include/utils.hpp
+++ b/include/utils.hpp
@@ -195,24 +195,83 @@ void setupPowerMatch(const std::shared_ptr<sdbusplus::asio::connection>& conn)
 inline void logDeviceAdded(const std::string& model, const std::string& type,
                            const std::string& sn)
 {
-    sd_journal_send("MESSAGE=%s", "Inventory Added", "PRIORITY=%i", LOG_ERR,
-                    "REDFISH_MESSAGE_ID=%s", "OpenBMC.0.1.InventoryAdded",
-                    "REDFISH_MESSAGE_ARGS=%s,%s,%s", model.c_str(),
-                    type.c_str(), sn.c_str(), NULL);
+    std::string severity =
+        "xyz.openbmc_project.Logging.Entry.Level.Informational";
+    auto bus = sdbusplus::bus::new_default_system();
+    sdbusplus::message::message m = bus.new_method_call(
+        "xyz.openbmc_project.Logging", "/xyz/openbmc_project/logging",
+        "xyz.openbmc_project.Logging.Create", "Create");
+    std::string messageID = "OpenBMC.0.1.InventoryAdded";
+    std::string journalMsg = messageID + "," + model + "," + type + "," + sn;
+
+    std::map<std::string, std::string> additionalData;
+    additionalData["REDFISH_MESSAGE_ID"] = messageID;
+    additionalData["REDFISH_MESSAGE_ARGS"] = model + "," + type + "," + sn;
+    additionalData["Additional_Data1"] = "Inventory Added";
+    additionalData["Additional_Data2"] =
+        model + " " + type + " with serial number " + sn + " was installed.";
+    m.append(journalMsg, severity, additionalData);
+    try
+    {
+        bus.call(m);
+    }
+    catch (const sdbusplus::exception_t& e)
+    {
+        std::cerr << "Failed to create log entry: " << e.what() << std::endl;
+    }
 }
 
 inline void logDeviceRemoved(const std::string& model, const std::string& type,
                              const std::string& sn)
 {
-    sd_journal_send("MESSAGE=%s", "Inventory Removed", "PRIORITY=%i", LOG_ERR,
-                    "REDFISH_MESSAGE_ID=%s", "OpenBMC.0.1.InventoryRemoved",
-                    "REDFISH_MESSAGE_ARGS=%s,%s,%s", model.c_str(),
-                    type.c_str(), sn.c_str(), NULL);
+    std::string severity =
+        "xyz.openbmc_project.Logging.Entry.Level.Informational";
+    auto bus = sdbusplus::bus::new_default_system();
+    sdbusplus::message::message m = bus.new_method_call(
+        "xyz.openbmc_project.Logging", "/xyz/openbmc_project/logging",
+        "xyz.openbmc_project.Logging.Create", "Create");
+    std::string messageID = "OpenBMC.0.1.InventoryRemoved";
+    std::string journalMsg = messageID + "," + model + "," + type + "," + sn;
+
+    std::map<std::string, std::string> additionalData;
+    additionalData["REDFISH_MESSAGE_ID"] = messageID;
+    additionalData["REDFISH_MESSAGE_ARGS"] = model + "," + type + "," + sn;
+    additionalData["Additional_Data1"] = "Inventory Removed";
+    additionalData["Additional_Data2"] =
+        model + " " + type + " with serial number " + sn + " was removed.";
+    m.append(journalMsg, severity, additionalData);
+    try
+    {
+        bus.call(m);
+    }
+    catch (const sdbusplus::exception_t& e)
+    {
+        std::cerr << "Failed to create log entry: " << e.what() << std::endl;
+    }
 }
 
 inline void logDriveError(const std::string& name)
 {
-    sd_journal_send("MESSAGE=%s", "Drive Error", "PRIORITY=%i", LOG_ERR,
-                    "REDFISH_MESSAGE_ID=%s", "OpenBMC.0.1.DriveError",
-                    "REDFISH_MESSAGE_ARGS=%s", name.c_str(), NULL);
+    std::string severity = "xyz.openbmc_project.Logging.Entry.Level.Warning";
+    auto bus = sdbusplus::bus::new_default_system();
+    sdbusplus::message::message m = bus.new_method_call(
+        "xyz.openbmc_project.Logging", "/xyz/openbmc_project/logging",
+        "xyz.openbmc_project.Logging.Create", "Create");
+    std::string messageID = "OpenBMC.0.1.DriveError";
+    std::string journalMsg = messageID + "," + name;
+
+    std::map<std::string, std::string> additionalData;
+    additionalData["REDFISH_MESSAGE_ID"] = messageID;
+    additionalData["REDFISH_MESSAGE_ARGS"] = name;
+    additionalData["Additional_Data1"] = "Drive Error";
+    additionalData["Additional_Data2"] = "Drive Error Occurred: " + name;
+    m.append(journalMsg, severity, additionalData);
+    try
+    {
+        bus.call(m);
+    }
+    catch (const sdbusplus::exception_t& e)
+    {
+        std::cerr << "Failed to create log entry: " << e.what() << std::endl;
+    }
 }
-- 
2.34.1

