From 0fc36f278c3c9aca986cacde17821bf22b9286e8 Mon Sep 17 00:00:00 2001
From: thrishalig <thrishalig@ami.com>
Date: Fri, 13 Oct 2023 13:26:54 +0530
Subject: [PATCH] converted index to 0 based and made pwm starts from 0

Signed-off-by: thrishalig <thrishalig@ami.com>
---
 src/FanMain.cpp | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/src/FanMain.cpp b/src/FanMain.cpp
index 11d2c22..9be9939 100644
--- a/src/FanMain.cpp
+++ b/src/FanMain.cpp
@@ -161,7 +161,7 @@ bool findPwmPath(const fs::path& directory, unsigned int pwm, fs::path& pwmPath)
     std::error_code ec;
 
     /* Assuming PWM file is appeared in the same directory as fanX_input */
-    auto path = directory / ("pwm" + std::to_string(pwm + 1));
+    auto path = directory / ("pwm" + std::to_string(pwm));
     bool exists = fs::exists(path, ec);
 
     if (ec || !exists)
@@ -291,7 +291,7 @@ void createSensors(
             std::string cfgIntf = configInterfaceName(sensorTypes[fanType]);
 
             // convert to 0 based
-            size_t index = std::stoul(indexStr) - 1;
+            size_t index = std::stoul(indexStr);
 
             const char* baseType = nullptr;
             const SensorData* sensorData = nullptr;
@@ -485,8 +485,8 @@ void createSensors(
                         continue;
                     }
 
-                    fs::path pwmEnableFile = "pwm" + std::to_string(pwm + 1) +
-                                             "_enable";
+                    fs::path pwmEnableFile =
+                        "pwm" + std::to_string(pwm) + "_enable";
                     fs::path enablePath = pwmPath.parent_path() / pwmEnableFile;
                     enablePwm(enablePath);
 
@@ -500,7 +500,7 @@ void createSensors(
                     }
                     else
                     {
-                        pwmName = "Pwm_" + std::to_string(pwm + 1);
+                        pwmName = "Pwm_" + std::to_string(pwm);
                     }
 
                     // Check PWM sensor mutability
-- 
2.25.1

