From 4e934f9ce356c1ce9471371df84ba04968fb4808 Mon Sep 17 00:00:00 2001
From: Vijaysankar R <vijaysankarr@ami.com>
Date: Wed, 23 Apr 2025 18:36:57 +0530
Subject: [PATCH 3/3] Added-Radius-In-Pamorder

---
 src/pam_helper.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/pam_helper.c b/src/pam_helper.c
index 3ab168b..8ae56d1 100644
--- a/src/pam_helper.c
+++ b/src/pam_helper.c
@@ -8,6 +8,7 @@
 #include <string.h>
 #include <syslog.h>
 #include <unistd.h>
+#define MAX_USER_NAME_LEN 255
 
 typedef enum nss_status nss_status_t;
 
@@ -128,7 +129,7 @@ nss_status_t _nss_radius_getpwnam_r(char *name, struct passwd *pwd,
     return result;
   }
   init_buf = buffer;
-  strncpy(pwd->pw_name, nss_res.usr.name, strlen(nss_res.usr.name));
+  snprintf(pwd->pw_name, MAX_USER_NAME_LEN, "%s", nss_res.usr.name);
   pwd->pw_uid = nss_res.usr.uid;
   pwd->pw_gid = nss_res.usr.priv;
 
@@ -145,7 +146,6 @@ enum nss_status _nss_radius_getpwuid_r(uid_t uid, struct passwd *pwd,
                                        char *buffer, size_t buflen,
                                        int *errnop) {
   enum nss_status result = NSS_STATUS_SUCCESS;
-  return result;
   int ret = 0;
   char *init_buf = NULL;
   pamusrpkt_t nss_req, nss_res;
-- 
2.43.0

