diff -Naur a/src/FanMain.cpp b/src/FanMain.cpp
--- a/src/FanMain.cpp	2024-03-13 15:21:40.886755655 +0530
+++ b/src/FanMain.cpp	2024-03-13 15:23:14.688023981 +0530
@@ -67,6 +67,7 @@
     {"aspeed,ast2400-pwm-tacho", FanTypes::aspeed},
     {"aspeed,ast2500-pwm-tacho", FanTypes::aspeed},
     {"aspeed,ast2600-pwm-tacho", FanTypes::aspeed},
+    {"aspeed,ast2700-pwm-tach", FanTypes::aspeed},
     {"nuvoton,npcm750-pwm-fan", FanTypes::nuvoton},
     {"nuvoton,npcm845-pwm-fan", FanTypes::nuvoton},
     {"hpe,gxp-fan-ctrl", FanTypes::hpe}
