From 714b26af563d3ac280cda1302df57d2e719da628 Mon Sep 17 00:00:00 2001
From: Sharath <sharathkumark@ami.com>
Date: Tue, 17 Jun 2025 05:02:41 +0000
Subject: [PATCH] OT-14429 oem update username validation check

Signed-off-by: Sharath <sharathkumark@ami.com>
---
 src/oemcommands.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/oemcommands.cpp b/src/oemcommands.cpp
index 45f882a..e1cd34f 100644
--- a/src/oemcommands.cpp
+++ b/src/oemcommands.cpp
@@ -6170,7 +6170,7 @@ bool isValidUserName(ipmi::Context::ptr ctx, const std::string& userName)
         return false;
     }
     if (!std::regex_match(userName.c_str(),
-                          std::regex("[a-zA-z_][a-zA-Z_0-9]*")))
+                          std::regex("^[a-zA-Z0-9_.*]{1,16}$")))
     {
         phosphor::logging::log<level::ERR>("Unsupported characters in string");
         return false;
-- 
2.43.0

