From 45689ff4b8de18c43a0fe5e54c5b066df03f0679 Mon Sep 17 00:00:00 2001
From: Nikhila Dande <nikhilad@ami.com>
Date: Fri, 3 Jul 2026 09:38:00 +0000
Subject: [PATCH] Fix for LED Coverity issue

Signed-off-by: Nikhila Dande <nikhilad@ami.com>
---
 utils.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/utils.cpp b/utils.cpp
index 04488b2..28e2e94 100644
--- a/utils.cpp
+++ b/utils.cpp
@@ -40,7 +40,7 @@ std::string DBusHandler::getService(const std::string& path,
 PropertyMap DBusHandler::getAllProperties(const std::string& objectPath,
                                           const std::string& interface)
 {
-    PropertyMap properties;
+    PropertyMap properties{};
 
     auto& bus = DBusHandler::getBus();
     auto service = getService(objectPath, interface);
-- 
2.43.0

