diff -Naur a/redfish-core/lib/cups_service.hpp b/redfish-core/lib/cups_service.hpp
--- a/redfish-core/lib/cups_service.hpp	2022-11-03 15:02:48.506266762 +0530
+++ b/redfish-core/lib/cups_service.hpp	2022-11-03 15:02:01.137574366 +0530
@@ -540,7 +540,7 @@
                         }
                         members.push_back(
                             {{"@odata.id", "/redfish/v1/Chassis/" + *chassis +
-                                               "/Sensors/" + *sensor}});
+                                               "/Sensors/utilization_" + *sensor}});
                     }
                 }
             }
