From e276ea5734018ab141f6dfca11a05cafad4be211 Mon Sep 17 00:00:00 2001
From: Abinaya L <abinayal@ami.com>
Date: Fri, 16 Jun 2023 12:17:45 +0530
Subject: [PATCH 2/2] Fix for Unauthorized OOB user in bmcweb

---
 config/pam-webserver | 1 -
 1 file changed, 1 deletion(-)

diff --git a/config/pam-webserver b/config/pam-webserver
index b7755f5..74b39f9 100644
--- a/config/pam-webserver
+++ b/config/pam-webserver
@@ -3,7 +3,6 @@
 auth     include  common-auth
 # skip redfish group check for non-local user (ldap)
 auth     [success=ok perm_denied=1 default=ignore] pam_localuser.so
-auth     required pam_succeed_if.so user ingroup redfish
 auth     sufficient pam_succeed_if.so user ingroup redfish
 auth     sufficient pam_succeed_if.so user ingroup redfish-hostiface
 account  include  common-account
-- 
2.25.1

