From 6ace5594a90f2c4c8f9dfa0528532e26b2ac3e62 Mon Sep 17 00:00:00 2001
From: Nibin Chacko <nibinc@ami.com>
Date: Fri, 1 Sep 2023 03:26:18 -0400
Subject: [PATCH 106/106] removing cupservice under openbmc

---
 redfish-core/lib/service_root.hpp | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/redfish-core/lib/service_root.hpp b/redfish-core/lib/service_root.hpp
index 51d652c3..99ce523a 100644
--- a/redfish-core/lib/service_root.hpp
+++ b/redfish-core/lib/service_root.hpp
@@ -95,11 +95,6 @@ inline void handleServiceRootGetImpl(
     asyncResp->res.jsonValue["Links"]["ManagerProvidingService"]["@odata.id"] =
         "/redfish/v1/Managers/bmc";
 
-    asyncResp->res.jsonValue["Oem"]["OpenBmc"]["@odata.type"] =
-        "#OemServiceRoot.v1_0_0.ServiceRoot";
-    asyncResp->res.jsonValue["Oem"]["OpenBmc"]["CupsService"] = {
-        {"@odata.id", "/redfish/v1/CupsService"}};
-
     asyncResp->res.jsonValue["Oem"]["OpenBmc"]["Pef"]["@odata.type"] =
         "#OemPefServiceRoot.v1_0_0.ServiceRoot";
     asyncResp->res.jsonValue["Oem"]["OpenBmc"]["Pef"]["PefService"] = {
-- 
2.34.1

