From 6202ea300696120d25a0fe0a729a13acca8ca930 Mon Sep 17 00:00:00 2001
From: pavithra <pavithrat@ami.com>
Date: Fri, 20 Feb 2026 20:17:51 +0530
Subject: [PATCH] Added fix for OBMCErrorNotification date types

---
 src/snmp_trap.cpp | 16 +++++++++++++---
 1 file changed, 13 insertions(+), 3 deletions(-)

diff --git a/src/snmp_trap.cpp b/src/snmp_trap.cpp
index 20793d9..5fb2295 100644
--- a/src/snmp_trap.cpp
+++ b/src/snmp_trap.cpp
@@ -50,15 +50,25 @@ void ErrorTrap::trap(sdbusplus::message_t& msg) const
     auto additionalData =
         std::get<std::vector<std::string>>(propMap.at("AdditionalData"));
 
-    std::string EventGenerateTime = timestampToFormattedString(timestamp);
+    // Convert severity string to int32_t
+    int32_t severityLevel = 0;
+    if (sev == "xyz.openbmc_project.Logging.Entry.Level.Critical")
+    {
+        severityLevel = 2;
+    }
+    else if (sev == "xyz.openbmc_project.Logging.Entry.Level.Warning")
+    {
+        severityLevel = 1;
+    }
+
     for (auto& s : additionalData)
     {
         message.append(" ");
         message.append(s);
     }
-    try
+   try
     {
-        sendTrap<OBMCErrorNotification>(errorID, EventGenerateTime, sev, message);
+        sendTrap<OBMCErrorNotification>(errorID, timestamp, severityLevel, message);
     }
     catch (const InternalFailure& e)
     {
-- 
2.34.1

