From 1ddffeafaf21759ae9ded5c08adb564a1ebea124 Mon Sep 17 00:00:00 2001
From: Vijaysankar R <vijaysankarr@ami.com>
Date: Tue, 3 Mar 2026 13:53:42 +0530
Subject: [PATCH 175/175] Refactored sendTrap

---
 src/snmp_trap.cpp | 23 ++++++++++++++++++-----
 1 file changed, 18 insertions(+), 5 deletions(-)

diff --git a/src/snmp_trap.cpp b/src/snmp_trap.cpp
index 921d628..4735dd7 100644
--- a/src/snmp_trap.cpp
+++ b/src/snmp_trap.cpp
@@ -20,6 +20,18 @@ using InternalFailure =
 
 static constexpr auto entry = "xyz.openbmc_project.Logging.Entry";
 
+std::string timestampToFormattedString(std::time_t timestamp)
+{
+    std::tm localTime;
+    localtime_r(&timestamp, &localTime); // Convert timestamp to local time
+
+    char buffer[100];
+    std::strftime(buffer, sizeof(buffer), "%a %b %d %H:%M:%S %Z %Y",
+                  &localTime);
+
+    return buffer;
+}
+
 void ErrorTrap::trap(sdbusplus::message_t& msg) const
 {
     sdbusplus::message::object_path path;
@@ -34,11 +46,11 @@ void ErrorTrap::trap(sdbusplus::message_t& msg) const
     auto& propMap = it->second;
     auto errorID = std::get<uint32_t>(propMap.at("Id"));
     auto timestamp = std::get<uint64_t>(propMap.at("Timestamp"));
-    auto sev = std::get<Entry::Level>(propMap.at("Severity"));
-    auto isev = static_cast<uint8_t>(sev);
+    auto sev = std::get<std::string>(propMap.at("Severity"));
     auto message = std::get<std::string>(propMap.at("Message"));
     auto additionalData = std::get<std::map<std::string, std::string>>(
         propMap.at("AdditionalData"));
+    std::string EventGenerateTime = timestampToFormattedString(timestamp);
     for (auto& [k, v] : additionalData)
     {
         message.append(" ");
@@ -46,14 +58,15 @@ void ErrorTrap::trap(sdbusplus::message_t& msg) const
     }
     try
     {
-        sendTrap<OBMCErrorNotification>(errorID, timestamp, isev, message);
+        sendTrap<OBMCErrorNotification>(errorID, EventGenerateTime, sev,
+                                        message);
     }
     catch (const InternalFailure& e)
     {
         lg2::error(
             "Failed to send SNMP trap: {ERROR}, ERROR_ID = {EID}, TIMESTAMP = {TSP}, SEVERITY = {SEVERITY}, MESSAGE = {MSG}",
-            "ERROR", e, "EID", errorID, "TSP", timestamp, "SEVERITY",
-            convertForMessage(sev), "MSG", message);
+            "ERROR", e, "EID", errorID, "TSP", timestamp, "SEVERITY", sev,
+            "MSG", message);
     }
 }
 } // namespace monitoring
-- 
2.43.0

