--- a/snmp_util.cpp	2023-07-13 14:19:48.000000000 -0400
+++ b/snmp_util.cpp	2023-12-22 15:50:07.012974952 -0500
@@ -90,6 +90,11 @@
                        "RC", isValid);
             elog<InternalFailure>();
         }
+        char ipTemp[INET6_ADDRSTRLEN]{0};
+        strcat(ipTemp,"[");
+        strcat(ipTemp,ipaddress);
+        strcat(ipTemp,"]");
+        strcpy(ipaddress,ipTemp);
     }
 
     return ipaddress;
