From 38dac855e8d6b19bf0f84dd3065cd891f3dbd128 Mon Sep 17 00:00:00 2001
From: pandiarajt <pandiarajt@ami.com>
Date: Wed, 21 Feb 2024 11:45:29 +0530
Subject: [PATCH 69/69] add to change for enable/disable SNMPTrap

---
 snmp_notification.cpp | 17 ++++++++++++++++-
 1 file changed, 16 insertions(+), 1 deletion(-)

diff --git a/snmp_notification.cpp b/snmp_notification.cpp
index 1bebdbc..e2fb98d 100644
--- a/snmp_notification.cpp
+++ b/snmp_notification.cpp
@@ -2,7 +2,7 @@
 
 #include "snmp_util.hpp"
 #include "xyz/openbmc_project/Common/error.hpp"
-
+#include <fstream>
 #include <phosphor-logging/elog-errors.hpp>
 #include <phosphor-logging/lg2.hpp>
 
@@ -63,6 +63,21 @@ void Notification::sendTrap()
     netsnmp_session session{};
     snmp_sess_init(&session);
 
+    bool snmpTrapStatus;
+    std::ifstream file("/etc/snmp/SnmpTrapStatus", std::ios::in);
+    if (!file.is_open()) {
+        lg2::error("Failed to open /etc/snmp/SnmpTrapStatus file");
+        return;
+    }
+    file >> std::boolalpha >> snmpTrapStatus;
+    file.close();
+
+    if(snmpTrapStatus == false)
+    {
+        lg2::error("SNMPTrap is not enabled");
+        return;
+    }
+
     init_snmp("snmpapp");
 
     // TODO: https://github.com/openbmc/openbmc/issues/3145
-- 
2.34.1

