From 5a2ef676015ec28dc390521aaeb39bcbe9efb02e Mon Sep 17 00:00:00 2001
From: pandiarajt <pandiarajt@ami.com>
Date: Thu, 29 Feb 2024 20:40:43 +0530
Subject: [PATCH 70/70] Added changes for SNMPTrap Bind variables

---
 snmp_notification.hpp | 57 +++++++++++++++++++++++++++++++++++++++----
 1 file changed, 52 insertions(+), 5 deletions(-)

diff --git a/snmp_notification.hpp b/snmp_notification.hpp
index b48bc78..32a48ad 100644
--- a/snmp_notification.hpp
+++ b/snmp_notification.hpp
@@ -149,9 +149,14 @@ class OBMCErrorNotification : public Notification
 {
   private:
     uint32_t OBMCErrorID = 0;
-    uint64_t OBMCErrorTimestamp = 0;
-    int32_t OBMCErrorSeverity = 0;
+    std::string OBMCErrorTimestamp;
+    std::string OBMCErrorSeverity;
     std::string OBMCErrorMessage;
+    std::string OBMCeventID ;
+    std::string OBMCEventStatus;
+    std::string OBMCEventSubjectSN;
+    std::string OBMCaddData ;
+    std::string OBMCIP;
 
   public:
     OBMCErrorNotification() = delete;
@@ -167,11 +172,16 @@ class OBMCErrorNotification : public Notification
      *  @param[in] sev - The severity of the error.
      *  @param[in] msg - The message of the error.
      */
-    OBMCErrorNotification(uint32_t id, uint64_t ts, int32_t sev,
+    OBMCErrorNotification(uint32_t id, std::string ts, std::string sev,
                           std::string msg) :
         OBMCErrorID(id),
         OBMCErrorTimestamp(ts), OBMCErrorSeverity(sev), OBMCErrorMessage(msg)
     {}
+ 
+    OBMCErrorNotification(uint32_t id, std::string ts, std::string sev, std::string msg, std::string eventID ,std::string EventStatus,std::string EventSubjectSN, std::string addData, std::string IP ):
+        OBMCErrorID(id),OBMCErrorTimestamp(ts), OBMCErrorSeverity(sev), OBMCErrorMessage(msg), OBMCeventID(eventID), OBMCEventStatus(EventStatus), OBMCEventSubjectSN(EventSubjectSN), OBMCaddData(addData), OBMCIP(IP)
+    {}
+
 
   protected:
     std::pair<OID, OID_LEN> getTrapOID() override
@@ -185,7 +195,7 @@ class OBMCErrorNotification : public Notification
     std::vector<Object> getFieldOIDList() override
     {
         std::vector<Object> objectList;
-        objectList.reserve(4);
+        objectList.reserve(9);
         {
             OID_LEN idLen = 11;
             OID id = {1, 3, 6, 1, 4, 1, 49871, 1, 0, 1, 1};
@@ -203,17 +213,54 @@ class OBMCErrorNotification : public Notification
         {
             OID_LEN idLen = 11;
             OID id = {1, 3, 6, 1, 4, 1, 49871, 1, 0, 1, 3};
+            u_char type = getASNType<decltype(OBMCeventID)>();
+
+            objectList.emplace_back(id, idLen, type, OBMCeventID);
+        }
+        {
+            OID_LEN idLen = 11;
+            OID id = {1, 3, 6, 1, 4, 1, 49871, 1, 0, 1, 4};
             u_char type = getASNType<decltype(OBMCErrorSeverity)>();
 
             objectList.emplace_back(id, idLen, type, OBMCErrorSeverity);
         }
         {
             OID_LEN idLen = 11;
-            OID id = {1, 3, 6, 1, 4, 1, 49871, 1, 0, 1, 4};
+            OID id = {1, 3, 6, 1, 4, 1, 49871, 1, 0, 1, 5};
             u_char type = getASNType<decltype(OBMCErrorMessage)>();
 
             objectList.emplace_back(id, idLen, type, OBMCErrorMessage);
         }
+        {
+            OID_LEN idLen = 11;
+            OID id = {1, 3, 6, 1, 4, 1, 49871, 1, 0, 1, 6};
+            u_char type = getASNType<decltype(OBMCIP)>();
+
+            objectList.emplace_back(id, idLen, type, OBMCIP);
+        }
+        {
+            OID_LEN idLen = 11;
+            OID id = {1, 3, 6, 1, 4, 1, 49871, 1, 0, 1, 7};
+            u_char type = getASNType<decltype(OBMCEventStatus)>();
+
+            objectList.emplace_back(id, idLen, type, OBMCEventStatus);
+        }
+	{
+            OID_LEN idLen = 11;
+            OID id = {1, 3, 6, 1, 4, 1, 49871, 1, 0, 1, 8};
+            u_char type = getASNType<decltype(OBMCEventSubjectSN)>();
+
+            objectList.emplace_back(id, idLen, type, OBMCEventSubjectSN);
+        }
+
+        {
+            OID_LEN idLen = 11;
+            OID id = {1, 3, 6, 1, 4, 1, 49871, 1, 0, 1, 9};
+            u_char type = getASNType<decltype(OBMCaddData)>();
+
+            objectList.emplace_back(id, idLen, type, OBMCaddData);
+        }
+
         return objectList;
     }
 
-- 
2.34.1

