From b36169a48d2b6906e814b4e310dd8f1ab35f8afc Mon Sep 17 00:00:00 2001
From: sanjays <sanjays@ami.com>
Date: Wed, 23 Jul 2025 17:50:32 +0530
Subject: [PATCH 70/70] fix for OOB error

---
 snmp_util.cpp | 17 +++++++++++------
 1 file changed, 11 insertions(+), 6 deletions(-)

diff --git a/snmp_util.cpp b/snmp_util.cpp
index 5015ea7..6f70410 100644
--- a/snmp_util.cpp
+++ b/snmp_util.cpp
@@ -46,6 +46,7 @@ namespace network
 
 std::string resolveAddress(const std::string& address)
 {
+    int ret = 0;
     addrinfo hints{};
     addrinfo* addr = nullptr;
 
@@ -64,7 +65,7 @@ std::string resolveAddress(const std::string& address)
     AddrPtr addrPtr{addr};
     addr = nullptr;
 
-    char ipaddress[INET6_ADDRSTRLEN]{0};
+    char ipaddress[INET6_ADDRSTRLEN+2]{0};
     result = getnameinfo(addrPtr->ai_addr, addrPtr->ai_addrlen, ipaddress,
                          sizeof(ipaddress), NULL, 0, NI_NUMERICHOST);
     if (result)
@@ -91,11 +92,15 @@ std::string resolveAddress(const std::string& address)
                        "RC", isValid);
             elog<InternalFailure>();
         }
-        char ipTemp[INET6_ADDRSTRLEN]{0};
-        strcat(ipTemp,"[");
-        strcat(ipTemp,ipaddress);
-        strcat(ipTemp,"]");
-        strcpy(ipaddress,ipTemp);
+        char ipTemp[INET6_ADDRSTRLEN+2]{0};
+	strncpy(ipTemp, ipaddress, sizeof(ipaddress));
+	ret = snprintf(ipaddress, sizeof(ipaddress), "[%s]", ipTemp);
+	if (!(ret >= 0 && ret < INET6_ADDRSTRLEN)) {
+		lg2::error("Failed to format IPv6 address {ADDRESS}: {RC}", "ADDRESS", ipaddress,
+				"RC", isValid);
+		elog<InternalFailure>();
+	}
+
     }
 
     return ipaddress;
-- 
2.34.1

