From e84aa69fb4e284dd3c9d9c6939bfa3d5a610429b Mon Sep 17 00:00:00 2001
From: madhankumars <madhankumars@ami.com>
Date: Thu, 11 Dec 2025 14:28:15 +0530
Subject: [PATCH 74/74] Decryption-of-SNMP-User-Password

---
 meson.build           | 2 ++
 snmp_notification.cpp | 4 ++++
 2 files changed, 6 insertions(+)

diff --git a/meson.build b/meson.build
index 4203295..db613b0 100644
--- a/meson.build
+++ b/meson.build
@@ -64,6 +64,7 @@ libsnmp_deps = [
 libsnmp_sources = files(
     'snmp_notification.cpp',
     'snmp_util.cpp',
+    'Encryption.cpp',
 )
 
 libsnmp_lib = library(
@@ -92,6 +93,7 @@ import('pkgconfig').generate(
 install_headers(
     'snmp.hpp',
     'snmp_notification.hpp',
+    'Encryption.hpp',
     subdir: '.',
 )
 
diff --git a/snmp_notification.cpp b/snmp_notification.cpp
index 0948e27..0bf5efa 100644
--- a/snmp_notification.cpp
+++ b/snmp_notification.cpp
@@ -7,6 +7,7 @@
 #include <grp.h>
 #include <map>
 #include <iostream>
+#include "Encryption.hpp"
 
 namespace phosphor
 {
@@ -231,6 +232,7 @@ namespace phosphor
 				std::string encryption;
 				bool isRead = false;
 				const int maxRetries = 3;
+				int outlen = 0;
 				for (int retry = 0; retry < maxRetries; ++retry)
 				{
 				    std::filesystem::file_time_type timeBeforeRead = std::filesystem::last_write_time(snmpUserFilePath);
@@ -243,6 +245,8 @@ namespace phosphor
 					    break;
 					}
 				    std::tie(passwd, encryption, algorithm, timeAfterRead) = getSnmpUserInfo(snmpUserFile, snmpUserFilePath);
+				    std::string DecryptedPswd = decryptString(passwd.c_str(),&outlen);
+				    passwd = DecryptedPswd;
 
 				    if (timeBeforeRead == timeAfterRead)
 					{
-- 
2.34.1

