From 0310ca81c709cdc4ec6d13fbc73b90154ffd7e08 Mon Sep 17 00:00:00 2001
From: Vijaysankar R <vijaysankarr@ami.com>
Date: Sun, 1 Jun 2025 23:35:25 +0530
Subject: [PATCH 80/80] Coverity-Fix

---
 snmp_main.cpp | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/snmp_main.cpp b/snmp_main.cpp
index 29c18c3..d3c29c8 100644
--- a/snmp_main.cpp
+++ b/snmp_main.cpp
@@ -21,6 +21,8 @@ using EventPtr = std::unique_ptr<sd_event, EventDeleter>;
 
 int main(int /*argc*/, char** /*argv[]*/)
 {
+    try
+    {
     auto bus = sdbusplus::bus::new_default();
 
     sd_event* event = nullptr;
@@ -47,4 +49,10 @@ int main(int /*argc*/, char** /*argv[]*/)
     manager->restoreClients();
 
     return sd_event_loop(eventPtr.get());
+    }
+    catch (const std::exception& e)
+    {
+	    lg2::error("Caught exception: {ERROR}", "ERROR", e.what());
+	    return -1;
+    }
 }
-- 
2.34.1

