From f3078905010c2a990cf0b9d234a297b84090e24d Mon Sep 17 00:00:00 2001
From: Vijaysankar R <vijaysankarr@ami.com>
Date: Thu, 8 May 2025 16:03:27 +0530
Subject: [PATCH 4/4] Added-Radius-In-Pamorder

---
 src/radiusclient.c | 23 ++++++++++++++++-------
 1 file changed, 16 insertions(+), 7 deletions(-)

diff --git a/src/radiusclient.c b/src/radiusclient.c
index 98c2d20..0ce79fd 100644
--- a/src/radiusclient.c
+++ b/src/radiusclient.c
@@ -307,6 +307,7 @@ static int nslcd_request_exists(pam_handle_t *pamh, struct pld_cfg *cfg,
     SKIP_STRING(fp);            /* shell */
   )
 #endif
+return 0;
 }
 
 /* perform an authentication call over nslcd */
@@ -344,6 +345,7 @@ static int nslcd_request_authc(pam_handle_t *pamh, struct pld_cfg *cfg,
     }
   )
 #endif
+return 0;
 }
 
 /* do a session open nslcd request */
@@ -366,6 +368,7 @@ static int nslcd_request_sess_o(pam_handle_t *pamh, struct pld_cfg *cfg,
     READ_STRING(fp, resp->msg)
   )
 #endif
+return 0;
 }
 
 /* do a session close nslcd request */
@@ -388,6 +391,7 @@ static int nslcd_request_sess_c(pam_handle_t *pamh, struct pld_cfg *cfg,
     /* no result entry to read */ ;
   )
 #endif
+return 0;
 }
 
 /* do a password modification nslcd call */
@@ -416,6 +420,7 @@ static int nslcd_request_pwmod(pam_handle_t *pamh, struct pld_cfg *cfg,
     READ_STRING(fp, resp->msg);
   )
 #endif
+return 0;
 }
 
 static int nslcd_request_config_get(pam_handle_t *pamh, struct pld_cfg *cfg,
@@ -431,6 +436,7 @@ static int nslcd_request_config_get(pam_handle_t *pamh, struct pld_cfg *cfg,
     READ_STRING(fp, resp->msg);
   )
 #endif
+return 0;
 }
 
 /* remap the return code based on the configuration */
@@ -654,7 +660,8 @@ PAM_EXTERN int pam_sm_authenticate(pam_handle_t *pamh, int flags, int argc,
     if (NULL != getRadiusEnableDbusProperty) {
       retval = getRadiusEnableDbusProperty(&radiusEnableStatus);
       if ((retval < 0) || (radiusEnableStatus == false)) {
-        retval = -1;
+        retval = PAM_AUTH_ERR;
+	dlclose(bnss_radius_handle);
         return retval;
       }
     }
@@ -663,12 +670,15 @@ PAM_EXTERN int pam_sm_authenticate(pam_handle_t *pamh, int flags, int argc,
     if (NULL != getRadiusPortDbusProperty) {
       retval = getRadiusPortDbusProperty(&radiusPortNo);
       if ((retval < 0) || (radiusPortNo != 1812)) {
-        retval = -1;
+        retval = PAM_AUTH_ERR;
+	dlclose(bnss_radius_handle);
         return retval;
       }
     }
   }
-
+  if(bnss_radius_handle != NULL){
+  dlclose(bnss_radius_handle);
+  }
   /*libnss_radius related variables declartions*/
   pamusrpkt_t pam_radius_req, pam_radius_res;
   memset(&pam_radius_req, 0, sizeof(pamusrpkt_t));
@@ -746,7 +756,6 @@ PAM_EXTERN int pam_sm_authenticate(pam_handle_t *pamh, int flags, int argc,
     if (rc_avpair_parse(rh, userNamePassword, vp) < 0) {
       retval = PAM_AUTH_ERR;
       return retval;
-      vp = &send->next;
     }
     retval = process(rh, send, acct, nas_port, &privilage);
     if (retval == 0) {
@@ -757,13 +766,13 @@ PAM_EXTERN int pam_sm_authenticate(pam_handle_t *pamh, int flags, int argc,
                      PAM_RADIUS_RES_Q);
       if (ret < 0 || ret >= (signed)sizeof(pam_radius_req.srcq)) {
         fprintf(stderr, "Buffer Overflow\n");
-        return -1;
+        return PAM_AUTH_ERR;
       }
       ret = snprintf(pam_radius_req.usr.name, sizeof(pam_radius_req.usr.name),
                      "%s", username);
       if (ret < 0 || ret >= (signed)sizeof(pam_radius_req.usr.name)) {
         fprintf(stderr, "Buffer Overflow\n");
-        return -1;
+        return PAM_AUTH_ERR;
       }
       // pam_radius_req.usr.priv = privilege;
       // this need to updated while sync
@@ -785,7 +794,7 @@ PAM_EXTERN int pam_sm_authenticate(pam_handle_t *pamh, int flags, int argc,
       return PAM_SUCCESS;
     } else {
       // authentication failure
-      return retval;
+      return PAM_AUTH_ERR;
     }
   }
   while (1 == 1) {
-- 
2.34.1

