From 920e279685b135ffbd83bb893c8cc8d7131437f6 Mon Sep 17 00:00:00 2001
From: ramsankarr <ramsankarr@ami.com>
Date: Thu, 7 Aug 2025 13:36:41 +0530
Subject: [PATCH 6/6] RADIUS Auth Init Info

---
 src/radiusclient.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/radiusclient.c b/src/radiusclient.c
index 2bace0c..6838125 100644
--- a/src/radiusclient.c
+++ b/src/radiusclient.c
@@ -625,8 +625,7 @@ void GetModuleCountEnv(pam_handle_t *pamh, int *cntr) {
 PAM_EXTERN int pam_sm_authenticate(pam_handle_t *pamh, int flags, int argc,
                                    const char **argv) {
   /*radius module related variable declarations*/
-  int i, nas_port, ch, acct, server, ecount, firstline, theend, ret = 0,
-                                                                pam_handle;
+  int i, nas_port, ch, acct, server, ecount, firstline, theend, ret = 0,pam_handle;
   void *rh;
   size_t len;
   VALUE_PAIR *send, **vp;
@@ -655,6 +654,7 @@ PAM_EXTERN int pam_sm_authenticate(pam_handle_t *pamh, int flags, int argc,
   int (*getRadiusPortDbusProperty)(int *radiusPortNo);
   bool radiusEnableStatus = false;
   int radiusPortNo = 0;
+  fprintf(stderr, "RADIUS User Authentication Started\n");
   bnss_radius_handle = dlopen("/usr/lib/libnss_radius.so", RTLD_LAZY);
   if (bnss_radius_handle) {
     getRadiusEnableDbusProperty =
-- 
2.34.1

