From 8c0c1efe5699e065d1921454efc1bb8c3471d270 Mon Sep 17 00:00:00 2001
From: madhankumars <madhankumars@ami.com>
Date: Thu, 11 Sep 2025 18:28:10 +0530
Subject: [PATCH 8/8] Encrypt and Decrypt the password

---
 lib/config.c       | 15 +++++++++++----
 src/radiusclient.c | 31 ++++++++++++++++++++++---------
 2 files changed, 33 insertions(+), 13 deletions(-)

diff --git a/lib/config.c b/lib/config.c
index d42e9d0..eeb533d 100644
--- a/lib/config.c
+++ b/lib/config.c
@@ -18,6 +18,7 @@
 #include <includes.h>
 #include <radiusclient-ng.h>
 #include <options.h>
+#include <Encryption.h>
 
 static int test_config(rc_handle *, char *);
 
@@ -631,12 +632,18 @@ int rc_find_server (rc_handle *rh, char *server_name, UINT4 *ip_addr, char *secr
 
 		memset (secret, '\0', MAX_SECRET_LENGTH);
 		len = strlen (s);
-		if (len > MAX_SECRET_LENGTH)
+		char *decryptedpwd = decryptString(s, len);
+		if (decryptedpwd != NULL)
 		{
-			len = MAX_SECRET_LENGTH;
+			strncpy(secret, decryptedpwd, MAX_SECRET_LENGTH);
+			secret[MAX_SECRET_LENGTH] = '\0';
+		}
+		else
+		{
+			rc_log(LOG_ERR, "Decryption failed");
+			fclose (clientfd);
+			return -1;
 		}
-		strncpy (secret, s, (size_t) len);
-		secret[MAX_SECRET_LENGTH] = '\0';
 
 		if (!strchr (hostnm, '/')) /* If single name form */
 		{
diff --git a/src/radiusclient.c b/src/radiusclient.c
index d6d2e19..673505d 100644
--- a/src/radiusclient.c
+++ b/src/radiusclient.c
@@ -794,13 +794,15 @@ PAM_EXTERN int pam_sm_authenticate(pam_handle_t *pamh, int flags, int argc,
       bnss_radius_handle = dlopen("/usr/lib/libnss_radius.so", RTLD_LAZY);
       if (bnss_radius_handle) {
         post_pam_userinfo = dlsym(bnss_radius_handle, "post_pam_userinfo");
-        if (NULL != post_pam_userinfo) {
-          retval = post_pam_userinfo(&pam_radius_req, PAM_HELPER_Q);
-          if (retval < 0) {
-            retval = PAM_AUTHINFO_UNAVAIL;
-            return retval;
-          }
-        }
+	if (NULL != post_pam_userinfo) {
+		pam_radius_req.checksum = 0;
+		pam_radius_req.checksum = calculate_checksum(&pam_radius_req);
+		retval = post_pam_userinfo(&pam_radius_req, PAM_HELPER_Q);
+		if (retval < 0) {
+			retval = PAM_AUTHINFO_UNAVAIL;
+			return retval;
+		}
+	}
       }
 
       return PAM_SUCCESS;
@@ -929,7 +931,7 @@ int process(void *rh, VALUE_PAIR *send, int acct, int nas_port,
   VALUE_PAIR *received;
   char msg[4096];
   char privlage[1024] = {0};
-  char groupName[1024] = {0};
+  char *groupName = NULL;
   char username[256] = {0};
   int i, retval = 0;
   char *serverRes;
@@ -997,7 +999,15 @@ int process(void *rh, VALUE_PAIR *send, int acct, int nas_port,
 		    }
 	    }
 	    serverRes = rc_avpair_log(rh, received);
+	    groupName = malloc((strlen(serverRes)+1) *sizeof(char));
+	    if(groupName == NULL)
+	    {
+		    retval = -1;
+		    rc_avpair_free(received);
+		    return retval;
+	    }
 	    memcpy(groupName, serverRes, strlen(serverRes));
+	    groupName[strlen(serverRes)] = '\0';
 	    bnss_radius_handle = dlopen("/usr/lib/libnss_radius.so", RTLD_LAZY);
 	    if (bnss_radius_handle) {
 		    getDbusProperty = dlsym(bnss_radius_handle, "getDbusProperty");
@@ -1007,11 +1017,12 @@ int process(void *rh, VALUE_PAIR *send, int acct, int nas_port,
 					    (strlen(privlage) < 1)) {
 				    retval = -1;
 				    rc_avpair_free(received);
+				    free(groupName);
 				    return retval;
 			    } else {
 				    for (iter = 0; iter < (sizeof(privBuf) / sizeof(privBuf[0]));
 						    iter++) {
-					    if (strstr(privBuf[iter].privStr, privlage)) {
+					    if ((strcmp(privBuf[iter].privStr, privlage)) == 0){
 						    *privilage = privBuf[iter].privNum;
 						    break;
 					    }
@@ -1019,11 +1030,13 @@ int process(void *rh, VALUE_PAIR *send, int acct, int nas_port,
 				    if (iter >= ((sizeof(privBuf) / sizeof(privBuf[0])))) {
 					    retval = -1;
 					    rc_avpair_free(received);
+					    free(groupName);
 					    return retval;
 				    }
 			    }
 		    }
 	    }
+	    free(groupName);
 	    rc_avpair_free(received);
 	    received = NULL;
     }
-- 
2.34.1

