From eaf8b303876de5fc5d765c7aa1bc31120bc4c121 Mon Sep 17 00:00:00 2001
From: arjuncr <arjuncr@ami.com>
Date: Tue, 23 Apr 2024 14:26:02 -0400
Subject: [PATCH] fix for legacy nm version

Signed-off-by: arjuncr <arjuncr@ami.com>
---
 NodeManagerProxy.hpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/NodeManagerProxy.hpp b/NodeManagerProxy.hpp
index e94001d..1474e64 100644
--- a/NodeManagerProxy.hpp
+++ b/NodeManagerProxy.hpp
@@ -367,7 +367,7 @@ class GetMeVer
                 "xyz.openbmc_project.Software.Version.VersionPurpose.ME"));
 
         iface->register_property(
-            "Version", std::string(""),
+            "Version", std::string("nan"),
             [](const std::string &newVal, std::string &oldVal) { return 1; },
             [this](const std::string &val) { return getDevId(); });
 
@@ -403,7 +403,7 @@ class GetMeVer
 
     std::string getDevId()
     {
-        constexpr const char *invalidMeVersion = "";
+        constexpr const char *invalidMeVersion = "nan";
         std::vector<uint8_t> dataToSend;
 
         IpmbDbusRspType ipmbResponse;
-- 
2.34.1

