From 5d399eea0175de845b29b0f9b57bb29c68542ed8 Mon Sep 17 00:00:00 2001
From: arjuncr <arjuncr@ami.com>
Date: Wed, 25 Mar 2026 03:46:00 -0400
Subject: [PATCH] update CMAKE_CXX_STANDARD to 23

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8a87655..b1ef32e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,6 +1,6 @@
 cmake_minimum_required (VERSION 3.5 FATAL_ERROR)
 project (node-manager-proxy CXX)
-set (CMAKE_CXX_STANDARD 20)
+set (CMAKE_CXX_STANDARD 23)
 set (CMAKE_CXX_STANDARD_REQUIRED ON)
 set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fno-rtti")
 set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fno-rtti")
-- 
2.34.1

