From 1c0ef00d0993e07d412fdaa7326bb6abeb3ebf6b Mon Sep 17 00:00:00 2001
From: Abinaya L <abinayal@ami.com>
Date: Fri, 19 May 2023 14:36:03 +0530
Subject: [PATCH 2/2] Fixed compilation error on USB gadget

---
 drivers/usb/gadget/function/f_rndis.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/usb/gadget/function/f_rndis.c b/drivers/usb/gadget/function/f_rndis.c
index 425db0a..117fc24 100644
--- a/drivers/usb/gadget/function/f_rndis.c
+++ b/drivers/usb/gadget/function/f_rndis.c
@@ -428,6 +428,7 @@ static void rndis_response_complete(struct usb_ep *ep, struct usb_request *req)
 		DBG(cdev, "RNDIS %s response error %d, %d/%d\n",
 			ep->name, status,
 			req->actual, req->length);
+		break;
 		/* FALLTHROUGH */
 	case 0:
 		if (ep != rndis->notify)
@@ -712,7 +713,7 @@ rndis_bind(struct usb_configuration *c, struct usb_function *f)
 					  sizeof(rndis->ethaddr));
 	if (status < 12) {
 		kfree(rndis);
-		return ERR_PTR(-EINVAL);
+		return -EINVAL;
 	}
 	rndis_string_defs[0].s = rndis->ethaddr;
 	us = usb_gstrings_attach(cdev, rndis_strings,
-- 
2.25.1

