From d1da9c513893dab0c41cfce2a015c4f41dce0079 Mon Sep 17 00:00:00 2001
From: Abinaya L <abinayal@ami.com>
Date: Fri, 21 Jul 2023 17:04:10 +0530
Subject: [PATCH 2/2] 0002-Fixed-compilation-error-on-USB-gadget

---
 drivers/usb/gadget/function/f_rndis.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/usb/gadget/function/f_rndis.c b/drivers/usb/gadget/function/f_rndis.c
index 60653cb..7a54222 100644
--- a/drivers/usb/gadget/function/f_rndis.c
+++ b/drivers/usb/gadget/function/f_rndis.c
@@ -431,6 +431,7 @@ static void rndis_response_complete(struct usb_ep *ep, struct usb_request *req)
 			ep->name, status,
 			req->actual, req->length);
                /* FALLTHROUGH */
+	       break;
 	case 0:
 		if (ep != rndis->notify)
 			break;
@@ -714,7 +715,7 @@ rndis_bind(struct usb_configuration *c, struct usb_function *f)
                                          sizeof(rndis->ethaddr));
        if (status < 12) {
                kfree(rndis);
-               return ERR_PTR(-EINVAL);
+               return -EINVAL;
        }
        rndis_string_defs[0].s = rndis->ethaddr;
 
-- 
2.25.1

