From 3f47b15c977fef619ab3a685a094c18dfe191ebe Mon Sep 17 00:00:00 2001
From: saikumaru <saikumaru@ami.com>
Date: Thu, 12 Sep 2024 11:19:35 +0530
Subject: [PATCH 03/19] Ported ADC driver changes from INTEL & AMI.

Signed-off-by: saikumaru <saikumaru@ami.com>
---
 drivers/iio/adc/aspeed_adc.c | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/drivers/iio/adc/aspeed_adc.c b/drivers/iio/adc/aspeed_adc.c
index 998e8bcc06e1..c16f8de7e7e2 100644
--- a/drivers/iio/adc/aspeed_adc.c
+++ b/drivers/iio/adc/aspeed_adc.c
@@ -82,6 +82,9 @@
  */
 #define ASPEED_ADC_DEF_SAMPLING_RATE	65000
 
+#define ASPEED_ADC_CHANNELS_MAX		16
+#define ASPEED_ADC_RAW_VALUE_MAX	GENMASK(ASPEED_RESOLUTION_BITS - 1, 0)
+
 struct aspeed_adc_trim_locate {
 	const unsigned int offset;
 	const unsigned int field;
@@ -120,6 +123,7 @@ struct aspeed_adc_data {
 	int			cv;
 	bool			battery_sensing;
 	struct adc_gain		battery_mode_gain;
+	int			channel_raw_value[ASPEED_ADC_CHANNELS_MAX];
 };
 
 #define ASPEED_CHAN(_idx, _data_reg_addr) {			\
@@ -307,6 +311,13 @@ static int aspeed_adc_read_raw(struct iio_dev *indio_dev,
 			       data->base + ASPEED_REG_ENGINE_CONTROL);
 		} else
 			*val = readw(data->base + chan->address);
+		if (*val == ASPEED_ADC_RAW_VALUE_MAX) {
+			*val = data->channel_raw_value[chan->channel];
+			pr_err("aspeed_adc: channel %d drop invalid raw reading %lu\n",
+			       chan->channel, ASPEED_ADC_RAW_VALUE_MAX);
+		} else {
+			data->channel_raw_value[chan->channel] = *val;
+		}
 		return IIO_VAL_INT;
 
 	case IIO_CHAN_INFO_OFFSET:
@@ -488,6 +499,7 @@ static int aspeed_adc_probe(struct platform_device *pdev)
 	u32 adc_engine_control_reg_val;
 	unsigned long scaler_flags = 0;
 	char clk_name[32], clk_parent_name[32];
+	int i;
 
 	indio_dev = devm_iio_device_alloc(&pdev->dev, sizeof(*data));
 	if (!indio_dev)
@@ -626,6 +638,10 @@ static int aspeed_adc_probe(struct platform_device *pdev)
 	}
 
 	aspeed_adc_compensation(indio_dev);
+
+	for (i = 0; i < ASPEED_ADC_CHANNELS_MAX; i++)
+		data->channel_raw_value[i] = ASPEED_ADC_RAW_VALUE_MAX;
+
 	/* Start all channels in normal mode. */
 	adc_engine_control_reg_val =
 		readl(data->base + ASPEED_REG_ENGINE_CONTROL);
-- 
2.43.2

