diff -Naur A/arch/arm/boot/dts/aspeed-bmc-intel-bhs.dtsi  B/arch/arm/boot/dts/aspeed-bmc-intel-bhs.dtsi
--- A/arch/arm/boot/dts/aspeed-bmc-intel-bhs.dtsi	2024-01-22 15:42:17.295613329 +0530
+++ B/arch/arm/boot/dts/aspeed-bmc-intel-bhs.dtsi	2024-01-22 15:44:00.144673622 +0530
@@ -138,10 +138,11 @@
 		 */
 		spi-max-frequency = <25000000>;
 		spi-rx-bus-width = <4>;
-		spi-tx-bus-width = <4>;
 		m25p,fast-read;
 #include "openbmc-flash-layout-intel-bhs-128MB.dtsi"
 	};
+
+
 };
 
 /* SPI1 controller - BIOS SPI flash */
