From 72b75af56449828be69a2ff63911aa8b20dfac38 Mon Sep 17 00:00:00 2001
From: mickhuang <mickhuang@ami.com>
Date: Thu, 5 Sep 2024 20:24:33 +0800
Subject: [PATCH] Fix probe regression for ASPEED UDC

Signed-off-by: mickhuang <mickhuang@ami.com>
---
 drivers/usb/gadget/udc/aspeed_udc.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/usb/gadget/udc/aspeed_udc.c b/drivers/usb/gadget/udc/aspeed_udc.c
index 01968e2167f9..86cdbbb7ced1 100644
--- a/drivers/usb/gadget/udc/aspeed_udc.c
+++ b/drivers/usb/gadget/udc/aspeed_udc.c
@@ -1301,6 +1301,7 @@ static int ast_udc_start(struct usb_gadget *gadget,
 	UDC_DBG(udc, "\n");
 	udc->driver = driver;
 	udc->gadget.dev.of_node = udc->pdev->dev.of_node;
+	udc->gadget.dev.of_node_reused = true;
 
 	for (i = 0; i < AST_UDC_NUM_ENDPOINTS; i++) {
 		ep = &udc->ep[i];
-- 
2.34.1

