From 38e2454c7d1301a4eb2195870ad276a57fa31890 Mon Sep 17 00:00:00 2001
From: seancheng1 <seancheng@ami.com>
Date: Tue, 26 Nov 2024 11:10:49 +0800
Subject: [PATCH] Fix send hid data will halt on hid queue.

---
 drivers/usb/gadget/function/f_hid.c | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/drivers/usb/gadget/function/f_hid.c b/drivers/usb/gadget/function/f_hid.c
index 2db01e03bfbf..eb75a0d75693 100644
--- a/drivers/usb/gadget/function/f_hid.c
+++ b/drivers/usb/gadget/function/f_hid.c
@@ -438,6 +438,7 @@ static ssize_t f_hidg_write(struct file *file, const char __user *buffer,
 	struct usb_request *req;
 	unsigned long flags;
 	ssize_t status = -ENOMEM;
+	long timeout = msecs_to_jiffies(5);
 
 	spin_lock_irqsave(&hidg->write_spinlock, flags);
 
@@ -454,9 +455,19 @@ static ssize_t f_hidg_write(struct file *file, const char __user *buffer,
 		if (file->f_flags & O_NONBLOCK)
 			return -EAGAIN;
 
+		/*
+		//Due to retry send data will be locked when write queue will busy or write failed.
+		//Modify to interruptible timeout and setup 5ms for daemon process needs.
 		if (wait_event_interruptible_exclusive(
 				hidg->write_queue, WRITE_COND))
 			return -ERESTARTSYS;
+		*/
+		status = wait_event_interruptible_timeout(hidg->write_queue, WRITE_COND,timeout);
+		if(status == 0) {
+			return -EAGAIN;
+		} else if (status == -ERESTARTSYS) {
+			return -ERESTARTSYS;
+		}
 
 		spin_lock_irqsave(&hidg->write_spinlock, flags);
 	}
-- 
2.34.1

