diff --git a/drivers/Makefile b/drivers/Makefile
index e1a0f44d5e62..94c430656ca9 100644
--- a/drivers/Makefile
+++ b/drivers/Makefile
@@ -200,3 +200,6 @@ obj-$(CONFIG_DRM_ACCEL)		+= accel/
 obj-$(CONFIG_CDX_BUS)		+= cdx/
 
 obj-$(CONFIG_S390)		+= s390/
+
+obj-$(CONFIG_JTAG)		+= jtag/
+
diff --git a/drivers/Kconfig b/drivers/Kconfig
index efb66e25fa2d..6b6c8ab87a79 100644
--- a/drivers/Kconfig
+++ b/drivers/Kconfig
@@ -241,6 +241,8 @@ source "drivers/peci/Kconfig"
 
 source "drivers/hte/Kconfig"
 
+source "drivers/jtag/Kconfig"
+
 source "drivers/cdx/Kconfig"
 
 endmenu
