From 60c59c3c94a4314a321a1b83bfc592acc6149c44 Mon Sep 17 00:00:00 2001
From: Pranoy Goru <gorumanip@ami.com>
Date: Fri, 13 Oct 2023 13:08:57 +0530
Subject: [PATCH] managers-add-factory-restore

---
 redfish-core/lib/managers.hpp | 37 +++++++++++++++++++++++++++++++----
 1 file changed, 33 insertions(+), 4 deletions(-)

diff --git a/redfish-core/lib/managers.hpp b/redfish-core/lib/managers.hpp
index 6c64ee5b..20b14a78 100644
--- a/redfish-core/lib/managers.hpp
+++ b/redfish-core/lib/managers.hpp
@@ -104,6 +104,33 @@ inline void
     });
 }
 
+inline void
+    writeRestoreOptions(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+                        const std::string& resetType)
+{
+    constexpr const char* restoreOpFname = "/tmp/.rwfs/.restore_op";
+    int option = 0;
+
+    if (resetType == "ResetAll")
+    {
+        option = 2; // full restore
+    }
+    else if (resetType == "ResetToDefaultButKeepReservedSettings")
+    {
+        option = 5; // reset to factory defaults but reserve user and lan
+                    // configuration
+    }
+
+    std::ofstream restoreFile(restoreOpFname, std::ios::trunc);
+    if (!restoreFile)
+    {
+        //BMCWEB_LOG_ERROR << "error in opring output stream " << restoreOpFname;
+        messages::internalError(asyncResp->res);
+        return;
+    }
+    restoreFile << option << "\n";
+}
+
 /**
  * ManagerResetAction class supports the POST method for the Reset (reboot)
  * action.
@@ -197,7 +224,8 @@ inline void requestRoutesManagerResetToDefaultsAction(App& app)
             return;
         }
 
-        if (resetType != "ResetAll")
+        if (resetType != "ResetAll" &&
+            resetType != "ResetToDefaultButKeepReservedSettings")
         {
             BMCWEB_LOG_DEBUG(
                 "Invalid property value for ResetToDefaultsType: {}",
@@ -208,13 +236,14 @@ inline void requestRoutesManagerResetToDefaultsAction(App& app)
         }
 
         crow::connections::systemBus->async_method_call(
-            [asyncResp](const boost::system::error_code& ec) {
+            [asyncResp, resetType](const boost::system::error_code& ec) {
             if (ec)
             {
                 BMCWEB_LOG_DEBUG("Failed to ResetToDefaults: {}", ec);
                 messages::internalError(asyncResp->res);
                 return;
             }
+            writeRestoreOptions(asyncResp, resetType);
             // Factory Reset doesn't actually happen until a reboot
             // Can't erase what the BMC is running on
             doBMCGracefulRestart(asyncResp);
@@ -1976,8 +2005,8 @@ inline void requestRoutesManager(App& app)
             asyncResp->res.jsonValue["Actions"]["#Manager.ResetToDefaults"];
         resetToDefaults["target"] =
             "/redfish/v1/Managers/bmc/Actions/Manager.ResetToDefaults";
-        resetToDefaults["ResetType@Redfish.AllowableValues"] =
-            nlohmann::json::array_t({"ResetAll"});
+        resetToDefaults["ResetType@Redfish.AllowableValues"] = {
+            "ResetAll", "ResetToDefaultButKeepReservedSettings"};
 
         std::pair<std::string, std::string> redfishDateTimeOffset =
             redfish::time_utils::getDateTimeOffsetNow();
-- 
2.34.1

