From 8fbd15fe4425f6348389ed2c82b971432e6f48e4 Mon Sep 17 00:00:00 2001
From: Pranoy Goru <gorumanip@ami.com>
Date: Tue, 18 Oct 2022 20:42:39 +0530
Subject: [PATCH] managers-add-factory-restore

---
 redfish-core/lib/managers.hpp | 37 ++++++++++++++++++++++++++++++++---
 1 file changed, 34 insertions(+), 3 deletions(-)

diff --git a/redfish-core/lib/managers.hpp b/redfish-core/lib/managers.hpp
index 195ca553..6078049c 100644
--- a/redfish-core/lib/managers.hpp
+++ b/redfish-core/lib/managers.hpp
@@ -104,6 +104,34 @@ inline void
         interfaceName, destProperty, dbusPropertyValue);
 }
 
+inline void
+    writeRestoreOptions(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+                        const std::string& resetType)
+{
+    constexpr const char* restoreOpFname = "/tmp/.rwfs/.restore_op";
+    int option = 0;
+
+    if (resetType == "ResetAll")
+    {
+        option = 2; // full restore
+    }
+    else if (resetType == "ResetToDefaultButKeepReservedSettings")
+    {
+        option = 5; // reset to factory defaults but reserve user and lan
+                    // configuration
+    }
+
+    std::ofstream restoreFile(restoreOpFname, std::ios::trunc);
+    if (!restoreFile)
+    {
+        BMCWEB_LOG_ERROR << "error in opring output stream " << restoreOpFname;
+        messages::internalError(asyncResp->res);
+        return;
+    }
+    restoreFile << option << "\n";
+}
+
+
 /**
  * ManagerResetAction class supports the POST method for the Reset (reboot)
  * action.
@@ -199,7 +227,8 @@ inline void requestRoutesManagerResetToDefaultsAction(App& app)
             return;
         }
 
-        if (resetType != "ResetAll")
+        if (resetType != "ResetAll" &&
+	           resetType != "ResetToDefaultButKeepReservedSettings")
         {
             BMCWEB_LOG_DEBUG
                 << "Invalid property value for ResetToDefaultsType: "
@@ -210,13 +239,14 @@ inline void requestRoutesManagerResetToDefaultsAction(App& app)
         }
 
         crow::connections::systemBus->async_method_call(
-            [asyncResp](const boost::system::error_code ec) {
+            [asyncResp, resetType](const boost::system::error_code ec) {
             if (ec)
             {
                 BMCWEB_LOG_DEBUG << "Failed to ResetToDefaults: " << ec;
                 messages::internalError(asyncResp->res);
                 return;
             }
+	    writeRestoreOptions(asyncResp, resetType);
             // Factory Reset doesn't actually happen until a reboot
             // Can't erase what the BMC is running on
             doBMCGracefulRestart(asyncResp);
@@ -2057,7 +2087,8 @@ inline void requestRoutesManager(App& app)
             asyncResp->res.jsonValue["Actions"]["#Manager.ResetToDefaults"];
         resetToDefaults["target"] =
             "/redfish/v1/Managers/bmc/Actions/Manager.ResetToDefaults";
-        resetToDefaults["ResetType@Redfish.AllowableValues"] = {"ResetAll"};
+        resetToDefaults["ResetType@Redfish.AllowableValues"] = {
+		"ResetAll", "ResetToDefaultButKeepReservedSettings"};
 
         std::pair<std::string, std::string> redfishDateTimeOffset =
             redfish::time_utils::getDateTimeOffsetNow();
