--- /dev/null
+++ b/redfish-core/lib/sensors.hpp	2022-09-15 15:01:35.552307883 +0530
@@ -57,7 +57,13 @@
     "/xyz/openbmc_project/sensors/current",
     "/xyz/openbmc_project/sensors/count",
     "/xyz/openbmc_project/sensors/airflow",
-               "/xyz/openbmc_project/sensors/humidity",
+    "/xyz/openbmc_project/sensors/humidity",
+    "/xyz/openbmc_project/sensors/utilization",
+    "/xyz/openbmc_project/sensors/altitude",
+    "/xyz/openbmc_project/sensors/voltage",
+    "/xyz/openbmc_project/sensors/fan_tach",
+    "/xyz/openbmc_project/sensors/temperature",
+    "/xyz/openbmc_project/sensors/fan_pwm",
 #ifdef BMCWEB_NEW_POWERSUBSYSTEM_THERMALSUBSYSTEM
     "/xyz/openbmc_project/sensors/voltage",
     "/xyz/openbmc_project/sensors/fan_tach",
@@ -66,7 +72,6 @@
     "/xyz/openbmc_project/sensors/altitude",
     "/xyz/openbmc_project/sensors/energy",
 #endif
-    "/xyz/openbmc_project/sensors/utilization"
 });
 
 auto thermalPaths = std::to_array<std::string_view>({
