diff -Naur bmcweb_b/redfish-core/lib/account_service.hpp bmcweb_a/redfish-core/lib/account_service.hpp
--- bmcweb_b/redfish-core/lib/account_service.hpp	2022-09-22 10:22:07.014147548 +0530
+++ bmcweb_a/redfish-core/lib/account_service.hpp	2022-09-22 10:25:14.035949165 +0530
@@ -1195,7 +1195,18 @@
                     dbus::utility::DbusVariantType{*enabled});
             }
 
-            if (roleId)
+	    if((username == "root") && roleId)
+	    {
+		BMCWEB_LOG_ERROR << "Not able to change privilage level for root user";
+               messages::accessDenied(
+                                asyncResp->res,
+                                crow::utility::urlFromPieces(
+                                    "redfish", "v1", "AccountService",
+				    "Accounts", username));
+
+		return;
+	    }
+	    else if (roleId)
             {
                 std::string priv = getPrivilegeFromRoleId(*roleId);
                 if (priv.empty())
@@ -2012,6 +2023,13 @@
                 tempObjPath /= username;
                 const std::string userPath(tempObjPath);
 
+		if (username == "root")
+		{
+		    BMCWEB_LOG_DEBUG << "Not able to delete root user\n" ; 
+		    messages::resourceCannotBeDeleted(asyncResp->res);
+                    return;
+		}
+
                 crow::connections::systemBus->async_method_call(
                     [asyncResp, username](const boost::system::error_code ec) {
                         if (ec)
