--- /dev/null
+++ b/redfish-core/lib/sensors.hpp	2023-01-03 00:29:37.520059174 +0530
@@ -61,7 +61,6 @@
 static auto sensorPaths = std::to_array<std::string_view>({
     "/xyz/openbmc_project/sensors/power",
     "/xyz/openbmc_project/sensors/current",
-    "/xyz/openbmc_project/sensors/count",
     "/xyz/openbmc_project/sensors/airflow",
     "/xyz/openbmc_project/sensors/humidity",
     "/xyz/openbmc_project/sensors/energy",
@@ -71,6 +70,8 @@
     "/xyz/openbmc_project/sensors/fan_tach",
     "/xyz/openbmc_project/sensors/temperature",
     "/xyz/openbmc_project/sensors/fan_pwm",
+    "/xyz/openbmc_project/sensors/cpu",
+    "/xyz/openbmc_project/sensors/watchdog",
 #ifdef BMCWEB_NEW_POWERSUBSYSTEM_THERMALSUBSYSTEM
     "/xyz/openbmc_project/sensors/voltage",
     "/xyz/openbmc_project/sensors/fan_tach",
