From 0a6b5879c77770cab919317d3e9a4609c0392990 Mon Sep 17 00:00:00 2001
From: nibinc <nibinc@ami.com>
Date: Wed, 7 Jun 2023 02:23:04 -0400
Subject: [PATCH 34/34] fix for the kvm session not able to open more than one

---
 include/kvm_websocket.hpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/kvm_websocket.hpp b/include/kvm_websocket.hpp
index a63bf8f1..3fa753e6 100644
--- a/include/kvm_websocket.hpp
+++ b/include/kvm_websocket.hpp
@@ -169,13 +169,13 @@ inline void requestRoutes(App& app)
 
             if (sessions.size() == maxSessions)
             {
+	        kvmActiveStatus = 1;
                 conn.close("Max sessions are already connected");
                 return;
             }
 
             sessions[&conn] = std::make_unique<KvmSession>(conn);
             conn.session->kvmConnections++;
-	     kvmActiveStatus = 1;
         })
         .onclose([](crow::websocket::Connection& conn, const std::string&) {
             sessions.erase(&conn);
-- 
2.17.1

