From cfc0f8324dc7c2e0696017b683d1267a3d757c30 Mon Sep 17 00:00:00 2001
From: Nibin Chacko <nibinc@ami.com>
Date: Fri, 14 Jul 2023 09:14:23 -0400
Subject: [PATCH 48/48] changing maximum supported kvm session value to 1

---
 include/kvm_websocket.hpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/kvm_websocket.hpp b/include/kvm_websocket.hpp
index e4d97c85..f723695c 100644
--- a/include/kvm_websocket.hpp
+++ b/include/kvm_websocket.hpp
@@ -13,7 +13,7 @@ namespace crow
 namespace obmc_kvm
 {
 
-static constexpr const uint maxSessions = 4;
+static constexpr const uint maxSessions = 1;
 
 class KvmSession
 {
-- 
2.25.1

