From dd45aa7fbd3326428942a113e5939089800ff89a Mon Sep 17 00:00:00 2001
From: vamsikrishnak <vamsikrishnak@ami.com>
Date: Thu, 22 Jun 2023 16:46:48 +0530
Subject: [PATCH] Removing KVM ServiceEnabled property under manager

Signed-off-by: vamsikrishnak <vamsikrishnak@ami.com>
---
 redfish-core/lib/managers.hpp | 9 ---------
 1 file changed, 9 deletions(-)

diff --git a/redfish-core/lib/managers.hpp b/redfish-core/lib/managers.hpp
index fbf7ca9..af82420 100644
--- a/redfish-core/lib/managers.hpp
+++ b/redfish-core/lib/managers.hpp
@@ -2350,15 +2350,6 @@ inline void requestRoutesManager(App& app)
         asyncResp->res.jsonValue["CommandShell"]["ConnectTypesSupported"] = {
             "SSH", "IPMI"};
 
-#ifdef BMCWEB_ENABLE_KVM
-        // Fill in GraphicalConsole info
-        asyncResp->res.jsonValue["GraphicalConsole"]["ServiceEnabled"] = true;
-        asyncResp->res.jsonValue["GraphicalConsole"]["MaxConcurrentSessions"] =
-            4;
-        asyncResp->res.jsonValue["GraphicalConsole"]["ConnectTypesSupported"] =
-            nlohmann::json::array_t({"KVMIP"});
-#endif // BMCWEB_ENABLE_KVM
-
         asyncResp->res.jsonValue["Links"]["ManagerForServers@odata.count"] = 1;
 
         nlohmann::json::array_t managerForServers;
-- 
2.9.5

