From 67e4f9422347235ce746a77c852f4558961d44c7 Mon Sep 17 00:00:00 2001
From: vamsikrishnak <vamsikrishnak@ami.com>
Date: Fri, 13 Oct 2023 13:40:57 +0530
Subject: [PATCH] Removing KVM ServiceEnabled property under manager

Signed-off-by: vamsikrishnak <vamsikrishnak@ami.com>
---
 redfish-core/lib/managers.hpp | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/redfish-core/lib/managers.hpp b/redfish-core/lib/managers.hpp
index 6c64ee5..71d4c42 100644
--- a/redfish-core/lib/managers.hpp
+++ b/redfish-core/lib/managers.hpp
@@ -2001,14 +2001,6 @@ inline void requestRoutesManager(App& app)
         asyncResp->res.jsonValue["CommandShell"]["ConnectTypesSupported"] = {
             "SSH", "IPMI"};
 
-#ifdef BMCWEB_ENABLE_KVM
-        // Fill in GraphicalConsole info
-        asyncResp->res.jsonValue["GraphicalConsole"]["ServiceEnabled"] = true;
-        asyncResp->res.jsonValue["GraphicalConsole"]["MaxConcurrentSessions"] =
-            4;
-        asyncResp->res.jsonValue["GraphicalConsole"]["ConnectTypesSupported"] =
-            nlohmann::json::array_t({"KVMIP"});
-#endif // BMCWEB_ENABLE_KVM
         if constexpr (!bmcwebEnableMultiHost)
         {
             asyncResp->res.jsonValue["Links"]["ManagerForServers@odata.count"] =
-- 
2.9.5

