From 812808e8d28b6bd96f790402896528179362c3d4 Mon Sep 17 00:00:00 2001
From: gokulp <gokulp@ami.com>
Date: Fri, 30 Jun 2023 12:08:33 +0530
Subject: [PATCH 2143/2143] Removed Deprecated properties redfish

Signed-off-by: gokulp <gokulp@ami.com>
---
 redfish-core/lib/chassis.hpp | 13 +++++++------
 1 file changed, 7 insertions(+), 6 deletions(-)

diff --git a/redfish-core/lib/chassis.hpp b/redfish-core/lib/chassis.hpp
index 0b1fe911..9ff838f1 100644
--- a/redfish-core/lib/chassis.hpp
+++ b/redfish-core/lib/chassis.hpp
@@ -346,7 +346,7 @@ inline void
                 if (std::find(interfaces2.begin(), interfaces2.end(),
                               interface) != interfaces2.end())
                 {
-                    getIndicatorLedState(asyncResp);
+                    // getIndicatorLedState(asyncResp);
                     getLocationIndicatorActive(asyncResp);
                     break;
                 }
@@ -406,15 +406,16 @@ inline void
 
                 asyncResp->res.jsonValue["Name"] = chassisId;
                 asyncResp->res.jsonValue["Id"] = chassisId;
-#ifdef BMCWEB_ALLOW_DEPRECATED_POWER_THERMAL
+
+//Fixing RedFish Validator issue
+/* #ifdef BMCWEB_ALLOW_DEPRECATED_POWER_THERMAL
                 asyncResp->res.jsonValue["Thermal"]["@odata.id"] =
                     "/redfish/v1/Chassis/" + chassisId + "/Thermal";
                 // Power object
                 asyncResp->res.jsonValue["Power"]["@odata.id"] =
                     "/redfish/v1/Chassis/" + chassisId + "/Power";
-#endif
-//Fixing RedFish Validator issue
-/* #ifdef BMCWEB_NEW_POWERSUBSYSTEM_THERMALSUBSYSTEM
+#endif */
+#ifdef BMCWEB_NEW_POWERSUBSYSTEM_THERMALSUBSYSTEM
                 asyncResp->res.jsonValue["ThermalSubsystem"]["@odata.id"] =
                     crow::utility::urlFromPieces("redfish", "v1", "Chassis",
                                                  chassisId, "ThermalSubsystem");
@@ -425,7 +426,7 @@ inline void
                     crow::utility::urlFromPieces("redfish", "v1", "Chassis",
                                                  chassisId,
                                                  "EnvironmentMetrics");
-#endif */
+#endif
                 // Network Adapters
                 asyncResp->res.jsonValue["NetworkAdapters"] = {
                     {"@odata.id", "/redfish/v1/Chassis/" +
-- 
2.34.1

