From b948c39676dabb5965b96d42c85d7997fd392cea Mon Sep 17 00:00:00 2001
From: Sandeep <sandeepap@ami.com>
Date: Sun, 15 Oct 2023 18:32:01 +0530
Subject: [PATCH] Adding 400 Bad request response for invalid MACAddress

Signed-off-by: Sandeep <sandeepap@ami.com>
---
 redfish-core/lib/ethernet.hpp | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/redfish-core/lib/ethernet.hpp b/redfish-core/lib/ethernet.hpp
index 111ac3c8..5ebc484e 100644
--- a/redfish-core/lib/ethernet.hpp
+++ b/redfish-core/lib/ethernet.hpp
@@ -1039,7 +1039,7 @@ inline void
         *crow::connections::systemBus, "xyz.openbmc_project.Network",
         "/xyz/openbmc_project/network/" + ifaceId,
         "xyz.openbmc_project.Network.MACAddress", "MACAddress", macAddress,
-        [asyncResp](const boost::system::error_code& ec,
+        [asyncResp,macAddress](const boost::system::error_code& ec,
                     const sdbusplus::message_t& msg) {
         if (ec)
         {
@@ -1054,7 +1054,8 @@ inline void
                 messages::propertyNotWritable(asyncResp->res, "MACAddress");
                 return;
             }
-            messages::internalError(asyncResp->res);
+	    messages::actionParameterNotSupported(asyncResp->res, macAddress,
+                                                  "MACAddress");
             return;
         }
     });
-- 
2.25.1

