From 2aa2d5b34741f581ff46f4e5bf670a3d540d4c4e Mon Sep 17 00:00:00 2001
From: gokulp <gokulp@ami.com>
Date: Tue, 18 Jul 2023 12:23:48 +0530
Subject: [PATCH 2/2] To get Status code as 200OK with NoOperation

Signed-off-by: ramadevip <ramadevip@ami.com>
---
 redfish-core/lib/chassis.hpp | 19 ++++++++++++++++++-
 1 file changed, 18 insertions(+), 1 deletion(-)

diff --git a/redfish-core/lib/chassis.hpp b/redfish-core/lib/chassis.hpp
index abb8e9b7..68d581e6 100644
--- a/redfish-core/lib/chassis.hpp
+++ b/redfish-core/lib/chassis.hpp
@@ -770,6 +770,23 @@ inline void
         });
 }
 
+/*
+ * Function to get the status code ad 200 Ok with message response as NoOperation
+ *
+ * @param[in] asyncResp - Shared pointer for completing asynchronous call
+ */
+inline void NoOperation(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp)
+{
+ asyncResp->res.result(boost::beast::http::status::ok);
+ asyncResp->res.jsonValue["@odata.type"] = " #Message.v1_1_1.Message";
+ asyncResp->res.jsonValue["MessageId"] = "Base.1.13.0.NoOperation";
+ asyncResp->res.jsonValue["Message"] = "The request body submitted contain no data to act upon "
+                                       "and no changes to the resource took place.";
+ asyncResp->res.jsonValue["MessageArgs"] = "[]";
+ asyncResp->res.jsonValue["MessageSeverity"] = "Warning";
+ asyncResp->res.jsonValue["Resolution"] = "Add properties in the JSON object and resubmit the request.";
+}
+
 const PropertyValue getOperatingSystemState(
                             const std::string& processName,
                             const std::string& objectPath,
@@ -1050,7 +1067,7 @@ inline void handleChassisResetActionInfoPost(
 
                             if (requestedOperatingSystemState != "xyz.openbmc_project.State.OperatingSystem.Status.OSStatus.Standby")
                             {
-                                    NoOperation(asyncResp);
+				    NoOperation(asyncResp);
                                     return;
                             }
 
-- 
2.34.1

