From 59186d2c62a0416e150ce9f6df9e82e282e2dff7 Mon Sep 17 00:00:00 2001
From: Abinaya L <abinayal@ami.com>
Date: Thu, 24 Aug 2023 18:15:33 +0530
Subject: [PATCH 95/95] Added-BIOS-Settings-support

---
 redfish-core/lib/bios.hpp | 11 ++++++-----
 1 file changed, 6 insertions(+), 5 deletions(-)

diff --git a/redfish-core/lib/bios.hpp b/redfish-core/lib/bios.hpp
index f301a91..974fd44 100644
--- a/redfish-core/lib/bios.hpp
+++ b/redfish-core/lib/bios.hpp
@@ -1524,14 +1524,15 @@ inline void
     asyncResp->res.jsonValue["Actions"]["#Bios.ResetBios"] = {
         {"target", "/redfish/v1/Systems/system/Bios/Actions/Bios.ResetBios"}};
 
-    asyncResp->res.jsonValue["Settings"]["@odata.id"] =
-            "/redfish/v1/Systems/system/Bios/Settings";
-
+    asyncResp->res.jsonValue["@Redfish.Settings"]["@odata.type"] = "#Settings.v1_2_2.Settings";
+    asyncResp->res.jsonValue["@Redfish.Settings"]["SettingsObject"] = {
+            {"@odata.id", "/redfish/v1/Systems/system/Bios/Settings"}};
+ 
     // Get the ActiveSoftwareImage and SoftwareImages
     sw_util::populateSoftwareInformation(asyncResp, sw_util::biosPurpose, "",
                                          true);
 
-    asyncResp->res.jsonValue["Attributes"] = nlohmann::json({});
+    asyncResp->res.jsonValue["Attributes"] = nlohmann::json({});
     // Get the BIOS Attributes
     bios::getBiosAttributes(asyncResp);
 
@@ -1596,7 +1597,7 @@ inline void
     asyncResp->res.jsonValue["Description"] = "BIOS Settings";
     asyncResp->res.jsonValue["Id"] = "BIOS_Settings";
 
-    asyncResp->res.jsonValue["Attributes"] = {};
+    asyncResp->res.jsonValue["Attributes"] = nlohmann::json({});
 
     // get the BIOS Attributes
     bios::getBiosSettingsAttr(asyncResp);
-- 
2.25.1

