From f1194bc2f37407640166772e8ac625eff212bca4 Mon Sep 17 00:00:00 2001
From: nibinc <nibinc@ami.com>
Date: Mon, 28 Aug 2023 13:03:08 +0530
Subject: [PATCH 97/97] time-offset-patch

---
 redfish-core/lib/managers.hpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/redfish-core/lib/managers.hpp b/redfish-core/lib/managers.hpp
index f8f5cd6d..340da8bf 100644
--- a/redfish-core/lib/managers.hpp
+++ b/redfish-core/lib/managers.hpp
@@ -2275,7 +2275,8 @@ inline void requestRoutesManager(App& app)
 		"ResetAll", "ResetToDefaultButKeepReservedSettings"};
 
         std::pair<std::string, std::string> redfishDateTimeOffset =
-            redfish::time_utils::getDateTimeOffsetNow();
+	    crow::utility::getDateTimeOffsetNow();
+           // redfish::time_utils::getDateTimeOffsetNow();
 
         asyncResp->res.jsonValue["DateTime"] = redfishDateTimeOffset.first;
         asyncResp->res.jsonValue["DateTimeLocalOffset"] =
-- 
2.25.1

