From cb295c33d7d3504deb0dfd5b24e8cbd68620068c Mon Sep 17 00:00:00 2001
From: prasannas <prasannas@ami.com>
Date: Wed, 30 Aug 2023 12:39:17 +0530
Subject: [PATCH 2422/2422] Adding Error Message for Invalid MTUSize

---
 redfish-core/lib/ethernet.hpp | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/redfish-core/lib/ethernet.hpp b/redfish-core/lib/ethernet.hpp
index 0d42372..a4ad123 100644
--- a/redfish-core/lib/ethernet.hpp
+++ b/redfish-core/lib/ethernet.hpp
@@ -34,6 +34,8 @@
 #include <regex>
 #include <string_view>
 #include <vector>
+#define MAX_MTU 1500
+#define MIN_MTU 68
 
 namespace redfish
 {
@@ -949,6 +951,13 @@ inline void
 {
     sdbusplus::message::object_path objPath = "/xyz/openbmc_project/network/" +
                                               ifaceId;
+    if ((mtuSize < MIN_MTU)||(mtuSize>MAX_MTU))
+    {
+            std::string mtu=std::to_string(mtuSize);
+            std::string_view mtuview(mtu);
+            messages::propertyValueOutOfRange(asyncResp->res,mtuview,"MTUSize");
+            return;
+    }
     crow::connections::systemBus->async_method_call(
         [asyncResp](const boost::system::error_code& ec) {
         if (ec)
-- 
2.9.5

