From 64bfe56817419e0fec924e36eafd4832077c9f9b Mon Sep 17 00:00:00 2001
From: prasannas <prasannas@ami.com>
Date: Fri, 8 Sep 2023 11:09:12 +0530
Subject: [PATCH] Fix For Pmt Sensor Not listed in Redfish

---
 redfish-core/lib/sensors.hpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/redfish-core/lib/sensors.hpp b/redfish-core/lib/sensors.hpp
index 4543f98..c511a3c 100644
--- a/redfish-core/lib/sensors.hpp
+++ b/redfish-core/lib/sensors.hpp
@@ -65,6 +65,7 @@ constexpr auto powerPaths = std::to_array<std::string_view>({
 });
 
 constexpr auto sensorPaths = std::to_array<std::string_view>({
+    "/xyz/openbmc_project/sensors/count",
     "/xyz/openbmc_project/sensors/power",
     "/xyz/openbmc_project/sensors/current",
     "/xyz/openbmc_project/sensors/airflow",
-- 
2.9.5

