From afae88204d765d0b2486f0784a275da0fbbc270a Mon Sep 17 00:00:00 2001
From: Sandeep <sandeepap@ami.com>
Date: Thu, 5 Oct 2023 15:28:39 +0530
Subject: [PATCH] Fixed IP Address showing as SessionType and vice-versa under
 Sessions URI

Signed-off-by: Sandeep <sandeepap@ami.com>
---
 include/sessions.hpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/sessions.hpp b/include/sessions.hpp
index 646dcf6b..2caa5417 100644
--- a/include/sessions.hpp
+++ b/include/sessions.hpp
@@ -264,7 +264,7 @@ class SessionStore
 
         auto session = std::make_shared<UserSession>(UserSession{
             uniqueId, sessionToken, std::string(username), csrfToken, clientId, 
-	    sessionType, redfish::ip_util::toString(clientIp),
+	    redfish::ip_util::toString(clientIp), sessionType,
             std::chrono::steady_clock::now(), persistence,
             isConfigureSelfOnly});
         auto it = authTokens.emplace(sessionToken, session);
-- 
2.25.1

