From 04e35b6fbe990fddc8ea1eed0d11048db0bbc389 Mon Sep 17 00:00:00 2001
From: Abinaya L <abinayal@ami.com>
Date: Tue, 10 Oct 2023 18:26:55 +0530
Subject: [PATCH 131/131] Added-BIOS-Put-Call-Support-in-Redfish

---
 redfish-core/lib/bios.hpp | 28 ++++++++++++++--------------
 1 file changed, 14 insertions(+), 14 deletions(-)

diff --git a/redfish-core/lib/bios.hpp b/redfish-core/lib/bios.hpp
index 4ff7a52..4f7aa7d 100644
--- a/redfish-core/lib/bios.hpp
+++ b/redfish-core/lib/bios.hpp
@@ -31,7 +31,7 @@ using BaseBIOSTable = boost::container::flat_map<
                std::variant<int64_t, std::string, bool>,
                std::variant<int64_t, std::string, bool>,
                std::vector<std::tuple<std::string,
-                                      std::variant<int64_t, std::string>>>>>;
+                                      std::variant<int64_t, std::string>, std::string>>>>;
 
 using BaseBIOSTableItem = std::pair<
     std::string,
@@ -39,7 +39,7 @@ using BaseBIOSTableItem = std::pair<
                std::variant<int64_t, std::string, bool>,
                std::variant<int64_t, std::string, bool>,
                std::vector<std::tuple<std::string,
-                                      std::variant<int64_t, std::string>>>>>;
+                                      std::variant<int64_t, std::string>, std::string>>>>;
 using PendingAttrType = boost::container::flat_map<
     std::string,
     std::tuple<std::string, std::variant<int64_t, std::string, bool>>>;
@@ -49,7 +49,7 @@ using PendingAttrItemType = std::pair<
     std::tuple<std::string, std::variant<int64_t, std::string, bool>>>;
 
 using AttrBoundType =
-    std::tuple<std::string, std::variant<int64_t, std::string>>;
+    std::tuple<std::string, std::variant<int64_t, std::string>, std::string>;
 
 enum BaseBiosTableIndex
 {
@@ -581,7 +581,7 @@ static void fillBiosTable(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
         std::string attrMenuPath;
         std::string attrType;
         bool attrReadOnly;
-        std::vector<std::tuple<std::string, std::variant<int64_t, std::string>>>
+        std::vector<std::tuple<std::string, std::variant<int64_t, std::string>, std::string>>
             attrValues;
 
         attr = attrJson["AttributeName"].get<std::string>();
@@ -601,9 +601,9 @@ static void fillBiosTable(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
             for (const auto& value :
                 attrJson["Values"].get<std::vector<std::string>>())
             {
-                attrValues.emplace_back(
+                attrValues.emplace_back(std::make_tuple(
                     "xyz.openbmc_project.BIOSConfig.Manager.BoundType.OneOf",
-                    value);
+                    value, ""));
             }
             attrType = getDbusBiosAttrType(attrType);
             baseBiosTable.insert(std::make_pair(
@@ -617,15 +617,15 @@ static void fillBiosTable(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
             int64_t defaultVal = attrJson["DefaultValue"].get<int64_t>();
 
             // read and update the bound values
-            attrValues.emplace_back(
+            attrValues.emplace_back(std::make_tuple(
                 "xyz.openbmc_project.BIOSConfig.Manager.BoundType.LowerBound",
-               attrJson["LowerBound"].get<int64_t>());
-            attrValues.emplace_back(
+               attrJson["LowerBound"].get<int64_t>(), ""));
+            attrValues.emplace_back(std::make_tuple(
                 "xyz.openbmc_project.BIOSConfig.Manager.BoundType.UpperBound",
-                attrJson["UpperBound"].get<int64_t>());
-           attrValues.emplace_back(
+                attrJson["UpperBound"].get<int64_t>(), ""));
+           attrValues.emplace_back(std::make_tuple(
                 "xyz.openbmc_project.BIOSConfig.Manager.BoundType.ScalarIncrement",
-               attrJson["ScalarIncrement"].get<int64_t>());
+               attrJson["ScalarIncrement"].get<int64_t>(), ""));
 
             attrType = getDbusBiosAttrType(attrType);
             baseBiosTable.insert(std::make_pair(
@@ -646,9 +646,9 @@ static void fillBiosTable(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
             for (const auto& value :
                  attrJson["Values"].get<std::vector<bool>>())
             {
-                attrValues.emplace_back(
+                attrValues.emplace_back(std::make_tuple(
                     "xyz.openbmc_project.BIOSConfig.Manager.BoundType.OneOf",
-                    (value == true ? 1 : 0));
+                    (value == true ? 1 : 0), ""));
             }
             attrType = getDbusBiosAttrType(attrType);
             baseBiosTable.insert(std::make_pair(
-- 
2.25.1

