From a9773402277a836f1781fddd6619fe6876151994 Mon Sep 17 00:00:00 2001
From: prasannas <prasannas@ami.com>
Date: Wed, 22 Nov 2023 11:11:23 +0530
Subject: [PATCH 145/145] Fix for Duplicate Etag value in Redfish

---
 http/http_response.hpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/http/http_response.hpp b/http/http_response.hpp
index d7b589d..1343d5a 100644
--- a/http/http_response.hpp
+++ b/http/http_response.hpp
@@ -285,6 +285,7 @@ struct Response
         }
         size_t hashval = std::hash<nlohmann::json>{}(jsonValue);
         std::string hexVal = "\"" + intToHexString(hashval, 8) + "\"";
+        clearHeader(boost::beast::http::field::etag);
         addHeader(boost::beast::http::field::etag, hexVal);
         if (expectedHash && hexVal == *expectedHash)
         {
-- 
2.9.5

