From e51bcf5858e2685bca849e6aad562f48efe5a225 Mon Sep 17 00:00:00 2001
From: prasannas <prasannas@ami.com>
Date: Wed, 24 Jan 2024 16:34:41 +0530
Subject: [PATCH 145/145] Fix for Duplicate Etag value in Redfish

---
 http/http_response.hpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/http/http_response.hpp b/http/http_response.hpp
index 43d3bb59..55bd5d48 100644
--- a/http/http_response.hpp
+++ b/http/http_response.hpp
@@ -343,6 +343,7 @@ struct Response
         }
         size_t hashval = std::hash<nlohmann::json>{}(jsonValue);
         std::string hexVal = "\"" + intToHexString(hashval, 8) + "\"";
+        clearHeader(http::field::etag);
         addHeader(http::field::etag, hexVal);
         if (expectedHash && hexVal == *expectedHash)
         {
-- 
2.25.1

