From e35e7bdb9f91c5d0a3a1a2a43070695ce408785f Mon Sep 17 00:00:00 2001
From: vamsikrishnak <vamsikrishnak@ami.com>
Date: Wed, 22 Nov 2023 18:22:54 +0530
Subject: [PATCH] NTP severs count fix under Network protocol

Signed-off-by: vamsikrishnak <vamsikrishnak@ami.com>
---
 redfish-core/lib/network_protocol.hpp | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/redfish-core/lib/network_protocol.hpp b/redfish-core/lib/network_protocol.hpp
index de51166..976bb56 100644
--- a/redfish-core/lib/network_protocol.hpp
+++ b/redfish-core/lib/network_protocol.hpp
@@ -286,6 +286,16 @@ inline void
 {
     std::vector<std::string>::iterator currentNtpServer =
         currentNtpServers.begin();
+
+    size_t limit = 3;
+
+    if(ntpServerObjects.size() > limit)
+    {
+       BMCWEB_LOG_DEBUG("out of Limit");
+       asyncResp->res.result(boost::beast::http::status::bad_request);
+       return;
+    }
+
     for (size_t index = 0; index < ntpServerObjects.size(); index++)
     {
         const nlohmann::json& ntpServer = ntpServerObjects[index];
-- 
2.9.5

