diff --git a/redfish-core/lib/event_service.hpp b/redfish-core/lib/event_service.hpp
index e4793866..69caaf32 100644
--- a/redfish-core/lib/event_service.hpp
+++ b/redfish-core/lib/event_service.hpp
@@ -439,7 +439,7 @@ inline void requestRoutesEventDestinationCollection(App& app)
 
         if (protocol == "SNMPv2c")
         {
-            if (context)
+            /*if (context)
             {
                 messages::propertyValueConflict(asyncResp->res, "Context",
                                                 "Protocol");
@@ -486,7 +486,7 @@ inline void requestRoutesEventDestinationCollection(App& app)
                 messages::propertyValueConflict(
                     asyncResp->res, "MetricReportDefinitions", "Protocol");
                 return;
-            }
+            }*/
 
             addSnmpTrapClient(asyncResp, host, port);
             return;
