From d61f619cc8b0d55ef93170c6b2b81c38f9dfa42e Mon Sep 17 00:00:00 2001
From: vamsikrishnak <vamsikrishnak@ami.com>
Date: Thu, 7 Dec 2023 07:17:29 -0500
Subject: [PATCH] Removing NMI Actions from Systems URI

Signed-off-by: vamsikrishnak <vamsikrishnak@ami.com>
---
 redfish-core/lib/systems.hpp | 30 ------------------------------
 1 file changed, 30 deletions(-)

diff --git a/redfish-core/lib/systems.hpp b/redfish-core/lib/systems.hpp
index a6ffc2d2..1d2a9826 100644
--- a/redfish-core/lib/systems.hpp
+++ b/redfish-core/lib/systems.hpp
@@ -3554,30 +3554,6 @@ inline void
         dbus::utility::DbusVariantType(hostTransitionTimeOut));
 }
 
-/**
- * Function transceives data with dbus directly.
- */
-inline void doNMI(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp)
-{
-    constexpr const char* serviceName = "xyz.openbmc_project.Control.Host.NMI";
-    constexpr const char* objectPath = "/xyz/openbmc_project/control/host0/nmi";
-    constexpr const char* interfaceName =
-        "xyz.openbmc_project.Control.Host.NMI";
-    constexpr const char* method = "NMI";
-
-    crow::connections::systemBus->async_method_call(
-        [asyncResp](const boost::system::error_code& ec) {
-        if (ec)
-        {
-            BMCWEB_LOG_ERROR(" Bad D-Bus request error: {}", ec);
-            messages::internalError(asyncResp->res);
-            return;
-        }
-        messages::success(asyncResp->res);
-    },
-        serviceName, objectPath, interfaceName, method);
-}
-
 /**
  * Handle error responses from d-bus for system power requests
  */
@@ -3723,11 +3699,6 @@ inline void
             hostCommand = true;
         }
     }
-    else if (resetType == "Nmi")
-    {
-        doNMI(asyncResp);
-        return;
-    }
     else
     {
         messages::actionParameterUnknown(asyncResp->res, "Reset", resetType);
@@ -4471,7 +4442,6 @@ inline void handleSystemCollectionResetActionGet(
     allowableValues.emplace_back("GracefulRestart");
     allowableValues.emplace_back("GracefulShutdown");
     allowableValues.emplace_back("PowerCycle");
-    allowableValues.emplace_back("Nmi");
     parameter["AllowableValues"] = std::move(allowableValues);
     parameters.emplace_back(std::move(parameter));
 
-- 
2.34.1

