From 1af5b2337faa8fd6ad26f12e6b0e12055ddc1a5c Mon Sep 17 00:00:00 2001
From: Ginu George <ginugeorge@ami.com>
Date: Tue, 2 Jan 2024 09:47:05 -0500
Subject: [PATCH 164/164] Support HostName FQDN Validation

---
 redfish-core/lib/ethernet.hpp | 52 ++++++++++++++++++++++++++++++-----
 1 file changed, 45 insertions(+), 7 deletions(-)

diff --git a/redfish-core/lib/ethernet.hpp b/redfish-core/lib/ethernet.hpp
index c5a5e740..86b68b4c 100644
--- a/redfish-core/lib/ethernet.hpp
+++ b/redfish-core/lib/ethernet.hpp
@@ -1040,6 +1040,28 @@ inline void
     });
 }
 
+inline bool validateFqdnHostName(const std::string& hostname, const std::string& fqdn,
+                            const std::shared_ptr<bmcweb::AsyncResp>& asyncResp)
+{
+    size_t pos = fqdn.find('.');
+    if (pos == std::string::npos)
+    {
+        messages::propertyValueFormatError(asyncResp->res, fqdn, "FQDN");
+        return false;
+    }
+
+    std::string fqdnhostname;
+    fqdnhostname = (fqdn).substr(0, pos);
+
+    if(fqdnhostname != hostname)
+    {
+        messages::propertyValueConflict(asyncResp->res, "FQDN","HostName");
+        return false;
+    }
+
+    return true;
+}
+
 inline void handleFqdnPatch(const std::string& ifaceId, const std::string& fqdn,
                             const std::shared_ptr<bmcweb::AsyncResp>& asyncResp)
 {
@@ -2484,9 +2506,30 @@ inline void handleEthernetInterfaceInstancePatch(
                     }
         }
 
-        if (hostname)
+        bool FqdnHostnameValidate = true;
+        if (hostname && fqdn)
         {
-            handleHostnamePatch(*hostname, asyncResp);
+            FqdnHostnameValidate = validateFqdnHostName(*hostname, *fqdn, asyncResp);
+            if(!FqdnHostnameValidate)
+            {
+                return;
+            }
+            else
+            {
+                handleFqdnPatch(ifaceId, *fqdn, asyncResp);
+            }
+        }
+        else
+        {
+            if (hostname)
+            {
+                handleHostnamePatch(*hostname, asyncResp);
+            }
+
+            if (fqdn)
+            {
+                handleFqdnPatch(ifaceId, *fqdn, asyncResp);
+            }
         }
 
         if (ipv6AutoConfigEnabled)
@@ -2495,11 +2538,6 @@ inline void handleEthernetInterfaceInstancePatch(
                                         asyncResp);
         }
 
-        if (fqdn)
-        {
-            handleFqdnPatch(ifaceId, *fqdn, asyncResp);
-        }
-
         if (macAddress)
         {
             handleMACAddressPatch(ifaceId, *macAddress, asyncResp);
-- 
2.34.1

