From 314b5d7922772f2161ad883cf3d61827f117f668 Mon Sep 17 00:00:00 2001
From: Nibin Chacko <nibinc@ami.com>
Date: Wed, 20 Dec 2023 09:07:02 -0500
Subject: [PATCH 167/167] task starts event redfish listner

---
 redfish-core/lib/task.hpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/redfish-core/lib/task.hpp b/redfish-core/lib/task.hpp
index 29344613..4714ad83 100644
--- a/redfish-core/lib/task.hpp
+++ b/redfish-core/lib/task.hpp
@@ -101,7 +101,7 @@ struct TaskData : std::enable_shared_from_this<TaskData>
         matchStr(matchIn), index(idx),
         startTime(std::chrono::system_clock::to_time_t(
             std::chrono::system_clock::now())),
-        status("OK"), state("New"), messages(nlohmann::json::array()),
+        status("OK"), state("Running"), messages(nlohmann::json::array()),
         timer(crow::connections::systemBus->get_io_context())
 
     {}
-- 
2.34.1

