From adcf233e27c8d8e15641bb0a38d97277ee99c026 Mon Sep 17 00:00:00 2001
From: rkarthika <rkarthika@ami.com>
Date: Tue, 26 Dec 2023 12:44:16 +0530
Subject: [PATCH 172/172] Provide delay for set SSH properties

---
 redfish-core/lib/network_protocol.hpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/redfish-core/lib/network_protocol.hpp b/redfish-core/lib/network_protocol.hpp
index f8dbdc0a..88155c70 100644
--- a/redfish-core/lib/network_protocol.hpp
+++ b/redfish-core/lib/network_protocol.hpp
@@ -431,6 +431,11 @@ inline void
         {
             if (boost::algorithm::starts_with(entry.first, netBasePath))
             {
+                if(protocolEnabled)
+                {
+                    BMCWEB_LOG_DEBUG("wait for get properties");
+                    sleep(5);
+                }
                 sdbusplus::asio::setProperty(
                     *crow::connections::systemBus, entry.second.begin()->first,
                     entry.first,
-- 
2.34.1

