From 3c07b09e5c0963acf1ab1d2e71380e7bf8f52307 Mon Sep 17 00:00:00 2001
From: prasannas <prasannas@ami.com>
Date: Fri, 29 Dec 2023 09:37:08 +0530
Subject: [PATCH 173/173] Fixed 500 InternalError in trigger_PostCall

---
 redfish-core/lib/trigger.hpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/redfish-core/lib/trigger.hpp b/redfish-core/lib/trigger.hpp
index f03fbf3..38e8c46 100644
--- a/redfish-core/lib/trigger.hpp
+++ b/redfish-core/lib/trigger.hpp
@@ -677,7 +677,7 @@ inline void afterCreateTrigger(
     }
     if (ec)
     {
-        messages::internalError(asyncResp->res);
+        messages::propertyValueFormatError(asyncResp->res, id, "Id");
         BMCWEB_LOG_ERROR("respHandler DBus error {}", ec);
         return;
     }
-- 
2.9.5

