From 62dc2c16c7276b8ea1d766d3a4572c3e531b33b1 Mon Sep 17 00:00:00 2001
From: muthulakshmin <muthulakshmin@ami.com>
Date: Wed, 27 Dec 2023 18:13:53 +0530
Subject: [PATCH 2620/2620] Fixed invalid subnetmask IP

---
 redfish-core/lib/ethernet.hpp | 19 ++++++++++++++++++-
 1 file changed, 18 insertions(+), 1 deletion(-)

diff --git a/redfish-core/lib/ethernet.hpp b/redfish-core/lib/ethernet.hpp
index 866218f0..7556286f 100644
--- a/redfish-core/lib/ethernet.hpp
+++ b/redfish-core/lib/ethernet.hpp
@@ -55,7 +55,8 @@ enum class Type
     GATEWAY4_ADDRESS,
     GATEWAY6_ADDRESS,
     IP4_ADDRESS,
-    IP6_ADDRESS
+    IP6_ADDRESS,
+    SUBNETMASK
 };
 
 /**
@@ -1406,6 +1407,14 @@ inline bool isValidIPv4Addr(const std::string& addr, Type type)
             return false;
         }
     }
+    else if (type == Type::SUBNETMASK)
+    {
+        if (ip[0] == 0 && ip[1] == 0 && ip[2] == 0 &&
+            ip[3] == 0) // checks if all four octets are zero
+        {
+            return false;
+        }
+    }
     return true;
 }
 
@@ -1442,6 +1451,7 @@ inline bool
             {
                 const std::string& ipAddress = *address;
                 const std::string& ipGateway = *gateway;
+                const std::string& ipSubnetMask = *subnetMask;
 
                 std::bitset<32> maskBinary(0);
                 if (subnetMask.has_value())
@@ -1494,6 +1504,13 @@ inline bool
                     messages::invalidip(asyncResp->res, "Gateway", ipGateway);
                     return false;
                 }
+                if (!isValidIPv4Addr(
+                        ipSubnetMask, Type::SUBNETMASK)) // checking the IPv4
+                                                            // subnetmask Address
+                {
+                    messages::invalidip(asyncResp->res, "subnetmask", ipSubnetMask);
+                    return false;
+                }
             }
         }
         entryIdx++;
-- 
2.34.1

