From c7cfb0e671be625edd86f69bebc22126c8abb1ed Mon Sep 17 00:00:00 2001
From: muthulakshmin <muthulakshmin@ami.com>
Date: Wed, 27 Dec 2023 20:48:44 +0530
Subject: [PATCH 2620/2620] Fixed RestoreOptions in EVB

---
 redfish-core/lib/managers.hpp | 13 +++----------
 1 file changed, 3 insertions(+), 10 deletions(-)

diff --git a/redfish-core/lib/managers.hpp b/redfish-core/lib/managers.hpp
index 049a9838..f3a62dcd 100644
--- a/redfish-core/lib/managers.hpp
+++ b/redfish-core/lib/managers.hpp
@@ -275,8 +275,7 @@ inline void setTimer(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
 }
 
 inline void
-    writeRestoreOptions(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
-                        const std::string& resetType)
+    writeRestoreOptions(const std::string& resetType)
 {
     constexpr const char* restoreOpFname = "/tmp/.rwfs/.restore_op";
     int option = 0;
@@ -292,14 +291,8 @@ inline void
     }
 
     std::ofstream restoreFile(restoreOpFname, std::ios::trunc);
-    if (!restoreFile)
-    {
-        // BMCWEB_LOG_ERROR << "error in opring output stream " <<
-        // restoreOpFname;
-        messages::internalError(asyncResp->res);
-        return;
-    }
     restoreFile << option << "\n";
+    return;
 }
 
 /**
@@ -512,7 +505,7 @@ inline void requestRoutesManagerResetToDefaultsAction(App& app)
                 messages::internalError(asyncResp->res);
                 return;
             }
-            writeRestoreOptions(asyncResp, resetType);
+            writeRestoreOptions(resetType);
             // Factory Reset doesn't actually happen until a reboot
             // Can't erase what the BMC is running on
             doBMCGracefulRestart(asyncResp);
-- 
2.34.1

