From 6b2c657e111d8297d8ef3f9a54d7722d3ae0258f Mon Sep 17 00:00:00 2001
From: prasannas <prasannas@ami.com>
Date: Tue, 20 Feb 2024 18:31:14 +0530
Subject: [PATCH 181/181] Fix for Duplicate Error Message in BIOS_URI

---
 redfish-core/lib/bios.hpp | 2 --
 1 file changed, 2 deletions(-)

diff --git a/redfish-core/lib/bios.hpp b/redfish-core/lib/bios.hpp
index e00b8368..bfbfab7c 100644
--- a/redfish-core/lib/bios.hpp
+++ b/redfish-core/lib/bios.hpp
@@ -1419,7 +1419,6 @@ inline void
                                   resetBiosToDefaultsPending))
     {
         BMCWEB_LOG_ERROR("No 'ResetBiosToDefaultsPending' found");
-        messages::unrecognizedRequestBody(asyncResp->res);
         return;
     }
 
@@ -1488,7 +1487,6 @@ inline void
                                            pendingAttrJson))
     {
         BMCWEB_LOG_ERROR("No 'Attributes' found");
-        messages::unrecognizedRequestBody(asyncResp->res);
         return;
     }
 
-- 
2.43.0

