From 5f5d8cdd98e250b3568391a416456a8b659d2707 Mon Sep 17 00:00:00 2001
From: Karthika R <rkarthika@ami.com>
Date: Mon, 29 Jan 2024 08:09:20 -0500
Subject: [PATCH 187/187] Receive SubmitTestEvent in SSE

---
 redfish-core/lib/event_service.hpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/redfish-core/lib/event_service.hpp b/redfish-core/lib/event_service.hpp
index 10cb78e8..88c89f17 100644
--- a/redfish-core/lib/event_service.hpp
+++ b/redfish-core/lib/event_service.hpp
@@ -266,6 +266,11 @@ inline void requestRoutesSubmitTestEvent(App& app)
         {
             return;
         }
+        if (!EventServiceManager::getInstance().sendTestEventLog())
+        {
+            messages::internalError(asyncResp->res);
+            return;
+        }
         // EventServiceManager::getInstance().readEventLogsFromFile();
         asyncResp->res.result(boost::beast::http::status::no_content);
     });
-- 
2.34.1

