From e2db1e82839e7e614cb4ffb38ff94e8628abf458 Mon Sep 17 00:00:00 2001
From: rkarthika <rkarthika@ami.com>
Date: Thu, 22 Feb 2024 12:39:39 +0530
Subject: [PATCH 187/187] Receive SubmitTestEvent in SSE

---
 redfish-core/lib/event_service.hpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/redfish-core/lib/event_service.hpp b/redfish-core/lib/event_service.hpp
index 88c89f17..6d401759 100644
--- a/redfish-core/lib/event_service.hpp
+++ b/redfish-core/lib/event_service.hpp
@@ -271,6 +271,11 @@ inline void requestRoutesSubmitTestEvent(App& app)
             messages::internalError(asyncResp->res);
             return;
         }
+        if (!EventServiceManager::getInstance().sendTestEventLog())
+        {
+            messages::internalError(asyncResp->res);
+            return;
+        }
         // EventServiceManager::getInstance().readEventLogsFromFile();
         asyncResp->res.result(boost::beast::http::status::no_content);
     });
-- 
2.34.1

