From 920f0fa47d41b76a29b25b5f8337f22d81dc5d1a Mon Sep 17 00:00:00 2001
From: Nibin Chacko <nibinc@ami.com>
Date: Mon, 29 Jan 2024 09:13:34 -0500
Subject: [PATCH 188/188] Arranging the error message

---
 .../registries/base_message_registry.hpp      | 23 +++++++++----------
 1 file changed, 11 insertions(+), 12 deletions(-)

diff --git a/redfish-core/include/registries/base_message_registry.hpp b/redfish-core/include/registries/base_message_registry.hpp
index feff8bbb..9aaf1f45 100644
--- a/redfish-core/include/registries/base_message_registry.hpp
+++ b/redfish-core/include/registries/base_message_registry.hpp
@@ -1321,6 +1321,16 @@ constexpr std::array registry =
 		},
 		"Provide Valid IP address",
 	}},
+    MessageEntry{
+	    "PasswordResetFailed",
+        {
+		"Indicates that we cannot use or reset the previous password for the particular user",
+		"Last password cannot be used to reset the redfish account password. Please change the password value and resubmit the request",
+		"Critical",
+		0,
+		{},
+		"Change the password value and resubmit the request",
+	}},
     MessageEntry{
 	    "differentIpSeries",
 	    {
@@ -1334,17 +1344,6 @@ constexpr std::array registry =
 		},
 		"Provide IP in the same series",
 	}},
-    MessageEntry{
-	    "PasswordResetFailed",
-        {
-		"Indicates that we cannot use or reset the previous password for the particular user",
-		"Last password cannot be used to reset the redfish account password. Please change the password value and resubmit the request",
-		"Critical",
-		0,
-		{},
-		"Change the password value and resubmit the request",
-	}},
-
 };
 
 enum class Index
@@ -1460,7 +1459,7 @@ enum class Index
     undeterminedFault = 108,
     unrecognizedRequestBody = 109,
     invalidip=110,
+    passwordResetFailed=111,
     differentIpSeries=112,
-    passwordResetFailed=113,
 };
 } // namespace redfish::registries::base
-- 
2.34.1

