From 510b3d4f01fddd36d7242560a58b7638be4c9dc9 Mon Sep 17 00:00:00 2001
From: Karthika R <rkarthika@ami.com>
Date: Sat, 3 Feb 2024 17:51:27 +0530
Subject: [PATCH 195/195] Verify Set DateTime With Invalid Data

---
 redfish-core/lib/managers.hpp | 21 +++++++++++++++++++++
 1 file changed, 21 insertions(+)

diff --git a/redfish-core/lib/managers.hpp b/redfish-core/lib/managers.hpp
index 0a727953..d5098b09 100644
--- a/redfish-core/lib/managers.hpp
+++ b/redfish-core/lib/managers.hpp
@@ -2156,6 +2156,20 @@ inline void setTimeZone(std::shared_ptr<bmcweb::AsyncResp> aResp,
     }
 }
 
+bool isValidDateTimeFormat(const std::string& datetime)
+{
+    // Define the regex pattern for the expected DateTime format
+    std::regex dateTimePattern(R"(\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}\+00:00)");
+
+    // Check if the datetime string matches the expected format
+    if (!std::regex_match(datetime, dateTimePattern))
+    {
+        return false;
+    }
+
+    return true;
+}
+
 inline void setDateTime(std::shared_ptr<bmcweb::AsyncResp> asyncResp,
                         std::string datetime)
 {
@@ -2586,6 +2600,13 @@ inline void requestRoutesManager(App& app)
         }
         if (datetime)
         {
+            if (!isValidDateTimeFormat(*datetime)) // datetime Format should be
+                                                   // yyyy-mm-ddThh:mm:ss+00:00
+            {
+                messages::propertyValueFormatError(asyncResp->res, *datetime,
+                                                   "DateTime");
+                return;
+            }
             setDateTime(asyncResp, std::move(*datetime));
         }
         if (locationIndicatorActive)
-- 
2.34.1

