From 67b0562d8ee1e68147256c65f39a3aee95957dd4 Mon Sep 17 00:00:00 2001
From: MuthuLakshmiN <muthulakshmin@ami.com>
Date: Mon, 15 Apr 2024 15:06:16 +0530
Subject: [PATCH 203/203] Fixed USB0 interface dhcp and MACAddress configs

Signed-off-by: MuthuLakshmiN <muthulakshmin@ami.com>
---
 redfish-core/lib/ethernet.hpp | 17 ++++++++++++++++-
 1 file changed, 16 insertions(+), 1 deletion(-)

diff --git a/redfish-core/lib/ethernet.hpp b/redfish-core/lib/ethernet.hpp
index e5a48e32..e2028df3 100644
--- a/redfish-core/lib/ethernet.hpp
+++ b/redfish-core/lib/ethernet.hpp
@@ -1090,7 +1090,14 @@ inline void
 {
     static constexpr std::string_view dbusNotAllowedError =
         "xyz.openbmc_project.Common.Error.NotAllowed";
-
+    if (strcmp(ifaceId.c_str(), "usb0") == 0)
+    {
+        BMCWEB_LOG_INFO(
+            "since usb0 interface is static by default, mac address modification is not allowed");
+        messages::actionNotSupported(asyncResp->res,
+                                     "MACAddress change in USB0 static");
+        return;
+    }
     sdbusplus::asio::setProperty(
         *crow::connections::systemBus, "xyz.openbmc_project.Network",
         "/xyz/openbmc_project/network/" + ifaceId,
@@ -2421,6 +2428,14 @@ inline void handleEthernetInterfaceInstancePatch(
 
         if (dhcpv4 || dhcpv6)
         {
+            if (strcmp(ifaceId.c_str(), "usb0") == 0)
+            {
+                BMCWEB_LOG_INFO(
+                    "since usb0 interface is static by default, DHCPEnable modification is not allowed");
+                messages::actionNotSupported(asyncResp->res,
+                                             "DHCPEnable in USB0 static");
+                return;
+            }
             if(dhcpv4)
             {
                 if(v4dhcpParms.dhcpv4Enabled)
-- 
2.43.0

