From ab09b0d8817bdd48b2679739239cf6523c158422 Mon Sep 17 00:00:00 2001
From: rkarthika <rkarthika@ami.com>
Date: Thu, 18 Apr 2024 17:53:18 +0530
Subject: [PATCH 205/205] List missed sensors in redfish call

---
 redfish-core/lib/sensors.hpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/redfish-core/lib/sensors.hpp b/redfish-core/lib/sensors.hpp
index 3d18be9e..a5bee645 100644
--- a/redfish-core/lib/sensors.hpp
+++ b/redfish-core/lib/sensors.hpp
@@ -83,6 +83,10 @@ constexpr auto sensorPaths = std::to_array<std::string_view>({
     "/xyz/openbmc_project/sensors/fan_pwm",
     "/xyz/openbmc_project/sensors/cpu",
     "/xyz/openbmc_project/sensors/watchdog",
+    "/xyz/openbmc_project/sensors/temperature",
+    "/xyz/openbmc_project/sensors/chassisstate",
+    "/xyz/openbmc_project/sensors/battery",
+    "/xyz/openbmc_project/sensors/acpidevice",    
 #ifdef BMCWEB_NEW_POWERSUBSYSTEM_THERMALSUBSYSTEM
     "/xyz/openbmc_project/sensors/voltage",
     "/xyz/openbmc_project/sensors/fan_tach",
-- 
2.25.1

