From 5a32c29c8ce69f4687af823cc8d47ebeb850886f Mon Sep 17 00:00:00 2001
From: rkarthika <rkarthika@ami.com>
Date: Fri, 16 Feb 2024 20:46:58 +0530
Subject: [PATCH 207/207] Set Sensor Reading fractional value as 4 digits

---
 redfish-core/lib/sensors.hpp | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/redfish-core/lib/sensors.hpp b/redfish-core/lib/sensors.hpp
index a87849ef..392d1e20 100644
--- a/redfish-core/lib/sensors.hpp
+++ b/redfish-core/lib/sensors.hpp
@@ -1100,7 +1100,17 @@ inline void objectPropertiesToJson(
             }
             else
             {
-                sensorJson[key] = *doubleValue;
+                if (key == nlohmann::json::json_pointer("/Reading")) {
+                        double roundedValue = std::round(*doubleValue * 10000.0) / 10000.0;
+
+                        std::stringstream ss;
+                        ss << std::fixed << std::setprecision(4) << roundedValue;
+                        std::string roundedStringValue = ss.str();
+                        sensorJson[key] = roundedStringValue;
+                }
+                else {
+                        sensorJson[key] = *doubleValue;
+                }
             }
         }
     }
-- 
2.34.1

