From 676e1ef2ca38b7dab964b24709aa208cc47333b6 Mon Sep 17 00:00:00 2001
From: KarthikaR <rkarthika@ami.com>
Date: Thu, 21 Mar 2024 18:42:37 +0530
Subject: [PATCH 219/219] Throw error when IPv4StaticAddresses Address equals
 Gateway

---
 redfish-core/lib/ethernet.hpp | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/redfish-core/lib/ethernet.hpp b/redfish-core/lib/ethernet.hpp
index 2d09270f..02d003f3 100644
--- a/redfish-core/lib/ethernet.hpp
+++ b/redfish-core/lib/ethernet.hpp
@@ -130,6 +130,8 @@ struct DHCPParameters
     std::optional<std::string> dhcpv6OperatingMode;
 };
 
+std::optional<std::string> defaultGatewayValue;
+
 // Helper function that changes bits netmask notation (i.e. /24)
 // into full dot notation
 inline std::string getNetmask(unsigned int bits)
@@ -1536,6 +1538,13 @@ inline void
             // current request.
             if (address)
             {
+                if (*address == *defaultGatewayValue)
+                {
+                    messages::propertyValueConflict(asyncResp->res, "Address",
+                                                    "DefaultGateway");
+                    return;
+                }
+
                 if (!ip_util::ipv4VerifyIpAndGetBitcount(*address))
                 {
                     messages::propertyValueFormatError(asyncResp->res, *address,
@@ -1598,6 +1607,7 @@ inline void
                                       "Address");
                     return;
                 }
+                defaultGatewayValue = gateway;
             }
             else if (nicIpEntry != ipv4Data.cend())
             {
-- 
2.25.1

