From 171622c10d78b396dc279de84f023aeb59706360 Mon Sep 17 00:00:00 2001
From: rkarthika <rkarthika@ami.com>
Date: Mon, 18 Mar 2024 13:01:13 +0530
Subject: [PATCH 220/220] Restrict special characters in NTPServers

---
 redfish-core/lib/network_protocol.hpp | 24 ++++++++++++++++++++++++
 1 file changed, 24 insertions(+)

diff --git a/redfish-core/lib/network_protocol.hpp b/redfish-core/lib/network_protocol.hpp
index 4d363c8c..cd51900c 100644
--- a/redfish-core/lib/network_protocol.hpp
+++ b/redfish-core/lib/network_protocol.hpp
@@ -316,6 +316,30 @@ inline void
         return;
     }
 
+    auto isValidNtpServer = [](const std::string& server) -> bool {
+        for (char c : server)
+        {
+            if (!isdigit(c) && c != '-' && c != '.')
+            {
+                return false; // Found an invalid character
+            }
+        }
+        return true; // All characters are valid
+    };
+
+    for (const auto& ntpServerObject : ntpServerObjects)
+    {
+        std::string ntpServerAddress = ntpServerObject.get<std::string>();
+
+        if (!isValidNtpServer(ntpServerAddress))
+        {
+            BMCWEB_LOG_DEBUG("Invalid character found in NTP server address.");
+            messages::propertyValueFormatError(asyncResp->res, ntpServerObjects,
+                                           "NTPServers");
+            return;
+        }
+    }
+
     for (size_t index = 0; index < ntpServerObjects.size(); index++)
     {
         const nlohmann::json& ntpServer = ntpServerObjects[index];
-- 
2.34.1

