From beb397df01fd2944cea82ec5d95b3cc0999e8623 Mon Sep 17 00:00:00 2001
From: vamsikrishnak <vamsikrishnak@ami.com>
Date: Tue, 19 Mar 2024 17:45:01 +0530
Subject: [PATCH] Fixed pipeline errors

Signed-off-by: vamsikrishnak <vamsikrishnak@ami.com>
---
 .../include/event_service_manager.hpp         | 20 +++++++++----------
 1 file changed, 10 insertions(+), 10 deletions(-)

diff --git a/redfish-core/include/event_service_manager.hpp b/redfish-core/include/event_service_manager.hpp
index 41edbd82..99923dd2 100644
--- a/redfish-core/include/event_service_manager.hpp
+++ b/redfish-core/include/event_service_manager.hpp
@@ -65,7 +65,7 @@ std::function<void(const std::string&)> retryExhaustCallback =
 static constexpr const uint8_t maxNoOfSubscriptions = 20;
 static constexpr const uint8_t maxNoOfSSESubscriptions = 10;
 
-#ifdef BMCWEB_ENABLE_REDFISH_DBUS_LOG_ENTRIES
+//#ifdef BMCWEB_ENABLE_REDFISH_DBUS_LOG_ENTRIES
 // NOLINTNEXTLINE(cppcoreguidelines-avoid-non-const-global-variables)
 static std::optional<boost::asio::posix::stream_descriptor> inotifyConn;
 static constexpr const char* redfishEventLogDir = "/var/log";
@@ -263,7 +263,7 @@ inline int formatEventLogEntry(const std::string& logEntryID,
 }
 
 } // namespace event_log
-#endif
+//#endif
 
 inline bool isFilterQuerySpecialChar(char c)
 {
@@ -581,7 +581,7 @@ class Subscription : public persistent_data::UserSubscription
         return true;
     }
 
-#ifdef BMCWEB_ENABLE_REDFISH_DBUS_LOG_ENTRIES
+//#ifdef BMCWEB_ENABLE_REDFISH_DBUS_LOG_ENTRIES
     void filterAndSendEventLogs(
         const std::vector<EventLogObjectsType>& eventRecords)
     {
@@ -647,7 +647,7 @@ class Subscription : public persistent_data::UserSubscription
         sendEvent(std::move(strMsg));
         eventSeqNum++;
     }
-#endif
+//#endif
 
     void filterAndSendReports(const std::string& reportId,
                               const telemetry::TimestampReadings& var)
@@ -874,11 +874,11 @@ class EventServiceManager
 
             updateNoOfSubscribersCount();
 
-#ifdef BMCWEB_ENABLE_REDFISH_DBUS_LOG_ENTRIES
+//#ifdef BMCWEB_ENABLE_REDFISH_DBUS_LOG_ENTRIES
 
             cacheRedfishLogFile();
 
-#endif
+//#endif
             // Update retry configuration.
             subValue->updateRetryConfig(retryAttempts, retryTimeoutInterval);
         }
@@ -1130,12 +1130,12 @@ class EventServiceManager
             persistSubscriptionData();
         }
 
-#ifdef BMCWEB_ENABLE_REDFISH_DBUS_LOG_ENTRIES
+//#ifdef BMCWEB_ENABLE_REDFISH_DBUS_LOG_ENTRIES
         if (redfishLogFilePosition != 0)
         {
             cacheRedfishLogFile();
         }
-#endif
+//#endif
         // Update retry configuration.
         subValue->updateRetryConfig(retryAttempts, retryTimeoutInterval);
 
@@ -1367,7 +1367,7 @@ class EventServiceManager
         }
     }
 
-#ifdef BMCWEB_ENABLE_REDFISH_DBUS_LOG_ENTRIES
+//#ifdef BMCWEB_ENABLE_REDFISH_DBUS_LOG_ENTRIES
 
     void resetRedfishFilePosition()
     {
@@ -1615,7 +1615,7 @@ class EventServiceManager
         return 0;
     }
 
-#endif
+//#endif
     static void getReadingsForReport(sdbusplus::message_t& msg)
     {
         if (msg.is_method_error())
-- 
2.43.2

