From 431261888cd55fb02a40bcfef661f1798628daa5 Mon Sep 17 00:00:00 2001
From: GokulprasathP <gokulp@ami.com>
Date: Thu, 4 Apr 2024 12:00:56 +0530
Subject: [PATCH 230/230] Fixed dateTime changes in patch

---
 redfish-core/lib/managers.hpp | 17 +++++++++++------
 1 file changed, 11 insertions(+), 6 deletions(-)

diff --git a/redfish-core/lib/managers.hpp b/redfish-core/lib/managers.hpp
index 3630a28c..62144953 100644
--- a/redfish-core/lib/managers.hpp
+++ b/redfish-core/lib/managers.hpp
@@ -2159,7 +2159,10 @@ inline void setTimeZone(std::shared_ptr<bmcweb::AsyncResp> aResp,
 bool isValidDateTimeFormat(const std::string& datetime)
 {
     // Define the regex pattern for the expected DateTime format
-    std::regex dateTimePattern(R"((\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}\+00:00)|(\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}\.\d{3}Z))");
+    std::regex dateTimePattern(
+        R"((\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}\+\d{2}:\d{2})|(\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}\.\d{3}Z))");
+    // Old (\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}\+00:00)
+
     // Check if the datetime string matches the expected format
     if (!std::regex_match(datetime, dateTimePattern))
     {
@@ -2199,8 +2202,8 @@ inline void setDateTime(std::shared_ptr<bmcweb::AsyncResp> asyncResp,
                     "/redfish/v1/Managers/bmc/NetworkProtocol"));
             return;
         }
-        //asyncResp->res.jsonValue["DateTime"] = datetime;
-	messages::success(asyncResp->res);
+        // asyncResp->res.jsonValue["DateTime"] = datetime;
+        asyncResp->res.result(boost::beast::http::status::no_content);
     });
 }
 
@@ -2537,7 +2540,7 @@ inline void requestRoutesManager(App& app)
             asyncResp->res.result(boost::beast::http::status::bad_request);
             return;
         }
-        //asyncResp->res.result(boost::beast::http::status::no_content);
+        // asyncResp->res.result(boost::beast::http::status::no_content);
         if (oem)
         {
             // #ifdef BMCWEB_ENABLE_REDFISH_OEM_MANAGER_FAN_DATA
@@ -2595,8 +2598,10 @@ inline void requestRoutesManager(App& app)
         }
         if (datetime)
         {
-            if (!isValidDateTimeFormat(*datetime)) // datetime Format should be
-                                                   // yyyy-mm-ddThh:mm:ss+00:00 or yyyy-mm-ddThh:mm:ss.sssz
+            if (!isValidDateTimeFormat(
+                    *datetime)) // datetime Format should be
+                                // yyyy-mm-ddThh:mm:ss+00:00 or
+                                // yyyy-mm-ddThh:mm:ss.sssz
             {
                 messages::propertyValueFormatError(asyncResp->res, *datetime,
                                                    "DateTime");
-- 
2.34.1

