From 135bd4fbbdf0e59277105f6331475c6897226fcd Mon Sep 17 00:00:00 2001
From: prasannas <prasannas@ami.com>
Date: Wed, 27 Mar 2024 18:45:08 +0530
Subject: [PATCH] Fix for Compilation Error in AST2700 build

---
 include/nbd_proxy.hpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/nbd_proxy.hpp b/include/nbd_proxy.hpp
index a0a3f3e2..087ad12c 100644
--- a/include/nbd_proxy.hpp
+++ b/include/nbd_proxy.hpp
@@ -50,7 +50,7 @@ struct NbdProxyServer : std::enable_shared_from_this<NbdProxyServer>
         connection(connIn)
     {
         std::filesystem::path endpointPath(endpointIdIn);
-        endpointIndex = std::stoul(endpointPath.filename().string());
+	endpointIndex = static_cast<unsigned int>(std::stoul(endpointPath.filename().string()));
     }
 
     NbdProxyServer(const NbdProxyServer&) = delete;
-- 
2.43.2

