From 74bc6589ea47f54618f2911a4753a8aa5b8f9c94 Mon Sep 17 00:00:00 2001
From: bharathbharathia <bharathbharathia@ami.com>
Date: Fri, 5 Apr 2024 13:00:16 +0530
Subject: [PATCH 238/238] update Message.ExtendedInfo for ComputerSystem.Reset

---
 redfish-core/lib/systems.hpp | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/redfish-core/lib/systems.hpp b/redfish-core/lib/systems.hpp
index 776d6328..54644443 100644
--- a/redfish-core/lib/systems.hpp
+++ b/redfish-core/lib/systems.hpp
@@ -3800,7 +3800,7 @@ inline void
     }
     else
     {
-        messages::actionParameterUnknown(asyncResp->res, "Reset", resetType);
+        messages::actionParameterNotSupported(asyncResp->res, resetType,"ResetType");
         return;
     }
 
@@ -3936,7 +3936,6 @@ inline void handleComputerSystemResetActionPost(
         !maintenanceWindowStartTime)
     {
         systemResetAction(asyncResp, resetType);
-        messages::success(asyncResp->res);
         return;
     }
 
-- 
2.34.1

