From 01de31e5e9159482b6b50267a33c7bc6311c1928 Mon Sep 17 00:00:00 2001
From: MuthuLakshmiN <muthulakshmin@ami.com>
Date: Thu, 18 Apr 2024 16:08:58 +0530
Subject: [PATCH 249/249] Fix for ipv4 disappears while Setting invalid subnet
 mask

Signed-off-by: MuthuLakshmiN <muthulakshmin@ami.com>
---
 redfish-core/lib/ethernet.hpp | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/redfish-core/lib/ethernet.hpp b/redfish-core/lib/ethernet.hpp
index 954f8959..776feed6 100644
--- a/redfish-core/lib/ethernet.hpp
+++ b/redfish-core/lib/ethernet.hpp
@@ -1477,6 +1477,12 @@ inline bool
                             return false;
                         }
                     }
+                    if(!ip_util::ipv4VerifyIpAndGetBitcount(*subnetMask, &prefixLength))
+                    {
+                        messages::propertyValueFormatError(asyncResp->res, *subnetMask,
+                                                           "SubnetMask");
+                        return false;
+                    }
                 }
             }
         }
-- 
2.34.1

