From 5122c347601a19edaf67960e9df4aec1c8db0237 Mon Sep 17 00:00:00 2001
From: Nibin Chacko <nibinc@ami.com>
Date: Thu, 2 May 2024 13:12:30 -0400
Subject: [PATCH 261/261] resolving nvidia 500 error from systems

---
 redfish-core/lib/led.hpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/redfish-core/lib/led.hpp b/redfish-core/lib/led.hpp
index 3dbbf140..d1ac579d 100644
--- a/redfish-core/lib/led.hpp
+++ b/redfish-core/lib/led.hpp
@@ -291,7 +291,7 @@ inline void getPhysicalLedState(const std::shared_ptr<bmcweb::AsyncResp>& aResp)
         if (ec)
         {
             BMCWEB_LOG_ERROR("Get Physical State Amber Led: DBus Error", ec);
-            messages::internalError(aResp->res);
+            //messages::internalError(aResp->res);
             return;
         }
         setPhysicalLedState(aResp, "AmberLED", amberLedState);
@@ -307,7 +307,7 @@ inline void getPhysicalLedState(const std::shared_ptr<bmcweb::AsyncResp>& aResp)
         if (ec)
         {
             BMCWEB_LOG_ERROR("Get Physical State Green Led: DBus Error", ec);
-            messages::internalError(aResp->res);
+           // messages::internalError(aResp->res);
             return;
         }
         setPhysicalLedState(aResp, "GreenLED", greenLedState);
@@ -323,7 +323,7 @@ inline void getPhysicalLedState(const std::shared_ptr<bmcweb::AsyncResp>& aResp)
         if (ec)
         {
             BMCWEB_LOG_ERROR("Get Physical State Susack Led: DBus Error", ec);
-            messages::internalError(aResp->res);
+            //messages::internalError(aResp->res);
             return;
         }
         setPhysicalLedState(aResp, "SusackLED", susackLedState);
-- 
2.34.1

