From 9de2141f771fbaabe18aa22aef0e521cf1f1afd0 Mon Sep 17 00:00:00 2001
From: prasannas <prasannas@ami.com>
Date: Wed, 15 May 2024 19:52:57 +0530
Subject: [PATCH 267/267] Added TwoFacEnableStatus Property in Login Uri

---
 include/login_routes.hpp | 24 ++++++++++++++++++++++++
 1 file changed, 24 insertions(+)

diff --git a/include/login_routes.hpp b/include/login_routes.hpp
index 65f74bd1..db9b62c3 100644
--- a/include/login_routes.hpp
+++ b/include/login_routes.hpp
@@ -180,6 +180,30 @@ inline void handleLogin(const crow::Request& req,
 
             // if content type is json, assume json token
             asyncResp->res.jsonValue["token"] = session->sessionToken;
+
+            if (std::filesystem::exists("/usr/lib/redfish/core/libami.so.1"))
+            {
+            std::string user(username);
+            sdbusplus::asio::getProperty<bool>(
+                *crow::connections::systemBus,
+                "xyz.openbmc_project.User.Manager",
+                "/xyz/openbmc_project/user/" + user,
+                "xyz.openbmc_project.User.Attributes", "TwoFacEnableStatus",
+                [asyncResp](const boost::system::error_code& ec,
+                            bool ServiceEnabled) {
+                if (ec)
+                {
+                    asyncResp->res.result(
+                        boost::beast::http::status::internal_server_error);
+                    return;
+                }
+                asyncResp->res.jsonValue["TwoFacEnableStatus"] = ServiceEnabled;
+            });
+            }
+            else
+            {
+                asyncResp->res.jsonValue["TwoFacEnableStatus"] = "N/A";
+            }
         }
     }
     else
-- 
2.43.2

