From f1550c093b4f8b4a177a8d00fd138bf0426cd2dd Mon Sep 17 00:00:00 2001
From: Sandeep <sandeepap@ami.com>
Date: Thu, 9 May 2024 16:36:54 +0530
Subject: [PATCH] powerSaveMode support removal from BMCWEB for KVM

Signed-off-by: Sandeep <sandeepap@ami.com>
---
 include/kvm_websocket.hpp | 2 --
 1 file changed, 2 deletions(-)

diff --git a/include/kvm_websocket.hpp b/include/kvm_websocket.hpp
index 0989acde..b4d20ee5 100644
--- a/include/kvm_websocket.hpp
+++ b/include/kvm_websocket.hpp
@@ -274,13 +274,11 @@ inline void requestRoutes(App& app)
         sessions[&conn] = std::make_shared<KvmSession>(conn);
         conn.session->kvmConnections++;
         kvmActiveStatus = 1;
-        redfish::powerSaveMode(POWER_SAVE_MODE_DISABLE);
     })
         .onclose([](crow::websocket::Connection& conn, const std::string&) {
         sessions.erase(&conn);
         conn.session->kvmConnections--;
         kvmActiveStatus = 0;
-        redfish::powerSaveMode(POWER_SAVE_MODE_ENABLE);
     })
         .onmessage([](crow::websocket::Connection& conn,
                       const std::string& data, bool) {
-- 
2.43.2

