From a54b4c3eb4a6a9e4341c863ae8597107cc77fc6b Mon Sep 17 00:00:00 2001
From: Vijaysankar R <vijaysankarr@ami.com>
Date: Thu, 8 May 2025 17:12:11 +0530
Subject: [PATCH 4/4] Deny-Disabled-User
Upstream-Status: Inappropriate [Deny Disabled User Access]
---
 modules/pam_unix/pam_unix_auth.c | 26 ++++++++++++++++++++++++++
 1 file changed, 26 insertions(+)

diff --git a/modules/pam_unix/pam_unix_auth.c b/modules/pam_unix/pam_unix_auth.c
index ffb6154..ff998ec 100644
--- a/modules/pam_unix/pam_unix_auth.c
+++ b/modules/pam_unix/pam_unix_auth.c
@@ -49,6 +49,9 @@
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <syslog.h>
+#include <shadow.h>
+#include <string.h>
+#include <stdbool.h>
 
 #include <security/_pam_macros.h>
 #include <security/pam_modules.h>
@@ -89,6 +92,25 @@ setcred_free (pam_handle_t *pamh UNUSED, void *ptr, int err UNUSED)
 	free (ptr);
 }
 
+bool isUserEnabled(char* userName)
+{
+    char buffer[4096] = {0};
+    struct spwd spwd;
+    struct spwd* resultPtr = NULL;
+
+    int status = getspnam_r(userName, &spwd, buffer, sizeof(buffer), &resultPtr);
+
+    if (!status && (&spwd == resultPtr))
+    {
+        if (resultPtr->sp_expire >= 0)
+        {
+            return false; // user locked out
+        }
+        return true;
+    }
+    return false; // assume user is disabled for any error.
+}
+
 int
 pam_sm_authenticate(pam_handle_t *pamh, int flags, int argc, const char **argv)
 {
@@ -120,6 +142,10 @@ pam_sm_authenticate(pam_handle_t *pamh, int flags, int argc, const char **argv)
 		 * '+' or '-' as the first character of a user name. Don't
 		 * allow these characters here.
 		 */
+		if (!isUserEnabled(name)){
+		retval = PAM_PERM_DENIED;
+		AUTH_RETURN;
+		}
 		if (name[0] == '-' || name[0] == '+') {
 			pam_syslog(pamh, LOG_NOTICE, "bad username [%s]", name);
 			retval = PAM_USER_UNKNOWN;
-- 
2.43.0

