From 288987324c83e72beebbbdb9ee3bf0f76f10b189 Mon Sep 17 00:00:00 2001
From: ramsankarr <ramsankarr@ami.com>
Date: Thu, 7 Aug 2025 13:39:28 +0530
Subject: [PATCH 5/5] Internal Auth Init Info
Upstream-Status: Inappropriate [Internal Auth Init Info]

---
 modules/pam_unix/pam_unix_auth.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/modules/pam_unix/pam_unix_auth.c b/modules/pam_unix/pam_unix_auth.c
index ff998ec..3a88f4a 100644
--- a/modules/pam_unix/pam_unix_auth.c
+++ b/modules/pam_unix/pam_unix_auth.c
@@ -120,6 +120,7 @@ pam_sm_authenticate(pam_handle_t *pamh, int flags, int argc, const char **argv)
 	const char *p;
 
 	D(("called."));
+    syslog(LOG_NOTICE, "Internal User Authentication Started");
 
 	ctrl = _set_ctrl(pamh, flags, NULL, NULL, NULL, argc, argv);
 
-- 
2.34.1

