From 449e9c92f986d2ae080becf9839f3bdc8706b1e3 Mon Sep 17 00:00:00 2001
From: Manoj Ashok <amanoj@ami.com>
Date: Fri, 30 Jan 2026 19:19:56 +0530
Subject: [PATCH] Remove deprecated boost/asio/io_service.hpp include

Signed-off-by: Manoj Ashok <amanoj@ami.com>
---
 PDKHW.hpp    | 2 +-
 PDKHooks.hpp | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/PDKHW.hpp b/PDKHW.hpp
index bd8815d..c93f3b0 100644
--- a/PDKHW.hpp
+++ b/PDKHW.hpp
@@ -1,7 +1,7 @@
 #pragma once
 #include <boost/asio.hpp>
 #include <boost/asio/io_context.hpp>
-#include <boost/asio/io_service.hpp>
+// #include <boost/asio/io_service.hpp>
 #include <phosphor-logging/log.hpp>
 #include <sdbusplus/asio/connection.hpp>
 #include <sdbusplus/asio/object_server.hpp>
diff --git a/PDKHooks.hpp b/PDKHooks.hpp
index b3073d4..cc83bc6 100644
--- a/PDKHooks.hpp
+++ b/PDKHooks.hpp
@@ -1,7 +1,7 @@
 #pragma once
 #include <boost/asio.hpp>
 #include <boost/asio/io_context.hpp>
-#include <boost/asio/io_service.hpp>
+// #include <boost/asio/io_service.hpp>
 #include <phosphor-logging/lg2.hpp>
 #include <phosphor-logging/log.hpp>
 #include <sdbusplus/asio/connection.hpp>
--
2.34.1

