From 2d421583fd5bea49a484fd68c663790dd21fb614 Mon Sep 17 00:00:00 2001
From: abishek <abishekb@ami.com>
Date: Thu, 13 Mar 2025 14:46:28 +0530
Subject: [PATCH 36/36] Fix for OT-coverity issue

---
 src/zlib_file.cpp | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/zlib_file.cpp b/src/zlib_file.cpp
index 73695b0..dda1bf1 100644
--- a/src/zlib_file.cpp
+++ b/src/zlib_file.cpp
@@ -30,7 +30,9 @@ void ZlibFile::close()
         const int rc = gzclose_w(fd);
         if (rc != Z_OK)
         {
-            throw ZlibException(ZlibException::close, rc, fd, fileName);
+            /* Reason for False Positive - fd still holds the valid reference */
+	    /* coverity[use_after_free : FALSE] */
+	    throw ZlibException(ZlibException::close, rc, fd, fileName);
         }
         fd = Z_NULL;
         fileName.clear();
-- 
2.34.1

