From 416e42fb5f771a496a64922151fb35520726722c Mon Sep 17 00:00:00 2001
From: monika <ubuntu@MSGIND-10039S.in.megatrends.com>
Date: Wed, 25 Feb 2026 15:18:18 +0000
Subject: [PATCH 2/2] Replaced io service with io context

---
 src/main.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/main.cpp b/src/main.cpp
index 6e2a963..a5c3ed9 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -25,7 +25,7 @@
 
 int main()
 {
-    boost::asio::io_service io;
+    boost::asio::io_context io;
     auto systemBus = std::make_shared<sdbusplus::asio::connection>(io);
 
     systemBus->request_name(bios_config::service);
-- 
2.43.0

