From c5ea44b3099380137986cb3337c6985dec63c0d0 Mon Sep 17 00:00:00 2001
From: Ethan Wei <ethanwei@ami.com>
Date: Thu, 18 Apr 2024 17:34:05 +0800
Subject: [PATCH] Change Conf File Location

---
 src/openssl-manager.cpp | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/src/openssl-manager.cpp b/src/openssl-manager.cpp
index 0d05bee..ff26db1 100644
--- a/src/openssl-manager.cpp
+++ b/src/openssl-manager.cpp
@@ -18,17 +18,17 @@ using NotAllowedArgument = xyz::openbmc_project::Common::NotAllowed;
 
 
 constexpr int REBOOT_DELAY=5;
-constexpr auto FIPS_CNF = "/usr/lib/ssl-3/fipsmodule.cnf";
-constexpr auto OPENSSL_CONFIG = "/usr/lib/ssl-3/openssl.conf";
+constexpr auto FIPS_CNF = "/etc/ssl/fipsmodule.cnf";
+constexpr auto OPENSSL_CONFIG = "/etc/ssl/openssl.conf";
 
 Manager::Manager(sdbusplus::bus_t& bus, const char* path) :
     ManagerIntf(bus, path)
 {
     if (AMI_OPENSSL_FIPS_SUPPORT) {
-        std::filesystem::path path(OPENSSL_CONFIG);
-        config::Parser parser(path);
+        std::filesystem::path confFile(OPENSSL_CONFIG);
+        config::Parser parser(confFile);
         std::filesystem::path fips(FIPS_CNF);
-        if (std::filesystem::exists(path)) {
+        if (std::filesystem::exists(confFile)) {
             auto str1 = parser.map.getLastValueString("FIPS", "FIPSEnable");
             ManagerIntf::fips((!str1->empty() && *str1 == "true") ? true : false);
         } // if
@@ -84,4 +84,4 @@ void Manager::writeConfiguration() {
     auto path(OPENSSL_CONFIG);
     config.writeFile(path);
     log<level::INFO>(fmt::format("Wrote openssl configuration file: {}", path).c_str());
-}
\ No newline at end of file
+}
-- 
2.25.1

