--- a/src/vlan_interface.hpp  2022-07-06 15:08:50.000000000 +0530
+++ b/src/vlan_interface.hpp       2022-09-02 11:47:45.634818971 +0530
@@ -68,6 +68,7 @@
   private:
     /** @brief VLAN Identifier. */
     using VlanIface::id;
+    using VlanIface::priority;

     EthernetInterface& parentInterface;

 diff --git a/src/vlan_interface.cpp b/src/vlan_interface.cpp
 index b893be4a..836f0110 100644
--- a/src/vlan_interface.cpp
+++ b/src/vlan_interface.cpp
@@ -31,6 +31,7 @@ VlanInterface::VlanInterface(sdbusplus::bus::bus& bus,
     parentInterface(intf)
 {
     id(vlanID);
+    priority(0);
     VlanIface::interfaceName(EthernetInterface::interfaceName());
     MacAddressIntf::macAddress(parentInterface.macAddress());

@@ -67,6 +68,7 @@ void VlanInterface::writeDeviceFile()
     stream << "Kind=vlan\n";
     stream << "[VLAN]\n";
     stream << "Id=" << id() << "\n";
+    stream << "Priority=" << priority() << "\n";
     stream.close();
 }

