diff -Naur a/src/vlan_interface.cpp b/src/vlan_interface.cpp
--- a/src/vlan_interface.cpp	2022-10-20 18:55:53.135322150 +0530
+++ b/src/vlan_interface.cpp	2022-10-20 19:09:07.902568389 +0530
@@ -27,6 +27,7 @@
     parentInterface(intf)
 {
     id(vlanID);
+    priority(0);
     VlanIface::interfaceName(EthernetInterface::interfaceName());
     MacAddressIntf::macAddress(parentInterface.macAddress());
 
@@ -46,6 +47,7 @@
     netdev["Name"].emplace_back(EthernetInterface::interfaceName());
     netdev["Kind"].emplace_back("vlan");
     config.map["VLAN"].emplace_back()["Id"].emplace_back(std::to_string(id()));
+    config.map["VLAN"].emplace_back()["Priority"].emplace_back(std::to_string(priority()));
     config.writeFile(config::pathForIntfDev(
         manager.getConfDir(), EthernetInterface::interfaceName()));
 }
diff -Naur a/src/vlan_interface.hpp b/src/vlan_interface.hpp
--- a/src/vlan_interface.hpp	2022-10-20 18:55:53.135322150 +0530
+++ b/src/vlan_interface.hpp	2022-10-20 19:09:24.834668415 +0530
@@ -69,6 +69,7 @@
   private:
     /** @brief VLAN Identifier. */
     using VlanIface::id;
+    using VlanIface::priority;
 
     EthernetInterface& parentInterface;
 
