diff -Naur a/utils/mctp-demux-daemon.c b/utils/mctp-demux-daemon.c
--- a/utils/mctp-demux-daemon.c	2024-11-07 14:16:33.625595509 +0800
+++ b/utils/mctp-demux-daemon.c	2024-11-07 14:48:07.733415801 +0800
@@ -385,7 +385,7 @@
 	int rc;
 
 	rc = mctp_message_tx(ctx->mctp, eid,
-			     (tag_owner_and_tag & LIBMCTP_TAG_OWNER_MASK),
+			     ((tag_owner_and_tag & LIBMCTP_TAG_OWNER_MASK) >> 3),
 			     (tag_owner_and_tag & LIBMCTP_TAG_MASK), msg, len);
 	if (rc)
 		warnx("Failed to send message: %d", rc);
