From 04d99475c23b9a2b6a298dc80fd6d93c233849de Mon Sep 17 00:00:00 2001
From: pavithra <pavithrat@ami.com>
Date: Thu, 9 Oct 2025 14:48:38 +0530
Subject: [PATCH] 0013-Fix-for-unclosed-sockets

---
 agent/snmp_agent.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/agent/snmp_agent.c b/agent/snmp_agent.c
index bd57efc..e2050bb 100644
--- a/agent/snmp_agent.c
+++ b/agent/snmp_agent.c
@@ -146,7 +146,7 @@ bool getSystemLockEnabled() {
     sd_bus_message *reply = NULL;
     sd_bus *bus = NULL;
     int ret = -1;
-
+    bool result = false;
 
     ret = sd_bus_open_system(&bus);
     if (ret < 0) {
@@ -182,14 +182,15 @@ bool getSystemLockEnabled() {
         goto cleanup;
     }
 
-    return true;
+    result = true;
+    goto cleanup;
 
 cleanup:
     sd_bus_error_free(&error);
     sd_bus_message_unref(reply);
     sd_bus_unref(bus);
     snmp_log(LOG_ERR, "END %d\n", __LINE__);
-    return false;
+    return result;
 }
 
 
-- 
2.34.1

