From c05120823375c4328da5dcda5db1e17aaf545257 Mon Sep 17 00:00:00 2001
From: thrishalig <thrishalig@ami.com>
Date: Tue, 29 Aug 2023 13:27:28 +0530
Subject: [PATCH] pwmtachtool , dutycycle is giving percentage instead of
value

Signed-off-by: thrishalig <thrishalig@ami.com>
---
 pwmtach.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pwmtach.c b/pwmtach.c
index d5eab93..8f67a64 100644
--- a/pwmtach.c
+++ b/pwmtach.c
@@ -138,7 +138,7 @@ static int GET_PWM_DUTYCYCLE ( pwmtach_ioctl_data  *ppwmtach_arg )
 	read(fd, duty_num, 5);
     dutycycle = atoi(duty_num);
 	ppwmtach_arg->dutycycle = dutycycle * 100 / 255;
-	printf("%s:dutycycle value %d to pwm %d\n",__FUNCTION__,ppwmtach_arg->dutycycle,ppwmtach_arg->pwmnumber);
+	printf("%s:dutycycle value %d to pwm %d\n",__FUNCTION__,dutycycle,ppwmtach_arg->pwmnumber);
 	(void)sigwrap_close(fd);
 
 	return retval;
-- 
2.25.1

