From fa6f34293055cfe4be86caa59b712a92915eab4a Mon Sep 17 00:00:00 2001
From: Ramya Sivakumar <sramya@ami.com>
Date: Fri, 27 Feb 2026 18:29:56 +0530
Subject: [PATCH] build_fix_pwmtachtool

---
 pwmtachtool/src/EINTR_wrappers.c | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/EINTR_wrappers.c b/EINTR_wrappers.c
index e8f02db..4542af9 100755
--- a/EINTR_wrappers.c
+++ b/EINTR_wrappers.c
@@ -15,13 +15,11 @@
 #endif
 #include <errno.h>
 #include <unistd.h>
+#include <stdbool.h>
+#include <sys/uio.h>
 
 static const int OneSecondasNS = 1000000000;
 
-#ifndef bool
-typedef int bool;
-#endif
-
 #ifndef TRUE
 #define TRUE (1)
 #endif
