From ec74e448e175ddd13cd93121e0e9d50504f70bef Mon Sep 17 00:00:00 2001
From: ramsankarr <ramsankarr@ami.com>
Date: Thu, 7 Aug 2025 13:38:12 +0530
Subject: [PATCH 2014/2014] LDAP/AD Auth Init Info

---
 pam/pam.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/pam/pam.c b/pam/pam.c
index ee020f6..043ca13 100644
--- a/pam/pam.c
+++ b/pam/pam.c
@@ -470,6 +470,7 @@ int pam_sm_authenticate(pam_handle_t *pamh, int flags,
   const char *ruser = NULL, *rhost = NULL, *tty = NULL;
   char *passwd = NULL;
   struct nslcd_resp resp;
+  syslog(LOG_NOTICE, "LDAP/AD User Authentication Started");
   /* set up configuration */
   cfg_init(pamh, flags, argc, argv, &cfg);
   rc = init(pamh, &cfg, &ctx, &username, &service, &ruser, &rhost, &tty);
-- 
2.34.1

