From 0f2be0fd2c3f4dc7a3b09be19aa1f5c8d91a1d5c Mon Sep 17 00:00:00 2001
From: Vijaysankar R <vijaysankarr@ami.com>
Date: Sun, 7 Sep 2025 09:11:10 +0530
Subject: [PATCH 2015/2015] Coverity-Fix

---
 nslcd/group.c  | 18 ++++++++++--------
 nslcd/shadow.c | 19 ++++++++++---------
 2 files changed, 20 insertions(+), 17 deletions(-)

diff --git a/nslcd/group.c b/nslcd/group.c
index 09dbb53..1a3227f 100644
--- a/nslcd/group.c
+++ b/nslcd/group.c
@@ -198,30 +198,32 @@ void group_init(void)
   }
   /* set up attribute list */
   set = set_new();
-  attmap_add_attributes(set, attmap_group_cn);
-  attmap_add_attributes(set, attmap_group_userPassword);
-  attmap_add_attributes(set, attmap_group_gidNumber);
+  set = attmap_add_attributes(set, attmap_group_cn);
+  set = attmap_add_attributes(set, attmap_group_userPassword);
+  set = attmap_add_attributes(set, attmap_group_gidNumber);
   if (!nslcd_cfg->nss_getgrent_skipmembers)
   {
-    attmap_add_attributes(set, attmap_group_memberUid);
-    attmap_add_attributes(set, attmap_group_member);
+    set = attmap_add_attributes(set, attmap_group_memberUid);
+    set = attmap_add_attributes(set, attmap_group_member);
   }
   group_attrs = set_tolist(set);
   if (group_attrs == NULL)
   {
     log_log(LOG_CRIT, "malloc() failed to allocate memory");
+    set_free(set);
     exit(EXIT_FAILURE);
   }
   set_free(set);
   /* set up bymember attribute list */
   set = set_new();
-  attmap_add_attributes(set, attmap_group_cn);
-  attmap_add_attributes(set, attmap_group_userPassword);
-  attmap_add_attributes(set, attmap_group_gidNumber);
+  set = attmap_add_attributes(set, attmap_group_cn);
+  set = attmap_add_attributes(set, attmap_group_userPassword);
+  set = attmap_add_attributes(set, attmap_group_gidNumber);
   group_bymember_attrs = set_tolist(set);
   if (group_bymember_attrs == NULL)
   {
     log_log(LOG_CRIT, "malloc() failed to allocate memory");
+    set_free(set);
     exit(EXIT_FAILURE);
   }
   set_free(set);
diff --git a/nslcd/shadow.c b/nslcd/shadow.c
index 5fe5a94..c6b79f8 100644
--- a/nslcd/shadow.c
+++ b/nslcd/shadow.c
@@ -97,19 +97,20 @@ void shadow_init(void)
     shadow_scope = nslcd_cfg->scope;
   /* set up attribute list */
   set = set_new();
-  attmap_add_attributes(set, attmap_shadow_uid);
-  attmap_add_attributes(set, attmap_shadow_userPassword);
-  attmap_add_attributes(set, attmap_shadow_shadowLastChange);
-  attmap_add_attributes(set, attmap_shadow_shadowMax);
-  attmap_add_attributes(set, attmap_shadow_shadowMin);
-  attmap_add_attributes(set, attmap_shadow_shadowWarning);
-  attmap_add_attributes(set, attmap_shadow_shadowInactive);
-  attmap_add_attributes(set, attmap_shadow_shadowExpire);
-  attmap_add_attributes(set, attmap_shadow_shadowFlag);
+  set = attmap_add_attributes(set, attmap_shadow_uid);
+  set = attmap_add_attributes(set, attmap_shadow_userPassword);
+  set = attmap_add_attributes(set, attmap_shadow_shadowLastChange);
+  set = attmap_add_attributes(set, attmap_shadow_shadowMax);
+  set = attmap_add_attributes(set, attmap_shadow_shadowMin);
+  set = attmap_add_attributes(set, attmap_shadow_shadowWarning);
+  set = attmap_add_attributes(set, attmap_shadow_shadowInactive);
+  set = attmap_add_attributes(set, attmap_shadow_shadowExpire);
+  set = attmap_add_attributes(set, attmap_shadow_shadowFlag);
   shadow_attrs = set_tolist(set);
   if (shadow_attrs == NULL)
   {
     log_log(LOG_CRIT, "malloc() failed to allocate memory");
+    set_free(set);
     exit(EXIT_FAILURE);
   }
   set_free(set);
-- 
2.34.1

