From 761a35a0999060abfc3a993e7feaa373b8c7f56e Mon Sep 17 00:00:00 2001
From: pavithra <pavithrat@ami.com>
Date: Thu, 13 Nov 2025 18:21:29 +0530
Subject: [PATCH] Added coverity fix in nss-pam-ldapd

---
 nslcd/cfg.c    |  2 ++
 nslcd/passwd.c | 15 ++++++++-------
 2 files changed, 10 insertions(+), 7 deletions(-)

diff --git a/nslcd/cfg.c b/nslcd/cfg.c
index bb620ee..ba16012 100644
--- a/nslcd/cfg.c
+++ b/nslcd/cfg.c
@@ -696,6 +696,8 @@ static void handle_base(const char *filename, int lnr,
     if (bases[i] == NULL)
     {
       bases[i] = value;
+      /* Reason for False Positive – domain should not be free due to its points to static buffer from getfqdn() */
+      /* coverity[leaked_storage : FALSE] */
       return;
     }
   /* no free spot found */
diff --git a/nslcd/passwd.c b/nslcd/passwd.c
index 5ca05bd..3173cb6 100644
--- a/nslcd/passwd.c
+++ b/nslcd/passwd.c
@@ -143,17 +143,18 @@ void passwd_init(void)
   /* set up attribute list */
   set = set_new();
   attmap_add_attributes(set, "objectClass"); /* for testing shadowAccount */
-  attmap_add_attributes(set, attmap_passwd_uid);
-  attmap_add_attributes(set, attmap_passwd_userPassword);
-  attmap_add_attributes(set, attmap_passwd_uidNumber);
-  attmap_add_attributes(set, attmap_passwd_gidNumber);
-  attmap_add_attributes(set, attmap_passwd_gecos);
-  attmap_add_attributes(set, attmap_passwd_homeDirectory);
-  attmap_add_attributes(set, attmap_passwd_loginShell);
+  set = attmap_add_attributes(set, attmap_passwd_uid);
+  set = attmap_add_attributes(set, attmap_passwd_userPassword);
+  set = attmap_add_attributes(set, attmap_passwd_uidNumber);
+  set = attmap_add_attributes(set, attmap_passwd_gidNumber);
+  set = attmap_add_attributes(set, attmap_passwd_gecos);
+  set = attmap_add_attributes(set, attmap_passwd_homeDirectory);
+  set = attmap_add_attributes(set, attmap_passwd_loginShell);
   passwd_attrs = set_tolist(set);
   if (passwd_attrs == NULL)
   {
     log_log(LOG_CRIT, "malloc() failed to allocate memory");
+    set_free(set);
     exit(EXIT_FAILURE);
   }
   set_free(set);
-- 
2.34.1

