From 4e4eb4c36eee1b8bb65e6a872ea7d25899e22c7c Mon Sep 17 00:00:00 2001
From: Vijaysankar R <vijaysankarr@ami.com>
Date: Thu, 25 Dec 2025 18:31:52 +0530
Subject: [PATCH 2017/2017] Reduced-Retry-Time

---
 nslcd/cfg.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/nslcd/cfg.c b/nslcd/cfg.c
index ba16012..a8339da 100644
--- a/nslcd/cfg.c
+++ b/nslcd/cfg.c
@@ -1248,7 +1248,7 @@ static void cfg_defaults(struct ldap_config *cfg)
   cfg->timelimit = LDAP_NO_LIMIT;
   cfg->idle_timelimit = 0;
   cfg->reconnect_sleeptime = 1;
-  cfg->reconnect_retrytime = 10;
+  cfg->reconnect_retrytime = 2;
 #ifdef LDAP_OPT_X_TLS
   cfg->ssl = SSL_OFF;
 #endif /* LDAP_OPT_X_TLS */
-- 
2.34.1

