From 8b2385ca9892bc1ae934b26153f2bc9c77448831 Mon Sep 17 00:00:00 2001
From: eddyhan <eddyhan@ami.com>
Date: Tue, 8 Apr 2025 18:08:13 +0800
Subject: [PATCH] CMakeLists64.


diff --git a/CMakeLists.txt b/CMakeLists.txt
index 174a88c5..20829212 100755
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -574,7 +574,16 @@ if(CMAKE_SYSTEM_NAME MATCHES "Linux")
 
         SET(CMAKE_C_LINK_EXECUTABLE "")
     elseif(TOOLCHAIN STREQUAL "OPENBMC_GCC")
-        SET(CMAKE_C_COMPILER arm-openbmc-linux-gnueabi-gcc)
+        if(ARCH STREQUAL "aarch64")
+            SET(CMAKE_C_COMPILER aarch64-openbmc-linux-gcc)
+            SET(CMAKE_AR aarch64-openbmc-linux-gcc-ar)
+            SET(CMAKE_LINKER aarch64-openbmc-linux-gcc)
+            SET(DISABLE_TESTS "1")
+        else()
+            SET(CMAKE_C_COMPILER arm-openbmc-linux-gnueabi-gcc)
+            SET(CMAKE_AR arm-openbmc-linux-gnueabi-gcc-ar)
+            SET(CMAKE_LINKER arm-openbmc-linux-gnueabi-gcc)
+        endif()
         SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=c99 -g  -fno-strict-aliasing -Wall -Werror -Wno-array-bounds -ffunction-sections -fdata-sections -fno-common -Wno-address -fPIC -fno-asynchronous-unwind-tables -Wno-maybe-uninitialized -Wno-uninitialized  -Wno-builtin-declaration-mismatch -Wno-nonnull-compare")
        if(NOT CMAKE_BUILD_TYPE STREQUAL "Release")
             ADD_COMPILE_OPTIONS(-g)
@@ -587,13 +596,11 @@ if(CMAKE_SYSTEM_NAME MATCHES "Linux")
         SET(OPENSSL_FLAGS -include base.h -Wno-error=maybe-uninitialized -Wno-error=format -Wno-format -Wno-format-security -Wno-error=unused-but-set-variable -Wno-cast-qual)
         SET(CMOCKA_FLAGS -std=gnu99 -Wpedantic -Wall -Wshadow -Wmissing-prototypes -Wcast-align -Werror=address -Wstrict-prototypes -Werror=strict-prototypes -Wwrite-strings -Werror=write-strings -Werror-implicit-function-declaration -Wpointer-arith -Werror=pointer-arith -Wdeclaration-after-statement -Werror=declaration-after-statement -Wreturn-type -Werror=return-type -Wuninitialized -Werror=uninitialized -Werror=strict-overflow -Wstrict-overflow=2 -Wno-format-zero-length -Wmissing-field-initializers -Wformat-security -Werror=format-security -fno-common -Wformat -fno-common -fstack-protector-strong -Wno-cast-qual)
 
-        SET(CMAKE_AR arm-openbmc-linux-gnueabi-gcc-ar)
 
         if(NOT CMAKE_C_COMPILER_VERSION VERSION_GREATER_EQUAL 9.3)
                 SET(CMAKE_C_ARCHIVE_FINISH true)
         endif()
 
-        SET(CMAKE_LINKER arm-openbmc-linux-gnueabi-gcc)
         SET(CMAKE_EXE_LINKER_FLAGS "$ENV{LDFLAGS} ${OBMCGCCOPTION} -flto -Wno-error -no-pie" )
         if(GCOV STREQUAL "ON")
         SET(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS}   --coverage -lgcov -fprofile-arcs -ftest-coverage")
-- 
2.34.1

