diff -Naur a/gen/xyz/openbmc_project/meson.build b/gen/xyz/openbmc_project/meson.build
--- a/gen/xyz/openbmc_project/meson.build	2023-12-28 00:43:32.794217674 -0500
+++ b/gen/xyz/openbmc_project/meson.build	2023-12-28 00:47:19.801236883 -0500
@@ -14,6 +14,7 @@
     ],
 )
 
+subdir('Snmp')
 subdir('Backup')
 subdir('BIOSConfig')
 subdir('Certs')
diff -Naur a/gen/xyz/openbmc_project/Snmp/meson.build b/gen/xyz/openbmc_project/Snmp/meson.build
--- a/gen/xyz/openbmc_project/Snmp/meson.build	1969-12-31 19:00:00.000000000 -0500
+++ b/gen/xyz/openbmc_project/Snmp/meson.build	2023-12-28 00:36:51.212106200 -0500
@@ -0,0 +1,16 @@
+# Generated file; do not modify.
+subdir('SnmpAgent')
+generated_others += custom_target(
+    'xyz/openbmc_project/Snmp/SnmpAgent__markdown'.underscorify(),
+    input: [ '../../../../yaml/xyz/openbmc_project/Snmp/SnmpAgent.interface.yaml',  ],
+    output: [ 'SnmpAgent.md' ],
+    depend_files: sdbusplusplus_depfiles,
+    command: [
+        sdbuspp_gen_meson_prog, '--command', 'markdown',
+        '--output', meson.current_build_dir(),
+        '--tool', sdbusplusplus_prog,
+        '--directory', meson.current_source_dir() / '../../../../yaml',
+        'xyz/openbmc_project/Snmp/SnmpAgent',
+    ],
+)
+
diff -Naur a/gen/xyz/openbmc_project/Snmp/SnmpAgent/meson.build b/gen/xyz/openbmc_project/Snmp/SnmpAgent/meson.build
--- a/gen/xyz/openbmc_project/Snmp/SnmpAgent/meson.build	1969-12-31 19:00:00.000000000 -0500
+++ b/gen/xyz/openbmc_project/Snmp/SnmpAgent/meson.build	2023-12-28 00:36:51.212106200 -0500
@@ -0,0 +1,15 @@
+# Generated file; do not modify.
+generated_sources += custom_target(
+    'xyz/openbmc_project/Snmp/SnmpAgent__cpp'.underscorify(),
+    input: [ '../../../../../yaml/xyz/openbmc_project/Snmp/SnmpAgent.interface.yaml',  ],
+    output: [ 'server.cpp', 'server.hpp', 'client.hpp',  ],
+    depend_files: sdbusplusplus_depfiles,
+    command: [
+        sdbuspp_gen_meson_prog, '--command', 'cpp',
+        '--output', meson.current_build_dir(),
+        '--tool', sdbusplusplus_prog,
+        '--directory', meson.current_source_dir() / '../../../../../yaml',
+        'xyz/openbmc_project/Snmp/SnmpAgent',
+    ],
+)
+
diff -Naur a/yaml/xyz/openbmc_project/Snmp/SnmpAgent.interface.yaml b/yaml/xyz/openbmc_project/Snmp/SnmpAgent.interface.yaml
--- a/yaml/xyz/openbmc_project/Snmp/SnmpAgent.interface.yaml	1969-12-31 19:00:00.000000000 -0500
+++ b/yaml/xyz/openbmc_project/Snmp/SnmpAgent.interface.yaml	2023-12-28 00:36:51.212106200 -0500
@@ -0,0 +1,48 @@
+description: >
+methods:
+    - name: CreateSnmpAgent
+      description: >
+          Create a SnmpAgent
+      errors:
+          - xyz.openbmc_project.Common.Error.NotAllowed
+      parameters:
+          - name: SnmpAgent
+            type: string
+            description: >
+                Name for the backup file
+      returns:
+          - name: SnmpAgentName
+            type: string
+            description: >
+                The path for the created backup file
+    - name: GetSnmpAgent
+      description: >
+          GetSnmp
+      errors:
+          - xyz.openbmc_project.Common.Error.NotAllowed
+      returns:
+          - name: SnmpAgent
+            type: string
+            description: >
+                Indicates if successful or not
+properties:
+    - name: amiSnmpSleeper
+      type: int32
+      description: >
+          This is a simple object which is a basic integer.  It's value
+           indicates the number of seconds that the agent will take in
+           responding to requests of this object.
+
+    - name: amiSnmpString
+      type: string
+      description: >
+          This is a simple object which merely houses a writable
+           string.  It's only purposes is to hold the value of a single
+           string.
+
+    - name: amiSnmpInteger
+      type: int32
+      description: >
+          This is a simple object which merely houses a writable
+           integer.  It's only purposes is to hold the value of a single
+           integer.
