--- /dev/null	2023-03-06 17:55:29.135707410 -0500
+++ a/yaml/xyz/openbmc_project/Backup/BackupRestore.interface.yaml	2023-04-04 15:18:11.953634633 -0400
@@ -0,0 +1,37 @@
+description: >
+methods:
+    - name: CreateBackup
+      description: >
+          Create a Backup and Restore.
+      errors:
+          - xyz.openbmc_project.Common.Error.NotAllowed
+      parameters:
+          - name: BackupName
+            type: string
+            description: >
+                Name for the backup file
+      returns:
+          - name: Path
+            type: string
+            description: >
+                The path for the created backup file
+    - name: RestoreBackup
+      description: >
+          Restore a Backup.
+      errors:
+          - xyz.openbmc_project.Common.Error.NotAllowed
+      parameters:
+          - name: BackupFilePath
+            type: string
+            description: >
+                The backup file path
+      returns:
+          - name: retvalue
+            type: boolean
+            description: >
+                Indicates if successful or not
+properties:
+    - name: BackupFlags
+      type: string
+      description: >
+          Backup Flags
--- a/gen/xyz/openbmc_project/meson.build	2023-03-23 16:29:45.862865771 -0400
+++ b/gen/xyz/openbmc_project/meson.build	2023-04-11 17:34:30.120686074 -0400
@@ -14,6 +14,7 @@
     ],
 )
 
+subdir('Backup')
 subdir('BIOSConfig')
 subdir('Certs')
 generated_others += custom_target(
--- /dev/null	2023-03-06 17:55:29.135707410 -0500
+++ a/gen/xyz/openbmc_project/Backup/meson.build	2023-04-06 16:14:36.608126253 -0400
@@ -0,0 +1,16 @@
+# Generated file; do not modify.
+subdir('BackupRestore')
+generated_others += custom_target(
+    'xyz/openbmc_project/Backup/BackupRestore__markdown'.underscorify(),
+    input: [ '../../../../yaml/xyz/openbmc_project/Backup/BackupRestore.interface.yaml',  ],
+    output: [ 'BackupRestore.md' ],
+    depend_files: sdbusplusplus_depfiles,
+    command: [
+        sdbuspp_gen_meson_prog, '--command', 'markdown',
+        '--output', meson.current_build_dir(),
+        '--tool', sdbusplusplus_prog,
+        '--directory', meson.current_source_dir() / '../../../../yaml',
+        'xyz/openbmc_project/Backup/BackupRestore',
+    ],
+)
+
--- /dev/null	2023-03-06 17:55:29.135707410 -0500
+++ a/gen/xyz/openbmc_project/Backup/BackupRestore/meson.build	2023-04-06 16:14:36.608126253 -0400
@@ -0,0 +1,15 @@
+# Generated file; do not modify.
+generated_sources += custom_target(
+    'xyz/openbmc_project/Backup/BackupRestore__cpp'.underscorify(),
+    input: [ '../../../../../yaml/xyz/openbmc_project/Backup/BackupRestore.interface.yaml',  ],
+    output: [ 'server.cpp', 'server.hpp', 'client.hpp',  ],
+    depend_files: sdbusplusplus_depfiles,
+    command: [
+        sdbuspp_gen_meson_prog, '--command', 'cpp',
+        '--output', meson.current_build_dir(),
+        '--tool', sdbusplusplus_prog,
+        '--directory', meson.current_source_dir() / '../../../../../yaml',
+        'xyz/openbmc_project/Backup/BackupRestore',
+    ],
+)
+
