From 6f64a1722c9482663b9d54b0916e4497a6a3a07d Mon Sep 17 00:00:00 2001
From: Vasanthakumar K <vasanthakumark@ami.com>
Date: Sat, 30 May 2026 14:15:03 +0530
Subject: [PATCH] revert uimage_flash_addr to 80400000

Signed-off-by: Vasanthakumar K <vasanthakumark@ami.com>
---
 include/configs/arbel.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/configs/arbel.h b/include/configs/arbel.h
index b928f0f2746..02064eb1eb3 100644
--- a/include/configs/arbel.h
+++ b/include/configs/arbel.h
@@ -24,7 +24,7 @@
 #define SPI3_END_ADDR			0xBFFFFFFF
 
 /* Default environemnt variables */
-#define CFG_EXTRA_ENV_SETTINGS   "uimage_flash_addr=80800000\0"   \
+#define CFG_EXTRA_ENV_SETTINGS   "uimage_flash_addr=80400000\0"   \
 		"stdin=serial\0"   \
 		"stdout=serial\0"   \
 		"stderr=serial\0"    \
-- 
2.34.1

