From 83e6cb35724df25450daf49712936e8c8d3785b4 Mon Sep 17 00:00:00 2001
From: santhoshs <santhoshs@ami.com>
Date: Thu, 17 Apr 2025 03:51:36 +0000
Subject: [PATCH] Enable-auto-negotiation-of-SGMII-mode-at-U-boot

Upstream-Status: Inappropriate [AMI: driver changes]
Signed-off-by: santhoshs <santhoshs@ami.com>
---
 drivers/net/designware.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/net/designware.c b/drivers/net/designware.c
index 656fb52f23..f302bff5d9 100644
--- a/drivers/net/designware.c
+++ b/drivers/net/designware.c
@@ -384,6 +384,11 @@ static int dw_adjust_link(struct dw_eth_dev *priv, struct eth_mac_regs *mac_p,
 			    default:
 				break;
 			}
+			#endif
+			writew(0x1F80, 0xF07801FE);           /* Get access to 0x3F... (VR_MII_MMD_DIG_CTRL1) */
+			writew(readw(0xF0780002) | 4, 0xF0780002);	/* Set PCS_Mode to SGMII */
+			writew(readw(0xF0780000) | (1 << 9), 0xF0780000);	/* Set MAC_AUTO_SW */
+			writew(0x1F00, 0xF07801FE);           /* Get access to 0x3E... (SR_MII_STS) */
 
 			start = get_timer(0);
 			printf("SGMII PHY Wait for link up \n");
-- 
2.34.1

