From 3627dd98d2c593d695bff7acdb8b97502b62d934 Mon Sep 17 00:00:00 2001
From: santhoshs <santhoshs@ami.com>
Date: Thu, 17 Apr 2025 03:54:59 +0000
Subject: [PATCH] Re-enable-sgmii-auto-neg-feature-after-set-PCS-mode

Upstream-Status: Inappropriate [AMI: driver changes]
Signed-off-by: santhoshs <santhoshs@ami.com>
---
 drivers/net/designware.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/net/designware.c b/drivers/net/designware.c
index f302bff5d9..877becd3df 100644
--- a/drivers/net/designware.c
+++ b/drivers/net/designware.c
@@ -362,6 +362,7 @@ static int dw_adjust_link(struct dw_eth_dev *priv, struct eth_mac_regs *mac_p,
 			/* Clear SGMII PHY default auto neg. */
 			writew(readw(0xF0780000) & ~(1 << SR_MII_CTRL_ANEN_BIT12), 0xF0780000);
 
+			#if 0
 			switch (phydev->speed) {
 			    case SPEED_1000:
 				/* Set SGMII PHY 10/100/1000   SS6=1  */
@@ -389,6 +390,8 @@ static int dw_adjust_link(struct dw_eth_dev *priv, struct eth_mac_regs *mac_p,
 			writew(readw(0xF0780002) | 4, 0xF0780002);	/* Set PCS_Mode to SGMII */
 			writew(readw(0xF0780000) | (1 << 9), 0xF0780000);	/* Set MAC_AUTO_SW */
 			writew(0x1F00, 0xF07801FE);           /* Get access to 0x3E... (SR_MII_STS) */
+			/* Enable SGMII PHY default auto neg. */
+			writew(readw(0xF0780000) | (1 << SR_MII_CTRL_ANEN_BIT12), 0xF0780000);
 
 			start = get_timer(0);
 			printf("SGMII PHY Wait for link up \n");
-- 
2.34.1

