From e83c9f2342b5213a49e62aaed2a8c747e7bca6c9 Mon Sep 17 00:00:00 2001
From: santhoshs <santhoshs@ami.com>
Date: Mon, 28 Apr 2025 10:39:12 +0000
Subject: [PATCH] Modified-i2c-eeprom-file

Upstream-Status: Inappropriate [AMI: driver changes]
Signed-off-by: santhoshs <santhoshs@ami.com>
---
 drivers/misc/i2c_eeprom.c | 14 +++++++++-----
 include/i2c_eeprom.h      | 33 ++++-----------------------------
 2 files changed, 13 insertions(+), 34 deletions(-)

diff --git a/drivers/misc/i2c_eeprom.c b/drivers/misc/i2c_eeprom.c
index bdd7e018cc..5926c91a2e 100644
--- a/drivers/misc/i2c_eeprom.c
+++ b/drivers/misc/i2c_eeprom.c
@@ -3,8 +3,6 @@
  * Copyright (c) 2014 Google, Inc
  */
 
-#define LOG_CATEGORY UCLASS_I2C_EEPROM
-
 #include <common.h>
 #include <eeprom.h>
 #include <linux/delay.h>
@@ -33,8 +31,7 @@ int i2c_eeprom_read(struct udevice *dev, int offset, uint8_t *buf, int size)
 	return ops->read(dev, offset, buf, size);
 }
 
-int i2c_eeprom_write(struct udevice *dev, int offset, const uint8_t *buf,
-		     int size)
+int i2c_eeprom_write(struct udevice *dev, int offset, uint8_t *buf, int size)
 {
 	const struct i2c_eeprom_ops *ops = device_get_ops(dev);
 
@@ -170,6 +167,13 @@ static const struct i2c_eeprom_drv_data eeprom_data = {
 	.offset_len = 1,
 };
 
+static const struct i2c_eeprom_drv_data mc24aa02e48_data = {
+	.size = 256,
+	.pagesize = 8,
+	.addr_offset_mask = 0,
+	.offset_len = 1,
+};
+
 static const struct i2c_eeprom_drv_data atmel24c01a_data = {
 	.size = 128,
 	.pagesize = 8,
@@ -257,7 +261,7 @@ static const struct i2c_eeprom_drv_data atmel24c512_data = {
 
 static const struct udevice_id i2c_eeprom_std_ids[] = {
 	{ .compatible = "i2c-eeprom", (ulong)&eeprom_data },
-	{ .compatible = "atmel,24c01", (ulong)&atmel24c01a_data },
+	{ .compatible = "microchip,24aa02e48", (ulong)&mc24aa02e48_data },
 	{ .compatible = "atmel,24c01a", (ulong)&atmel24c01a_data },
 	{ .compatible = "atmel,24c02", (ulong)&atmel24c02_data },
 	{ .compatible = "atmel,24c04", (ulong)&atmel24c04_data },
diff --git a/include/i2c_eeprom.h b/include/i2c_eeprom.h
index 32dcb03497..cd620d519f 100644
--- a/include/i2c_eeprom.h
+++ b/include/i2c_eeprom.h
@@ -6,8 +6,6 @@
 #ifndef __I2C_EEPROM
 #define __I2C_EEPROM
 
-struct udevice;
-
 struct i2c_eeprom_ops {
 	int (*read)(struct udevice *dev, int offset, uint8_t *buf, int size);
 	int (*write)(struct udevice *dev, int offset, const uint8_t *buf,
@@ -22,7 +20,6 @@ struct i2c_eeprom {
 	unsigned long size;
 };
 
-#if CONFIG_IS_ENABLED(I2C_EEPROM)
 /*
  * i2c_eeprom_read() - read bytes from an I2C EEPROM chip
  *
@@ -31,7 +28,7 @@ struct i2c_eeprom {
  * @buf:	Place to put data
  * @size:	Number of bytes to read
  *
- * Return: 0 on success, -ve on failure
+ * @return 0 on success, -ve on failure
  */
 int i2c_eeprom_read(struct udevice *dev, int offset, uint8_t *buf, int size);
 
@@ -43,39 +40,17 @@ int i2c_eeprom_read(struct udevice *dev, int offset, uint8_t *buf, int size);
  * @buf:	Buffer containing data to write
  * @size:	Number of bytes to write
  *
- * Return: 0 on success, -ve on failure
+ * @return 0 on success, -ve on failure
  */
-int i2c_eeprom_write(struct udevice *dev, int offset, const uint8_t *buf,
-		     int size);
+int i2c_eeprom_write(struct udevice *dev, int offset, uint8_t *buf, int size);
 
 /*
  * i2c_eeprom_size() - get size of I2C EEPROM chip
  *
  * @dev:	Chip to query
  *
- * Return: +ve size in bytes on success, -ve on failure
+ * @return +ve size in bytes on success, -ve on failure
  */
 int i2c_eeprom_size(struct udevice *dev);
 
-#else /* !I2C_EEPROM */
-
-static inline int i2c_eeprom_read(struct udevice *dev, int offset, uint8_t *buf,
-				  int size)
-{
-	return -ENOSYS;
-}
-
-static inline int i2c_eeprom_write(struct udevice *dev, int offset,
-				   const uint8_t *buf, int size)
-{
-	return -ENOSYS;
-}
-
-static inline int i2c_eeprom_size(struct udevice *dev)
-{
-	return -ENOSYS;
-}
-
-#endif /* I2C_EEPROM */
-
 #endif
-- 
2.34.1

