From bfeb0259621743e0a367daed068c192d3c5cb80e Mon Sep 17 00:00:00 2001
From: kirubagharanas <kirubagharanas@ami.com>
Date: Tue, 23 Sep 2025 12:31:46 +0530
Subject: [PATCH] enable fieldmode in uboot and save the env before auto-boot

Upstream-Status: Inappropriate [AMI: driver changes]
---
 common/main.c           | 1 +
 include/configs/arbel.h | 1 +
 2 files changed, 2 insertions(+)

diff --git a/common/main.c b/common/main.c
index 7c70de2e59a..b258a3b54bd 100644
--- a/common/main.c
+++ b/common/main.c
@@ -65,6 +65,7 @@ void main_loop(void)
 	if (cli_process_fdt(&s))
 		cli_secure_boot_cmd(s);
 
+	env_save();
 	autoboot_command(s);
 
 	cli_loop();
diff --git a/include/configs/arbel.h b/include/configs/arbel.h
index ce94a944768..596196f6565 100644
--- a/include/configs/arbel.h
+++ b/include/configs/arbel.h
@@ -45,6 +45,7 @@
 		"tftptimeoutcountmax=50\0" \
 		"bootpretryperiod=60000\0" \
 		"stopwdt=wdt dev watchdog@901c; wdt stop\0" \
+		"fieldmode=true\0" \
 		"\0"
 
 #endif
-- 
2.34.1

