From d364135a1cbf1ad3a90df8535818a7905457fc97 Mon Sep 17 00:00:00 2001
From: kirubagharanas <kirubagharanas@ami.com>
Date: Tue, 8 Oct 2024 13:46:05 +0530
Subject: [PATCH] add new interface for partial preserve config support

---
 src/entity_manager.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/entity_manager.cpp b/src/entity_manager.cpp
index 929bfea..7a2aae5 100644
--- a/src/entity_manager.cpp
+++ b/src/entity_manager.cpp
@@ -68,8 +68,8 @@ const boost::container::flat_map<const char*, probe_type_codes, CmpStr>
                 {"FOUND", probe_type_codes::FOUND},
                 {"MATCH_ONE", probe_type_codes::MATCH_ONE}}};
 
-static constexpr std::array<const char*, 6> settableInterfaces = {
-    "FanProfile", "Pid", "Pid.Zone", "Stepwise", "Thresholds", "Polling"};
+static constexpr std::array<const char*, 7> settableInterfaces = {
+    "FanProfile", "Pid", "Pid.Zone", "Stepwise", "Thresholds", "Polling", "Preserve"};
 using JsonVariantType =
     std::variant<std::vector<std::string>, std::vector<double>, std::string,
                  int64_t, uint64_t, double, int32_t, uint32_t, int16_t,
-- 
2.34.1

