From 40b53efab897f17bad9413aaf333db5a408e78d6 Mon Sep 17 00:00:00 2001
From: pavithra <pavithrat@ami.com>
Date: Mon, 21 Jul 2025 10:40:34 +0530
Subject: [PATCH] CoverityFixFinal

---
 src/fru_device.cpp | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/fru_device.cpp b/src/fru_device.cpp
index 46b7774..ea20b4c 100644
--- a/src/fru_device.cpp
+++ b/src/fru_device.cpp
@@ -817,6 +817,7 @@ static void findI2CDevices(const std::vector<fs::path>& i2cBuses,
         {
             std::cerr << "Error: Can't use SMBus Receive Byte command bus "
                       << bus << "\n";
+	    close(file);
             continue;
         }
         auto& device = busmap[bus];
@@ -1131,8 +1132,8 @@ std::vector<uint8_t> getFruConfig(const uint16_t& fruBus,
         }
         auto configBus = properties.find("Bus");
         auto configAddr = properties.find("Address");
-        uint8_t confBus;
-        uint8_t confAddr;
+        uint8_t confBus = 0;
+        uint8_t confAddr = 0;
 
         if (configBus == properties.end() || configAddr == properties.end())
         {
-- 
2.34.1

