From e1fa3d6822a0af546aba579342a93eb915c4935a Mon Sep 17 00:00:00 2001
From: Vasanthakumar K <vasanthakumark@ami.com>
Date: Thu, 3 Jul 2025 23:16:38 +0530
Subject: [PATCH] MCTP driver memoryleak fix

Signed-off-by: Vasanthakumar K <vasanthakumark@ami.com>
---
 drivers/soc/aspeed/aspeed-mctp.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/soc/aspeed/aspeed-mctp.c b/drivers/soc/aspeed/aspeed-mctp.c
index c962f4ef6662..0617929ae772 100644
--- a/drivers/soc/aspeed/aspeed-mctp.c
+++ b/drivers/soc/aspeed/aspeed-mctp.c
@@ -2352,7 +2352,7 @@ static int aspeed_mctp_probe(struct platform_device *pdev)
 		return id;
 	priv->mctp_miscdev.parent = priv->dev;
 	priv->mctp_miscdev.minor = MISC_DYNAMIC_MINOR;
-	priv->mctp_miscdev.name = kasprintf(GFP_KERNEL, "aspeed-mctp%d", id);
+	priv->mctp_miscdev.name = devm_kasprintf(priv->dev, GFP_KERNEL, "aspeed-mctp%d", id);
 	priv->mctp_miscdev.fops = &aspeed_mctp_fops;
 	ret = misc_register(&priv->mctp_miscdev);
 	if (ret) {
@@ -2384,7 +2384,7 @@ static int aspeed_mctp_probe(struct platform_device *pdev)
 		aspeed_mctp_rx_trigger(&priv->rx);
 	}
 
-	name = kasprintf(GFP_KERNEL, "peci-mctp%d", id);
+	name = devm_kasprintf(priv->dev, GFP_KERNEL, "peci-mctp%d", id);
 	priv->peci_mctp =
 		platform_device_register_data(priv->dev, name, PLATFORM_DEVID_NONE, NULL, 0);
 	if (IS_ERR(priv->peci_mctp))
@@ -2506,4 +2506,4 @@ module_exit(aspeed_mctp_exit)
 MODULE_DEVICE_TABLE(of, aspeed_mctp_match_table);
 MODULE_LICENSE("GPL");
 MODULE_AUTHOR("Iwona Winiarska <iwona.winiarska@intel.com>");
-MODULE_DESCRIPTION("Aspeed MCTP driver");
\ No newline at end of file
+MODULE_DESCRIPTION("Aspeed MCTP driver");
-- 
2.34.1

