From af1ccdd471c09005567b67214b83466ab62f3671 Mon Sep 17 00:00:00 2001
From: roycheng <roycheng@ami.com>
Date: Thu, 18 Sep 2025 13:56:49 +0800
Subject: [PATCH] OneTree 2.1 MCTP coverity issues

---
 drivers/soc/aspeed/aspeed-mctp.c | 17 ++++++++++-------
 1 file changed, 10 insertions(+), 7 deletions(-)

diff --git a/drivers/soc/aspeed/aspeed-mctp.c b/drivers/soc/aspeed/aspeed-mctp.c
index 8902667978e3..befdb39d1752 100644
--- a/drivers/soc/aspeed/aspeed-mctp.c
+++ b/drivers/soc/aspeed/aspeed-mctp.c
@@ -181,13 +181,14 @@
 #define MCTP_HDR_SOM			BIT(7)
 #define MCTP_HDR_EOM			BIT(6)
 #define MCTP_HDR_SOM_EOM		(MCTP_HDR_SOM | MCTP_HDR_EOM)
-#define MCTP_HDR_TYPE_OFFSET		16
+#define MCTP_PAYLOAD_TYPE_OFFSET		0
 #define MCTP_HDR_TYPE_CONTROL		0
 #define MCTP_HDR_TYPE_VDM_PCI		0x7e
 #define MCTP_HDR_TYPE_SPDM		0x5
 #define MCTP_HDR_TYPE_BASE_LAST		MCTP_HDR_TYPE_SPDM
-#define MCTP_HDR_VENDOR_OFFSET		17
-#define MCTP_HDR_VDM_TYPE_OFFSET	19
+#define MCTP_PAYLOAD_VENDOR_OFFSET		1
+#define MCTP_PAYLOAD_VDM_TYPE_OFFSET	3
+
 
 /* MCTP header DW little endian mask definitions */
 /* 0th DW */
@@ -722,6 +723,7 @@ aspeed_mctp_find_handler(struct aspeed_mctp *priv,
 {
 	struct mctp_type_handler *handler;
 	u8 *hdr = (u8 *)packet->data.hdr;
+	u8 *payload = (u8 *)packet->data.payload;
 	struct mctp_client *client = NULL;
 	u8 mctp_type, som_eom;
 	u16 vendor = 0;
@@ -737,10 +739,10 @@ aspeed_mctp_find_handler(struct aspeed_mctp *priv,
 	if (som_eom != MCTP_HDR_SOM_EOM)
 		return NULL;
 
-	mctp_type = hdr[MCTP_HDR_TYPE_OFFSET];
+	mctp_type = payload[MCTP_PAYLOAD_TYPE_OFFSET];
 	if (mctp_type == MCTP_HDR_TYPE_VDM_PCI) {
-		vendor = *((u16 *)&hdr[MCTP_HDR_VENDOR_OFFSET]);
-		vdm_type = *((u16 *)&hdr[MCTP_HDR_VDM_TYPE_OFFSET]);
+		vendor = *((u16 *)&payload[MCTP_PAYLOAD_VENDOR_OFFSET]);
+		vdm_type = *((u16 *)&payload[MCTP_PAYLOAD_VDM_TYPE_OFFSET]);
 	}
 
 	list_for_each_entry(handler, &priv->mctp_type_handlers, link) {
@@ -1252,6 +1254,7 @@ int aspeed_mctp_send_packet(struct mctp_client *client,
 	struct aspeed_mctp *priv = client->priv;
 	u32 *hdr_dw = (u32 *)packet->data.hdr;
 	u8 *hdr = (u8 *)packet->data.hdr;
+	u8 *payload = (u8 *)packet->data.payload;
 	u16 packet_data_sz_dw;
 	u16 pci_data_len_dw;
 	int ret;
@@ -1282,7 +1285,7 @@ int aspeed_mctp_send_packet(struct mctp_client *client,
 	 * XXX Don't update EID for MCTP Control messages - old EID may
 	 * interfere with MCTP discovery flow.
 	 */
-	if (priv->eid && hdr[MCTP_HDR_TYPE_OFFSET] != MCTP_HDR_TYPE_CONTROL)
+	if (priv->eid && payload[MCTP_PAYLOAD_TYPE_OFFSET] != MCTP_HDR_TYPE_CONTROL)
 		hdr[MCTP_HDR_SRC_EID_OFFSET] = priv->eid;
 
 	ret = ptr_ring_produce_bh(&client->tx_queue, packet);
-- 
2.34.1

