From ca87f8c660ea01456ba2dc5297f851b78b41be8a Mon Sep 17 00:00:00 2001
From: fredhsieh <fredhsieh@ami.com>
Date: Fri, 19 Sep 2025 21:33:41 +0800
Subject: [PATCH] Fix common kernel I2C patch error in chalupa

Upstream-Status: Inappropriate [AMI: driver changes]
---
 drivers/i2c/i2c-smbus.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/i2c/i2c-smbus.c b/drivers/i2c/i2c-smbus.c
index 6520e0974391..87a12da6c7cc 100644
--- a/drivers/i2c/i2c-smbus.c
+++ b/drivers/i2c/i2c-smbus.c
@@ -274,6 +274,7 @@ static int i2c_slave_host_notify_cb(struct i2c_client *client,
 			i2c_handle_smbus_host_notify(client->adapter,
 						     status->addr);
 		fallthrough;
+	case I2C_SLAVE_GCALL_REQUESTED:
 	case I2C_SLAVE_WRITE_REQUESTED:
 		status->index = 0;
 		break;
-- 
2.34.1

