From 5e77bedcdbbb6630590b24d454525bc53c538373 Mon Sep 17 00:00:00 2001
From: Vijay K S <vijayks@ami.com>
Date: Wed, 15 Jul 2026 11:43:14 +0530
Subject: [PATCH] Fixed the bootup issue by removing the func. call added twice

Signed-off-by: Vijay K S <vijayks@ami.com>

Upstream-Status: Inappropriate [AMI: driver changes]

---
 drivers/i2c/i2c-core-smbus.c | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/drivers/i2c/i2c-core-smbus.c b/drivers/i2c/i2c-core-smbus.c
index 1b60dc2b92e5..046ab889018e 100644
--- a/drivers/i2c/i2c-core-smbus.c
+++ b/drivers/i2c/i2c-core-smbus.c
@@ -540,10 +540,6 @@ s32 i2c_smbus_xfer(struct i2c_adapter *adapter, u16 addr,
 	bool do_bus_lock = true;
 	s32 res;
 
-	res = __i2c_lock_bus_helper(adapter);
-	if (res)
-		return res;
-
 	/*
 	 * Do not lock a bus for delivering an unhold msg to a mux adpater.
 	 * This is just for a single length unhold msg case.
-- 
2.34.1

