From b98edfeca96d31702b65df5f99740a3dd7fdc2d4 Mon Sep 17 00:00:00 2001
From: roycheng <roycheng@ami.com>
Date: Fri, 31 Oct 2025 11:40:14 +0800
Subject: [PATCH] 0023-Receive-MCTP-Broadcast-Package.patch

---
 net/mctp/route.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/net/mctp/route.c b/net/mctp/route.c
index da12b5bc3fe1..05cd755eedd7 100644
--- a/net/mctp/route.c
+++ b/net/mctp/route.c
@@ -789,7 +789,7 @@ static struct mctp_route *mctp_route_lookup_null(struct net *net,
 	rcu_read_lock();
 
 	list_for_each_entry_rcu(tmp, &net->mctp.routes, list) {
-		if (tmp->dev->dev == dev && tmp->type == RTN_LOCAL &&
+		if (tmp->type == RTN_LOCAL &&
 		    refcount_inc_not_zero(&tmp->refs)) {
 			rt = tmp;
 			break;
@@ -1207,7 +1207,7 @@ static int mctp_pkttype_receive(struct sk_buff *skb, struct net_device *dev,
 	rt = mctp_route_lookup(net, cb->net, mh->dest);
 
 	/* NULL EID, but addressed to our physical address */
-	if (!rt && mh->dest == MCTP_ADDR_NULL && skb->pkt_type == PACKET_HOST)
+	if (!rt && (mh->dest == MCTP_ADDR_ANY || (mh->dest == MCTP_ADDR_NULL && skb->pkt_type == PACKET_HOST)))
 		rt = mctp_route_lookup_null(net, dev);
 
 	if (!rt)
-- 
2.34.1

