From 963a72c3b30dbb06cc54f3647ee7b51f881ef745 Mon Sep 17 00:00:00 2001
From: Ethan Wei <ethanwei@ami.com>
Date: Thu, 2 Jan 2025 10:24:30 +0800
Subject: [PATCH] Turn on NCSI Messages for Link State

---
 net/ncsi/ncsi-aen.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/net/ncsi/ncsi-aen.c b/net/ncsi/ncsi-aen.c
index ba637619ebb4..9fbfdba67f5c 100644
--- a/net/ncsi/ncsi-aen.c
+++ b/net/ncsi/ncsi-aen.c
@@ -74,7 +74,7 @@ static int ncsi_aen_handler_lsc(struct ncsi_dev_priv *ndp,
 	had_link = !!(old_data & 0x1);
 	has_link = !!(data & 0x1);
 
-	netdev_dbg(ndp->ndev.dev, "NCSI: LSC AEN - channel %u state %s\n",
+	netdev_info(ndp->ndev.dev, "NCSI: LSC AEN - channel %u state %s\n",
 		   nc->id, data & 0x1 ? "up" : "down");
 
 	//update carrier in sysfs path, which helps dhcp
-- 
2.34.1


