From 9858921fb83ca2dcd32328d4193995ce63f07b13 Mon Sep 17 00:00:00 2001
From: saikumaru <saikumaru@ami.com>
Date: Tue, 9 Sep 2025 10:32:39 +0530
Subject: [PATCH] rtl8211f: customized led

Signed-off-by: saikumaru <saikumaru@ami.com>
Upstream-Status: Backport [Nuvoton SDK/ASPEED SDK migration (git commit cb052d637)]
---
 drivers/net/phy/realtek/realtek_main.c | 21 +++++++++++++++++++++
 1 file changed, 21 insertions(+)

diff --git a/drivers/net/phy/realtek/realtek_main.c b/drivers/net/phy/realtek/realtek_main.c
index 7c3d277efaf0..7087396c512f 100644
--- a/drivers/net/phy/realtek/realtek_main.c
+++ b/drivers/net/phy/realtek/realtek_main.c
@@ -1737,6 +1737,26 @@ static irqreturn_t rtl9000a_handle_interrupt(struct phy_device *phydev)
 	return IRQ_HANDLED;
 }
 
+static void rtl8211f_link_change_notify(struct phy_device *phydev)
+{
+	if (phydev->link && phydev->state == PHY_RUNNING) {
+		phy_write(phydev, RTL821x_PAGE_SELECT, 0xd04);
+
+		if (phydev->speed == SPEED_1000) {
+			phy_write(phydev, 0x10, 0x6008);
+			phy_write(phydev, 0x11, 0x6008);
+		} else if (phydev->speed == SPEED_100) {
+			phy_write(phydev, 0x10, 0x4840);
+			phy_write(phydev, 0x11, 0x6008);
+		} else if (phydev->speed == SPEED_10) {
+			phy_write(phydev, 0x10, 0x4400);
+			phy_write(phydev, 0x11, 0x6008);
+		}
+
+		phy_write(phydev, RTL821x_PAGE_SELECT, 0);
+	}
+}
+
 static struct phy_driver realtek_drvs[] = {
 	{
 		PHY_ID_MATCH_EXACT(0x00008201),
@@ -1815,6 +1835,7 @@ static struct phy_driver realtek_drvs[] = {
 		.name		= "RTL8211F Gigabit Ethernet",
 		.probe		= rtl8211f_probe,
 		.config_init	= &rtl8211f_config_init,
+		.link_change_notify = &rtl8211f_link_change_notify,
 		.read_status	= rtlgen_read_status,
 		.config_intr	= &rtl8211f_config_intr,
 		.handle_interrupt = rtl8211f_handle_interrupt,
-- 
2.34.1

