From 9858921fb83ca2dcd32328d4193995ce63f07b13 Mon Sep 17 00:00:00 2001
From: saikumaru <saikumaru@ami.com>
Date: Tue, 9 Sep 2025 10:32:39 +0530
Subject: [PATCH] rtl8211f: customized led

Signed-off-by: saikumaru <saikumaru@ami.com>
---
 drivers/net/phy/realtek.c | 21 +++++++++++++++++++++
 1 file changed, 21 insertions(+)

diff --git a/drivers/net/phy/realtek.c b/drivers/net/phy/realtek.c
index 2604d9663a5b..b89ed3c824ea 100644
--- a/drivers/net/phy/realtek.c
+++ b/drivers/net/phy/realtek.c
@@ -313,6 +313,26 @@ static irqreturn_t rtl8211f_handle_interrupt(struct phy_device *phydev)
 	return IRQ_HANDLED;
 }
 
+static void rtl8211f_link_change_notify(struct phy_device *phydev)
+{
+	if (phydev->link && phydev->state == PHY_RUNNING) {
+		phy_write(phydev, RTL821x_PAGE_SELECT, 0xd04);
+
+		if (phydev->speed == SPEED_1000) {
+			phy_write(phydev, 0x10, 0x6008);
+			phy_write(phydev, 0x11, 0x6008);
+		} else if (phydev->speed == SPEED_100) {
+			phy_write(phydev, 0x10, 0x4840);
+			phy_write(phydev, 0x11, 0x6008);
+		} else if (phydev->speed == SPEED_10) {
+			phy_write(phydev, 0x10, 0x4400);
+			phy_write(phydev, 0x11, 0x6008);
+		}
+
+		phy_write(phydev, RTL821x_PAGE_SELECT, 0);
+	}
+}
+
 static int rtl8211_config_aneg(struct phy_device *phydev)
 {
 	int ret;
@@ -954,6 +974,7 @@ static struct phy_driver realtek_drvs[] = {
 		.name		= "RTL8211F Gigabit Ethernet",
 		.probe		= rtl821x_probe,
 		.config_init	= &rtl8211f_config_init,
+		.link_change_notify     = &rtl8211f_link_change_notify,
 		.read_status	= rtlgen_read_status,
 		.config_intr	= &rtl8211f_config_intr,
 		.handle_interrupt = rtl8211f_handle_interrupt,
-- 
2.47.1

