From 88f4cfe9b92f62c46ccfea28efc0207314f5acf6 Mon Sep 17 00:00:00 2001
From: saikumaru <saikumaru@ami.com>
Date: Tue, 9 Sep 2025 10:36:24 +0530
Subject: [PATCH] Add workaround for using header file in user space without
 compile error

Signed-off-by: saikumaru <saikumaru@ami.com>
---
 include/linux/compiler.h | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/include/linux/compiler.h b/include/linux/compiler.h
index 5a4054f17cbc..351f14b525d3 100644
--- a/include/linux/compiler.h
+++ b/include/linux/compiler.h
@@ -258,6 +258,9 @@ static inline void *offset_to_ptr(const int *off)
  */
 #define prevent_tail_call_optimization()	mb()
 
-#include <asm/rwonce.h>
+//Add workaround for using header file in user space without error
+#ifdef CONFIG_ARM
+ #include <asm/rwonce.h>
+#endif
 
 #endif /* __LINUX_COMPILER_H */
-- 
2.47.1

