From e10aa3a58397d14f4b630d3b26cad68956edfc16 Mon Sep 17 00:00:00 2001
From: rashidmp <rashidmp@ami.com>
Date: Wed, 3 Sep 2025 16:26:13 +0530
Subject: [PATCH] net: ethernet: stmmac: add sgmii support

Signed-off-by: rashidmp <rashidmp@ami.com>
---
 .../ethernet/stmicro/stmmac/stmmac_platform.c | 21 +++++++++++++++++++
 1 file changed, 21 insertions(+)

diff --git a/drivers/net/ethernet/stmicro/stmmac/stmmac_platform.c b/drivers/net/ethernet/stmicro/stmmac/stmmac_platform.c
index 6c684f6ee84b..e69f0cc344c9 100644
--- a/drivers/net/ethernet/stmicro/stmmac/stmmac_platform.c
+++ b/drivers/net/ethernet/stmicro/stmmac/stmmac_platform.c
@@ -20,6 +20,8 @@
 #include "stmmac.h"
 #include "stmmac_platform.h"
 
+#define IND_AC_INDX	0x1FE
+#define SR_MII_CTRL	0x003E0000
 #ifdef CONFIG_OF
 
 /**
@@ -420,6 +422,8 @@ stmmac_probe_config_dt(struct platform_device *pdev, u8 *mac)
 	struct plat_stmmacenet_data *plat;
 	struct stmmac_dma_cfg *dma_cfg;
 	static int bus_id = -ENODEV;
+	void __iomem *base;
+	u16 RegValue;
 	int phy_mode;
 	void *ret;
 	int rc;
@@ -515,6 +519,7 @@ stmmac_probe_config_dt(struct platform_device *pdev, u8 *mac)
 	if (of_device_is_compatible(np, "st,spear600-gmac") ||
 		of_device_is_compatible(np, "snps,dwmac-3.50a") ||
 		of_device_is_compatible(np, "snps,dwmac-3.70a") ||
+		of_device_is_compatible(np, "nuvoton,npcm-dwmac") ||
 		of_device_is_compatible(np, "snps,dwmac")) {
 		/* Note that the max-frame-size parameter as defined in the
 		 * ePAPR v1.1 spec is defined as max-frame-size, it's
@@ -649,6 +654,22 @@ stmmac_probe_config_dt(struct platform_device *pdev, u8 *mac)
 		goto error_hw_init;
 	}
 
+	if (of_device_is_compatible(np, "nuvoton,npcm-dwmac")) {
+		base = devm_platform_ioremap_resource(pdev, 1);
+		if (IS_ERR(base)) {
+			dev_warn(&pdev->dev, "devm_platform_ioremap_resource failed\n");
+		}
+		iowrite16((u16)(SR_MII_CTRL >> 9), base + IND_AC_INDX);
+		RegValue = ioread16(base + 0x2);
+		RegValue = ioread16(base + 0x0);
+		RegValue |= BIT(15);
+		iowrite16(RegValue, base + 0x0);
+		while (RegValue & BIT(15))
+			RegValue = ioread16(base + 0x0);
+		RegValue &= ~(BIT(12));
+		iowrite16(RegValue, base + 0x0);
+	}
+
 	return plat;
 
 error_hw_init:
-- 
2.34.1

